Index: kdegames-4.4.4/kiriki/src/kiriki.desktop
===================================================================
--- kdegames-4.4.4.orig/kiriki/src/kiriki.desktop
+++ kdegames-4.4.4/kiriki/src/kiriki.desktop
@@ -91,7 +91,7 @@ GenericName[sr@ijekavian]=Јамболи
 GenericName[sr@ijekavianlatin]=Jambolika igra sa kockicama
 GenericName[sr@latin]=Jambolika igra sa kockicama
 GenericName[sv]=Yatzy-liknande tärningsspel
-GenericName[tr]=Yahtzee-benzeri Zar Oyunu
+GenericName[tr]=Yahtzee benzeri Zar Oyunu
 GenericName[uk]=Гра схожа на покер з кістками
 GenericName[wa]=On djeu d' dés ki ravize li djeu Yahtzee
 GenericName[x-test]=xxYahtzee-like Dice Gamexx
