Index: opera-11.00-1156.i386.linux/share/applications/opera-browser.desktop
===================================================================
--- opera-11.00-1156.i386.linux.orig/share/applications/opera-browser.desktop
+++ opera-11.00-1156.i386.linux/share/applications/opera-browser.desktop
@@ -21,6 +21,7 @@ GenericName[pt_BR]=Navegador
 GenericName[ro]=Navigator de web
 GenericName[ru]=Веб-браузер
 GenericName[sl]=Spletni brskalnik
+GenericName[tr]=Web Tarayıcı
 GenericName[ven]=Buronza ya Webu
 GenericName[xh]=Umkhangeli Zincwadi Zokubhaliweyo
 GenericName[zu]=Umkhangeli zincwadi we Web
