Index: opera-11.11-2109.x86_64.linux/share/applications/opera-browser.desktop
===================================================================
--- opera-11.11-2109.x86_64.linux.orig/share/applications/opera-browser.desktop
+++ opera-11.11-2109.x86_64.linux/share/applications/opera-browser.desktop
@@ -21,6 +21,7 @@ GenericName[pt_BR]=Navegador
 GenericName[ro]=Navigator de web
 GenericName[ru]=Веб-браузер
 GenericName[sl]=Spletni brskalnik
+GenericName[tr]=Web Tarayıcı
 GenericName[ven]=Buronza ya Webu
 GenericName[xh]=Umkhangeli Zincwadi Zokubhaliweyo
 GenericName[zu]=Umkhangeli zincwadi we Web
Index: opera-11.11-2109.x86_64.linux/install
===================================================================
--- opera-11.11-2109.x86_64.linux.orig/install
+++ opera-11.11-2109.x86_64.linux/install
@@ -18,7 +18,7 @@ iterate()
 	"$@" NX 7b4daa2168e8ad899bbc6b9c856ecdaa lib/opera@@{SUFFIX}/operapluginwrapper-ia32-linux
 	"$@" NX 9ad7fe367dd9dc01e78de7cd2324702a lib/opera@@{SUFFIX}/operapluginwrapper-native
 	"$@" NF 0a8389b127c477aa2c55ee5d234473ca lib/opera@@{SUFFIX}/plugins/README
-	"$@" PF e081fdf465b85cf9cb7a032e1aee5539 share/applications/opera-browser@@{SUFFIX}.desktop
+	"$@" PF 82afca33b2978e0ceada4db49349150c share/applications/opera-browser@@{SUFFIX}.desktop
 	"$@" PF c32d6b1cdaa99944767e83a59fe27b4d share/applications/opera-widget-installer@@{SUFFIX}.desktop
 	"$@" PF bf9c8c97f9432e69936000fc9db64290 share/applications/opera-widget-manager@@{SUFFIX}.desktop
 	"$@" NF 3214f080875748938ba060314b4f727d share/doc/opera@@{SUFFIX}/LGPL-2
