Index: opera-11.10-2092.x86_64.linux/share/applications/opera-browser.desktop
===================================================================
--- opera-11.10-2092.x86_64.linux.orig/share/applications/opera-browser.desktop
+++ opera-11.10-2092.x86_64.linux/share/applications/opera-browser.desktop
@@ -21,6 +21,7 @@ GenericName[pt_BR]=Navegador
 GenericName[ro]=Navigator de web
 GenericName[ru]=Веб-браузер
 GenericName[sl]=Spletni brskalnik
+GenericName[tr]=Web Tarayıcı
 GenericName[ven]=Buronza ya Webu
 GenericName[xh]=Umkhangeli Zincwadi Zokubhaliweyo
 GenericName[zu]=Umkhangeli zincwadi we Web
Index: opera-11.10-2092.x86_64.linux/install
===================================================================
--- opera-11.10-2092.x86_64.linux.orig/install
+++ opera-11.10-2092.x86_64.linux/install
@@ -18,7 +18,7 @@ iterate()
 	"$@" NX 8b08c6ad6b9db689877896d3a22d2798 lib/opera@@{SUFFIX}/operapluginwrapper-ia32-linux
 	"$@" NX 57d050f349d22c0f1cc6760bd311ab56 lib/opera@@{SUFFIX}/operapluginwrapper-native
 	"$@" NF 0a8389b127c477aa2c55ee5d234473ca lib/opera@@{SUFFIX}/plugins/README
-	"$@" PF 4f94950abb9e8a521213260054561503 share/applications/opera-browser@@{SUFFIX}.desktop
+	"$@" PF 3c565f85023fc2a1b0f35b05dd5773da share/applications/opera-browser@@{SUFFIX}.desktop
 	"$@" PF c32d6b1cdaa99944767e83a59fe27b4d share/applications/opera-widget-installer@@{SUFFIX}.desktop
 	"$@" PF bf9c8c97f9432e69936000fc9db64290 share/applications/opera-widget-manager@@{SUFFIX}.desktop
 	"$@" NF 3214f080875748938ba060314b4f727d share/doc/opera@@{SUFFIX}/LGPL-2
