diff -Nur pyopenssl-0.8-3/work/pyOpenSSL-0.8/test/test_crypto.py pyOpenSSL-0.8/test/test_crypto.py
--- pyopenssl-0.8-3/work/pyOpenSSL-0.8/test/test_crypto.py	2008-12-16 16:44:33.000000000 +0200
+++ pyOpenSSL-0.8/test/test_crypto.py	2008-12-16 16:42:37.000000000 +0200
@@ -653,3 +653,8 @@
         self.assertTrue(isinstance(loadedKey, PKeyType))
         self.assertEqual(loadedKey.type(), key.type())
         self.assertEqual(loadedKey.bits(), key.bits())
+
+if __name__ == '__main__':
+    import unittest
+    unittest.main()
+
diff -Nur pyopenssl-0.8-3/work/pyOpenSSL-0.8/test/test_ssl.py pyOpenSSL-0.8/test/test_ssl.py
--- pyopenssl-0.8-3/work/pyOpenSSL-0.8/test/test_ssl.py	2008-12-16 16:44:33.000000000 +0200
+++ pyOpenSSL-0.8/test/test_ssl.py	2008-12-16 16:42:08.000000000 +0200
@@ -115,3 +115,7 @@
 
         # Kind of lame.  Just make sure it got called somehow.
         self.assertTrue(called)
+
+if __name__ == '__main__':
+    import unittest
+    unittest.main()
