Index: smb4k-0.10.66/smb4k/smb4k.desktop
===================================================================
--- smb4k-0.10.66.orig/smb4k/smb4k.desktop
+++ smb4k-0.10.66/smb4k/smb4k.desktop
@@ -18,7 +18,9 @@ Name[tr]=Smb4K
 Name[uk]=Smb4K
 Name[x-test]=xxSmb4Kxx
 Name[zh_CN]=Smb4K
+GenericName[tr]=Samba Paylaşım Yöneticisi
 Comment=Advanced Network Neighborhood Browser
+Comment[tr]=Gelişmiş Çevre Ağ Tarayıcısı
 Comment[da]=Den avancerede browser til netværksnabolag
 Comment[de]=Erweiterter Browser für die Netzwerkumgebung
 Comment[en_GB]=Advanced Network Neighbourhood Browser
@@ -36,6 +38,6 @@ Icon=smb4k
 X-DocPath=smb4k/index.html
 Type=Application
 Terminal=false
-Categories=Qt;KDE;Utility;
+Categories=Qt;KDE;Network;Utility;
 X-DBUS-StartupType=Unique
 X-DBUS-ServiceName=de.berlios.smb4k
