diff -Nur texlive-core-20091107-old/texlive-20091107-source/texk/web2c/pdftexdir/pdftoepdf.cc texlive-core-20091107/texlive-20091107-source/texk/web2c/pdftexdir/pdftoepdf.cc
--- texlive-core-20091107-old/texlive-20091107-source/texk/web2c/pdftexdir/pdftoepdf.cc	2010-08-24 10:17:31.194000001 +0300
+++ texlive-core-20091107/texlive-20091107-source/texk/web2c/pdftexdir/pdftoepdf.cc	2010-08-24 10:19:05.960000217 +0300
@@ -650,7 +650,6 @@
     }
     for (r = encodingList; r != 0; r = n) {
         n = r->next;
-        delete r->font;
         delete r;
     }
 }
