Index: texlive-core-20091107/texlive-20091107-source/texk/texlive/linked_scripts/Makefile.am
===================================================================
--- texlive-core-20091107.orig/texlive-20091107-source/texk/texlive/linked_scripts/Makefile.am
+++ texlive-core-20091107/texlive-20091107-source/texk/texlive/linked_scripts/Makefile.am
@@ -17,11 +17,9 @@ scriptxdir = ${prefix}/texmf/scripts
 nobase_dist_scriptx_SCRIPTS = \
 	a2ping/a2ping.pl \
 	getnonfreefonts/getnonfreefonts.pl \
-	simpdftex/simpdftex \
 	tetex/e2pall.pl \
 	tetex/texdoctk.pl \
 	texdoc/texdoc.tlu \
-	texlive/rungs.tlu \
 	texlive/tlmgr.pl
 
 scriptdir = ${prefix}/texmf-dist/scripts
@@ -29,32 +27,8 @@ nobase_dist_script_SCRIPTS = \
 	accfonts/mkt1font \
 	accfonts/vpl2ovp \
 	accfonts/vpl2vpl \
-	bengali/ebong.py \
-	cachepic/cachepic.tlu \
-	context/lua/luatools.lua \
-	context/lua/mtxrun.lua \
-	context/ruby/texmfstart.rb \
-	context/stubs/unix/context \
-	context/stubs/unix/ctxtools \
-	context/stubs/unix/makempy \
-	context/stubs/unix/mpstools \
-	context/stubs/unix/mptopdf \
-	context/stubs/unix/mtxtools \
-	context/stubs/unix/pdftools \
-	context/stubs/unix/pstopdf \
-	context/stubs/unix/rlxtools \
-	context/stubs/unix/runtools \
-	context/stubs/unix/texexec \
-	context/stubs/unix/texfont \
-	context/stubs/unix/textools \
-	context/stubs/unix/texutil \
-	context/stubs/unix/tmftools \
-	context/stubs/unix/xmltools \
 	dviasm/dviasm.py \
-	epspdf/epspdf \
-	epspdf/epspdftk \
 	epstopdf/epstopdf.pl \
-	fig4latex/fig4latex \
 	findhyph/findhyph \
 	fontools/afm2afm \
 	fontools/autoinst \
@@ -64,32 +38,19 @@ nobase_dist_script_SCRIPTS = \
 	fontools/pfm2kpx \
 	fontools/showglyphs \
 	fragmaster/fragmaster.pl \
-	glossaries/makeglossaries \
 	latex2man/latex2man \
 	latexmk/latexmk.pl \
 	listings-ext/listings-ext.sh \
 	mkjobtexmf/mkjobtexmf.pl \
-	mkgrkindex/mkgrkindex \
-	oberdiek/pdfatfi.pl \
-	pax/pdfannotextractor.pl \
 	pdfcrop/pdfcrop.pl \
 	perltex/perltex \
 	pkfix/pkfix.pl \
 	pkfix-helper/pkfix-helper \
-	ppower4/pdfthumb.tlu \
-	ppower4/ppower4.tlu \
-	pst-pdf/ps4pdf \
-	pst2pdf/pst2pdf.pl \
 	purifyeps/purifyeps \
-	splitindex/perl/splitindex.pl \
-	svn-multi/svn-multi.pl \
 	texcount/texcount.pl \
 	texdiff/texdiff \
 	texdirflatten/texdirflatten \
 	texloganalyser/texloganalyser \
-	thumbpdf/thumbpdf.pl \
-	ulqda/ulqda.pl \
-	vpe/vpe.pl
 
 ## The idea is to install the scripts themselves in texmf*/scripts, and
 ## have bin/arch/foo be a symlink to, say,
Index: texlive-core-20091107/texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in
===================================================================
--- texlive-core-20091107.orig/texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in
+++ texlive-core-20091107/texlive-20091107-source/texk/texlive/linked_scripts/Makefile.in
@@ -160,11 +160,9 @@ scriptxdir = ${prefix}/texmf/scripts
 nobase_dist_scriptx_SCRIPTS = \
 	a2ping/a2ping.pl \
 	getnonfreefonts/getnonfreefonts.pl \
-	simpdftex/simpdftex \
 	tetex/e2pall.pl \
 	tetex/texdoctk.pl \
 	texdoc/texdoc.tlu \
-	texlive/rungs.tlu \
 	texlive/tlmgr.pl
 
 scriptdir = ${prefix}/texmf-dist/scripts
@@ -172,30 +170,7 @@ nobase_dist_script_SCRIPTS = \
 	accfonts/mkt1font \
 	accfonts/vpl2ovp \
 	accfonts/vpl2vpl \
-	bengali/ebong.py \
-	cachepic/cachepic.tlu \
-	context/lua/luatools.lua \
-	context/lua/mtxrun.lua \
-	context/ruby/texmfstart.rb \
-	context/stubs/unix/context \
-	context/stubs/unix/ctxtools \
-	context/stubs/unix/makempy \
-	context/stubs/unix/mpstools \
-	context/stubs/unix/mptopdf \
-	context/stubs/unix/mtxtools \
-	context/stubs/unix/pdftools \
-	context/stubs/unix/pstopdf \
-	context/stubs/unix/rlxtools \
-	context/stubs/unix/runtools \
-	context/stubs/unix/texexec \
-	context/stubs/unix/texfont \
-	context/stubs/unix/textools \
-	context/stubs/unix/texutil \
-	context/stubs/unix/tmftools \
-	context/stubs/unix/xmltools \
 	dviasm/dviasm.py \
-	epspdf/epspdf \
-	epspdf/epspdftk \
 	epstopdf/epstopdf.pl \
 	fig4latex/fig4latex \
 	findhyph/findhyph \
@@ -207,32 +182,19 @@ nobase_dist_script_SCRIPTS = \
 	fontools/pfm2kpx \
 	fontools/showglyphs \
 	fragmaster/fragmaster.pl \
-	glossaries/makeglossaries \
 	latex2man/latex2man \
 	latexmk/latexmk.pl \
 	listings-ext/listings-ext.sh \
 	mkjobtexmf/mkjobtexmf.pl \
-	mkgrkindex/mkgrkindex \
-	oberdiek/pdfatfi.pl \
-	pax/pdfannotextractor.pl \
 	pdfcrop/pdfcrop.pl \
 	perltex/perltex \
 	pkfix/pkfix.pl \
 	pkfix-helper/pkfix-helper \
-	ppower4/pdfthumb.tlu \
-	ppower4/ppower4.tlu \
-	pst-pdf/ps4pdf \
-	pst2pdf/pst2pdf.pl \
 	purifyeps/purifyeps \
-	splitindex/perl/splitindex.pl \
-	svn-multi/svn-multi.pl \
 	texcount/texcount.pl \
 	texdiff/texdiff \
 	texdirflatten/texdirflatten \
 	texloganalyser/texloganalyser \
-	thumbpdf/thumbpdf.pl \
-	ulqda/ulqda.pl \
-	vpe/vpe.pl
 
 all: all-am
 
