diff -Nur texlive-core-20091107-old/texlive-20091107-source/texk/kpathsea/texmf.cnf texlive-core-20091107/texlive-20091107-source/texk/kpathsea/texmf.cnf
--- texlive-core-20091107-old/texlive-20091107-source/texk/kpathsea/texmf.cnf	2010-08-19 15:03:25.143117121 +0000
+++ texlive-core-20091107/texlive-20091107-source/texk/kpathsea/texmf.cnf	2010-08-19 15:04:12.666366989 +0000
@@ -498,7 +498,7 @@
 % 
 % Unfortunately we found too many problems with shell_escape=p, so it
 % should be treated as though it were just as dangerous as shell_escape=t.
-shell_escape_commands = \
+shell_escape_commands = \\
 false
 
 % These programs are our first goal when we can enable it.
