Index: zlib-1.2.5/Makefile.in
===================================================================
--- zlib-1.2.5.orig/Makefile.in
+++ zlib-1.2.5/Makefile.in
@@ -176,7 +176,6 @@ install-libs: $(LIBS)
 	  rm -f $(SHAREDLIB) $(SHAREDLIBM); \
 	  ln -s $(SHAREDLIBV) $(SHAREDLIB); \
 	  ln -s $(SHAREDLIBV) $(SHAREDLIBM); \
-	  ($(LDCONFIG) || true)  >/dev/null 2>&1; \
 	fi
 	cp zlib.3 $(DESTDIR)$(man3dir)
 	chmod 644 $(DESTDIR)$(man3dir)/zlib.3
