--- zlib-1.2.1.old/configure	2004-08-28 08:19:49.000000000 -0400
+++ zlib-1.2.1/configure	2004-08-28 08:33:57.745864064 -0400
@@ -73,7 +73,11 @@
 
 if test "$gcc" -eq 1 && ($cc -c $cflags $test.c) 2>/dev/null; then
   CC="$cc"
-  SFLAGS=${CFLAGS-"-fPIC -O3"}
+  #SFLAGS=${CFLAGS-"-fPIC -O3"}
+  # the above is horribly wrong on a few archs where -fPIC should ALWAYS be
+  # used in the creation of shared libraries. without the following, the
+  # shared lib test will sometimes fail even when shared libs -can- be created.
+  SFLAGS="${CFLAGS-"-O3"} -fPIC"
   CFLAGS="$cflags"
   case `(uname -s || echo unknown) 2>/dev/null` in
   Linux | linux | GNU | GNU/*)
