--- configure.orig	2003-12-02 12:48:03.882675272 -0800
+++ configure	2003-12-02 13:00:18.728031272 -0800
@@ -76,7 +76,9 @@
   SFLAGS=${CFLAGS-"-fPIC -O3"}
   CFLAGS="$cflags"
   case `(uname -s || echo unknown) 2>/dev/null` in
-  Linux | linux | GNU | GNU/*) LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1"};;
+  Linux | linux | GNU | GNU/*)
+          LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1"}
+          MAPFILE=${MAPFILE-"-Wl,--version-script,mapfile"};;
   CYGWIN* | Cygwin* | cygwin* )
              EXE='.exe';;
   QNX*)  # This is for QNX6. I suppose that the QNX rule below is for QNX2,QNX4
@@ -123,6 +125,7 @@
              LDSHARED=${LDSHARED-"cc -dy -KPIC -G"};;
   SunOS\ 5*) SFLAGS=${CFLAGS-"-fast -xcg89 -KPIC -R."}
              CFLAGS=${CFLAGS-"-fast -xcg89"}
+             MAPFILE=${MAPFILE-"-M mapfile"}
              LDSHARED=${LDSHARED-"cc -G"};;
   SunOS\ 4*) SFLAGS=${CFLAGS-"-O2 -PIC"}
              CFLAGS=${CFLAGS-"-O2"}
@@ -169,6 +172,7 @@
     echo 'No shared library support; try without defining CC and CFLAGS'
     shared=0;
   fi
+  LDSHARED="${LDSHARED} ${MAPFILE}"
 fi
 if test $shared -eq 0; then
   LDSHARED="$CC"
--- /dev/null	2003-10-15 22:31:52.000000000 -0700
+++ mapfile	2003-12-02 12:59:22.275926800 -0800
@@ -0,0 +1,56 @@
+{
+	global:
+		zlibVersion ;
+		deflateInit_ ;
+		deflateInit2_ ;
+		deflate ;
+		deflateSetDictionary ;
+		deflateCopy ;
+		deflateReset ;
+		deflateParams ;
+		deflateEnd ;
+		deflateBound ;
+		deflatePrime ;
+		inflateInit_ ;
+		inflateInit2_ ;
+		inflate ;
+		inflateSetDictionary ;
+		inflateSync ;
+		inflateReset ;
+		inflateEnd ;
+		inflateCopy ;
+		inflateBackInit_ ;
+		inflateBack ;
+		inflateBackEnd ;
+		zlibCompileFlags ;
+		compressBound ;
+		compress ;
+		compress2 ;
+		uncompress ;
+		gzopen ;
+		gzdopen ;
+		gzsetparams ;
+		gzread ;
+		gzwrite ;
+		gzprintf ;
+		gzputs ;
+		gzgets ;
+		gzputc ;
+		gzgetc ;
+		gzungetc ;
+		gzflush ;
+		gzseek ;
+		gzrewind ;
+		gztell ;
+		gzeof ;
+		gzclose ;
+		gzerror ;
+		gzclearerr ;
+		adler32 ;
+		crc32 ;
+		zError ;
+		inflateSyncPoint ;
+		get_crc_table ;
+	local:
+		*;
+};
