Security Bugfixes (CAN-2004-0797, OpenPKG-SA-2004.038-zlib):

Index: infback.c
--- infback.c.orig	2003-08-12 01:48:06 +0200
+++ infback.c	2004-08-25 12:37:07 +0200
@@ -434,6 +434,9 @@
                 }
             }
 
+            if (state->mode == BAD)
+                break;
+
             /* build code tables */
             state->next = state->codes;
             state->lencode = (code const FAR *)(state->next);
Index: inflate.c
--- inflate.c.orig	2003-10-26 07:15:36 +0100
+++ inflate.c	2004-08-25 12:37:07 +0200
@@ -861,6 +861,9 @@
                 }
             }
 
+            if (state->mode == BAD)
+                break;
+
             /* build code tables */
             state->next = state->codes;
             state->lencode = (code const FAR *)(state->next);
