--- CMakeLists.txt.orig	2010-06-22 14:45:15.458448408 +0300
+++ CMakeLists.txt	2010-06-22 14:45:41.759198911 +0300
@@ -324,7 +324,7 @@
         CACHE STRING "The folder where CGAL CMake modules will be installed, relative to CMAKE_INSTALL_PREFIX" 
       )
       
-  set ( CGAL_INSTALL_DOC_DIR "share/doc/${CGAL_VERSION_DIR}"  
+  set ( CGAL_INSTALL_DOC_DIR "share/doc/CGAL"
         CACHE STRING "The folder where CGAL documentation and license files will be installed, relative to CMAKE_INSTALL_PREFIX" 
       )
 
