Index: xpcom/io/nsAppFileLocationProvider.cpp
===================================================================
--- xpcom/io/nsAppFileLocationProvider.cpp.orig
+++ xpcom/io/nsAppFileLocationProvider.cpp
@@ -107,6 +107,10 @@
 #define PLUGINS_DIR_NAME            NS_LITERAL_CSTRING("plugins")
 #define SEARCH_DIR_NAME             NS_LITERAL_CSTRING("searchplugins")
 
+#ifndef PARDUS_NS_PLUGINS_DIR
+#define PARDUS_NS_PLUGINS_DIR "/usr/lib/nsbrowser/plugins/"
+#endif
+
 //*****************************************************************************
 // nsAppFileLocationProvider::Constructor/Destructor
 //*****************************************************************************
@@ -258,6 +262,10 @@ nsAppFileLocationProvider::GetFile(const
         rv = NS_NewNativeLocalFile(nsDependentCString(sysLPlgDir),
                                    PR_FALSE, getter_AddRefs(localFile));
     }
+    else if (nsCRT::strcmp(prop, PARDUS_NS_PLUGINS_DIR) == 0)
+    {
+        rv = NS_NewNativeLocalFile(nsDependentCString(PARDUS_NS_PLUGINS_DIR), PR_TRUE, getter_AddRefs(localFile));
+    }
 #endif
 #endif
     else if (nsCRT::strcmp(prop, NS_APP_SEARCH_DIR) == 0)
@@ -593,7 +601,7 @@ nsAppFileLocationProvider::GetFiles(cons
         *_retval = new nsAppDirectoryEnumerator(this, keys);
 #else
 #ifdef XP_UNIX
-        static const char* keys[] = { nsnull, NS_USER_PLUGINS_DIR, NS_APP_PLUGINS_DIR, NS_SYSTEM_PLUGINS_DIR, nsnull };
+        static const char* keys[] = { nsnull, PARDUS_NS_PLUGINS_DIR, NS_USER_PLUGINS_DIR, NS_APP_PLUGINS_DIR, NS_SYSTEM_PLUGINS_DIR, nsnull };
 #else
         static const char* keys[] = { nsnull, NS_USER_PLUGINS_DIR, NS_APP_PLUGINS_DIR, nsnull };
 #endif
