--- xscreensaver-5.11/hacks/fadeplot.c.undef	2008-08-03 05:11:04.000000000 +0900
+++ xscreensaver-5.11/hacks/fadeplot.c	2010-07-08 18:16:02.000000000 +0900
@@ -202,11 +202,11 @@
 	if ((fp->temps % (fp->angles / 2)) == 0) {
 		fp->temps = fp->temps % fp->angles * 5;
 		if ((fp->temps % (fp->angles)) == 0)
-			fp->speed.y = (fp->speed.y++) % 30 + 1;
+			fp->speed.y = (fp->speed.y + 1) % 30 + 1;
 		if ((fp->temps % (fp->angles * 2)) == 0)
 			fp->speed.x = (fp->speed.x) % 20;
 		if ((fp->temps % (fp->angles * 3)) == 0)
-			fp->step.y = (fp->step.y++) % 2 + 1;
+			fp->step.y = (fp->step.y + 1) % 2 + 1;
 
 		MI_CLEARWINDOW(mi);
 	}
