--- gcc-4.5.1.orig/gcc/config/avr/libgcc.S	2009-05-23 17:16:07 +1000
+++ gcc-4.5.1.orig/gcc/config/avr/libgcc.S	2010-08-12 09:38:05 +1000
@@ -802,7 +802,9 @@ 
 	mov_h	r31, r29
 	mov_l	r30, r28
 	out     __RAMPZ__, r20
+	push	r20
 	XCALL	__tablejump_elpm__
+	pop	r20
 .L__do_global_ctors_start:
 	cpi	r28, lo8(__ctors_start)
 	cpc	r29, r17
@@ -843,7 +845,9 @@ 
 	mov_h	r31, r29
 	mov_l	r30, r28
 	out     __RAMPZ__, r20
+	push	r20
 	XCALL	__tablejump_elpm__
+	pop	r20
 .L__do_global_dtors_start:
 	cpi	r28, lo8(__dtors_end)
 	cpc	r29, r17
