Index: bin/mudur.py
===================================================================
--- bin/mudur.py	(revision 31612)
+++ bin/mudur.py	(revision 31613)
@@ -56,12 +56,12 @@
 
 def load_file(path, ignore_comments=False):
     """Reads the contents of a file and returns it."""
+    data = ""
     try:
         with open(path, "r") as _file:
             data = _file.read()
     except IOError:
         return ""
-    data = ""
     if ignore_comments:
         data = filter(lambda x: not (x.startswith("#") or x == ""), data)
     return data
