Index: bin/mudur.py
===================================================================
--- bin/mudur.py	(revision 31592)
+++ bin/mudur.py	(revision 31593)
@@ -56,9 +56,12 @@
 
 def load_file(path, ignore_comments=False):
     """Reads the contents of a file and returns it."""
+    try:
+        with open(path, "r") as _file:
+            data = _file.read()
+    except IOError:
+        return ""
     data = ""
-    with open(path, "r") as _file:
-        data = _file.read()
     if ignore_comments:
         data = filter(lambda x: not (x.startswith("#") or x == ""), data)
     return data
