diff -Naur fontypython-0.4.2.3.orig//fontypython.desktop fontypython-0.4.2.3//fontypython.desktop
--- fontypython-0.4.2.3.orig//fontypython.desktop	2009-09-28 19:50:37.000000000 +0300
+++ fontypython-0.4.2.3//fontypython.desktop	2011-01-24 21:14:34.305000001 +0200
@@ -1,8 +1,11 @@
 [Desktop Entry]
 Name=Fonty Python Font Manager
+Name[tr]=fontypython
+GenericName=Font Manager
+GenericName[tr]=Yazıtipi Görüntüleyici
 Comment=View and temporarily install all kinds of fonts.
 Comment[fr]=Fonty Python est aussi approprié pour visionner vos fichiers TTF où qu'ils soient sur votre ordinateur.
-Comment[it]=
+Comment[tr]=Yazıtiplerini görüntülemenizi sağlayan bir program.
 
 Type=Application
 Categories=Graphics;
