From be78cc4a136f8ec63dc6d7efd8356625c639a877 Mon Sep 17 00:00:00 2001
From: Alon Levy <alevy@redhat.com>
Date: Tue, 3 Aug 2010 11:37:51 +0300
Subject: [PATCH 30/39] qxl: add 800x480 resolution to qxl_modes (n900 native)

---
 hw/qxl.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/hw/qxl.c b/hw/qxl.c
index 7bd4467..86c0e03 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -64,6 +64,7 @@

 static QXLMode qxl_modes[] = {
     QXL_MODE_EX(640, 480),
+    QXL_MODE_EX(800, 480),
     QXL_MODE_EX(800, 600),
     QXL_MODE_EX(832, 624),
     QXL_MODE_EX(1024, 768),
-- 
1.7.2.3

