diff -up bash-4.0/tests/read.tests.tty_tests bash-4.0/tests/read.tests
--- bash-4.0/tests/read.tests.tty_tests	2008-09-06 19:09:11.000000000 +0200
+++ bash-4.0/tests/read.tests	2009-01-21 16:21:48.000000000 +0100
@@ -82,9 +82,6 @@ echo " foo" | { IFS=$':' ; read line; re
 # test read -d delim behavior
 ${THIS_SH} ./read1.sub
 
-# test read -t timeout behavior
-${THIS_SH} ./read2.sub
-
 # test read -n nchars behavior
 ${THIS_SH} ./read3.sub
 
diff -up bash-4.0/tests/read.right.tty_tests bash-4.0/tests/read.right
--- bash-4.0/tests/read.right.tty_tests	2008-09-06 19:10:14.000000000 +0200
+++ bash-4.0/tests/read.right	2009-01-21 16:30:03.000000000 +0100
@@ -33,14 +33,6 @@ a = abcdefg
 a = xyz
 a = -xyz 123-
 a = abc
-timeout 1: ok
-
-timeout 2: ok
-
-./read2.sub: line 23: read: -3: invalid timeout specification
-1
-
-abcde
 ./read3.sub: line 4: read: -1: invalid number
 abc
 ab
