From 2d05df9a985801a6e7efcfe576a7faa33d49a866 Mon Sep 17 00:00:00 2001
From: Ben Skeggs <bskeggs@redhat.com>
Date: Mon, 8 Feb 2010 12:51:47 +1000
Subject: [PATCH 2/2] set canDoBGNoneRoot

---
 src/nv_driver.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/nv_driver.c b/src/nv_driver.c
index 4b00e3d..ef9e902 100644
--- a/src/nv_driver.c
+++ b/src/nv_driver.c
@@ -652,6 +652,7 @@ NVPreInit(ScrnInfoPtr pScrn, int flags)
 
 	/* Set pScrn->monitor */
 	pScrn->monitor = pScrn->confScreen->monitor;
+	pScrn->canDoBGNoneRoot = 1;
 
 	/*
 	 * The first thing we should figure out is the depth, bpp, etc.
-- 
1.7.0.1

