--- desktop-data/bluefish.desktop.in.old	2010-02-21 23:49:27.391314186 +0200
+++ desktop-data/bluefish.desktop.in	2010-02-21 23:54:45.447320495 +0200
@@ -1,7 +1,15 @@
 [Desktop Entry]
 Version=1.0
 _Name=Bluefish Editor
+Name[de]=Bluefish Editor
+Name[fr]=Éditeur Bluefish
+Name[tr]=Bluefish Editörü
+GenericName=Create web pages
+GenericName[tr]=Web sayfası yaratmaya yarar
 _Comment=Web Development Editor
+Comment[de]=Editor für Web-Entwickler
+Comment[fr]=Éditeur pour développeurs web
+Comment[tr]=Web tasarımcıları ve programcıları için güçlü bir editör
 Exec=bluefish %F
 Icon=bluefish
 Terminal=false
