Index: xserver-xorg-input-synaptics/src/synaptics.c
===================================================================
--- xserver-xorg-input-synaptics.orig/src/synaptics.c
+++ xserver-xorg-input-synaptics/src/synaptics.c
@@ -691,6 +691,9 @@
 
     DBG(3, ErrorF("Synaptics DeviceOn called\n"));
 
+    if (xf86Screens[0]->vtSema == FALSE)
+	    return !Success;
+
     SetDeviceAndProtocol(local);
     local->fd = xf86OpenSerial(local->options);
     if (local->fd == -1) {
