Index: xfree86-driver-synaptics-0.99.3/src/synaptics.c
===================================================================
--- xfree86-driver-synaptics-0.99.3.orig/src/synaptics.c	2008-12-15 00:33:15.000000000 +0100
+++ xfree86-driver-synaptics-0.99.3/src/synaptics.c	2009-01-24 15:48:29.000000000 +0100
@@ -464,8 +464,8 @@
     pars->guestmouse_off = xf86SetBoolOption(opts, "GuestMouseOff", FALSE);
     pars->locked_drags = xf86SetBoolOption(opts, "LockedDrags", FALSE);
     pars->locked_drag_time = xf86SetIntOption(opts, "LockedDragTimeout", 5000);
-    pars->tap_action[RT_TAP] = xf86SetIntOption(opts, "RTCornerButton", 0);
-    pars->tap_action[RB_TAP] = xf86SetIntOption(opts, "RBCornerButton", 0);
+    pars->tap_action[RT_TAP] = xf86SetIntOption(opts, "RTCornerButton", 2);
+    pars->tap_action[RB_TAP] = xf86SetIntOption(opts, "RBCornerButton", 3);
     pars->tap_action[LT_TAP] = xf86SetIntOption(opts, "LTCornerButton", 0);
     pars->tap_action[LB_TAP] = xf86SetIntOption(opts, "LBCornerButton", 0);
     pars->tap_action[F1_TAP] = xf86SetIntOption(opts, "TapButton1",     tapButton1);
