--- doc/Makefile.am~	2009-05-11 18:59:19.964310487 +0300
+++ doc/Makefile.am	2009-05-11 18:59:19.969312970 +0300
@@ -73,11 +73,11 @@
         $(LIBGUPNP_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
+# include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += version.xml.in
+EXTRA_DIST = version.xml.in
 
 # Files not to distribute
 # for --rebuild-types in $(SCAN_OPTIONS), e.g. $(DOC_MODULE).types
--- configure.ac~	2009-05-11 19:03:04.037319032 +0300
+++ configure.ac	2009-05-11 19:03:04.038315517 +0300
@@ -16,8 +16,6 @@
 GLIB_GENMARSHAL=`pkg-config --variable=glib_genmarshal glib-2.0`
 AC_SUBST(GLIB_GENMARSHAL)
 
-GTK_DOC_CHECK([1.0])
-
 PKG_CHECK_MODULES(GTest, glib-2.0 >= 2.16, [gtest=yes], [gtest=no])
 
 AM_CONDITIONAL(HAVE_GTEST, [test x$gtest = xyes])
