diff -Nur irrlicht-1.7.2-old//source/Irrlicht/CImageLoaderPNG.cpp irrlicht-1.7.2/source/Irrlicht/CImageLoaderPNG.cpp
--- irrlicht-1.7.2-old//source/Irrlicht/CImageLoaderPNG.cpp	2011-03-08 16:49:00.708001316 +0200
+++ irrlicht-1.7.2/source/Irrlicht/CImageLoaderPNG.cpp	2011-03-08 16:49:33.019001316 +0200
@@ -162,7 +162,7 @@
 	if (BitDepth < 8)
 	{
 		if (ColorType==PNG_COLOR_TYPE_GRAY || ColorType==PNG_COLOR_TYPE_GRAY_ALPHA)
-			png_set_gray_1_2_4_to_8(png_ptr);
+			png_set_expand_gray_1_2_4_to_8(png_ptr);
 		else
 			png_set_packing(png_ptr);
 	}
