Index: work/xcursorgen-1.0.3/configure.ac
===================================================================
--- work.orig/xcursorgen-1.0.3/configure.ac
+++ work/xcursorgen-1.0.3/configure.ac
@@ -18,7 +18,7 @@ AC_PROG_INSTALL
 XORG_DEFAULT_OPTIONS
 
 # Checks for pkg-config packages
-PKG_CHECK_MODULES(XCURSORGEN, x11 xcursor libpng12)
+PKG_CHECK_MODULES(XCURSORGEN, x11 xcursor libpng)
 AC_SUBST(XCURSORGEN_CFLAGS)
 AC_SUBST(XCURSORGEN_LIBS)
 
