Index: work/xsm-1.0.1_20081204/configure.ac
===================================================================
--- work.orig/xsm-1.0.1_20081204/configure.ac
+++ work/xsm-1.0.1_20081204/configure.ac
@@ -31,19 +31,6 @@ AM_CONFIG_HEADER(config.h)
 AC_PROG_CC
 AC_PROG_INSTALL
 
-if [[ -z $RSH ]] ; then
-   AC_PATH_PROG(RSH,rsh)
-fi
-if [[ -z $RSH ]] ; then
-   AC_PATH_PROG(RSH,rcmd)
-fi
-if [[ -z $RSH ]] ; then
-   AC_PATH_PROG(RSH,remsh)
-fi
-if [[ -z $RSH ]] ; then
-   AC_MSG_ERROR(No rsh like program found)
-fi
-
 #if !defined(RshCmd)
 #if SystemV
 #ifdef HPArchitecture
