--- libsofia-sip-ua-glib/Makefile.am~	2008-11-27 17:09:17.000000000 -0600
+++ libsofia-sip-ua-glib/Makefile.am	2008-12-09 10:04:07.000000000 -0600
@@ -18,7 +18,7 @@
 endif
 
 libsofia_sip_ua_glib_la_SOURCES =
-libsofia_sip_ua_glib_la_LIBADD = $(GLIB_TARGETS) $(GLIB_LIBS)
+libsofia_sip_ua_glib_la_LIBADD = $(GLIB_TARGETS) $(GLIB_LIBS) ../libsofia-sip-ua/libsofia-sip-ua.la
 
 # set the libtool version info version:revision:age for libsofia-sip-ua-glib
 # - soname to 'libsofia-sip-ua-glib.so.(CUR-AGE)'
