Index: Makefile.am
===================================================================
RCS file: /cvsroot/suspend/suspend/Makefile.am,v
retrieving revision 1.6
diff -u -p -r1.6 Makefile.am
--- Makefile.am	16 Apr 2008 15:27:36 -0000	1.6
+++ Makefile.am	16 Apr 2008 16:43:25 -0000
@@ -176,3 +176,9 @@ install-resume-new-initrd:      resume
 
 install-resume-on-initrd:       resume
 	BOOT_DIR=$(DESTDIR)$(bootdir) ./scripts/install-resume.sh
+
+whitelist-test:
+	gcc -shared -o /dev/null -Werror -W -Wall whitelist.c
+
+all-local:
+	make whitelist-test
