--- src/kaptan/screens/scrSummary.py	2010/12/28 14:41:30	34485
+++ src/kaptan/screens/scrSummary.py	2010/12/29 14:05:49	34523
@@ -155,6 +155,17 @@
                         if str(launcher).find('launcher') >= 0:
                             subg2.writeEntry('plugin', self.menuSettings["selectedMenu"] )
 
+
+        def removeFolderViewWidget():
+            config = KConfig("plasma-desktop-appletsrc")
+            group = config.group("Containments")
+            for each in list(group.groupList()):
+                subgroup = group.group(each)
+                plugin = subgroup.readEntry('plugin')
+                if plugin == 'folderview' and subgroup.hasGroup("Applets") and not subgroup.hasKey("wallpaper"):
+                    subgroup.deleteGroup()
+                    dir(each)
+
         # Desktop Type
         if self.styleSettings["hasChangedDesktopType"]:
             hasChanged = True
@@ -169,6 +180,17 @@
                     if int(subcomponent2) == 0:
                         subgroup.writeEntry('plugin', self.styleSettings["desktopType"])
 
+            # Remove folder widget - normally this would be done over dbus but thanks to improper naming of the plasma interface
+            # this is not possible
+            # ValueError: Invalid interface or error name 'org.kde.plasma-desktop': contains invalid character '-'
+            #
+            # Related Bug:
+            # Bug 240358 - Invalid D-BUS interface name 'org.kde.plasma-desktop.PlasmaApp' found while parsing introspection
+            # https://bugs.kde.org/show_bug.cgi?id=240358
+
+            if self.styleSettings["desktopType"] == "folderview":
+                removeFolderViewWidget()
+
             config.sync()
 
         # Number of Desktops
