--- a/networkmanager/networkmanager.py	2010-07-13 14:52:40.395432703 +0300
+++ b/networkmanager/networkmanager.py	2010-07-13 14:50:23.299431832 +0300
@@ -406,39 +406,39 @@
 
     @property
     def flags(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "Flags")
 
     @property
     def wpaflags(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "WpaFlags")
 
     @property
     def rsnflags(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "RsnFlags")
 
     @property
     def ssid(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "Ssid")
 
     @property
     def frequency(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "Frequency")
 
     @property
     def hwaddress(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "HwAddress")
 
     @property
     def mode(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "Mode")
 
     @property
     def maxbitrate(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "MaxBitrate")
 
     @property
     def strength(self):
-        pass
+        return self.proxy.Get("org.freedesktop.NetworkManager.AccessPoint", "Strength")
 
 
 class NetworkManager(object):
