Index: networkmanager/networkmanager.py
===================================================================
--- networkmanager.orig/networkmanager.py
+++ networkmanager/networkmanager.py
@@ -731,7 +731,10 @@ def Settings(settings):
         if settings.id is None:
             settings.id = 'CDMA Connection'
     else:
-        raise UnsupportedConnectionType("Unknown connection type: '%s'" % conn_type)
+        #raise UnsupportedConnectionType("Unknown connection type: '%s'" % conn_type)
+        settings = BaseSettings(settings)
+        if settings.id is None:
+            settings.id = "Unknown Connection"
 
     return settings
 
