diff --git a/exec.c b/exec.c
index 4bae1d9..586f53f 100644
--- a/exec.c
+++ b/exec.c
@@ -2546,6 +2546,14 @@ ram_addr_t qemu_ram_alloc(ram_addr_t size)
 #else
         new_block->host = qemu_vmalloc(size);
 #endif
+#ifndef MADV_MERGEABLE
+/* This is a hack to enable functionality even when compiled with old
+ * kernel headers.  On real old kernel it will have no effect (the
+ * syscall below will fail with ENOSYS, but on more recent headers
+ * it will work.  This is in order to keep -stable with more recent
+ * kernel happy, mostly for -bpo. */
+#define MADV_MERGEABLE 12
+#endif
 #ifdef MADV_MERGEABLE
         madvise(new_block->host, size, MADV_MERGEABLE);
 #endif
