--- setup.py.link	2011-05-05 10:12:29.000000000 +0200
+++ setup.py	2011-05-05 10:13:38.000000000 +0200
@@ -263,7 +263,7 @@
             # _fastmath (uses GNU mp library)
             Extension("Crypto.PublicKey._fastmath",
                       include_dirs=['src/'],
-                      libraries=['gmp'],
+                      libraries=['gmp','m'],
                       sources=["src/_fastmath.c"]),
 
             # Hash functions
