--- src/RIPEMD.c.agriffis	2005-06-13 20:14:57.000000000 -0400
+++ src/RIPEMD.c	2006-02-02 08:59:43.000000000 -0500
@@ -99,15 +99,9 @@
       (c) = ROL((c), 10);\
    }
 
-typedef    unsigned char        byte;   /* unsigned 8-bit integer */
-#ifdef __alpha__
-typedef    unsigned int        word;   /* unsigned 32-bit integer */
-#elif defined(__amd64__)
-typedef    uint32_t            word;   /* unsigned 32-bit integer */
-#else
-typedef    unsigned long        word;   /* unsigned 32-bit integer */
-#endif
-typedef unsigned char   BYTE;
+typedef    unsigned char        byte;
+typedef    uint32_t             word;
+
 #define RMD_DATASIZE    64
 #define RMD_DIGESTSIZE  20
 #define RMDsize 160
@@ -137,7 +131,7 @@
 {
 	word tmp;
 	int dataCount, i;
-	BYTE *p;
+	byte *p;
 
 	/* Update bitcount */
 	tmp = shsInfo->countLo;
@@ -151,7 +145,7 @@
 	/* Handle any leading odd-sized chunks */
 	if(dataCount)
         {
-		p = ( BYTE * ) shsInfo->data + dataCount;
+		p = ( byte * ) shsInfo->data + dataCount;
 
 		dataCount = RMD_DATASIZE - dataCount;
 		if( count < dataCount )
