=== modified file 'test/test_chaffing.py'
--- test/test_chaffing.py	
+++ test/test_chaffing.py	
@@ -4,7 +4,7 @@
 
 __revision__ = "$Id: pycrypto-2.0.1-sancho-package-rename.patch,v 1.1 2006/02/13 21:26:38 marienz Exp $"
 
-from sancho.unittest import TestScenario, parse_args, run_scenarios
+from sancho0.unittest import TestScenario, parse_args, run_scenarios
 from Crypto.Protocol import Chaffing
 
 tested_modules = [ "Crypto.Protocol.Chaffing" ]

=== modified file 'test/test_hashes.py'
--- test/test_hashes.py	
+++ test/test_hashes.py	
@@ -5,7 +5,7 @@
 __revision__ = "$Id: pycrypto-2.0.1-sancho-package-rename.patch,v 1.1 2006/02/13 21:26:38 marienz Exp $"
 
 import time, string, binascii
-from sancho.unittest import TestScenario, parse_args, run_scenarios
+from sancho0.unittest import TestScenario, parse_args, run_scenarios
 
 from Crypto.Hash import *
 import testdata

=== modified file 'test/test_number.py'
--- test/test_number.py	
+++ test/test_number.py	
@@ -4,7 +4,7 @@
 
 __revision__ = "$Id: pycrypto-2.0.1-sancho-package-rename.patch,v 1.1 2006/02/13 21:26:38 marienz Exp $"
 
-from sancho.unittest import TestScenario, parse_args, run_scenarios
+from sancho0.unittest import TestScenario, parse_args, run_scenarios
 from Crypto.Util import number
 
 tested_modules = [ "Crypto.Util.number" ]

=== modified file 'test/test_publickey.py'
--- test/test_publickey.py	
+++ test/test_publickey.py	
@@ -5,7 +5,7 @@
 __revision__ = "$Id: pycrypto-2.0.1-sancho-package-rename.patch,v 1.1 2006/02/13 21:26:38 marienz Exp $"
 
 import sys, cPickle
-from sancho.unittest import TestScenario, parse_args, run_scenarios
+from sancho0.unittest import TestScenario, parse_args, run_scenarios
 from Crypto.PublicKey import *
 from Crypto.Util.randpool import RandomPool
 from Crypto.Util import number

=== modified file 'test/test_randpool.py'
--- test/test_randpool.py	
+++ test/test_randpool.py	
@@ -4,7 +4,7 @@
 
 __revision__ = "$Id: pycrypto-2.0.1-sancho-package-rename.patch,v 1.1 2006/02/13 21:26:38 marienz Exp $"
 
-from sancho.unittest import TestScenario, parse_args, run_scenarios
+from sancho0.unittest import TestScenario, parse_args, run_scenarios
 from Crypto.Hash import SHA
 from Crypto.Util import randpool
 

=== modified file 'test/test_rfc1751.py'
--- test/test_rfc1751.py	
+++ test/test_rfc1751.py	
@@ -5,7 +5,7 @@
 __revision__ = "$Id: pycrypto-2.0.1-sancho-package-rename.patch,v 1.1 2006/02/13 21:26:38 marienz Exp $"
 
 import binascii
-from sancho.unittest import TestScenario, parse_args, run_scenarios
+from sancho0.unittest import TestScenario, parse_args, run_scenarios
 from Crypto.Util import RFC1751
 
 tested_modules = [ "Crypto.Util.RFC1751" ]

