--- rdesktop-1.5.0.orig/configure	2007-02-10 23:33:06.000000000 +0300
+++ rdesktop-1.5.0/configure	2007-02-10 15:55:51.000000000 +0300
@@ -1982,13 +1982,13 @@
   CFLAGS=$ac_save_CFLAGS
 elif test $ac_cv_prog_cc_g = yes; then
   if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
+cflags="$cflags ${CFLAGS}"
   else
     CFLAGS="-g"
   fi
 else
   if test "$GCC" = yes; then
-    CFLAGS="-O2"
+cflags="$cflags ${CFLAGS}"
   else
     CFLAGS=
   fi
