--- apps/dolphin/src/dolphinui.rc	2009/01/19 10:23:45	1.1
+++ apps/dolphin/src/dolphinui.rc	2009/01/19 10:24:34
@@ -117,6 +117,7 @@
         <text context="@title:menu">Main Toolbar</text>
         <Action name="go_back" />
         <Action name="go_forward" />
+        <Action name="go_up" />
         <Separator name="separator_1" />
         <Action name="icons" />
         <Action name="details" />
