diff -Nur screen-4.0.3.orig/config.h.in screen-4.0.3/config.h.in
--- screen-4.0.3.orig/config.h.in	2006-10-23 16:06:32.000000000 +0300
+++ screen-4.0.3/config.h.in	2008-03-19 13:58:21.000000000 +0200
@@ -563,7 +563,7 @@
  * If you are on a SYS V machine that restricts filename length to 14 
  * characters, you may need to enforce that by setting NAME_MAX to 14
  */
-#undef NAME_MAX		/* KEEP_UNDEF_HERE override system value */
+#undef NAME_MAX
 #undef NAME_MAX
 
 /*
