Index: kdetoys-4.5.4/amor/src/amor.desktop
===================================================================
--- kdetoys-4.5.4.orig/amor/src/amor.desktop
+++ kdetoys-4.5.4/amor/src/amor.desktop
@@ -130,4 +130,4 @@ GenericName[zh_CN]=屏幕生物
 GenericName[zh_TW]=畫面擷取程式
 Terminal=false
 X-DBUS-StartupType=Unique
-Categories=Qt;KDE;Game;Amusement;
+Categories=Qt;KDE;Amusement;
Index: kdetoys-4.5.4/kteatime/src/kteatime.desktop
===================================================================
--- kdetoys-4.5.4.orig/kteatime/src/kteatime.desktop
+++ kdetoys-4.5.4/kteatime/src/kteatime.desktop
@@ -128,4 +128,4 @@ Terminal=false
 X-DocPath=kteatime/index.html
 X-KDE-StartupNotify=true
 X-DBUS-StartupType=Multi
-Categories=Qt;KDE;Game;Amusement;
+Categories=Qt;KDE;Amusement;
