Index: kdetoys-4.5.5/kteatime/src/kteatime.notifyrc
===================================================================
--- kdetoys-4.5.5.orig/kteatime/src/kteatime.notifyrc
+++ kdetoys-4.5.5/kteatime/src/kteatime.notifyrc
@@ -169,7 +169,7 @@ Comment[x-test]=xxTea is readyxx
 Comment[zh_CN]=茶沏好了
 Comment[zh_TW]=茶煮好了
 Sound=KDE-Sys-App-Message.ogg
-Action=Sound
+Action=Sound|Popup
 
 [Event/reminder]
 Name=Tea is getting lonely
@@ -273,4 +273,4 @@ Comment[x-test]=xxTea is getting lonely.
 Comment[zh_CN]=茶都凉了
 Comment[zh_TW]=茶煮好了，怎麼沒人喝
 Sound=KDE-Sys-App-Special.ogg
-Action=Sound
+Action=Sound|Popup
Index: kdetoys-4.5.5/kteatime/src/settings.cpp
===================================================================
--- kdetoys-4.5.5.orig/kteatime/src/settings.cpp
+++ kdetoys-4.5.5/kteatime/src/settings.cpp
@@ -76,7 +76,7 @@ SettingsDialog::SettingsDialog(TopLevel
     x = qMin( qMax( 0, y ), desktop.screenGeometry().height() - height() );
     move( QPoint( x, y ) );
 
-    bool popup=group.readEntry( "UsePopup", true );
+    bool popup=group.readEntry( "UsePopup", false );
     bool autohide=group.readEntry( "PopupAutoHide", false );
     int autohidetime=group.readEntry( "PopupAutoHideTime", 30 );
     bool reminder=group.readEntry( "UseReminder", false );
