# HG changeset patch
# Parent bad58443f421fd09ef6366d359b8405ed50c3731
--- a/mozilla/toolkit/system/gnome/Makefile.in
+++ b/mozilla/toolkit/system/gnome/Makefile.in
@@ -77,20 +77,22 @@ endif
 EXTRA_DSO_LDOPTS += \
 		$(XPCOM_GLUE_LDOPTS) \
 		$(XPCOM_FROZEN_LDOPTS) \
 		$(NSPR_LIBS) \
 		$(MOZ_GCONF_LIBS) \
 		$(MOZ_GNOMEVFS_LIBS) \
 		$(GLIB_LIBS) \
 		$(MOZ_LIBNOTIFY_LIBS) \
+		$(MOZ_GTK2_LIBS) \
 		$(NULL)
 
 LOCAL_INCLUDES += -I$(topsrcdir)/toolkit/components/build/
 
 include $(topsrcdir)/config/rules.mk
 
 CXXFLAGS += \
 	$(MOZ_GCONF_CFLAGS) \
 	$(MOZ_GNOMEVFS_CFLAGS) \
 	$(GLIB_CFLAGS) \
 	$(MOZ_LIBNOTIFY_CFLAGS) \
+	$(MOZ_GTK2_CFLAGS) \
 	$(NULL)
