diff -up kdebase-4.6.2/konsole/src/Session.h.kde#267896 kdebase-4.6.2/konsole/src/Session.h
--- kdebase-4.6.2/konsole/src/Session.h.kde#267896	2011-04-01 20:35:03.000000000 +0200
+++ kdebase-4.6.2/konsole/src/Session.h	2011-04-06 17:44:09.000000000 +0200
@@ -376,7 +376,8 @@ public slots:
    * then the terminal connection (the pty) is closed and Konsole waits for the 
    * process to exit.
    */
-  Q_SCRIPTABLE void close();
+  //Q_SCRIPTABLE void close(); // This cause the menu issues bko 185466
+  void close();
 
   /**
    * Changes the session title or other customizable aspects of the terminal
diff -up kdebase-4.6.2/konsole/src/ViewContainer.cpp.kde#267896 kdebase-4.6.2/konsole/src/ViewContainer.cpp
--- kdebase-4.6.2/konsole/src/ViewContainer.cpp.kde#267896	2011-04-01 20:35:03.000000000 +0200
+++ kdebase-4.6.2/konsole/src/ViewContainer.cpp	2011-04-06 17:45:36.000000000 +0200
@@ -443,13 +443,11 @@ TabbedViewContainer::TabbedViewContainer
     _closeTabButton->setIcon(KIcon("tab-close"));
     _closeTabButton->adjustSize();
     _closeTabButton->setHidden(true);
-    _closeTabButton->setEnabled(false);
 
     connect( _tabBar , SIGNAL(currentChanged(int)) , this , SLOT(currentTabChanged(int)) );
     connect( _tabBar , SIGNAL(tabDoubleClicked(int)) , this , SLOT(tabDoubleClicked(int)) );
     connect( _tabBar , SIGNAL(newTabRequest()) , this , SIGNAL(newViewRequest()) );
     connect( _tabBar , SIGNAL(wheelDelta(int)) , this , SLOT(wheelScrolled(int)) );
-    connect( _tabBar , SIGNAL(tabCloseRequested(int)) , this , SLOT(closeTab(int)) );
     connect( _tabBar , SIGNAL(initiateDrag(int)) , this , SLOT(startTabDrag(int)) );
     connect( _tabBar, SIGNAL(contextMenu(int, const QPoint&)), this,
             SLOT(openTabContextMenu(int, const QPoint&)) );
@@ -495,11 +493,11 @@ TabbedViewContainer::TabbedViewContainer
     _contextPopupMenu->addAction(KIcon(),
         i18nc("@action:inmenu", "&Rename Tab..."), this,
         SLOT(tabContextMenuRenameTab()));
-/*
+
     _contextPopupMenu->addAction(KIcon("tab-close"),
         i18nc("@action:inmenu", "&Close Tab"), this,
         SLOT(tabContextMenuCloseTab()));
-*/
+
 }
 void TabbedViewContainer::setNewViewMenu(QMenu* menu)
 {
@@ -521,16 +519,10 @@ void TabbedViewContainer::closeCurrentTa
 {
     if (_stackWidget->currentIndex() != -1)
     {
-        closeTab(_stackWidget->currentIndex());
+        emit closeTab(this, _stackWidget->widget(_stackWidget->currentIndex()));
     }
 }
-void TabbedViewContainer::closeTab(int tab)
-{
-    Q_ASSERT(tab >= 0 && tab < _stackWidget->count());
-    
-    if (viewProperties(_stackWidget->widget(tab))->confirmClose())
-        removeView(_stackWidget->widget(tab));
-}
+
 void TabbedViewContainer::setTabBarVisible(bool visible)
 {
     _tabBar->setVisible(visible);
@@ -655,7 +647,8 @@ void TabbedViewContainer::openTabContext
 
 void TabbedViewContainer::tabContextMenuCloseTab()
 {
-    closeTab(_contextMenuTabIndex);
+    _tabBar->setCurrentIndex(_contextMenuTabIndex);// Required for this to work
+    emit closeTab(this, _stackWidget->widget(_contextMenuTabIndex));
 }
 
 void TabbedViewContainer::tabContextMenuDetachTab()
diff -up kdebase-4.6.2/konsole/src/ViewContainer.h.kde#267896 kdebase-4.6.2/konsole/src/ViewContainer.h
--- kdebase-4.6.2/konsole/src/ViewContainer.h.kde#267896	2011-02-28 13:50:53.000000000 +0100
+++ kdebase-4.6.2/konsole/src/ViewContainer.h	2011-04-06 17:44:09.000000000 +0200
@@ -427,7 +427,6 @@ private slots:
     void updateIcon(ViewProperties* item);
     void updateActivity(ViewProperties* item);
     void currentTabChanged(int index);
-    void closeTab(int index);
     void closeCurrentTab();
     void wheelScrolled(int delta);
    
@@ -440,6 +439,7 @@ private slots:
 
 signals:
     void detachTab(ViewContainer * self, QWidget * activeView);
+    void closeTab(ViewContainer * self, QWidget * activeView);
 
 private:
     void dynamicTabBarVisibility();
diff -up kdebase-4.6.2/konsole/src/ViewManager.cpp.kde#267896 kdebase-4.6.2/konsole/src/ViewManager.cpp
--- kdebase-4.6.2/konsole/src/ViewManager.cpp.kde#267896	2011-04-06 17:44:09.000000000 +0200
+++ kdebase-4.6.2/konsole/src/ViewManager.cpp	2011-04-06 17:44:09.000000000 +0200
@@ -639,6 +639,11 @@ ViewContainer* ViewManager::createContai
                     this,
                     SLOT(detachView(ViewContainer*, QWidget*))
                     );
+                connect(container,
+                    SIGNAL(closeTab(ViewContainer*, QWidget*)),
+                    this,
+                    SLOT(closeTabFromContainer(ViewContainer*, QWidget*)));
+
             }
             break;
         case NoNavigation:
@@ -1084,5 +1089,13 @@ void ViewManager::moveSessionRight()
     this->moveActiveViewRight();
 }
 
+void ViewManager::closeTabFromContainer(ViewContainer *container, QWidget *tab)
+{
+    SessionController *controller = dynamic_cast<SessionController*>(container->viewProperties(tab));
+    Q_ASSERT(controller);
+    if (controller && controller->confirmClose())
+        controller->session()->close();
+}
+
 #include "ViewManager.moc"
 
diff -up kdebase-4.6.2/konsole/src/ViewManager.h.kde#267896 kdebase-4.6.2/konsole/src/ViewManager.h
--- kdebase-4.6.2/konsole/src/ViewManager.h.kde#267896	2011-02-28 13:50:53.000000000 +0100
+++ kdebase-4.6.2/konsole/src/ViewManager.h	2011-04-06 17:44:09.000000000 +0200
@@ -313,6 +313,8 @@ private slots:
 
     void detachView(ViewContainer* container, QWidget* view);
 
+    void closeTabFromContainer(ViewContainer *container, QWidget *view);
+    
 private:
     void createView(Session* session, ViewContainer* container, int index);
     const ColorScheme* colorSchemeForProfile(const Profile::Ptr profile) const;
