Index: xorg-server/hw/xfree86/modes/xf86RandR12.c
===================================================================
--- xorg-server.orig/hw/xfree86/modes/xf86RandR12.c	2009-12-02 23:06:15.000000000 +0200
+++ xorg-server/hw/xfree86/modes/xf86RandR12.c	2009-12-02 23:06:30.000000000 +0200
@@ -940,7 +940,8 @@
     for (c = 0; c < config->num_crtc; c++) {
 	xf86CrtcPtr    crtc = config->crtc[c];
 
-	RRCrtcSetRotations (crtc->randr_crtc, rotations);
+        if (crtc != NULL)
+            RRCrtcSetRotations (crtc->randr_crtc, rotations);
     }
 #endif
     randrp->supported_rotations = rotations;
