Index: bluez-4.56/Makefile.am
===================================================================
--- bluez-4.56.orig/Makefile.am
+++ bluez-4.56/Makefile.am
@@ -90,7 +90,8 @@ endif
 if NETLINK
 plugin_LTLIBRARIES += plugins/netlink.la
 
-plugins_netlink_la_LIBADD = @NETLINK_LIBS@
+plugins_netlink_la_SOURCES = plugins/netlink.c src/log.h src/log.c
+plugins_netlink_la_LIBADD = @GLIB_LIBS@ @NETLINK_LIBS@
 plugins_netlink_la_LDFLAGS = -module -avoid-version -no-undefined
 plugins_netlink_la_CFLAGS = -fvisibility=hidden @DBUS_CFLAGS@ \
 					@GLIB_CFLAGS@ @NETLINK_CFLAGS@
