Index: bluez-4.48/src/bluetooth.conf
===================================================================
--- bluez-4.48.orig/src/bluetooth.conf
+++ bluez-4.48/src/bluetooth.conf
@@ -17,9 +17,9 @@
     <allow send_destination="org.bluez"/>
   </policy>
 
-  <!-- allow users of lp group (printing subsystem) to 
+  <!-- allow users of pnp group (printing subsystem) to 
        communicate with bluetoothd -->
-  <policy group="lp">
+  <policy group="pnp">
     <allow send_destination="org.bluez"/>
   </policy>
 
