From 2eac19af5c9a275a78fb98c9e8f232bf88f4a040 Mon Sep 17 00:00:00 2001
From: Pascal Terjan <pterjan@mandriva.com>
Date: Wed, 23 Dec 2009 15:07:41 +0100
Subject: [PATCH] Fix a oops on rpm -e openoffice.org-voikko

Signed-off-by: Pascal Terjan <pterjan@mandriva.com>

diff --git a/fs/unionfs/unlink.c b/fs/unionfs/unlink.c
index bd5a3b3..6b2f16f 100644
--- a/fs/unionfs/unlink.c
+++ b/fs/unionfs/unlink.c
@@ -144,8 +144,6 @@
 	if (!err) {
 		unionfs_postcopyup_release(dentry);
 		unionfs_postcopyup_setmnt(parent);
-		if (inode->i_nlink == 0) /* drop lower inodes */
-			iput_lowers_all(inode, false);
 		d_drop(dentry);
 		/*
 		 * if unlink/whiteout succeeded, parent dir mtime has
-- 
1.6.2.5
