---
 sysdeps/generic/framestate.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/sysdeps/generic/framestate.c
+++ b/sysdeps/generic/framestate.c
@@ -25,6 +25,10 @@
 #include <unwind-dw2.c>
 #undef __frame_state_for
 
+#ifndef LIBGCC_S_SO
+#define LIBGCC_S_SO "libgcc_s.so.1"
+#endif
+
 typedef struct frame_state * (*framesf)(void *pc, struct frame_state *);
 struct frame_state *__frame_state_for (void *pc,
 				       struct frame_state *frame_state);
@@ -36,7 +40,7 @@
 
   if (frame_state_for == NULL)
     {
-      void *handle = __libc_dlopen ("libgcc_s.so.1");
+      void *handle = __libc_dlopen (LIBGCC_S_SO);
 
       if (handle == NULL
 	  || (frame_state_for
