diff -Nur libv4l-0.6.3-old/libv4lconvert/control/libv4lcontrol.c libv4l-0.6.3/libv4lconvert/control/libv4lcontrol.c
--- libv4l-0.6.3-old/libv4lconvert/control/libv4lcontrol.c	2009-12-03 19:56:20.121419537 +0200
+++ libv4l-0.6.3/libv4lconvert/control/libv4lcontrol.c	2009-12-03 20:07:14.553398740 +0200
@@ -187,6 +187,8 @@
     V4LCONTROL_HFLIPPED|V4LCONTROL_VFLIPPED },
   { 0x090c, 0xe370, 0, "ASUSTeK Computer Inc.        ", "U6S       ",
     V4LCONTROL_HFLIPPED|V4LCONTROL_VFLIPPED },
+  { 0x0c45, 0x624f, 0, "NEC COMPUTERS SAS            ", "MS-1642X  ",
+    V4LCONTROL_HFLIPPED|V4LCONTROL_VFLIPPED },
   { 0x174f, 0x5a35, 0, "ASUSTeK Computer Inc.        ", "F3Ka      ",
     V4LCONTROL_HFLIPPED|V4LCONTROL_VFLIPPED },
   { 0x174f, 0x5a35, 0, "ASUSTeK Computer Inc.        ", "F3Ke      ",
