diff -Nur v4l-utils-0.8.1-old//utils/keytable/Makefile v4l-utils-0.8.1/utils/keytable/Makefile
--- v4l-utils-0.8.1-old//utils/keytable/Makefile	2010-12-24 16:17:23.895000097 +0200
+++ v4l-utils-0.8.1/utils/keytable/Makefile	2010-12-24 16:17:51.617000005 +0200
@@ -53,8 +53,9 @@
 install: $(TARGETS)
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
 	install -m 755 $(TARGETS) $(DESTDIR)$(PREFIX)/bin
-	install -m 644 rc_maps.cfg.example /etc
-	install -m 755 -d /etc/rc_keymaps
-	install -m 644 rc_keymaps/* /etc/rc_keymaps
+	install -m 755 -d $(DESTDIR)/etc
+	install -m 644 rc_maps.cfg.example $(DESTDIR)/etc
+	install -m 755 -d $(DESTDIR)/etc/rc_keymaps
+	install -m 644 rc_keymaps/* $(DESTDIR)/etc/rc_keymaps
 
 include ../../Make.rules
