diff -Nuar redland-1.0.10.orig/configure.ac redland-1.0.10/configure.ac
--- redland-1.0.10.orig/configure.ac	2010-03-07 01:30:22.595657894 +0200
+++ redland-1.0.10/configure.ac	2010-03-07 01:48:47.156408670 +0200
@@ -1246,7 +1246,6 @@
 
 dnl Shave "make autotools output sane" - disabled by default
 dnl http://git.lespiau.name/cgit/shave/  git clone git://git.lespiau.name/shave
-SHAVE_INIT([build], [disable])
 
 dnl when automake 1.11 is more widespread can use this:
 dnl m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])],)
