diff -up redland-1.0.10/src/Makefile.am.no_undefined redland-1.0.10/src/Makefile.am
--- redland-1.0.10/src/Makefile.am.no_undefined	2009-11-23 00:38:32.000000000 -0600
+++ redland-1.0.10/src/Makefile.am	2010-01-04 12:51:17.245649696 -0600
@@ -152,7 +152,7 @@ if STORAGE_VIRTUOSO
 plugin_LTLIBRARIES += librdf_storage_virtuoso.la 
 librdf_storage_virtuoso_la_SOURCES = rdf_storage_virtuoso.c rdf_storage_virtuoso_internal.h
 librdf_storage_virtuoso_la_CPPFLAGS = @ODBC_CFLAGS@
-librdf_storage_virtuoso_la_LIBADD = @ODBC_LIBS@
+librdf_storage_virtuoso_la_LIBADD = @ODBC_LIBS@ librdf.la
 librdf_storage_virtuoso_la_LDFLAGS = -module -avoid-version
 endif
 
diff -up redland-1.0.10/src/Makefile.in.no_undefined redland-1.0.10/src/Makefile.in
--- redland-1.0.10/src/Makefile.in.no_undefined	2009-12-14 00:53:52.000000000 -0600
+++ redland-1.0.10/src/Makefile.in	2010-01-04 12:56:01.389896396 -0600
@@ -214,7 +214,7 @@ librdf_storage_tstore_la_LINK = $(LIBTOO
 	$(librdf_storage_tstore_la_LDFLAGS) $(LDFLAGS) -o $@
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@am_librdf_storage_tstore_la_rpath =  \
 @MODULAR_LIBRDF_TRUE@@STORAGE_TSTORE_TRUE@	-rpath $(plugindir)
-librdf_storage_virtuoso_la_DEPENDENCIES =
+librdf_storage_virtuoso_la_DEPENDENCIES = librdf.la
 am__librdf_storage_virtuoso_la_SOURCES_DIST = rdf_storage_virtuoso.c \
 	rdf_storage_virtuoso_internal.h
 @MODULAR_LIBRDF_TRUE@@STORAGE_VIRTUOSO_TRUE@am_librdf_storage_virtuoso_la_OBJECTS = librdf_storage_virtuoso_la-rdf_storage_virtuoso.lo
@@ -570,7 +570,7 @@ plugindir = $(libdir)/redland
 @MODULAR_LIBRDF_TRUE@@STORAGE_POSTGRESQL_TRUE@librdf_storage_postgresql_la_LDFLAGS = -module -avoid-version
 @MODULAR_LIBRDF_TRUE@@STORAGE_VIRTUOSO_TRUE@librdf_storage_virtuoso_la_SOURCES = rdf_storage_virtuoso.c rdf_storage_virtuoso_internal.h
 @MODULAR_LIBRDF_TRUE@@STORAGE_VIRTUOSO_TRUE@librdf_storage_virtuoso_la_CPPFLAGS = @ODBC_CFLAGS@
-@MODULAR_LIBRDF_TRUE@@STORAGE_VIRTUOSO_TRUE@librdf_storage_virtuoso_la_LIBADD = @ODBC_LIBS@
+@MODULAR_LIBRDF_TRUE@@STORAGE_VIRTUOSO_TRUE@librdf_storage_virtuoso_la_LIBADD = @ODBC_LIBS@ librdf.la
 @MODULAR_LIBRDF_TRUE@@STORAGE_VIRTUOSO_TRUE@librdf_storage_virtuoso_la_LDFLAGS = -module -avoid-version
 librdf_la_LIBADD = \
 @LTLIBOBJS@ @DIGEST_OBJS@ @HASH_OBJS@ \
