diff -up redland-1.0.10/redland-config.in.libs redland-1.0.10/redland-config.in
diff -up redland-1.0.10/redland.pc.in.libs redland-1.0.10/redland.pc.in
--- redland-1.0.10/redland.pc.in.libs	2009-04-12 20:01:30.000000000 -0500
+++ redland-1.0.10/redland.pc.in	2010-01-03 17:55:23.791897828 -0600
@@ -9,7 +9,7 @@ swig=${prefix}/share/redland/Redland.i
 
 Name: Redland RDF API Library
 Description: Redland RDF API and triple store library
-Requires: raptor >= @RAPTOR_MIN_VERSION@, rasqal >= @RASQAL_MIN_VERSION@, rasqal <= @RASQAL_MAX_VERSION@
+Requires.private: raptor >= @RAPTOR_MIN_VERSION@, rasqal >= @RASQAL_MIN_VERSION@, rasqal <= @RASQAL_MAX_VERSION@
 Version: @VERSION@
 Libs: -L${libdir} -lrdf
 Libs.private: @LIBRDF_LDFLAGS@ @LIBRDF_PKGCONFIG_PRIVATE_LIBS@
