diff -up gc-7.1/include/include.am.dup_headers gc-7.1/include/include.am
--- gc-7.1/include/include.am.dup_headers	2007-12-21 18:20:15.000000000 -0600
+++ gc-7.1/include/include.am	2009-07-18 20:34:58.812906860 -0500
@@ -21,10 +21,8 @@ pkginclude_HEADERS += \
 	include/gc_typed.h \
 	include/gc_inline.h \
 	include/gc_mark.h \
-	include/gc_cpp.h \
 	include/weakpointer.h \
 	include/new_gc_alloc.h \
-	include/gc_allocator.h \
 	include/gc_backptr.h \
 	include/gc_gcj.h \
 	include/leak_detector.h \
