Index: cups-1.4.2/scheduler/conf.c
===================================================================
--- cups-1.4.2.orig/scheduler/conf.c
+++ cups-1.4.2/scheduler/conf.c
@@ -975,8 +975,10 @@ cupsdReadConfiguration(void)
 			     Group, 1, 1) < 0 ||
        cupsdCheckPermissions(ServerRoot, "ssl", 0700, RunUser,
 			     Group, 1, 0) < 0 ||
+      /* Never alter permissions of central conffile
        cupsdCheckPermissions(ServerRoot, "cupsd.conf", ConfigFilePerm, RunUser,
 			     Group, 0, 0) < 0 ||
+      */
        cupsdCheckPermissions(ServerRoot, "classes.conf", 0600, RunUser,
 			     Group, 0, 0) < 0 ||
        cupsdCheckPermissions(ServerRoot, "printers.conf", 0600, RunUser,
