Index: wacomtablet-1.2.3b/src/kded/devicehandler.cpp
===================================================================
--- wacomtablet-1.2.3b.orig/src/kded/devicehandler.cpp
+++ wacomtablet-1.2.3b/src/kded/devicehandler.cpp
@@ -166,7 +166,7 @@ bool DeviceHandler::findXInputDevice()
 
     for (int i = 0; i < ndevices; i++)
     {
-        if (info[i].use == IsXPointer || info[i].use == IsXKeyboard || info[i].use == IsXExtensionPointer)
+        if (info[i].use == IsXPointer || info[i].use == IsXKeyboard || info[i].use == IsXExtensionKeyboard)
             continue;
 
         uint wacom_prop = XInternAtom(QX11Info::display(), "Wacom Tool Type", True);
