diff -Nuar wacomtablet-1.2.3b.orig//po/CMakeLists.txt wacomtablet-1.2.3b/po/CMakeLists.txt
--- wacomtablet-1.2.3b.orig//po/CMakeLists.txt	2010-10-21 11:12:13.797000003 -0400
+++ wacomtablet-1.2.3b/po/CMakeLists.txt	2010-10-21 11:12:33.995000003 -0400
@@ -17,3 +17,4 @@
 add_subdirectory(eo)
 add_subdirectory(ja)
 add_subdirectory(en_GB)
+add_subdirectory(tr)
diff -Nuar wacomtablet-1.2.3b.orig//src/applet/plasma-applet-wacomtabletsettings.desktop wacomtablet-1.2.3b/src/applet/plasma-applet-wacomtabletsettings.desktop
--- wacomtablet-1.2.3b.orig//src/applet/plasma-applet-wacomtabletsettings.desktop	2010-10-21 11:12:13.802000003 -0400
+++ wacomtablet-1.2.3b/src/applet/plasma-applet-wacomtabletsettings.desktop	2010-10-21 11:33:11.811000004 -0400
@@ -13,6 +13,7 @@
 Comment[sv]=Anslutning under körning och profilstöd för Wacom ritplattor
 Comment[uk]=Підтримка з’єднання та профілювання планшетів Wacom
 Comment[x-test]=xxHotplugging and Profile support for Wacom Tabletsxx
+Comment[tr]=Wacom Tablet Ayarları
 Icon=input-tablet
 Type=Service
 X-KDE-ServiceTypes=Plasma/Applet
diff -Nuar wacomtablet-1.2.3b.orig//src/kcmodule/kcm_wacomtablet.desktop wacomtablet-1.2.3b/src/kcmodule/kcm_wacomtablet.desktop
--- wacomtablet-1.2.3b.orig//src/kcmodule/kcm_wacomtablet.desktop	2010-10-21 11:12:13.801000003 -0400
+++ wacomtablet-1.2.3b/src/kcmodule/kcm_wacomtablet.desktop	2010-10-21 11:32:34.771000007 -0400
@@ -18,6 +18,7 @@
 Name[sv]=Grafisk ritplatta
 Name[uk]=Графічний планшет
 Name[x-test]=xxGraphic Tabletxx
+Name[tr]=Grafik Tablet
 
 Comment=Wacom Tablet Settings
 Comment[en_GB]=Wacom Tablet Settings
@@ -26,5 +27,6 @@
 Comment[sv]=Inställningar av Wacom ritplatta
 Comment[uk]=Параметри планшетів Wacom
 Comment[x-test]=xxWacom Tablet Settingsxx
+Comment[tr]=Wacom Tablet Ayarları
 
 X-KDE-Keywords=Tablet,Input Devices,Wacom,Stylus
diff -Nuar wacomtablet-1.2.3b.orig//src/kded/wacomtablet.desktop wacomtablet-1.2.3b/src/kded/wacomtablet.desktop
--- wacomtablet-1.2.3b.orig//src/kded/wacomtablet.desktop	2010-10-21 11:12:13.800000003 -0400
+++ wacomtablet-1.2.3b/src/kded/wacomtablet.desktop	2010-10-21 11:31:41.778000008 -0400
@@ -16,6 +16,7 @@
 Name[sv]=Wacom ritplatta
 Name[uk]=Планшет Wacom
 Name[x-test]=xxWacom Tabletxx
+Name[tr]=Wacom Tablet
 Comment=Wacom tablet control daemon
 Comment[en_GB]=Wacom tablet control dæmon
 Comment[pt]=Servidor de controlo das tabletes Wacom
@@ -23,4 +24,5 @@
 Comment[sv]=Styrdemon för Wacom ritplatta
 Comment[uk]=Фонова служба керування планшетними пристроями Wacom
 Comment[x-test]=xxWacom tablet control daemonxx
+Comment[tr]=Wacom tablet aygıtı yönetim hizmeti
 
