Index: wacomtablet-v1.3.1/src/kded/xdeviceeventnotifier.cpp
===================================================================
--- wacomtablet-v1.3.1.orig/src/kded/xdeviceeventnotifier.cpp
+++ wacomtablet-v1.3.1/src/kded/xdeviceeventnotifier.cpp
@@ -66,6 +66,12 @@ bool XDeviceEventNotifier::x11Event(XEve
         //if(XGetEventData(QX11Info::display(), cookie))
         {
             XIHierarchyEvent * hev = (XIHierarchyEvent *)cookie->data;
+
+            if (!hev)
+            {
+                return false;
+            }
+
             XIHierarchyInfo *info = (XIHierarchyInfo *)hev->info;
 
             for (int i = 0; i < hev->num_info; i++)
