Index: VirtualBox-3.1.6_OSE/src/VBox/Frontends/VirtualBox/Makefile.kmk
===================================================================
--- VirtualBox-3.1.6_OSE.orig/src/VBox/Frontends/VirtualBox/Makefile.kmk
+++ VirtualBox-3.1.6_OSE/src/VBox/Frontends/VirtualBox/Makefile.kmk
@@ -32,7 +32,7 @@ VBOX_VIRTUALBOX4_OUT_DIR := $(PATH_TARGE
 
 # Ask the user to register at Sun. If this setting is disabled the user can
 # still register using the menu if desired.
-# VBOX_WITH_REGISTRATION_REQUEST := 1
+VBOX_WITH_REGISTRATION_REQUEST :=
 
 # For now this should be disabled in any case
 ifdef VBOX_WITH_REGISTRATION
Index: VirtualBox-3.1.6_OSE/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
===================================================================
--- VirtualBox-3.1.6_OSE.orig/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
+++ VirtualBox-3.1.6_OSE/src/VBox/Frontends/VirtualBox/src/VBoxUpdateDlg.cpp
@@ -104,7 +104,7 @@ QStringList VBoxUpdateData::list()
 
 VBoxUpdateData::VBoxUpdateData (const QString &aData)
     : mData (aData)
-    , mPeriodIndex (Period1Day)
+    , mPeriodIndex (PeriodNever)
     , mBranchIndex (BranchStable)
 {
     decode();
