--- client/config.py.orig	2009-12-12 12:36:12.433533820 +0200
+++ client/config.py	2009-12-12 12:37:55.853517891 +0200
@@ -3,15 +3,15 @@
 import commands
 import os_detect
 
-SMOON_URL = "http://www.smolts.org/"
+SMOON_URL = "http://smolt.pardus.org.tr:8090"
 SECURE = 0
 
 
 #Only a fool would edit what lays beyond here
 #Are you that fool?
 
-HW_UUID = "/etc/smolt/hw-uuid"
-PUB_UUID = "/etc/smolt/pub-uuid"
+#HW_UUID = "/etc/smolt/hw-uuid"
+#PUB_UUID = "/etc/smolt/pub-uuid"
 #ADMIN_TOKEN = "/etc/sysconfig/smolt-token"
 
 
@@ -106,5 +106,6 @@
 
 
 #For non RH Distros
-#HW_UUID = "/etc/smolt/hw-uuid"
-
+HW_UUID = "/etc/hw-uuid"
+PUB_UUID = "/etc/smolt/pub-uuid"
+ADMIN_TOKEN = "/etc/smolt/smolt-token"
