--- hdf5-1.8.5/hl/fortran/test/Makefile.in.tstlite	2010-06-04 12:26:00.000000000 -0600
+++ hdf5-1.8.5/hl/fortran/test/Makefile.in	2010-06-23 15:04:42.461913836 -0600
@@ -367,7 +367,7 @@
 	dsetf[1-4].h5 f1img.h5 f1tab.h5
 
 # Our main target, the test programs
-TEST_PROG = tstlite tstimage tsttable
+TEST_PROG = tstimage tsttable
 LDADD = $(LIBH5F_HL) $(LIBH5F) $(LIBH5_HL)  $(LIBHDF5)
 
 # Source files for the programs
