diff -Naur hdf5-1.8.5.orig/c++/examples/Makefile.am hdf5-1.8.5/c++/examples/Makefile.am
--- hdf5-1.8.5.orig/c++/examples/Makefile.am	2010-06-04 17:24:26.000000000 +0300
+++ hdf5-1.8.5/c++/examples/Makefile.am	2010-07-08 11:35:15.202887886 +0300
@@ -39,7 +39,7 @@
 CXX_API=yes
 
 # Where to install examples
-EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/hdf5_examples/c++
+EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/doc/hdf5/examples/c++
 
 # How to build programs using h5c++
 $(EXTRA_PROG): $(H5CPP)
diff -Naur hdf5-1.8.5.orig/c++/examples/run-c++-ex.sh.in hdf5-1.8.5/c++/examples/run-c++-ex.sh.in
--- hdf5-1.8.5.orig/c++/examples/run-c++-ex.sh.in	2010-06-04 17:24:26.000000000 +0300
+++ hdf5-1.8.5/c++/examples/run-c++-ex.sh.in	2010-07-08 11:35:16.688012624 +0300
@@ -20,7 +20,7 @@
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 #                                                                               #
 # This script will compile and run the c++ examples from source files           #
-# installed in .../share/hdf5_examples/c++ using h5c++.  The                    #
+# installed in .../share/doc/hdf5/examples/c++ using h5c++.  The                    #
 # order for running programs with RunTest in the MAIN section below is taken    #
 # from the Makefile.  The order is important since some of the test programs    #
 # use data files created by earlier test programs.  Any future additions should #
diff -Naur hdf5-1.8.5.orig/examples/Makefile.am hdf5-1.8.5/examples/Makefile.am
--- hdf5-1.8.5.orig/examples/Makefile.am	2010-06-04 17:24:27.000000000 +0300
+++ hdf5-1.8.5/examples/Makefile.am	2010-07-08 11:35:16.687012605 +0300
@@ -69,8 +69,8 @@
 CHECK_CLEANFILES+=$(EXTLINK_DIRS)
 
 # Example directory
-EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/hdf5_examples/c
-EXAMPLETOPDIR=${DESTDIR}$(exec_prefix)/share/hdf5_examples
+EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/doc/hdf5/examples/c
+EXAMPLETOPDIR=${DESTDIR}$(exec_prefix)/share/doc/hdf5/examples
 
 # List dependencies for each program.  Normally, automake would take
 # care of this for us, but if we tell automake about the programs it
diff -Naur hdf5-1.8.5.orig/examples/run-c-ex.sh.in hdf5-1.8.5/examples/run-c-ex.sh.in
--- hdf5-1.8.5.orig/examples/run-c-ex.sh.in	2010-06-04 17:24:27.000000000 +0300
+++ hdf5-1.8.5/examples/run-c-ex.sh.in	2010-07-08 11:35:16.693012698 +0300
@@ -20,7 +20,7 @@
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 #                                                                               #
 # This script will compile and run the c examples from source files installed   #
-# in .../share/hdf5_examples/c using h5cc or h5pc.  The order for running       #
+# in .../share/doc/hdf5/examples/c using h5cc or h5pc.  The order for running       #
 # programs with RunTest in the MAIN section below is taken from the Makefile.   #
 # The order is important since some of the test programs use data files created #
 # by earlier test programs.  Any future additions should be placed accordingly. #
diff -Naur hdf5-1.8.5.orig/fortran/examples/Makefile.am hdf5-1.8.5/fortran/examples/Makefile.am
--- hdf5-1.8.5.orig/fortran/examples/Makefile.am	2010-06-04 17:24:27.000000000 +0300
+++ hdf5-1.8.5/fortran/examples/Makefile.am	2010-07-08 11:35:16.687012605 +0300
@@ -65,7 +65,7 @@
 endif
 
 # Tell automake how to install examples
-EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/hdf5_examples/fortran
+EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/doc/hdf5/examples/fortran
 
 # List dependencies for each example.  Normally, automake would take
 # care of this for us, but if we tell automake about the programs it
diff -Naur hdf5-1.8.5.orig/fortran/examples/run-fortran-ex.sh.in hdf5-1.8.5/fortran/examples/run-fortran-ex.sh.in
--- hdf5-1.8.5.orig/fortran/examples/run-fortran-ex.sh.in	2010-06-04 17:24:27.000000000 +0300
+++ hdf5-1.8.5/fortran/examples/run-fortran-ex.sh.in	2010-07-08 11:35:16.691012603 +0300
@@ -20,7 +20,7 @@
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 #                                                                               #
 # This script will compile and run the fortran examples from source files       #
-# installed in .../share/hdf5_examples/fortran using h5fc or h5pfc.  The        #
+# installed in .../share/doc/hdf5/examples/fortran using h5fc or h5pfc.  The        #
 # order for running programs with RunTest in the MAIN section below is taken    #
 # from the Makefile.  The order is important since some of the test programs    #
 # use data files created by earlier test programs.  Any future additions should #
diff -Naur hdf5-1.8.5.orig/hl/c++/examples/Makefile.am hdf5-1.8.5/hl/c++/examples/Makefile.am
--- hdf5-1.8.5.orig/hl/c++/examples/Makefile.am	2010-06-04 17:24:24.000000000 +0300
+++ hdf5-1.8.5/hl/c++/examples/Makefile.am	2010-07-08 11:35:16.243012546 +0300
@@ -33,7 +33,7 @@
 CXX_API=yes
 
 # Where to install examples
-EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/hdf5_examples/hl/c++
+EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/doc/hdf5/examples/hl/c++
 
 # How to build programs using h5c++
 $(EXTRA_PROG): $(H5CPP)
diff -Naur hdf5-1.8.5.orig/hl/c++/examples/run-hlc++-ex.sh.in hdf5-1.8.5/hl/c++/examples/run-hlc++-ex.sh.in
--- hdf5-1.8.5.orig/hl/c++/examples/run-hlc++-ex.sh.in	2010-06-04 17:24:24.000000000 +0300
+++ hdf5-1.8.5/hl/c++/examples/run-hlc++-ex.sh.in	2010-07-08 11:35:16.689012509 +0300
@@ -20,7 +20,7 @@
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 #                                                                               #
 # This script will compile and run the c++ examples from source files           #
-# installed in .../share/hdf5_examples/hl/c++ using h5c++.  The                 #
+# installed in .../share/doc/hdf5/examples/hl/c++ using h5c++.  The                 #
 # order for running programs with RunTest in the MAIN section below is taken    #
 # from the Makefile.  The order is important since some of the test programs    #
 # use data files created by earlier test programs.  Any future additions should #
diff -Naur hdf5-1.8.5.orig/hl/examples/Makefile.am hdf5-1.8.5/hl/examples/Makefile.am
--- hdf5-1.8.5.orig/hl/examples/Makefile.am	2010-06-04 17:24:25.000000000 +0300
+++ hdf5-1.8.5/hl/examples/Makefile.am	2010-07-08 11:35:16.251012752 +0300
@@ -26,8 +26,8 @@
 endif
 
 # Example directory
-EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/hdf5_examples/hl/c
-EXAMPLETOPDIR=${DESTDIR}$(exec_prefix)/share/hdf5_examples/hl
+EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/doc/hdf5/examples/hl/c
+EXAMPLETOPDIR=${DESTDIR}$(exec_prefix)/share/doc/hdf5/examples/hl
 
 # Example programs.
 # Don't tell automake about them, because if it knew they were programs,
diff -Naur hdf5-1.8.5.orig/hl/examples/run-hlc-ex.sh.in hdf5-1.8.5/hl/examples/run-hlc-ex.sh.in
--- hdf5-1.8.5.orig/hl/examples/run-hlc-ex.sh.in	2010-06-04 17:24:25.000000000 +0300
+++ hdf5-1.8.5/hl/examples/run-hlc-ex.sh.in	2010-07-08 11:35:16.690012914 +0300
@@ -20,7 +20,7 @@
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 #                                                                               #
 # This script will compile and run the c examples from source files installed   #
-# in .../share/hdf5_examples/hl/c using h5cc or h5pc.  The order for running    #
+# in .../share/doc/hdf5/examples/hl/c using h5cc or h5pc.  The order for running    #
 # programs with RunTest in the MAIN section below is taken from the Makefile.   #
 # The order is important since some of the test programs use data files created #
 # by earlier test programs.  Any future additions should be placed accordingly. #
diff -Naur hdf5-1.8.5.orig/hl/fortran/examples/Makefile.am hdf5-1.8.5/hl/fortran/examples/Makefile.am
--- hdf5-1.8.5.orig/hl/fortran/examples/Makefile.am	2010-06-04 17:24:24.000000000 +0300
+++ hdf5-1.8.5/hl/fortran/examples/Makefile.am	2010-07-08 11:35:16.250012534 +0300
@@ -50,7 +50,7 @@
 endif
 
 # Tell automake how to install examples
-EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/hdf5_examples/hl/fortran
+EXAMPLEDIR=${DESTDIR}$(exec_prefix)/share/doc/hdf5/examples/hl/fortran
 
 # List dependencies for each example.  Normally, automake would take
 # care of this for us, but if we tell automake about the programs it
diff -Naur hdf5-1.8.5.orig/hl/fortran/examples/run-hlfortran-ex.sh.in hdf5-1.8.5/hl/fortran/examples/run-hlfortran-ex.sh.in
--- hdf5-1.8.5.orig/hl/fortran/examples/run-hlfortran-ex.sh.in	2010-06-04 17:24:24.000000000 +0300
+++ hdf5-1.8.5/hl/fortran/examples/run-hlfortran-ex.sh.in	2010-07-08 11:35:16.689012509 +0300
@@ -20,7 +20,7 @@
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 #                                                                               #
 # This script will compile and run the fortran examples from source files       #
-# installed in .../share/hdf5_examples/hl/fortran using h5fc or h5pfc.  The     #
+# installed in .../share/doc/hdf5/examples/hl/fortran using h5fc or h5pfc.  The     #
 # order for running programs with RunTest in the MAIN section below is taken    #
 # from the Makefile.  The order is important since some of the test programs    #
 # use data files created by earlier test programs.  Any future additions should #
