diff -Nur syslinux-4.04-old/com32/MCONFIG syslinux-4.04/com32/MCONFIG > myown-syslinux-4.04-nopie.patch
--- syslinux-4.04-old/com32/MCONFIG	2011-04-18 23:24:17.000000000 +0200
+++ syslinux-4.04/com32/MCONFIG	2011-08-08 15:16:42.125048741 +0200
@@ -21,6 +21,8 @@
 GCCOPT += $(call gcc_ok,-m32,)
 GCCOPT += $(call gcc_ok,-fno-stack-protector,)
 GCCOPT += $(call gcc_ok,-fwrapv,)
+GCCOPT += $(call gcc_ok,-nopie,)
+GCCOPT += $(call gcc_ok,-fno-pie,)
 GCCOPT += $(call gcc_ok,-freg-struct-return,)
 GCCOPT += -mregparm=3 -DREGPARM=3 -march=i386 -Os
 GCCOPT += $(call gcc_ok,-fPIE,-fPIC)
