diff -Nur gst-plugins-base-0.10.30-old//common/gtk-doc.mak gst-plugins-base-0.10.30/common/gtk-doc.mak
--- gst-plugins-base-0.10.30-old//common/gtk-doc.mak	2010-10-12 19:10:41.304000143 +0300
+++ gst-plugins-base-0.10.30/common/gtk-doc.mak	2010-10-12 19:11:18.961000044 +0300
@@ -232,6 +232,6 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
diff -Nur gst-plugins-base-0.10.30-old//common/gtk-doc-plugins.mak gst-plugins-base-0.10.30/common/gtk-doc-plugins.mak
--- gst-plugins-base-0.10.30-old//common/gtk-doc-plugins.mak	2010-10-12 19:10:41.303000153 +0300
+++ gst-plugins-base-0.10.30/common/gtk-doc-plugins.mak	2010-10-12 19:11:28.928999943 +0300
@@ -332,7 +332,7 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
