diff -ur bootsplash-3.3.orig/Utilities/splash.c bootsplash-3.3/Utilities/splash.c
--- bootsplash-3.3.orig/Utilities/splash.c	2006-03-24 17:01:25.000000000 +0200
+++ bootsplash-3.3/Utilities/splash.c	2008-01-12 12:59:33.000000000 +0200
@@ -980,7 +980,7 @@
 
   unsigned char quant[4][64];
   struct dec_hufftbl dhuff[4];
-  struct jpginfo info;
+  struct jpginfo info = {};
   struct scan dscans[MAXCOMP];
 
   setinput(&in, bp);
