--- octave-3.2.4/run-octave.in.orig	2010-12-31 14:29:56.943000062 +0200
+++ octave-3.2.4/run-octave.in	2010-12-31 14:30:30.280000059 +0200
@@ -30,9 +30,9 @@
 top_srcdir='%abs_top_srcdir%'
 builddir='%builddir%'
 
-liboctinterp="$builddir/src/%liboctinterp%"
-liboctave="$builddir/liboctave/%liboctave%"
-libcruft="$builddir/libcruft/%libcruft%"
+liboctinterp="$builddir/src/%liboctinterp.so.3.2.4%"
+liboctave="$builddir/liboctave/%liboctave.so.3.2.4%"
+libcruft="$builddir/libcruft/%libcruft.so.3.2.4%"
 
 d1="$top_srcdir/test"
 d2="$top_srcdir/scripts"
