diff -ur octave-3.2.3.orig/doc/Makefile.in octave-3.2.3/doc/Makefile.in
--- octave-3.2.3.orig/doc/Makefile.in	2009-06-25 05:36:29.000000000 +0100
+++ octave-3.2.3/doc/Makefile.in	2009-10-26 18:05:30.152146720 +0000
@@ -35,7 +35,7 @@
 	$(MAKE) -C interpreter ../$@
 .PHONY: ../BUGS ../INSTALL.OCTAVE
 
-$(SUBDIRS):
+$(SUBDIRS): conf.texi
 	$(MAKE) -C $@ all
 .PHONY: $(SUBDIRS)
 
diff -ur octave-3.2.3.orig/doc/faq/Makefile.in octave-3.2.3/doc/faq/Makefile.in
--- octave-3.2.3.orig/doc/faq/Makefile.in	2009-05-25 07:04:58.000000000 +0100
+++ octave-3.2.3/doc/faq/Makefile.in	2009-11-02 06:30:49.350407797 +0000
@@ -98,3 +98,5 @@
 dist: all
 	ln $(DISTFILES) ../../`cat ../../.fname`/doc/faq
 .PHONY: dist
+
+.NOTPARALLEL:
diff -ur octave-3.2.3.orig/doc/interpreter/Makefile.in octave-3.2.3/doc/interpreter/Makefile.in
--- octave-3.2.3.orig/doc/interpreter/Makefile.in	2009-05-25 07:04:58.000000000 +0100
+++ octave-3.2.3/doc/interpreter/Makefile.in	2009-11-02 06:33:30.179572944 +0000
@@ -394,3 +394,5 @@
 	done
 	for dir in $(DISTSUBDIRS); do mkdir ../../`cat ../../.fname`/doc/interpreter/$$dir; $(MAKE) -C $$dir $@; done
 .PHONY: dist
+
+.NOTPARALLEL:
diff -ur octave-3.2.3.orig/doc/liboctave/Makefile.in octave-3.2.3/doc/liboctave/Makefile.in
--- octave-3.2.3.orig/doc/liboctave/Makefile.in	2009-05-25 07:04:58.000000000 +0100
+++ octave-3.2.3/doc/liboctave/Makefile.in	2009-11-02 06:31:35.267093213 +0000
@@ -123,3 +123,5 @@
 	  ln ../../doc/liboctave/$$dir/* ../../`cat ../../.fname`/doc/liboctave/$$dir; \
 	done
 .PHONY: dist
+
+.NOTPARALLEL:
