Index: emesene-1.6.3/emesenelib/ProfileManager.py
===================================================================
--- emesene-1.6.3.orig/emesenelib/ProfileManager.py
+++ emesene-1.6.3/emesenelib/ProfileManager.py
@@ -152,8 +152,8 @@ class ProfileManager(gobject.GObject):
             except:
                 dpurl = ""
 
-        photo = response.body.split('</Photo>')[0].split('<Photo>')[1]
         try:
+            photo = response.body.split('</Photo>')[0].split('<Photo>')[1]
             self.dpid = photo.split('</ResourceID>')[0].split('<ResourceID>')[1]
         except:
             self.dpid = ''
