diff -urN autoopts.orig/autoopts-config.in autoopts/autoopts-config.in
--- autoopts.orig/autoopts-config.in	2010-09-08 02:54:30.000000000 +0200
+++ autoopts/autoopts-config.in	2010-09-11 05:54:15.000000000 +0200
@@ -1,4 +1,4 @@
-#! /old-home/ROOT/usr/local/bin/bash
+#! /bin/bash
 ## ---------------------------------------------------------------------
 ## autoopts-config.in -- Describe AutoOpts configuration
 ##
diff -urN autoopts.orig/mk-autoopts-pc.in autoopts/mk-autoopts-pc.in
--- autoopts.orig/mk-autoopts-pc.in	2010-09-08 02:54:30.000000000 +0200
+++ autoopts/mk-autoopts-pc.in	2010-09-11 05:54:07.000000000 +0200
@@ -1,4 +1,4 @@
-#! /old-home/ROOT/usr/local/bin/bash
+#! /bin/bash
 ## ---------------------------------------------------------------------
 ## mk-autoopts-pc.in -- Describe AutoOpts configuration
 ##
