--- a/src/hotplug_libhal.c	2009-07-04 11:10:31.000000000 +0300
+++ b/src/hotplug_libhal.c	2010-03-23 11:59:15.814045888 +0200
@@ -335,9 +335,9 @@
 		}
 	}
 
-	/* if we found a specific driver */
-	if (driver)
-		return driver;
+	/* if we found a specific driver other than openct */
+    if (driver && ((0 != strcmp(driver->CFBundleName, "openct-ifd") || !classdriver)))
+        return driver;
 
 	/* else return the Class driver */
 	return classdriver;
