Index: oxygen-gtk-1.0.0/rc/argb-apps.conf
===================================================================
--- oxygen-gtk-1.0.0.orig/rc/argb-apps.conf
+++ oxygen-gtk-1.0.0/rc/argb-apps.conf
@@ -57,3 +57,5 @@ disable:dockbarx_factory.py:liferea:mido
 disable:npviewer.bin:miro.real:gflashplayer:swfdec-player:xaralx:notecase:codeblocks:ekiga:xfce4-panel
 disable:chrome:<unknown>:eclipse:Eclipse:.:xmind:audacious2
 disable:nautilus:pcmanfm:emacs:AptanaStudio:dia:pgadmin3:kino:winff:mmc:acroread:filezilla
+disable:openshot:jamin
+disable:sylpheed
