--- oxygen-gtk-1.0.1/rc/argb-apps.conf	2011-01-13 14:16:20.000000000 +0100
+++ oxygen-gtk-1.0.1/rc/argb-apps.confnew	2011-01-13 23:27:00.000000000 +0100
@@ -43,7 +43,7 @@
 # enable all applications, by default
 # note that this line is not necessary since applications not found
 # in this file have argb support enabled anyway
-enable:all
+disable:all
 
 # mozilla apps
 disable:firefox:firefox-3.5:firefox-bin:thunderbird-bin:thunderbird:seamonkey:seamonkey-bin
@@ -60,3 +60,6 @@
 disable:nautilus:pcmanfm:emacs:AptanaStudio:dia:pgadmin3:kino:winff:mmc:acroread:filezilla
 disable:stardict:icecat:openshot:sylpheed:jamin:vmware:vmplayer:eboard
 disable:envy24control:snes9x-gtk:guayadeque:komodo-bin:SWT:lazarus
+
+#short list of known good apps
+enable:gimp:gqview
