From cc15beb12927e10ac16d02ec417dd3a215065255 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?C=C3=A9dric=20Bellegarde?= <gnumdk@gmail.com>
Date: Fri, 29 Oct 2010 09:46:19 +0200
Subject: [PATCH] Fix firefox 4 menu

---
 src/engine.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/engine.cpp b/src/engine.cpp
index 32cb491..e47e957 100644
--- a/src/engine.cpp
+++ b/src/engine.cpp
@@ -559,7 +559,7 @@ void Engine::drawMenuBarItem()
 void Engine::drawMenuFrame()
 {
     GTK_QT_DEBUG_FUNC
-    ENGINE_SETUP(QStyleOptionFrameV3, true, false, QPalette::Window)
+    ENGINE_SETUP(QStyleOptionFrameV3, false, false, QPalette::Window)
     option.state = QStyle::State_Raised;
     option.lineWidth = 3;
     option.midLineWidth = 3;
-- 
1.6.1

