diff -Nuar guake-0.4.2-orig/data/guake.desktop.in guake-0.4.2/data/guake.desktop.in
--- guake-0.4.2-orig/data/guake.desktop.in	2009-12-01 13:51:08.000000000 +0200
+++ guake-0.4.2/data/guake.desktop.in	2010-08-18 18:01:39.987893913 +0300
@@ -1,14 +1,20 @@
 [Desktop Entry]
 Encoding=UTF-8
-Name=Guake Terminal
-Name[pt]=Guake Terminal
-Name[pt_BR]=Guake Terminal
+Name=Guake
+Name[pt]=Guake
+Name[pt_BR]=Guake
+Name[tr]=Guake
+GenericName=Guake Terminal
+GenericName[pt]=Guake Terminal
+GenericName[pt_BR]=Guake Terminal
+GenericName[tr]=Guake Uçbirim
 Comment=Use the command line in a Quake-like terminal
 Comment[pt]=Utilizar a linha de comando em um terminal estilo Quake
 Comment[pt_BR]=Utilizar a linha de comando em um terminal estilo Quake
+Comment[tr]=Quake benzeri uçbirimde komut satırı kullanın
 TryExec=guake
 Exec=guake
-Icon=@DATADIR@/pixmaps/guake/guake.png
+Icon=guake
 Type=Application
 Categories=GNOME;GTK;Utility;TerminalEmulator;
 StartupNotify=true
