Index: OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegAudioStream.cpp
===================================================================
--- OpenSceneGraph-2.8.3.orig/src/osgPlugins/ffmpeg/FFmpegAudioStream.cpp
+++ OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegAudioStream.cpp
@@ -1,4 +1,4 @@
-
+#include "FFmpegHeaders.hpp"
 #include "FFmpegAudioStream.hpp"
 
 #include <OpenThreads/ScopedLock>
Index: OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegHeaders.hpp
===================================================================
--- OpenSceneGraph-2.8.3.orig/src/osgPlugins/ffmpeg/FFmpegHeaders.hpp
+++ OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegHeaders.hpp
@@ -6,6 +6,7 @@
 extern "C"
 {
 #define __STDC_CONSTANT_MACROS
+#define FF_API_OLD_SAMPLE_FMT 0
 #include <errno.h>    // for error codes defined in avformat.h
 #include <stdint.h>
 #include <avcodec.h>
Index: OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp
===================================================================
--- OpenSceneGraph-2.8.3.orig/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp
+++ OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/FFmpegImageStream.cpp
@@ -1,4 +1,4 @@
-
+#include "FFmpegHeaders.hpp"
 #include "FFmpegImageStream.hpp"
 #include "FFmpegAudioStream.hpp"
 
Index: OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/ReaderWriterFFmpeg.cpp
===================================================================
--- OpenSceneGraph-2.8.3.orig/src/osgPlugins/ffmpeg/ReaderWriterFFmpeg.cpp
+++ OpenSceneGraph-2.8.3/src/osgPlugins/ffmpeg/ReaderWriterFFmpeg.cpp
@@ -10,12 +10,12 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
  * OpenSceneGraph Public License for more details.
 */
+#include "FFmpegHeaders.hpp"
 
 #include <osgDB/Registry>
 #include <osgDB/FileNameUtils>
 #include <osgDB/FileUtils>
 
-#include "FFmpegHeaders.hpp"
 #include "FFmpegImageStream.hpp"
 
 
