From 9c940ee21009abb338182f5774d6ece0e9dd6eb9 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Thu, 26 Nov 2009 10:39:05 +0000
Subject: [PATCH 26/45] If /usr/sbin/hald does not exist then exit the haldaemon init script with retval 1 for LSB compliance

---
 hald/haldaemon.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hald/haldaemon.in b/hald/haldaemon.in
index cd6dfde..33ee31a 100755
--- a/hald/haldaemon.in
+++ b/hald/haldaemon.in
@@ -12,7 +12,7 @@
 #
 
 # Sanity checks.
-[ -x @sbindir@/hald ] || exit 0
+[ -x @sbindir@/hald ] || exit 1
 
 # Source function library.
 . @sysconfdir@/rc.d/init.d/functions
-- 
1.7.2.1

