From 8861cfc7b4a8a3278773db766b836bbdce33e718 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Aur=C3=A9lien=20G=C3=A2teau?= <agateau@kde.org>
Date: Wed, 17 Jun 2009 15:26:14 +0200
Subject: [PATCH] Removed event-id

---
--- a/plasma/applets/systemtray/core/notification.cpp
+++ b/plasma/applets/systemtray/core/notification.cpp
@@ -41,7 +41,6 @@ public:
 
     QString applicationName;
     QIcon applicationIcon;
-    QString eventId;
     QString message;
     QString summary;
     int timeout;
@@ -90,18 +89,6 @@ void Notification::setApplicationIcon(const QIcon &applicationIcon)
 }
 
 
-QString Notification::eventId() const
-{
-    return d->eventId;
-}
-
-
-void Notification::setEventId(const QString &eventId)
-{
-    d->eventId = eventId;
-}
-
-
 QString Notification::message() const
 {
     return d->message;
--- a/plasma/applets/systemtray/core/notification.h
+++ b/plasma/applets/systemtray/core/notification.h
@@ -42,7 +42,6 @@ public:
 
     QString applicationName() const;
     QIcon applicationIcon() const;
-    QString eventId() const;
     QString message() const;
     QString summary() const;
     int timeout() const;
@@ -66,7 +65,6 @@ signals:
 protected:
     void setApplicationName(const QString &applicationName);
     void setApplicationIcon(const QIcon &applicationIcon);
-    void setEventId(const QString &eventId);
     void setMessage(const QString &message);
     void setSummary(const QString &summary);
     void setTimeout(int timeout);
--- a/plasma/applets/systemtray/protocols/notifications/dbusnotificationprotocol.cpp
+++ b/plasma/applets/systemtray/protocols/notifications/dbusnotificationprotocol.cpp
@@ -89,7 +89,6 @@ void DBusNotificationProtocol::dataUpdated(const QString &source, const Plasma::
     DBusNotification* notification = m_notifications[source];
     notification->setApplicationName(data.value("appName").toString());
     notification->setApplicationIcon(KIcon(data.value("appIcon").toString()));
-    notification->setEventId(data.value("eventId").toString());
     notification->setSummary(data.value("summary").toString());
     notification->setMessage(data.value("body").toString());
     notification->setTimeout(data.value("expireTimeout").toInt());
--- a/plasma/dataengines/notifications/notificationsengine.cpp
+++ b/plasma/dataengines/notifications/notificationsengine.cpp
@@ -50,7 +50,7 @@ void NotificationsEngine::init()
 {
 }
 
-uint NotificationsEngine::Notify(const QString &app_name, uint replaces_id, const QString &event_id,
+uint NotificationsEngine::Notify(const QString &app_name, uint replaces_id,
                                  const QString &app_icon, const QString &summary, const QString &body,
                                  const QStringList &actions, const QVariantMap &hints, int timeout)
 {
@@ -77,7 +77,6 @@ uint NotificationsEngine::Notify(const QString &app_name, uint replaces_id, cons
     notificationData.insert("id", QString::number(id));
     notificationData.insert("appName", appname_str);
     notificationData.insert("appIcon", app_icon);
-    notificationData.insert("eventId", event_id);
     notificationData.insert("summary", summary);
     notificationData.insert("body", body);
     notificationData.insert("actions", actions);
--- a/plasma/dataengines/notifications/notificationsengine.h
+++ b/plasma/dataengines/notifications/notificationsengine.h
@@ -44,7 +44,7 @@ public:
      *  This function implements part of Notifications DBus interface.
      *  Once called, will add notification source to the engine
      */
-    uint Notify(const QString &app_name, uint replaces_id, const QString &event_id, const QString &app_icon, const QString &summary, const QString &body, const QStringList &actions, const QVariantMap &hints, int timeout);
+    uint Notify(const QString &app_name, uint replaces_id, const QString &app_icon, const QString &summary, const QString &body, const QStringList &actions, const QVariantMap &hints, int timeout);
 
     void CloseNotification( uint id );
 
--- a/plasma/dataengines/notifications/org.kde.VisualNotifications.xml
+++ b/plasma/dataengines/notifications/org.kde.VisualNotifications.xml
@@ -10,11 +10,10 @@
       <arg name="action_key" type="s" direction="out"/>
     </signal>
     <method name="Notify">
-      <annotation name="com.trolltech.QtDBus.QtTypeName.In7" value="QVariantMap"/>
+      <annotation name="com.trolltech.QtDBus.QtTypeName.In6" value="QVariantMap"/>
       <arg type="u" direction="out"/>
       <arg name="app_name" type="s" direction="in"/>
       <arg name="replaces_id" type="u" direction="in"/>
-      <arg name="event_id" type="s" direction="in"/>
       <arg name="app_icon" type="s" direction="in"/>
       <arg name="summary" type="s" direction="in"/>
       <arg name="body" type="s" direction="in"/>
