diff -ru kdebase-workspace-4.5.1/kstyles/oxygen/oxygenstyle.cpp kdebase-workspace-4.5.1/kstyles/oxygen/oxygenstyle.cpp
--- kdebase-workspace-4.5.1/kstyles/oxygen/oxygenstyle.cpp	2010-09-19 14:55:18.000000000 +0200
+++ kdebase-workspace-4.5.1/kstyles/oxygen/oxygenstyle.cpp	2010-09-19 14:58:00.000000000 +0200
@@ -4520,6 +4520,9 @@
 
         // draw bar
         sub_opt.rect = subElementRect( QStyle::SE_ProgressBarContents, cbOption, widget);
+        if (sub_opt.rect.height() < 3) {
+            sub_opt.rect.adjust(0, 0, 0, 2);
+        }
         drawControl( QStyle::CE_ProgressBarContents, &sub_opt, p, widget);
 
         // draw label
