Index: kdebase-workspace-4.4.92/krunner/screensaver/saverengine.cpp
===================================================================
--- kdebase-workspace-4.4.92.orig/krunner/screensaver/saverengine.cpp	2010-04-15 09:59:22.000000000 +0100
+++ kdebase-workspace-4.4.92/krunner/screensaver/saverengine.cpp	2010-07-15 17:33:35.000000000 +0100
@@ -202,7 +202,7 @@
     if (mState != Waiting)
         return false;
 
-    if (e)
+    if (e && KScreenSaverSettings::screenSaverEnabled() )
     {
         if (!mXAutoLock)
         {
