--- kdebase-workspace-4.5.1/kwin/scene_opengl.cpp	2010/09/18 17:40:41	1176786
+++ kdebase-workspace-4.5.1/kwin/scene_opengl.cpp	2010/09/18 17:41:07	1176787
@@ -141,6 +141,12 @@
         return; // error
     // Initialize OpenGL
     initGL();
+    if( QString((const char*)glGetString( GL_RENDERER )) == "Software Rasterizer" )
+        {
+        kError( 1212 ) << "OpenGL Software Rasterizer detected. Falling back to XRender.";
+        QTimer::singleShot( 0, Workspace::self(), SLOT( fallbackToXRenderCompositing()));
+        return;
+        }
     if( !hasGLExtension( "GL_ARB_texture_non_power_of_two" )
         && !hasGLExtension( "GL_ARB_texture_rectangle" ))
         {
