From 6e01fe489d0ad14acbdd508abfaafecfa0e6ef1d Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Aur=C3=A9lien=20G=C3=A2teau?= <agateau@kde.org>
Date: Mon, 22 Jun 2009 14:53:36 +0200
Subject: [PATCH] Handle case where timeout is -1

---
--- a/plasma/dataengines/notifications/notificationsengine.cpp
+++ b/plasma/dataengines/notifications/notificationsengine.cpp
@@ -62,6 +62,18 @@ uint NotificationsEngine::Notify(const QString &app_name, uint replaces_id,
         appname_str = i18n("Unknown Application");
     }
 
+    if (timeout == -1) {
+        const int AVERAGE_WORD_LENGTH = 6;
+        const int WORD_PER_MINUTE = 250;
+        int count = summary.length() + body.length();
+        timeout = 60000 * count / AVERAGE_WORD_LENGTH / WORD_PER_MINUTE;
+
+        // Add two seconds for the user to notice the notification, and ensure
+        // it last at least five seconds, otherwise all the user see is a
+        // flash
+        timeout = 2000 + qMin(timeout, 3000);
+    }
+
     const QString source = QString("notification %1").arg(id);
     if (replaces_id) {
         Plasma::DataContainer *container = containerForSource(source);
