Index: kdebase-workspace-4.2.95/plasma/applets/kickoff/core/leavemodel.cpp
===================================================================
--- kdebase-workspace-4.2.95.orig/plasma/applets/kickoff/core/leavemodel.cpp	2009-06-26 00:11:39.378869275 +0200
+++ kdebase-workspace-4.2.95/plasma/applets/kickoff/core/leavemodel.cpp	2009-06-26 00:13:26.954207539 +0200
@@ -74,13 +74,13 @@
         item->setIcon(KIcon("system-suspend"));
         item->setData(i18n("Pause without logging out"), Kickoff::SubTitleRole);
     } else if (basename == "suspenddisk") {
-        item->setText(i18n("Suspend to Disk"));
+        item->setText(i18n("Hibernate"));
         item->setIcon(KIcon("system-suspend-hibernate"));
-        item->setData(i18n("Pause without logging out"), Kickoff::SubTitleRole);
+        item->setData(i18n("Suspend to disk"), Kickoff::SubTitleRole);
     } else if (basename == "suspendram") {
-        item->setText(i18n("Suspend to RAM"));
+        item->setText(i18n("Sleep"));
         item->setIcon(KIcon("system-suspend"));
-        item->setData(i18n("Pause without logging out"), Kickoff::SubTitleRole);
+        item->setData(i18n("Suspend to RAM"), Kickoff::SubTitleRole);
     } else {
         item->setText(basename);
         item->setData(url, Kickoff::SubTitleRole);
