Index: mrtg-2.16.3/Makefile.in
===================================================================
--- mrtg-2.16.3.orig/Makefile.in
+++ mrtg-2.16.3/Makefile.in
@@ -30,7 +30,7 @@ MKINSTALLDIRS = $(SHELL) $(top_srcdir)/m
 all:	 bin/rateup subst
 
 bin/rateup: bin/rateup.o
-	LD_RUN_PATH=$(LD_RUN_PATH) $(CC) bin/rateup.o -o bin/rateup $(LDFLAGS)  $(LIBS) 
+	$(CC) bin/rateup.o -o bin/rateup $(LDFLAGS)  $(LIBS) 
 
 bin/rateup.o: $(top_srcdir)/src/rateup.c
 	$(CC) $(CPPFLAGS) $(CFLAGS) -c $(top_srcdir)/src/rateup.c -o bin/rateup.o
