--- support/apachectl.in.orig   2007-03-18 00:49:20.000000000 +0200
+++ support/apachectl.in        2007-03-18 00:49:43.000000000 +0200
@@ -41,7 +41,8 @@
 # --------------------                              --------------------
 #
 # the path to your httpd binary, including options if necessary
-HTTPD='@exp_sbindir@/@progname@'
+. /etc/conf.d/apache2
+HTTPD="/usr/sbin/apache2 ${APACHE2_OPTS}"
 #
 # pick up any necessary environment variables
 if test -f @exp_sbindir@/envvars; then
