diff -urp audit-2.0.5.orig/lib/libaudit.c audit-2.0.5/lib/libaudit.c
--- audit-2.0.5.orig/lib/libaudit.c	2009-12-09 09:49:10.000000000 -0500
+++ audit-2.0.5/lib/libaudit.c	2010-01-13 14:28:52.000000000 -0500
@@ -1130,7 +1130,12 @@ int audit_rule_fieldpair_data(struct aud
 			if (!isdigit((char)*(v)))
 				return -21;
 
-			rule->values[rule->field_count] = strtol(v, NULL, 0);
+			if (field == AUDIT_INODE)
+				rule->values[rule->field_count] =
+					strtoul(v, NULL, 0);
+			else
+				rule->values[rule->field_count] =
+					strtol(v, NULL, 0);
 			break;
 	}
 	rule->field_count++;
