diff -urp audit-2.0.5.orig/lib/libaudit.c audit-2.0.5/lib/libaudit.c
--- audit-2.0.5.orig/lib/libaudit.c	2009-12-09 09:49:10.000000000 -0500
+++ audit-2.0.5/lib/libaudit.c	2010-01-13 19:28:10.000000000 -0500
@@ -34,6 +34,7 @@
 #include <errno.h>
 #include <sys/poll.h>
 #include <sys/utsname.h>
+#include <sys/stat.h>
 #include <fcntl.h>	/* O_NOFOLLOW needs gnu defined */
 #include <limits.h>	/* for PATH_MAX */
 
