diff -Nur audit-2.0.4-old/audisp/plugins/Makefile.am audit-2.0.4/audisp/plugins/Makefile.am
--- audit-2.0.4-old/audisp/plugins/Makefile.am	2010-08-07 01:03:33.898396971 +0300
+++ audit-2.0.4/audisp/plugins/Makefile.am	2010-08-07 01:04:07.523279974 +0300
@@ -22,7 +22,7 @@
 
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig
 
-SUBDIRS = builtins zos-remote remote
+SUBDIRS = builtins remote
 #SUBDIRS = builtins zos-remote
 if HAVE_PRELUDE
 SUBDIRS += prelude
diff -Nur audit-2.0.4-old/configure.ac audit-2.0.4/configure.ac
--- audit-2.0.4-old/configure.ac	2010-08-07 01:03:33.898396971 +0300
+++ audit-2.0.4/configure.ac	2010-08-07 01:06:17.572271932 +0300
@@ -231,7 +231,7 @@
 AC_SUBST(LIBPRELUDE_CFLAGS)
 AC_SUBST(LIBPRELUDE_LDFLAGS)
 
-AC_OUTPUT(Makefile lib/Makefile lib/test/Makefile auparse/Makefile auparse/test/Makefile src/Makefile src/mt/Makefile src/libev/Makefile src/test/Makefile swig/Makefile docs/Makefile init.d/Makefile audisp/Makefile audisp/plugins/Makefile audisp/plugins/builtins/Makefile audisp/plugins/prelude/Makefile audisp/plugins/remote/Makefile audisp/plugins/zos-remote/Makefile bindings/Makefile bindings/python/Makefile tools/Makefile tools/aulast/Makefile tools/aulastlog/Makefile tools/ausyscall/Makefile)
+AC_OUTPUT(Makefile lib/Makefile lib/test/Makefile auparse/Makefile auparse/test/Makefile src/Makefile src/mt/Makefile src/libev/Makefile src/test/Makefile swig/Makefile docs/Makefile init.d/Makefile audisp/Makefile audisp/plugins/Makefile audisp/plugins/builtins/Makefile audisp/plugins/prelude/Makefile audisp/plugins/remote/Makefile bindings/Makefile bindings/python/Makefile tools/Makefile tools/aulast/Makefile tools/aulastlog/Makefile tools/ausyscall/Makefile)
 
 echo .
 echo "
