diff -Nur audit-1.7.9-old/audisp/plugins/Makefile.am audit-1.7.9/audisp/plugins/Makefile.am
--- audit-1.7.9-old/audisp/plugins/Makefile.am	2008-11-14 15:14:24.000000000 +0200
+++ audit-1.7.9/audisp/plugins/Makefile.am	2008-11-14 15:14:39.000000000 +0200
@@ -22,7 +22,7 @@
 
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig
 
-SUBDIRS = builtins zos-remote remote
+SUBDIRS = builtins remote
 #SUBDIRS = builtins zos-remote
 if HAVE_PRELUDE
 SUBDIRS += prelude
diff -Nur audit-1.7.9-old/configure.ac audit-1.7.9/configure.ac
--- audit-1.7.9-old/configure.ac	2008-11-14 15:14:24.000000000 +0200
+++ audit-1.7.9/configure.ac	2008-11-14 15:14:51.000000000 +0200
@@ -201,8 +201,7 @@
 AC_SUBST(LIBPRELUDE_CFLAGS)
 AC_SUBST(LIBPRELUDE_LDFLAGS)
 
-AC_CONFIG_SUBDIRS([system-config-audit])
-AC_OUTPUT(Makefile lib/Makefile lib/test/Makefile auparse/Makefile auparse/test/Makefile src/Makefile src/mt/Makefile src/libev/Makefile src/test/Makefile swig/Makefile docs/Makefile init.d/Makefile audisp/Makefile audisp/plugins/Makefile audisp/plugins/builtins/Makefile audisp/plugins/prelude/Makefile audisp/plugins/remote/Makefile audisp/plugins/zos-remote/Makefile bindings/Makefile bindings/python/Makefile tools/Makefile tools/aulast/Makefile tools/aulastlog/Makefile tools/ausyscall/Makefile)
+AC_OUTPUT(Makefile lib/Makefile lib/test/Makefile auparse/Makefile auparse/test/Makefile src/Makefile src/mt/Makefile src/libev/Makefile src/test/Makefile swig/Makefile docs/Makefile init.d/Makefile audisp/Makefile audisp/plugins/Makefile audisp/plugins/builtins/Makefile audisp/plugins/prelude/Makefile audisp/plugins/remote/Makefile bindings/Makefile bindings/python/Makefile tools/Makefile tools/aulast/Makefile tools/aulastlog/Makefile tools/ausyscall/Makefile)
 
 echo .
 echo "
diff -Nur audit-1.7.9-old/Makefile.am audit-1.7.9/Makefile.am
--- audit-1.7.9-old/Makefile.am	2008-11-14 15:14:24.000000000 +0200
+++ audit-1.7.9/Makefile.am	2008-11-14 15:14:39.000000000 +0200
@@ -22,7 +22,7 @@
 #
 
 SUBDIRS = lib auparse src/mt src/libev src audisp tools swig bindings init.d \
-	docs system-config-audit
+	docs
 EXTRA_DIST = ChangeLog AUTHORS NEWS README INSTALL audit.spec \
 	contrib/capp.rules contrib/nispom.rules contrib/lspp.rules \
 	contrib/stig.rules contrib/skeleton.c contrib/avc_snap \
@@ -31,5 +31,5 @@
 CONFIG_CLEAN_FILES = debug*.list config/*
 
 clean-generic:
-	rm -rf autom4te*.cache zos-remote-policy
+	rm -rf autom4te*.cache
 	rm -f *.rej *.orig *.lang
