diff -ur gtk+-2.22.1.orig//gdk/Makefile.am gtk+-2.22.1/gdk/Makefile.am
--- gtk+-2.22.1.orig//gdk/Makefile.am	2010-11-15 18:49:19.000000000 +0200
+++ gtk+-2.22.1/gdk/Makefile.am	2010-11-25 00:52:12.804000025 +0200
@@ -198,7 +198,7 @@
         gdkenumtypes.h
 
 Gdk-2.0.gir: $(gdktargetlib) Makefile
-Gdk_2_0_gir_SCANNERFLAGS = --warn-all
+Gdk_2_0_gir_SCANNERFLAGS = --warn-all --strip-prefix=Gdk
 Gdk_2_0_gir_INCLUDES = Gio-2.0 GdkPixbuf-2.0 Pango-1.0 cairo-1.0
 Gdk_2_0_gir_LIBS = $(gdktargetlib)
 Gdk_2_0_gir_FILES = $(introspection_files)
@@ -246,8 +246,6 @@
 GdkX11_2_0_gir_CFLAGS = $(INCLUDES) -I$(top_srcdir)/gdk -L$(top_builddir)/gdk
 INTROSPECTION_GIRS += GdkX11-2.0.gir
 
-introspection_files += $(filter-out x11/gdkx.h, $(x11_introspection_files))
-
 endif # USE_X11
 
 girdir = $(datadir)/gir-1.0
