--- modules/printbackends/cups/gtkprintbackendcups.c	2009-10-28 16:35:06.000000000 +0100
+++ modules/printbackends/cups/gtkprintbackendcups.c	2009-10-28 16:35:45.000000000 +0100
@@ -1405,7 +1405,6 @@ cups_request_printer_list_cb (GtkPrintBa
           "media-low",
           "media-empty",
           "offline",
-          "connecting-to-device",
           "other"
         };
       static const char * reasons_descs[] =
@@ -1425,7 +1424,6 @@ cups_request_printer_list_cb (GtkPrintBa
           N_("Printer '%s' is low on paper."),
           N_("Printer '%s' is out of paper."),
           N_("Printer '%s' is currently off-line."),
-          N_("Printer '%s' may not be connected."),
           N_("There is a problem on printer '%s'.")
         };
       gboolean is_paused = FALSE;
