Index: kdemultimedia-4.5.5/kscd/mbmanager.cpp
===================================================================
--- kdemultimedia-4.5.5.orig/kscd/mbmanager.cpp
+++ kdemultimedia-4.5.5/kscd/mbmanager.cpp
@@ -112,22 +112,21 @@ void MBManager::discLookup()
 		m_validInfo = false;
 	}
 
-	// TODO if multiple entries found
-	if (MB.DataInt(MBE_GetNumAlbums) > 1)
-	{
-		kDebug() << MB.DataInt(MBE_GetNumAlbums) << " entries found";
-	}
-
-	// TODO manage multiple entries
-	// Select the first album
-	MB.Select(MBS_SelectAlbum, 1);
-
-	// Get the number of tracks
-	numTracks = MB.DataInt(MBE_AlbumGetNumTracks);
-	//kDebug() << "NumTracks: " << numTracks << endl;
-
 	if (m_validInfo == true)
 	{
+		// TODO if multiple entries found
+		if (MB.DataInt(MBE_GetNumAlbums) > 1)
+		{
+			kDebug() << MB.DataInt(MBE_GetNumAlbums) << " entries found";
+		}
+		// TODO manage multiple entries
+		// Select the first album
+		MB.Select(MBS_SelectAlbum, 1);
+
+		// Get the number of tracks
+		numTracks = MB.DataInt(MBE_AlbumGetNumTracks);
+		//kDebug() << "NumTracks: " << numTracks << endl;
+
 		// Sets info
 		m_discInfo.Title = MB.Data(MBE_AlbumGetAlbumName).c_str();
 		m_discInfo.Artist = MB.Data(MBE_AlbumGetAlbumArtistName).c_str();
@@ -145,6 +144,14 @@ void MBManager::discLookup()
 	}
 	else
 	{
+		ret = MB.Query(string(MBQ_GetCDTOC));
+		if (!ret)
+		{
+			MB.GetQueryError(error);
+			printf("TOC query failed: %s\n", error.c_str());
+		}
+		numTracks = MB.DataInt(MBE_TOCGetLastTrack);
+
 		// If invalid data, fill the information with something
 		// Sets info
 		m_discInfo.Title = i18n("Unknown album");
@@ -154,10 +161,12 @@ void MBManager::discLookup()
 		MBTrackInfo track;
 		for(int i = 1; i <= numTracks; i++)
 		{
-
 			track.Title = i18n("Unknown title");
 			track.Artist = m_discInfo.Artist;
-			track.Duration = MB.Data(MBE_AlbumGetTrackDuration, i).c_str();
+
+			int duration = MB.DataInt(MBE_TOCGetTrackNumSectors, i + 1);
+			duration = duration * 2352 / (44100 / (8 * 16 * 2));
+			track.Duration = duration;
 
 			m_trackList << track;
 		}
