Index: kdemultimedia/kmix/apps/kmix.cpp
===================================================================
--- kdemultimedia/kmix/apps/kmix.cpp	(revision 1224425)
+++ kdemultimedia/kmix/apps/kmix.cpp	(revision 1224426)
@@ -95,6 +95,11 @@
     initActionsAfterInitMixer(); // init actions that require initialized mixer backend(s).
 
     recreateGUI(false);
+    if ( m_wsMixers->count()  < 1 )
+    {
+        // Something is wrong. Perhaps a hardware or driver or backend change. Let KMix search harder
+        recreateGUI(false, QString(), true);
+    }
 
     if ( !kapp->isSessionRestored() ) // done by the session manager otherwise
         setInitialSize();
@@ -528,7 +533,7 @@
         }
         // No TAB YET => This should mean KMix is just started, or the user has just plugged in a card
         bool profileListHasKey = pconfig.hasKey( mixer->id() ); // <<< SHOULD be before the following line
-        QStringList profileList = pconfig.readEntry( mixer->id(), QStringList() ); // Hint: Default is a list with ONE entry (an empty string). Important for GUIProfile::find()
+        QStringList profileList = pconfig.readEntry( mixer->id(), QStringList() );
 
         bool aProfileWasAddedSucesufully = false;
         foreach ( QString profileId, profileList)
@@ -547,7 +552,7 @@
 
         // The we_need_a_fallback case is a bit tricky. Please ask the author (cesken) before even considering to change the code.
         bool we_need_a_fallback = !aProfileWasAddedSucesufully;  // we *possibly* want a fallback, if we couldn't add one
-        bool thisMixerShouldBeForced = forceNewTab && ( mixer->id() == mixerId  );
+        bool thisMixerShouldBeForced = forceNewTab && ( mixerId.isEmpty() || (mixer->id() == mixerId) );
         we_need_a_fallback = we_need_a_fallback && ( thisMixerShouldBeForced || !profileListHasKey ); // Additional requirement: "forced-tab-for-this-mixer" OR "no key stored in kmixrc yet"
         if ( we_need_a_fallback )
         {
