Index: kdemultimedia-4.5.5/libkcddb/cache.cpp
===================================================================
--- kdemultimedia-4.5.5.orig/libkcddb/cache.cpp
+++ kdemultimedia-4.5.5/libkcddb/cache.cpp
@@ -112,7 +112,10 @@ namespace KCDDB
       newInfo.set("discid", id);
     }
 
-    cacheDir = c.cacheLocations().first() + cacheDir;
+    if (!c.cacheLocations().isEmpty())
+        cacheDir = c.cacheLocations().first() + cacheDir;
+    else
+        return;
 
     QDir dir;
 
