diff -up kdemultimedia-4.5.80/CMakeLists.txt.nomplayerthumbs kdemultimedia-4.5.80/CMakeLists.txt
--- kdemultimedia-4.5.80/CMakeLists.txt.nomplayerthumbs	2010-11-11 08:42:16.000000000 -0600
+++ kdemultimedia-4.5.80/CMakeLists.txt	2010-11-21 17:57:05.152363749 -0600
@@ -42,7 +42,7 @@ configure_file (config.h.cmake ${CMAKE_C
 
 add_subdirectory(libkcddb)
 add_subdirectory(libkcompactdisc)
-macro_optional_add_subdirectory(mplayerthumbs)
+#macro_optional_add_subdirectory(mplayerthumbs)
 
 if (FFMPEG_FOUND AND SWSCALE_FOUND)
      macro_optional_add_subdirectory(ffmpegthumbs)
