diff -up kdemultimedia-4.2.85/CMakeLists.txt.nomplayerthumbs kdemultimedia-4.2.85/CMakeLists.txt
--- kdemultimedia-4.2.85/CMakeLists.txt.nomplayerthumbs	2009-05-06 19:04:42.000000000 -0500
+++ kdemultimedia-4.2.85/CMakeLists.txt	2009-05-17 10:23:16.895948032 -0500
@@ -39,7 +39,7 @@ add_subdirectory(cmake)
 macro_optional_add_subdirectory(doc)
 add_subdirectory(libkcddb)
 add_subdirectory(libkcompactdisc)
-add_subdirectory(mplayerthumbs)
+#add_subdirectory(mplayerthumbs)
 
 if (TAGLIB_FOUND)
   macro_optional_add_subdirectory(juk)
