Index: kdemultimedia-4.6.1/kmix/gui/kmixdockwidget.cpp
===================================================================
--- kdemultimedia-4.6.1.orig/kmix/gui/kmixdockwidget.cpp
+++ kdemultimedia-4.6.1/kmix/gui/kmixdockwidget.cpp
@@ -33,10 +33,13 @@
 #include <kwindowsystem.h>
 #include <kactioncollection.h>
 #include <ktoggleaction.h>
+#include <kprocess.h>
+#include <kmessagebox.h>
 #include <qapplication.h>
 #include <qcursor.h>
 #include <QDesktopWidget>
 #include <QMouseEvent>
+#include <QFile>
 
 #ifdef Q_WS_X11
 #include <fixx11h.h>
@@ -122,6 +125,16 @@ void KMixDockWidget::createActions()
     menu->addAction( action );
 }
 
+  if (QFile::exists("/usr/bin/pavucontrol"))
+{
+  // Add pavucontrol shortcut to the context menu
+  QAction *paction = actionCollection()->addAction( "run_pavucontrol" );
+  paction->setText( i18n ("Advanced Sound Control") );
+  connect(paction, SIGNAL(triggered(bool) ), SLOT (runPavucontrol() ));
+  QAction *a1 = actionCollection()->action ("run_pavucontrol" );
+  if ( a1 ) menu->addAction( a1 );
+}
+
   // Put "Select Master Channel" dialog in context menu
   if ( m_mixer != 0 ) {
   QAction *action = actionCollection()->addAction( "select_master" );
@@ -181,6 +194,17 @@ void KMixDockWidget::selectMaster()
     // !! The dialog is modal. Does it delete itself?
 }
 
+void KMixDockWidget::runPavucontrol()
+{
+    KProcess *pulse_proc = new KProcess();
+    pulse_proc->setProgram("/usr/bin/pavucontrol");
+
+    if ( pulse_proc->startDetached() == 0 ) {
+        // Error
+        KMessageBox::error(0, i18n("Failed launching pavucontrol, make sure the pavucontrol package is correctly installed."),
+                              i18n("Error launching pavucontrol"));
+    }
+}
 
 void KMixDockWidget::handleNewMaster(QString& /*mixerID*/, QString& /*control_id*/)
 {
Index: kdemultimedia-4.6.1/kmix/gui/kmixdockwidget.h
===================================================================
--- kdemultimedia-4.6.1.orig/kmix/gui/kmixdockwidget.h
+++ kdemultimedia-4.6.1/kmix/gui/kmixdockwidget.h
@@ -87,6 +87,7 @@ class KMixDockWidget : public KStatusNot
    void trayWheelEvent(int delta);
    //void kmixSystrayAction(QSystemTrayIcon::ActivationReason);
    void selectMaster();
+   void runPavucontrol();
    void handleNewMaster(QString& soundcard_id, QString& channel_id);
    void contextMenuAboutToShow();
    void activateMenuOrWindow(bool, const QPoint &);
