diff -Naur kscd2/kscd.cpp kscd/kscd.cpp
--- kscd2/kscd.cpp	2009-08-14 10:44:25.161142971 +0300
+++ kscd/kscd.cpp	2009-08-14 10:46:04.082012455 +0300
@@ -845,13 +845,13 @@
             fprintf(stderr, "kscd is already running\n");
             if (args->count() > 0 || args->isSet("start"))
             {
-                QDBusInterface kscd("org.kde.kscd", "/CDPlayer", "org.kde.kscd.CDPlayer");
+                QDBusInterface kscd("org.kde.kscd", "/Player", "org.freedesktop.MediaPlayer");
                 if(kscd.isValid())
                 {
                     // Forward the command line args to the running instance.
                     if (args->isSet("start"))
                     {
-                        kscd.call("play");
+                        kscd.call("Play");
                     }
                 }
                 args->clear();
