diff -Naur kscd-orig/kscd-play-audiocd.desktop kscd/kscd-play-audiocd.desktop
--- kscd-orig/kscd-play-audiocd.desktop	2009-08-14 10:50:14.619012172 +0300
+++ kscd/kscd-play-audiocd.desktop	2009-08-14 10:51:42.740012228 +0300
@@ -2,6 +2,7 @@
 X-KDE-Solid-Predicate=[ StorageVolume.ignored == false AND OpticalDisc.availableContent == 'Audio' ]
 Type=Service
 Actions=open;
+InitialPreference=15
 
 [Desktop Action open]
 Exec=kscd -s
