Index: libuser-0.56.17/Makefile.am
===================================================================
--- libuser-0.56.17.orig/Makefile.am
+++ libuser-0.56.17/Makefile.am
@@ -178,7 +178,7 @@ modules_libuser_sasldb_la_LIBADD = lib/l
 python_libusermodule_la_SOURCES = python/libusermodule.c python/common.h \
 	python/debug.h
 python_libusermodule_la_CPPFLAGS = $(AM_CPPFLAGS) $(PYTHON_CPPFLAGS)
-python_libusermodule_la_LDFLAGS = -module -avoid-version $(GOBJECT_LIBS)
+python_libusermodule_la_LDFLAGS = -module -avoid-version $(GOBJECT_LIBS) -lpython$(PYTHON_VERSION)
 python_libusermodule_la_LIBADD = apps/libapputil.la lib/libuser.la
 
 samples_enum_CPPFLAGS = $(AM_CPPFLAGS) $(LOCALEDIR_CPPFLAGS)
