Index: cups-1.4.6/desktop/cups.desktop.in
===================================================================
--- cups-1.4.6.orig/desktop/cups.desktop.in
+++ cups-1.4.6/desktop/cups.desktop.in
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Categories=System;Printing;HardwareSettings;X-Red-Hat-Base;
+Categories=Settings;HardwareSettings;System;Printing;
 Exec=@CUPS_HTMLVIEW@ http://localhost:631/
 Icon=cups
 StartupNotify=false
