Index: cups-1.3.11/Makefile
===================================================================
--- cups-1.3.11.orig/Makefile
+++ cups-1.3.11/Makefile
@@ -167,11 +167,6 @@ install:	installhdrs
 		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(XINETD); \
 		$(INSTALL_DATA) init/cups-lpd $(BUILDROOT)$(XINETD)/cups-lpd; \
 	fi
-	if test "x$(MENUDIR)" != x; then \
-		echo Installing desktop menu...; \
-		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(MENUDIR); \
-		$(INSTALL_DATA) desktop/cups.desktop $(BUILDROOT)$(MENUDIR); \
-	fi
 	if test "x$(ICONDIR)" != x; then \
 		echo Installing desktop icons...; \
 		$(INSTALL_DIR) -m 755 $(BUILDROOT)$(ICONDIR)/hicolor/16x16/apps; \
