Index: kiwi-1.9.26/setup.py
===================================================================
--- kiwi-1.9.26.orig/setup.py
+++ kiwi-1.9.26/setup.py
@@ -57,12 +57,8 @@ setup(name="kiwi",
                              'kiwiwidgets', '*.png')),
                   (get_site_packages_dir('gazpacho', 'widgets'),
                    listfiles('gazpacho-plugin', 'kiwiwidgets.py')),
-                  ('share/doc/kiwi',
+                  ('share/doc/python-kiwi',
                    ('AUTHORS', 'ChangeLog', 'NEWS', 'README')),
-                  ('share/doc/kiwi/howto',
-                   listfiles('doc/howto/', '*')),
-                  ('share/doc/kiwi/api',
-                   listfiles('doc/api/', '*')),
                   ],
       scripts=['bin/kiwi-i18n',
                'bin/kiwi-ui-test'],
