--- ipheth-pair/Makefile.orig	2010-01-01 20:22:26.000000000 +0200
+++ ipheth-pair/Makefile	2010-01-11 18:00:36.793250548 +0200
@@ -6,9 +6,8 @@
 ipheth-pair: ipheth-pair.o
 
 install:
-	install -m 0755 ipheth-pair /lib/udev/
-	install -m 0644 90-iphone-tether.rules /etc/udev/rules.d
-	/sbin/udevadm control --reload-rules
+	install -m 0755 ipheth-pair /usr/bin
+	install -m 0644 90-iphone-tether.rules /lib/udev/rules.d
 
 clean:
 	rm -f *.o ipheth-pair
