--- ipheth-driver/Makefile.orig	2010-01-01 20:22:26.000000000 +0200
+++ ipheth-driver/Makefile	2010-01-11 17:49:06.572251045 +0200
@@ -1,12 +1,10 @@
 obj-m += ipheth.o
+KVERSION = $(shell uname -r)
 
 all:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
-
-clean:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
+	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD)/ipheth-driver modules
 
 install:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules_install
+	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD)/ipheth-driver modules_install
 
 
