--- ipheth-driver/ipheth-modprobe.conf.orig	2010-01-01 20:22:26.000000000 +0200
+++ ipheth-driver/ipheth-modprobe.conf	2010-01-01 23:20:48.743705881 +0200
@@ -3,5 +3,5 @@
 # This configuration forces modprobe to execute device pairing script after
 # loading the driver.
 
-install ipheth /sbin/modprobe --ignore-install ipheth && /usr/local/bin/ipheth-pair.py
+install ipheth /sbin/modprobe --ignore-install ipheth && /usr/bin/ipheth-pair.py
 
