--- ipheth-pair/90-iphone-tether.rules.orig	2010-01-01 20:22:26.619055733 +0200
+++ ipheth-pair/90-iphone-tether.rules	2010-01-01 21:16:00.981035308 +0200
@@ -2,6 +2,6 @@
 ATTR{idVendor}!="05ac", GOTO="ipheth_rules_end"
 
 # Execute pairing program when appropriate
-ACTION=="add", SUBSYSTEM=="net", ENV{ID_USB_DRIVER}=="ipheth", SYMLINK+="iphone", RUN+="ipheth-pair"
+ACTION=="add", SUBSYSTEM=="net", ENV{ID_USB_DRIVER}=="ipheth", SYMLINK+="iphone", RUN+="/usr/bin/ipheth-pair"
 
 LABEL="ipheth_rules_end"
