diff -ur gst-python-0.10.15.orig/gst/Makefile.am gst-python-0.10.15/gst/Makefile.am
--- gst-python-0.10.15.orig/gst/Makefile.am	2009-04-21 23:07:24.000000000 +0300
+++ gst-python-0.10.15/gst/Makefile.am	2009-06-30 02:40:22.353845730 +0300
@@ -1,6 +1,6 @@
 common_cflags = $(PYGOBJECT_CFLAGS) $(GST_CFLAGS) $(GST_OPTION_CFLAGS) -fno-strict-aliasing
 common_libadd = $(GST_LIBS) $(GST_OPTION_LIBS)
-common_ldflags = -module -avoid-version
+common_ldflags = -module -avoid-version `python-config --libs`
 
 pkgpyexecdir = $(pyexecdir)/gst-$(GST_MAJORMINOR)/gst
 
diff -ur gst-python-0.10.15.orig/Makefile.am gst-python-0.10.15/Makefile.am
--- gst-python-0.10.15.orig/Makefile.am	2009-04-21 23:07:24.000000000 +0300
+++ gst-python-0.10.15/Makefile.am	2009-06-30 02:40:22.342847051 +0300
@@ -2,7 +2,7 @@
 
 common_cflags = $(PYTHON_INCLUDES) $(PYGOBJECT_CFLAGS) $(GST_CFLAGS) $(GST_OPTION_CFLAGS) -fno-strict-aliasing
 common_libadd = $(GST_LIBS) $(GST_OPTION_LIBS)
-common_ldflags = -module -avoid-version
+common_ldflags = -module -avoid-version `python-config --libs`
 
 # include before EXTRA_DIST for win32 assignment
 include $(top_srcdir)/common/win32.mak
