diff -rup virt-manager-0.8.1/src/virtManager/uihelpers.py new/src/virtManager/uihelpers.py
--- virt-manager-0.8.1/src/virtManager/uihelpers.py	2009-12-03 16:15:01.000000000 -0500
+++ new/src/virtManager/uihelpers.py	2009-12-03 17:15:09.101994000 -0500
@@ -38,7 +38,7 @@ OPTICAL_MEDIA_KEY = 4
 # may use a nonroot user, so simply changing this will cause several UI
 # pieces to attempt to verify that permissions are correct. Eventually this
 # should be exposed via capabilities so we can determine this programmatically.
-QEMU_SYSTEM_EMULATOR_USER = "root"
+QEMU_SYSTEM_EMULATOR_USER = "qemu"
 
 ##############################################################
 # Initialize an error object to use for validation functions #
