diff -up openbabel-2.3.0/scripts/CMakeLists.txt.r openbabel-2.3.0/scripts/CMakeLists.txt
--- openbabel-2.3.0/scripts/CMakeLists.txt.r	2010-10-25 19:37:42.000000000 +0200
+++ openbabel-2.3.0/scripts/CMakeLists.txt	2010-11-24 22:53:29.250004683 +0100
@@ -82,6 +82,7 @@ if (DO_PYTHON_BINDINGS)
             COMMAND ${CMAKE_COMMAND} -E copy ob.py ${openbabel_SOURCE_DIR}/scripts/python/openbabel.py
             COMMAND ${CMAKE_COMMAND} -E remove ob.py
             VERBATIM)
+        add_dependencies(_openbabel openbabel)
 
         if (PYTHON_PREFIX)
             install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${openbabel_SOURCE_DIR}/scripts/python/setup.py install --prefix=${PYTHON_PREFIX} WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts)")
@@ -89,7 +90,7 @@ if (DO_PYTHON_BINDINGS)
             if (${CMAKE_INSTALL_PREFIX} STREQUAL "/usr/local")
                 install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${openbabel_SOURCE_DIR}/scripts/python/setup.py install WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts)")
             else (${CMAKE_INSTALL_PREFIX} STREQUAL "/usr/local")
-                install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${openbabel_SOURCE_DIR}/scripts/python/setup.py install --prefix=${CMAKE_INSTALL_PREFIX} WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts)")
+                install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${openbabel_SOURCE_DIR}/scripts/python/setup.py install --prefix=$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX} WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts)")
             endif (${CMAKE_INSTALL_PREFIX} STREQUAL "/usr/local")
         endif (PYTHON_PREFIX)
 
@@ -228,8 +229,8 @@ if (DO_PERL_BINDINGS)
             DEPENDS ${openbabel_SOURCE_DIR}/scripts/perl/openbabel-perl.cpp
             WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/perl
         )
-      add_dependencies(bindings_perl pre_bindings_perl)
-      install(CODE "execute_process(COMMAND make install WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/perl)")
+      add_dependencies(bindings_perl pre_bindings_perl openbabel)
+      install(CODE "execute_process(COMMAND make install_vendor WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/perl)")
     
     else (NOT MSVC)
       add_library(bindings_perl MODULE ${openbabel_SOURCE_DIR}/scripts/perl/openbabel-perl.cpp)
@@ -282,20 +283,20 @@ if (DO_RUBY_BINDINGS)
         COMMAND ${CMAKE_COMMAND} -E make_directory ${openbabel_BINARY_DIR}/scripts/ruby
         )
       add_custom_target(bindings_ruby ALL
-        COMMAND ${RUBY_EXECUTABLE} ${openbabel_SOURCE_DIR}/scripts/ruby/extconf.rb --with-openbabel-lib=${BABEL_LIBRARY} --with-openbabel-include=${openbabel_BINARY_DIR}/include
+        COMMAND ${RUBY_EXECUTABLE} ${openbabel_SOURCE_DIR}/scripts/ruby/extconf.rb --with-openbabel-lib=${openbabel_BINARY_DIR}/lib${LIB_SUFFIX} --with-openbabel-include=${openbabel_BINARY_DIR}/include
         COMMAND ${CMAKE_COMMAND} -E echo "Compiling Ruby bindings"
         COMMAND make
         DEPENDS ${openbabel_SOURCE_DIR}/scripts/ruby/openbabel-ruby.cpp
         WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/ruby
         )
-        add_dependencies(bindings_ruby pre_bindings_ruby)
+      add_dependencies(bindings_ruby pre_bindings_ruby openbabel)
       if (RUBY_PREFIX)
         install(CODE "execute_process(COMMAND DESTDIR=${RUBY_PREFIX} make install WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/ruby)")
       else (RUBY_PREFIX)
         if (${CMAKE_INSTALL_PREFIX} STREQUAL "/usr/local")
           install(CODE "execute_process(COMMAND make install WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/ruby)")
         else (${CMAKE_INSTALL_PREFIX} STREQUAL "/usr/local")
-          install(CODE "execute_process(COMMAND DESTDIR=${CMAKE_INSTALL_PREFIX} make install WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/ruby)")
+          install(CODE "execute_process(COMMAND make DESTDIR=$ENV{DESTDIR} install WORKING_DIRECTORY ${openbabel_BINARY_DIR}/scripts/ruby)")
         endif (${CMAKE_INSTALL_PREFIX} STREQUAL "/usr/local")
       endif (RUBY_PREFIX)
     else(NOT MSVC)
diff -up openbabel-2.3.0/scripts/perl/Makefile.PL.r openbabel-2.3.0/scripts/perl/Makefile.PL
--- openbabel-2.3.0/scripts/perl/Makefile.PL.r	2010-10-23 03:44:58.000000000 +0200
+++ openbabel-2.3.0/scripts/perl/Makefile.PL	2010-11-24 22:53:46.417249628 +0100
@@ -12,6 +12,9 @@ $cmakebindir = @ENV{"OBJ_DIR"};
 if (-e "$cmakebindir/lib/libopenbabel.so" || -e "$cmakebindir/lib/libopenbabel.dylib") {
   check_lib_or_exit( lib => 'openbabel' , libpath => "$cmakebindir/lib");
 }
+elsif (-e "$cmakebindir/lib64/libopenbabel.so") {
+  check_lib_or_exit( lib => 'openbabel' , libpath => "$cmakebindir/lib64");
+}
 else {
   check_lib_or_exit( lib => 'openbabel' , libpath => @ENV{"LD_LIBRARY_PATH"});
 }
@@ -21,6 +24,8 @@ $ldfrom = "\$(OBJECT) -L$cmakebindir/lib
     if (-r "$cmakebindir/lib/libopenbabel.dylib") and (-s _) and (-B _);
 $ldfrom = "\$(OBJECT) -L$cmakebindir/lib -lopenbabel"
     if (-r "$cmakebindir/lib/libopenbabel.so") and (-s _) and (-B _);
+$ldfrom = "\$(OBJECT) -L$cmakebindir/lib64 -lopenbabel"
+    if (-r "$cmakebindir/lib64/libopenbabel.so") and (-s _) and (-B _);
 
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
diff -up openbabel-2.3.0/scripts/python/setup.py.r openbabel-2.3.0/scripts/python/setup.py
--- openbabel-2.3.0/scripts/python/setup.py.r	2010-10-26 16:39:01.000000000 +0200
+++ openbabel-2.3.0/scripts/python/setup.py	2010-11-23 01:56:55.083627521 +0100
@@ -17,7 +17,9 @@ obExtension = Extension('_openbabel',
                  [os.path.join(srcdir, "openbabel-python.cpp")],
                  include_dirs=[os.path.join(srcdir, "..", "..", "include"),
                                os.path.join("..", "include")],
-                 library_dirs=[os.path.join("..", "lib")],
+                 library_dirs=[os.path.join(srcdir, "..", "..", "lib"),
+                               os.path.join(srcdir, "..", "..", "lib64"),
+                               os.path.join("..", "lib")],
                  libraries=['openbabel']
                  )
 
