--- scripts/python/setup.py.orig	2011-01-31 12:12:03.457000069 +0200
+++ scripts/python/setup.py	2011-01-31 12:12:13.072000073 +0200
@@ -25,7 +25,7 @@
 
 if "build" in sys.argv:
     shutil.copyfile(os.path.join(srcdir, "pybel_py%dx.py" % sys.version_info[0]), "pybel.py")
-    shutil.copyfile(os.path.join(srcdir, "openbabel.py"), "openbabel.py")
+    #shutil.copyfile(os.path.join(srcdir, "openbabel.py"), "openbabel.py")
 
 setup(name='openbabel',
       version='1.6',
