diff --git a/src/LexerStorage.cpp b/src/LexerStorage.cpp
index ef342b0..77adcac 100644
--- a/src/LexerStorage.cpp
+++ b/src/LexerStorage.cpp
@@ -553,7 +553,7 @@ void LSInterior::applyCustomStyle(const QString& name, const QFont& font) {
 			Log::debug(QString("Lexer is not 'none'"));
 			
 			if ( schemes_.contains(name) ) {
-				Log::debug(QString("Found sceme").arg(name));
+				Log::debug(QString("Found scheme %1").arg(name));
 				Scheme* scheme = schemes_[name];
 			
 				QFont f(font);
