diff --git a/juffed.desktop b/juffed.desktop
index cc25b99..1454e6b 100644
--- a/juffed.desktop
+++ b/juffed.desktop
@@ -5,12 +5,15 @@ Name[pl]=Edytor tekstu JuffEd
 Name[de]=Texteditor JuffEd
 Name[fr]=Éditeur de texte JuffEd
 Name[pt]=Editor de texto JuffEd
+Name[tr]=Metin editörü JuffEd
 Name[zh_CN]=文本编辑器 JuffEd
 Comment=Advanced text editor
 Comment[ru]=Продвинутый текстовый редактор
 Comment[pl]=Zaawansowany edytor tekstu
 Comment[de]=Ein erweiterter Texteditor
 Comment[fr]=Un avancé éditeur de texte
+Comment[pt]=Editor de texto avançado
+Comment[tr]=Ileri metin editörü
 Comment[zh_CN]=先进的文本编辑器
 Exec=juffed
 Terminal=false
@@ -24,6 +27,7 @@ GenericName[ru]=Продвинутый текстовый редактор
 GenericName[pl]=Zaawansowany edytor tekstu
 GenericName[de]=Ein erweiterter Texteditor
 GenericName[fr]=Un avancé éditeur de texte
-#GenericName[pt]=Editor de texto simples
+GenericName[pt]=Editor de texto avançado
+GenericName[tr]=Ileri metin editörü
 GenericName[zh_CN]=先进的文本编辑器
 X-SuSE-translate=true
