diff --git a/src/LexerStorage.cpp b/src/LexerStorage.cpp
index 31a4751..ad32f20 100644
--- a/src/LexerStorage.cpp
+++ b/src/LexerStorage.cpp
@@ -498,12 +498,11 @@ void LSInterior::readCustomStyle(const QString& name) {
 		scheme = new Scheme();
 		scheme->defaultStyle = styles["default"];
 		scheme->rules 
-				<< Rule(styles["comment"], QList<int>() << QsciLexerPascal::Comment << QsciLexerPascal::CommentLine)
-//				<< Rule(styles["commentdoc"], QList<int>() << QsciLexerPascal::CommentDoc)
+				<< Rule(styles["comment"], QList<int>() << QsciLexerPascal::Comment << QsciLexerPascal::CommentLine << QsciLexerPascal::CommentParenthesis)
 				<< Rule(styles["number"], QList<int>() << QsciLexerPascal::Number)
 				<< Rule(styles["keyword"], QList<int>() << QsciLexerPascal::Keyword)
 				<< Rule(styles["string"], QList<int>() << QsciLexerPascal::SingleQuotedString)
-				<< Rule(styles["preprocessor"], QList<int>() << QsciLexerPascal::PreProcessor)
+				<< Rule(styles["preprocessor"], QList<int>() << QsciLexerPascal::PreProcessor << QsciLexerPascal::PreProcessorParenthesis)
 				<< Rule(styles["operator"], QList<int>() << QsciLexerPascal::Operator)
 				<< Rule(styles["identifier"], QList<int>() << QsciLexerPascal::Identifier)
 				<< Rule(styles["asm"], QList<int>() << QsciLexerPascal::Asm);
