diff --git a/src/gui/SettingsDlg.cpp b/src/gui/SettingsDlg.cpp
index 3824259..6d3aec9 100644
--- a/src/gui/SettingsDlg.cpp
+++ b/src/gui/SettingsDlg.cpp
@@ -284,6 +284,7 @@ SettingsDlg::~SettingsDlg() {
 void SettingsDlg::init() {
 	JUFFENTRY;
 	
+	if (SettingsItem::notifier() != 0)
 	connect(SettingsItem::notifier(), SIGNAL(hasChangedItems(bool)), SLOT(somethingChanged(bool)));
 
 	pageMain_->init(items_);
