diff -up ntfsprogs-2.0.0/ntfsprogs/ntfsundelete.c.BAD ntfsprogs-2.0.0/ntfsprogs/ntfsundelete.c
--- ntfsprogs-2.0.0/ntfsprogs/ntfsundelete.c.BAD	2008-09-12 12:19:02.000000000 -0400
+++ ntfsprogs-2.0.0/ntfsprogs/ntfsundelete.c	2008-09-12 12:19:28.000000000 -0400
@@ -1779,7 +1779,7 @@ static int undelete_file(ntfs_volume *vo
 							"%lld clusters.\n",
 							(long long)rl[i].length);
 					memset(buffer, opts.fillbyte, bufsize);
-					for (k = 0; k < rl[k].length * vol->cluster_size; k += bufsize) {
+					for (k = 0; k < rl[i].length * vol->cluster_size; k += bufsize) {
 						if (write_data(fd, buffer, bufsize) < bufsize) {
 							ntfs_log_perror("Write failed");
 							close(fd);
