diff -Naur kdepim-4.1.80/libkleo/CMakeLists.txt kdepim-4.1.80.libqgpgme-link-fix/libkleo/CMakeLists.txt
--- kdepim-4.1.80/libkleo/CMakeLists.txt	2008-11-04 18:12:56.000000000 +0100
+++ kdepim-4.1.80.libqgpgme-link-fix/libkleo/CMakeLists.txt	2008-11-28 18:56:14.000000000 +0100
@@ -108,7 +108,7 @@
 if(WIN32)
 target_link_libraries(kleo ${kleo_LIB_LIBS}  ${GPGME_VANILLA_LIBRARIES} )
 else(WIN32)
-target_link_libraries(kleo ${kleo_LIB_LIBS} )
+target_link_libraries(kleo ${kleo_LIB_LIBS} ${QGPGME_LIBRARIES} )
 endif(WIN32)
 set_target_properties(kleo PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS kleo ${INSTALL_TARGETS_DEFAULT_ARGS})
