Index: alsa-driver/alsa-kernel/usb/card.c
===================================================================
--- alsa-driver.orig/alsa-kernel/usb/card.c
+++ alsa-driver/alsa-kernel/usb/card.c
@@ -72,7 +72,7 @@ MODULE_LICENSE("GPL");
 MODULE_SUPPORTED_DEVICE("{{Generic,USB Audio}}");
 
 
-static int index[SNDRV_CARDS] = SNDRV_DEFAULT_IDX;	/* Index 0-MAX */
+static int index[SNDRV_CARDS] = {[0 ... (SNDRV_CARDS-1)] = -2};	/* exclude the 1st slot */
 static char *id[SNDRV_CARDS] = SNDRV_DEFAULT_STR;	/* ID for this card */
 static int enable[SNDRV_CARDS] = SNDRV_DEFAULT_ENABLE_PNP;/* Enable this card */
 /* Vendor/product IDs for this card */
