diff -up arora-0.10.2/src/webview.cpp.declare-qwebelement-metatype arora-0.10.2/src/webview.cpp
--- arora-0.10.2/src/webview.cpp.declare-qwebelement-metatype	2009-12-05 23:45:07.000000000 +0100
+++ arora-0.10.2/src/webview.cpp	2010-05-17 17:32:44.048050041 +0200
@@ -87,7 +87,9 @@
 #include <qwebframe.h>
 
 #if QT_VERSION >= 0x040600 || defined(WEBKIT_TRUNK)
+#if !defined(QTWEBKIT_VERSION) || QTWEBKIT_VERSION < 0x020000
 Q_DECLARE_METATYPE(QWebElement)
+#endif
 #include <qinputdialog.h>
 #include <qlabel.h>
 #include <qmessagebox.h>
