diff -Nuar R-2.8.1.old/src/library/tools/R/packageshtml.R R-2.8.1/src/library/tools/R/packageshtml.R
--- R-2.8.1.old/src/library/tools/R/packageshtml.R	2008-09-22 01:05:12.000000000 +0300
+++ R-2.8.1/src/library/tools/R/packageshtml.R	2009-02-15 14:27:44.000000000 +0200
@@ -36,7 +36,7 @@
                 paste(drive, lib, sep="") else lib
             lib0 <- paste("file:///", lib0, sep="")
         } else {
-            lib0 <- "../../library"
+            lib0 <- "../../../../usr/lib/R/library"
             libname <- "the standard library"
         }
         if(length(lib.loc) > 1)
@@ -104,7 +104,7 @@
             libname <- lib
             lib0 <- paste("file:///", lib, sep="")
         } else {
-            lib0 <- "../../library"
+            lib0 <- "../../../../usr/lib/R/library"
             libname <- "the standard library"
         }
         cat("<p><h3>Packages in ", libname,
