diff -up aMule-2.2.6/src/amule.cpp.new aMule-2.2.6/src/amule.cpp
--- aMule-2.2.6/src/amule.cpp.new	2009-08-18 18:24:58.000000000 +0200
+++ aMule-2.2.6/src/amule.cpp	2010-10-14 16:10:55.045018587 +0200
@@ -480,7 +481,7 @@ bool CamuleApp::OnInit()
 #endif
 
 	if ( !cmdline.Found(wxT("disable-fatal")) ) {
-#ifndef __WXMSW__
+#if wxUSE_ON_FATAL_EXCEPTION
 		// catch fatal exceptions
 		wxHandleFatalExceptions(true);
 #endif
diff -up aMule-2.2.6/src/utils/wxCas/src/wxcasframe.cpp.new aMule-2.2.6/src/utils/wxCas/src/wxcasframe.cpp
--- aMule-2.2.6/src/utils/wxCas/src/wxcasframe.cpp.new	2008-09-06 17:39:00.000000000 +0200
+++ aMule-2.2.6/src/utils/wxCas/src/wxcasframe.cpp	2010-10-14 15:22:20.874269526 +0200
@@ -289,7 +289,7 @@ WxCasFrame::GetStatImage () const
 #else
 
 	memdc.
-	SetFont ( wxFont::wxFont ( 8, wxSWISS, wxNORMAL, wxBOLD ) );
+	SetFont ( wxFont ( 8, wxSWISS, wxNORMAL, wxBOLD ) );
 #endif
 
 	memdc.
