Index: aMule-2.2.6/amule.desktop
===================================================================
--- aMule-2.2.6.orig/amule.desktop.orig
+++ aMule-2.2.6/amule.desktop
@@ -2,7 +2,8 @@
 Name=aMule
 Exec=amule
 Icon=amule
+Comment=Clone of eMule file sharing application
+Comment[tr]=eMule klonu dosya paylaşım programı
 Terminal=false
 Type=Application
 Categories=Network;P2P;
-Comment=A client for the eD2k network
Index: aMule-2.2.6/src/utils/aLinkCreator/alc.desktop
===================================================================
--- aMule-2.2.6.orig/src/utils/aLinkCreator/alc.desktop
+++ aMule-2.2.6/src/utils/aLinkCreator/alc.desktop
@@ -9,3 +9,4 @@ Comment=aMule ed2k link creator
 Comment[de]=ed2k-Linkersteller
 Comment[fr]=Créateur de liens ed2k d'aMule
 Comment[hu]=ed2k hivatkozás készítő
+Comment[tr]=aMule ed2k bağ oluşturucu
Index: aMule-2.2.6/src/utils/wxCas/wxcas.desktop
===================================================================
--- aMule-2.2.6.orig/src/utils/wxCas/wxcas.desktop
+++ aMule-2.2.6/src/utils/wxCas/wxcas.desktop
@@ -9,3 +9,4 @@ Comment=aMule online statistics
 Comment[de]=aMule-Onlinestatistik
 Comment[fr]=Statistiques d'aMule en ligne
 Comment[hu]=aMule online statisztikák
+Comment[tr]=aMule çevrimiçi istatistik
