Index: gpac/modules/ffmpeg_in/ffmpeg_decode.c
===================================================================
--- gpac.orig/modules/ffmpeg_in/ffmpeg_decode.c
+++ gpac/modules/ffmpeg_in/ffmpeg_decode.c
@@ -644,7 +644,7 @@ redecode:
         	                NULL, NULL, NULL);
 			
 			if (ffd->sws_ctx)
-				sws_scale(ffd->sws_ctx, ffd->frame->data, ffd->frame->linesize, 0, ffd->ctx->height->codec->height, pict.data, pict.linesize);
+				sws_scale(ffd->sws_ctx, ffd->frame->data, ffd->frame->linesize, 0, ffd->ctx->height, pict.data, pict.linesize);
 
 #endif
 
