Index: gpac/configure
===================================================================
--- gpac.orig/configure
+++ gpac/configure
@@ -1503,7 +1503,6 @@ fi
 if test "$cpu" = "x86" ; then
   echo "TARGET_ARCH_X86=yes" >> config.mak
 elif test "$cpu" = "armv4l" ; then
-echo "CONFIG_PULSEAUDIO=$has_pulseaudio" >> config.mak
   echo "TARGET_ARCH_ARMV4L=yes" >> config.mak
 elif test "$cpu" = "alpha" ; then
   echo "TARGET_ARCH_ALPHA=yes" >> config.mak
@@ -1588,6 +1587,7 @@ echo "FFMPEG_CFLAGS=$FFMPEG_CFLAGS" >> c
 echo "CONFIG_OSS_AUDIO=$has_oss_audio" >> config.mak
 echo "CONFIG_ALSA=$has_alsa" >> config.mak
 echo "CONFIG_JACK=$has_jack" >> config.mak
+echo "CONFIG_PULSEAUDIO=$has_pulseaudio" >> config.mak
 echo "CONFIG_A52=$has_a52" >> config.mak
 
 echo "DISABLE_SVG=$disable_svg" >> config.mak
