Index: work/gpac/src/compositor/visual_manager_3d_gl.c
===================================================================
--- work.orig/gpac/src/compositor/visual_manager_3d_gl.c
+++ work/gpac/src/compositor/visual_manager_3d_gl.c
@@ -70,6 +70,7 @@ void gf_sc_load_opengl_extensions(GF_Com
 #elif defined (WIN32)
 #define GET_GLFUN(__name) (PFNGLARBMULTITEXTUREPROC) wglGetProcAddress(__name) 
 #else
+extern void (*glXGetProcAddress(const GLubyte *procname))( void );
 #define GET_GLFUN(__name) (PFNGLARBMULTITEXTUREPROC) glXGetProcAddress(__name) 
 #endif
 
