diff -Nur gpac-old/Makefile gpac/Makefile
--- gpac-old/Makefile	2009-04-23 23:37:44.000000000 +0300
+++ gpac/Makefile	2009-04-23 23:38:07.000000000 +0300
@@ -68,7 +68,6 @@
 else
 	install $(INSTFLAGS) -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac-$(VERSION).$(DYN_LIB_SUFFIX)
 	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac.$(DYN_LIB_SUFFIX)
-	ldconfig || true
 endif
 endif
 	install -d "$(DESTDIR)$(mandir)"
