diff -Nur gpac-old/configure gpac/configure
--- gpac-old/configure	2010-02-25 22:07:40.639525995 +0200
+++ gpac/configure	2010-02-25 22:08:21.700496525 +0200
@@ -223,7 +223,7 @@
     if [ x"$canon_arch" = x"x86_64" -o x"$canon_arch" = x"amd64" ]; then
       if [ -z "`echo $CFLAGS | grep -- -m32`"  ]; then
         cpu="x86_64"
-	libdir="lib64"
+	libdir="lib"
         #that's a bit crude...
         PIC_CFLAGS="-fPIC -DPIC"
         want_pic="yes"
@@ -1686,7 +1686,7 @@
 fi
 
 if test "$is_64" = "yes"; then
-echo "X11_LIB_PATH=$X11_PATH/lib64" >> config.mak
+echo "X11_LIB_PATH=$X11_PATH/lib" >> config.mak
 else
 echo "X11_LIB_PATH=$X11_PATH/lib" >> config.mak
 fi
