diff -up freeglut-2.6.0/src/freeglut_main.c.noxwarn freeglut-2.6.0/src/freeglut_main.c
--- freeglut-2.6.0/src/freeglut_main.c.noxwarn	2010-05-31 10:49:29.002037712 +0200
+++ freeglut-2.6.0/src/freeglut_main.c	2010-05-31 10:50:58.986037800 +0200
@@ -1413,7 +1413,9 @@ void FGAPIENTRY glutMainLoopEvent( void 
             break;
 
         default:
-            fgWarning ("Unknown X event type: %d\n", event.type);
+            /* Fedora patch: the unknown events seem to be quite frequent with
+             * the new xorg. Let's not spam the user with the messages */
+            /* fgWarning ("Unknown X event type: %d\n", event.type); */
             break;
         }
     }
