diff -up freeglut-2.6.0/progs/demos/CallbackMaker/Makefile.in.fixld freeglut-2.6.0/progs/demos/CallbackMaker/Makefile.in
--- freeglut-2.6.0/progs/demos/CallbackMaker/Makefile.in.fixld	2010-02-15 13:34:53.000000000 +0100
+++ freeglut-2.6.0/progs/demos/CallbackMaker/Makefile.in	2010-02-15 13:35:37.000000000 +0100
@@ -107,7 +107,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBXI = @LIBXI@
diff -up freeglut-2.6.0/progs/demos/Fractals/Makefile.in.fixld freeglut-2.6.0/progs/demos/Fractals/Makefile.in
--- freeglut-2.6.0/progs/demos/Fractals/Makefile.in.fixld	2010-02-15 13:36:39.000000000 +0100
+++ freeglut-2.6.0/progs/demos/Fractals/Makefile.in	2010-02-15 13:36:51.000000000 +0100
@@ -107,7 +107,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBXI = @LIBXI@
diff -up freeglut-2.6.0/progs/demos/Fractals_random/Makefile.in.fixld freeglut-2.6.0/progs/demos/Fractals_random/Makefile.in
--- freeglut-2.6.0/progs/demos/Fractals_random/Makefile.in.fixld	2010-02-15 13:37:00.000000000 +0100
+++ freeglut-2.6.0/progs/demos/Fractals_random/Makefile.in	2010-02-15 13:37:11.000000000 +0100
@@ -108,7 +108,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBXI = @LIBXI@
diff -up freeglut-2.6.0/progs/demos/Lorenz/Makefile.in.fixld freeglut-2.6.0/progs/demos/Lorenz/Makefile.in
--- freeglut-2.6.0/progs/demos/Lorenz/Makefile.in.fixld	2010-02-15 13:37:26.000000000 +0100
+++ freeglut-2.6.0/progs/demos/Lorenz/Makefile.in	2010-02-15 13:37:37.000000000 +0100
@@ -107,7 +107,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL -lm
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBXI = @LIBXI@
diff -up freeglut-2.6.0/progs/demos/One/Makefile.in.fixld freeglut-2.6.0/progs/demos/One/Makefile.in
--- freeglut-2.6.0/progs/demos/One/Makefile.in.fixld	2010-02-15 13:37:43.000000000 +0100
+++ freeglut-2.6.0/progs/demos/One/Makefile.in	2010-02-15 13:37:54.000000000 +0100
@@ -107,7 +107,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBXI = @LIBXI@
diff -up freeglut-2.6.0/progs/demos/shapes/Makefile.in.fixld freeglut-2.6.0/progs/demos/shapes/Makefile.in
--- freeglut-2.6.0/progs/demos/shapes/Makefile.in.fixld	2010-02-15 13:38:02.000000000 +0100
+++ freeglut-2.6.0/progs/demos/shapes/Makefile.in	2010-02-15 13:38:17.000000000 +0100
@@ -107,7 +107,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBXI = @LIBXI@
diff -up freeglut-2.6.0/progs/demos/smooth_opengl3/Makefile.in.fixld freeglut-2.6.0/progs/demos/smooth_opengl3/Makefile.in
--- freeglut-2.6.0/progs/demos/smooth_opengl3/Makefile.in.fixld	2010-02-15 13:38:27.000000000 +0100
+++ freeglut-2.6.0/progs/demos/smooth_opengl3/Makefile.in	2010-02-15 13:38:38.000000000 +0100
@@ -107,7 +107,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBXI = @LIBXI@
diff -up freeglut-2.6.0/progs/demos/spaceball/Makefile.in.fixld freeglut-2.6.0/progs/demos/spaceball/Makefile.in
--- freeglut-2.6.0/progs/demos/spaceball/Makefile.in.fixld	2010-02-15 13:40:44.000000000 +0100
+++ freeglut-2.6.0/progs/demos/spaceball/Makefile.in	2010-02-15 13:41:00.000000000 +0100
@@ -108,7 +108,7 @@ LDFLAGS = @LDFLAGS@
 LIBM = @LIBM@
 LIBOBJS = @LIBOBJS@
 LIBRARY = @LIBRARY@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lm -lGL
 LIBTOOL = @LIBTOOL@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 LIBXI = @LIBXI@
