Index: thunderbird/mozilla/build/unix/run-mozilla.sh
===================================================================
--- thunderbird.orig/mozilla/build/unix/run-mozilla.sh
+++ thunderbird/mozilla/build/unix/run-mozilla.sh
@@ -108,7 +108,7 @@ moz_get_debugger()
 	done="no"
 	for d in $debuggers
 	do
-		moz_test_binary /bin/which
+		moz_test_binary /usr/bin/which
 		if [ $? -eq 1 ]
 		then
 			dpath=`which ${d}`	
@@ -154,7 +154,7 @@ moz_debug_program()
 	fi
 	if [ -n "$moz_debugger" ]
 	then
-		moz_test_binary /bin/which
+		moz_test_binary /usr/bin/which
 		if [ $? -eq 1 ]
 		then	
 			debugger=`which $moz_debugger` 
