--- a/l10n/tr/mail/chrome/messenger/smtpEditOverlay.dtd	2008-09-23 14:19:10.000000000 +0300
+++ b/l10n/tr/mail/chrome/messenger/smtpEditOverlay.dtd	2008-09-23 14:29:24.000000000 +0300
@@ -1,10 +1,10 @@
-<!ENTITY alwaysSecure.accesskey "z">
-<!ENTITY alwaysSecure.label "Her zaman">
+<!ENTITY alwaysSecure.accesskey "T">
+<!ENTITY alwaysSecure.label "TLS">
 <!ENTITY alwaysSmtpS.accesskey "l">
 <!ENTITY alwaysSmtpS.label "SSL">
 <!ENTITY alwaysUseUsername.accesskey "s">
 <!ENTITY alwaysUseUsername.label "İsim ve parola kullan">
-<!ENTITY isSecure.label "Güvenli bağlantı kullan (SSL):">
+<!ENTITY isSecure.label "Güvenli bağlantı kullan:">
 <!ENTITY neverSecure.accesskey "a">
 <!ENTITY neverSecure.label "Asla">
 <!ENTITY serverName.accesskey "S">
@@ -13,8 +13,8 @@
 <!ENTITY serverPort.label "Port:">
 <!ENTITY serverPortDefault.label "Öntanımlı:">
 <!ENTITY smtpEditTitle.label "SMTP sunucusu">
-<!ENTITY sometimesSecure.accesskey "R">
-<!ENTITY sometimesSecure.label "Eğer varsa">
+<!ENTITY sometimesSecure.accesskey "S">
+<!ENTITY sometimesSecure.label "TLS, eğer varsa">
 <!ENTITY userName.accesskey "l">
 <!ENTITY userName.label "Kullanıcı İsmi">
 <!ENTITY settings.caption "Ayarlar">
