Index: thunderbird/l10n/tr/mail/chrome/messenger/messengercompose/composeMsgs.properties
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/messengercompose/composeMsgs.properties
+++ thunderbird/l10n/tr/mail/chrome/messenger/messengercompose/composeMsgs.properties
@@ -325,8 +325,8 @@ sendMessageCheckSendButtonLabel=Gönder
 
 ## reply header in composeMsg
 ## <author> wrote:
-mailnews.reply_header_authorwrote=%s yazmış
-mailnews.reply_header_ondate=%s
+mailnews.reply_header_authorwrote=%s yazdı
+mailnews.reply_header_ondate=%s tarihinde
 
 ## reply header in composeMsg
 ## user specified 
Index: thunderbird/l10n/tr/editor/ui/chrome/composer/editorOverlay.dtd
===================================================================
--- thunderbird.orig/l10n/tr/editor/ui/chrome/composer/editorOverlay.dtd
+++ thunderbird/l10n/tr/editor/ui/chrome/composer/editorOverlay.dtd
@@ -174,7 +174,7 @@
 <!ENTITY fontfaceMenu.accesskey "Y">
 <!ENTITY fontVarWidth.label "Değişken genişlik">
 <!ENTITY fontVarWidth.accesskey "D">
-<!ENTITY fontFixedWidth.label "Sabit en">
+<!ENTITY fontFixedWidth.label "Sabit genişlik">
 <!ENTITY fontFixedWidth.accesskey "S">
 <!ENTITY fontFixedWidth.key "T">
 <!ENTITY fontHelvetica.label "Helvetica, Arial">
Index: thunderbird/l10n/tr/mail/chrome/messenger/smtpEditOverlay.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/smtpEditOverlay.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/smtpEditOverlay.dtd
@@ -1,5 +1,5 @@
 <!ENTITY settings.caption "Ayarlar">
-<!ENTITY security.caption "Güvenlik ve kimlik doğrulama">
+<!ENTITY security.caption "Güvenlik ve Kimlik Doğrulama">
 <!ENTITY serverName.label "Sunucu adı:">
 <!ENTITY serverName.accesskey "S">
 <!ENTITY serverDescription.label "Tanım:">
@@ -11,10 +11,10 @@
 <!ENTITY connectionSecurity.label "Bağlantı güvenliği:">
 <!ENTITY connectionSecurity.accesskey "n">
 <!ENTITY connectionSecurityType-0.label "Hiçbiri">
-<!ENTITY connectionSecurityType-1.label "STARTTLS, mevcutsa">
+<!ENTITY connectionSecurityType-1.label "Varsa STARTTLS">
 <!ENTITY connectionSecurityType-2.label "STARTTLS">
 <!ENTITY connectionSecurityType-3.label "SSL/TLS">
 <!ENTITY smtpEditTitle.label "SMTP sunucusu">
 <!ENTITY serverPortDefault.label "Öntanımlı:">
-<!ENTITY authMethod.label "Authentication method:">
+<!ENTITY authMethod.label "Kimlik Doğrulama Yöntemi:">
 <!ENTITY authMethod.accesskey "i">
Index: thunderbird/l10n/tr/dom/chrome/netError.dtd
===================================================================
--- thunderbird.orig/l10n/tr/dom/chrome/netError.dtd
+++ thunderbird/l10n/tr/dom/chrome/netError.dtd
@@ -24,8 +24,8 @@
 <!ENTITY netInterrupt.title "Veri aktarma hatası">
 <!ENTITY netInterrupt.longDesc "<p>Tarayıcı başarıyla bağlantı kurmasına karşın veri aktarımı sırasında kesinti meydana geldi. Lütfen yeniden deneyin.</p><ul><li>Diğer siteleri de açamıyor olabilirsiniz. Bilgisayarınızın ağ bağlantısını gözden geçirin.</li><li>Hâlâ sorun yaşıyorsanız ağ yöneticinize ya da internet servis sağlayıcınıza başvurun.</li></ul>">
 
-<!ENTITY netOffline.title "Çevrim dışı bağlantı hatası">
-<!ENTITY netOffline.longDesc "<p>Tarayıcı şu an çevrim dışı ve istenen öğeye ulaşması mümkün değil.</p><ul><li>Bilgisayarınız etkin bir ağ bağlantısına sahip olmayabilir.</li><li>Tarayıcıyı çevrim içi hâle getirdikten sonra sayfayı açmayı yeniden deneyin.</li></ul>">
+<!ENTITY netOffline.title "Çevrimdışı bağlantı hatası">
+<!ENTITY netOffline.longDesc "<p>Tarayıcı şu an çevrimdışı ve istenen öğeye ulaşması mümkün değil.</p><ul><li>Bilgisayarınız etkin bir ağ bağlantısına sahip olmayabilir.</li><li>Tarayıcıyı çevrimiçi hâle getirdikten sonra sayfayı açmayı yeniden deneyin.</li></ul>">
 
 <!ENTITY contentEncodingError.title "İçerik kodlama hatası">
 <!ENTITY contentEncodingError.longDesc "<p>Bakmak istediğiniz sayfa gösterilemiyor, çünkü geçersiz ya da desteklenmeyen sıkıştırma biçimi kullanıyor.</p><ul><li>Site sahipleriyle iletişim kurup onları bu sorundan haberdar edin.</li></ul>">
Index: thunderbird/l10n/tr/editor/ui/chrome/composer/editor.properties
===================================================================
--- thunderbird.orig/l10n/tr/editor/ui/chrome/composer/editor.properties
+++ thunderbird/l10n/tr/editor/ui/chrome/composer/editor.properties
@@ -47,7 +47,7 @@ FileNotFound=%file% bulunamadı.
 SubdirDoesNotExist="%dir%" alt dizini bu sayfada mevcut değil veya "%file%" dosyası başka bir alt dizin tarafından kullanılıyor olabilir.
 FilenameIsSubdir="%file%" dosyası, bir başka alt dizin tarafından kullanılıyor.
 ServerNotAvailable=Sunucu mevcut değil. Bağlantınızı kontrol edip daha sonra tekrar deneyin.
-Offline=Şuanda çevrim dışı çalışıyorsunuz. Çevrim içi olmak için pencerenin sağ alt kısmındaki işarete tıklayın.
+Offline=Şuanda çevrimdışı çalışıyorsunuz. Çevrimiçi olmak için pencerenin sağ alt kısmındaki işarete tıklayın.
 DiskFull="%file%" dosyalarını kaydedecek yeterli boş yeriniz yok.
 NameTooLong=Dosya veya alt dizin isimleri çok uzun.
 AccessDenied=Bu adresde yayınlamak için yeterli yetkiniz yok.
Index: thunderbird/l10n/tr/mail/chrome/messenger/addressbook/addressBook.properties
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/addressbook/addressBook.properties
+++ thunderbird/l10n/tr/mail/chrome/messenger/addressbook/addressBook.properties
@@ -147,7 +147,7 @@ invalidName=Lütfen geçerli bir isim gi
 invalidHostname=Lütfen geçerli sunucu ismini giriniz.
 invalidPortNumber=Lütfen geçerli bir port numarası giriniz.
 invalidResults=Lütfen sonuçlar alanında geçerli bir numara giriniz.
-abReplicationOfflineWarning=LDAP yönetimini çalıştırabilmeniz için çevrim içi olmanız gereklidir.
+abReplicationOfflineWarning=LDAP yönetimini çalıştırabilmeniz için çevrimiçi olmanız gereklidir.
 abReplicationSaveSettings=Dizini indirmeden önce tercihlerin kayıt edilmesi gereklidir.
 
 # For importing / exporting
Index: thunderbird/l10n/tr/mail/chrome/messenger/addressbook/pref-directory-add.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/addressbook/pref-directory-add.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/addressbook/pref-directory-add.dtd
@@ -49,7 +49,7 @@
 <!ENTITY directoryLogin.label "Kullanıcı adı ve parola ile oturum aç">
 <!ENTITY directoryLogin.accesskey "K">
 <!ENTITY General.tab "Genel">
-<!ENTITY Offline.tab "Çevrim dışı">
+<!ENTITY Offline.tab "Çevrimdışı">
 <!ENTITY Advanced.tab "Gelişmiş">
 <!ENTITY portNumber.label "Port numarası:">
 <!ENTITY portNumber.accesskey "P">
@@ -64,7 +64,7 @@
 <!ENTITY return.label "Daha fazla döndürme">
 <!ENTITY return.accesskey "h">
 <!ENTITY results.label "sonuç">
-<!ENTITY offlineText.label "Çevrim dışı çalıştığınızda kullanabilmek için bu dizinin yerel bir kopyasını bilgisayarınıza alabilirsiniz.">
+<!ENTITY offlineText.label "Çevrimdışı çalıştığınızda kullanabilmek için bu dizinin yerel bir kopyasını bilgisayarınıza alabilirsiniz.">
 <!ENTITY saslMechanism.label "Giriş yöntemı: ">
 <!ENTITY saslMechanism.accesskey "y">
 <!ENTITY saslOff.label "Basit (Simple)">
Index: thunderbird/l10n/tr/mail/chrome/messenger/am-offline.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/am-offline.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/am-offline.dtd
@@ -1,6 +1,6 @@
 <!ENTITY doNotDownloadPop3Movemail.label "Yer sıkıntısı çekmemek için indirilmeyenler:">
-<!ENTITY doNotDownloadNntp.label "Yer sıkıntısı çekmemek için çevrim dışı kullanımı için indirilmeyenler:">
-<!ENTITY doNotDownloadImap.label "Disk alanından tasarruf etmek için sunucudaki iletilerin indirilmesi ve çevrim dışı için yerel kopyalarının bulundurması boyut veya yaş ile sınırlandırılabilir.">
+<!ENTITY doNotDownloadNntp.label "Yer sıkıntısı çekmemek için çevrimdışı kullanımı için indirilmeyenler:">
+<!ENTITY doNotDownloadImap.label "Disk alanından tasarruf etmek için sunucudaki iletilerin indirilmesi ve çevrimdışı için yerel kopyalarının bulundurması boyut veya yaş ile sınırlandırılabilir.">
 <!ENTITY allFoldersOffline.label "Hesaba ait iletileri bu bilgisayarda tut">
 <!ENTITY allFoldersOffline.accesskey "H">
 <!ENTITY offlineNotDownload.label "İleti">
@@ -29,7 +29,7 @@
 <!ENTITY retentionApplyToFlagged.accesskey "Y">
 <!ENTITY nntpRemoveMsgBody.label "İleti gövdesi">
 <!ENTITY nntpRemoveMsgBody.accesskey "ö">
-<!ENTITY offlineSelectNntp.label "Çevrim dışı kullanım için haber grubu seçin…">
+<!ENTITY offlineSelectNntp.label "Çevrimdışı kullanım için haber grubu seçin…">
 <!ENTITY offlineSelectNntp.accesskey "s">
 <!ENTITY offlineImapAdvancedOffline.label "Gelişmiş…">
 <!ENTITY offlineImapAdvancedOffline.accesskey "G">
Index: thunderbird/l10n/tr/mail/chrome/messenger/featureConfigurator.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/featureConfigurator.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/featureConfigurator.dtd
@@ -1,24 +1,24 @@
 <!-- LOCALIZATION NOTE (featureConfigurator.title): Title for the Migration Assistant page. It should match featureConfiguratorCmd.label in baseMenuOverlay.dtd -->
-<!ENTITY featureConfigurator.title "Geçiş yardımcısı">
+<!ENTITY featureConfigurator.title "Göç Asistanı">
 <!-- LOCALIZATION NOTE (featureConfigurator.heading1): Heading for the page -->
 
-<!ENTITY featureConfigurator.heading1 "New &brandShortName; features worth knowing about">
+<!ENTITY featureConfigurator.heading1 "Bilinmesi gereken yeni &brandShortName; özellikleri">
 <!-- LOCALIZATION NOTE (featureConfigurator.para.1): The tone of this paragraph is important: we tried for a tone that promotes the new features (which we believe are good and worth users learning about), but also emphasizing that we think it's important to preserve user choice. -->
 
-<!ENTITY featureConfigurator.para.1 "This new version of &brandShortName; includes powerful new features which we're quite proud of, and think most people will love. People's tastes and needs vary, so we made it easy to control and customize &brandShortName;.">
+<!ENTITY featureConfigurator.para.1 "&brandShortName; uygulamasının bu yeni sürümü, kullanıcıların hoşuna gideceğini düşündüğümüz ve gurur duyduğumuz bazı yeni ve güçlü özellikler içeriyor. Kullanıcıların kişisel zevkleri farklılık gösterebildiğinden bu özelliklerin kolayca yapılandırılmasını sağladık.">
 <!-- LOCALIZATION NOTE (featureConfigurator.para.2): This paragraph explains that this page is active (the buttons change the UI), and that they can get back to it if they change their mind later.-->
 
-<!ENTITY featureConfigurator.para.2 "Use this page to configure these new settings. You can always get back to the Migration Assistant from the Help menu.">
+<!ENTITY featureConfigurator.para.2 "Bu yeni özellikleri yapılandırmak için bu sayfayı kullanabilirsiniz. Yardım menüsünden Göç Asistanı'na her zaman ulaşabilirsiniz.">
 
-<!ENTITY featureConfigurator.toolbar.heading "﻿Yeni ileti araç çubuğu">
-<!ENTITY featureConfigurator.toolbar.new "Message Buttons Toolbar">
-<!ENTITY featureConfigurator.toolbar.new.status.explanation "A clean toolbar with all your messaging buttons at your fingertips.">
+<!ENTITY featureConfigurator.toolbar.heading "Yeni İleti Araç Çubuğu">
+<!ENTITY featureConfigurator.toolbar.new "İleti Düğmeleri Araç Çubuğu">
+<!ENTITY featureConfigurator.toolbar.new.status.explanation "Tüm ileti düğmelerini bir arada tutan yeni ve temiz bir araç çubuğu hizmetinizde.">
 <!-- LOCALIZATION NOTE (featureConfigurator.toolbar.original): The word 'original' is chosen deliberately to be neutral, vs. 'old' which in English at least had a negative connotation.-->
 
-<!ENTITY featureConfigurator.toolbar.original "Keep your original Toolbar buttons">
-<!ENTITY featureConfigurator.toolbar.original.status.explanation "Keep any custom toolbar buttons in addition to the new Message Buttons.">
+<!ENTITY featureConfigurator.toolbar.original "Mevcut araç çubuğu düğmelerini koru">
+<!ENTITY featureConfigurator.toolbar.original.status.explanation "Yeni ileti düğmelerine ek olarak mevcut araç çubuğu düğmelerini de göster.">
 
-<!ENTITY featureConfigurator.toolbarBenefits.para "Message actions such as reply and forward have moved from the toolbar and onto the message header itself. This ties the action buttons directly to the message, which leaves more room in the toolbar for the richer search textbox, as well as add-ons.">
+<!ENTITY featureConfigurator.toolbarBenefits.para "Yanıtla ve ilet gibi ileti eylemleri araç çubuğundan ileti başlığına taşındı. Bu değişiklik ile, ileti eylemleri doğrudan iletiyle ilişkilendirilerek, araç çubuğunda eklentilere ve gelişmiş metin arama özelliğine daha fazla yer açıldı.">
 <!-- LOCALIZATION NOTE (featureConfigurator.addOn.installing): The text to display while installing the addon. -->
 
 <!ENTITY featureConfigurator.addOn.installing "Kuruluyor…">
@@ -27,63 +27,63 @@
 <!ENTITY featureConfigurator.addOn.installed "Kuruldu">
 <!-- LOCALIZATION NOTE (featureConfigurator.addOn.alreadyInstalled): The label to display when the addon is already installed. -->
 
-<!ENTITY featureConfigurator.addOn.alreadyInstalled "﻿Bu eklenti zaten yüklü.">
+<!ENTITY featureConfigurator.addOn.alreadyInstalled "Bu eklenti zaten yüklü.">
 <!-- LOCALIZATION NOTE (featureConfigurator.collapsedHeaders.heading): The heading for the Compact Header page. -->
 
-<!ENTITY featureConfigurator.collapsedHeaders.heading "Kompakt başlık">
+<!ENTITY featureConfigurator.collapsedHeaders.heading "Kompakt Başlık">
 <!-- LOCALIZATION NOTE (featureConfigurator.collapsedHeaders.para.weak): The paragraph to display if the user hadn't previously used the collapsed headers feature. -->
 
-<!ENTITY featureConfigurator.collapsedHeaders.para.weak "The compact message reader view has been moved into its own extension where it is now getting much more attention and options.">
+<!ENTITY featureConfigurator.collapsedHeaders.para.weak "Kompakt ileti okuyucu görünümü artık bir eklenti olarak sunulmaktadır.">
 <!-- LOCALIZATION NOTE (featureConfigurator.collapsedHeaders.para.strong): The paragraph to display if the user had previously used the collapsed headers feature. -->
 
-<!ENTITY featureConfigurator.collapsedHeaders.para.strong "We noticed that you were using the compact message reader header.  This function has moved into its own extension and we recommend you install it here if you would like to continue using it.">
+<!ENTITY featureConfigurator.collapsedHeaders.para.strong "Kompakt ileti okuyucu görünümünü kullandığınızı tespit ettik. Bu işlev artık bir eklenti olarak sunulduğundan bu eklentiyi yüklemenizi tavsiye ediyoruz.">
 <!-- LOCALIZATION NOTE (featureConfigurator.collapsedHeaders.install): The text for the button that will install the Compact Headers addon. -->
 
-<!ENTITY featureConfigurator.collapsedHeaders.install "Kompakt başlık eklentisini yükleyin…">
+<!ENTITY featureConfigurator.collapsedHeaders.install "Kompakt Başlık eklentisini Yükle…">
 <!-- LOCALIZATION NOTE (featureConfigurator.folderColumns.heading): The heading for the Folder Columns page. -->
 
-<!ENTITY featureConfigurator.folderColumns.heading "Gelişmiş dizin başlıkları">
+<!ENTITY featureConfigurator.folderColumns.heading "Gelişmiş Dizin Sütunları">
 <!-- LOCALIZATION NOTE (featureConfigurator.folderColumns.para.weak): The paragraph to display if the user hadn't previously used the folder columns feature. -->
 
-<!ENTITY featureConfigurator.folderColumns.para.weak "The Extra Folder Columns extension replaces what was initially an advanced preference.  To keep tabs on the exact count of messages and disk space each folder is using you'll want to install this extension.">
+<!ENTITY featureConfigurator.folderColumns.para.weak "Eskiden gelişmiş bir özellik olarak sunulan Gelişmiş Dizin Sütunları artık bir eklenti halinde sunulmaktadır. Eğer dizinlerde kaç ileti tutulduğunu ve ne kadar disk alanı harcandığını ayrı sütunlar halinde görmek istiyorsanız bu eklentiyi kurabilirsiniz.">
 <!-- LOCALIZATION NOTE (featureConfigurator.folderColumns.para.strong): The paragraph to display if the user had previously used the folder columns feature. -->
 
-<!ENTITY featureConfigurator.folderColumns.para.strong "We noticed that you were using the Advanced Folder columns.  This feature has moved into its own extension, and we recommend you install here.">
+<!ENTITY featureConfigurator.folderColumns.para.strong "Gelişmiş Dizin Sütunları özelliğini kullandığınızı tespit ettik. Bu özellik artık bir eklenti olarak sunulduğundan bu eklentiyi yüklemenizi tavsiye ediyoruz.">
 <!-- LOCALIZATION NOTE (featureConfigurator.folderColumns.install): The text for the button that will install the Extra Folder Columns addon. -->
 
-<!ENTITY featureConfigurator.folderColumns.install "Install Extra Columns…">
+<!ENTITY featureConfigurator.folderColumns.install "Gelişmiş Dizin Sütunlarını Yükle…">
 <!-- LOCALIZATION NOTE (featureConfigurator.autoSync.heading): The heading for the Disk space page. -->
 
-<!ENTITY featureConfigurator.autoSync.heading "Disk alanı">
+<!ENTITY featureConfigurator.autoSync.heading "Disk Alanı">
 <!-- LOCALIZATION NOTE (featureConfigurator.autoSync.para): The paragraph that explains why the user would want to turn on autosync. -->
 
-<!ENTITY featureConfigurator.autoSync.para "This new version of &brandShortName; comes with a new and better search functionality.  To experience this new search you'll need to synchronize your email messages.">
+<!ENTITY featureConfigurator.autoSync.para "&brandShortName; uygulamasının bu yeni sürümü ile sunulan yeni ve gelişmiş arama altyapısından faydalanabilmek için e-posta iletilerinizi eşzamanlamanız gerekmektedir.">
 <!-- LOCALIZATION NOTE (featureConfigurator.autoSync.on): The label for the radio button for synchronizing all the accounts. -->
 
-<!ENTITY featureConfigurator.autoSync.on "Eşzamanlama yap">
-<!ENTITY featureConfigurator.autoSync.on.explanation "You'll see faster message loading and better search results.">
+<!ENTITY featureConfigurator.autoSync.on "Eşzamanlama Yap">
+<!ENTITY featureConfigurator.autoSync.on.explanation "Daha hızlı ileti yükleme ve daha iyi arama sonuçları elde edeceksiniz.">
 <!-- LOCALIZATION NOTE (featureConfigurator.autoSync.off): The label for the radio button that will synchronize none of the accounts. -->
 
-<!ENTITY featureConfigurator.autoSync.off "Eşzamanlama yapma">
-<!ENTITY featureConfigurator.autoSync.off.explanation "This will mean slower message loading and incomplete search results.">
+<!ENTITY featureConfigurator.autoSync.off "Eşzamanlama Yapma">
+<!ENTITY featureConfigurator.autoSync.off.explanation "Daha yavaş ileti yükleme başarımı ve eksik arama sonuçlarıyla karşılaşabilirsiniz.">
 <!-- LOCALIZATION NOTE (featureConfigurator.autoSync.some): The label for the radio button that will let you customize the synchronize for each account. -->
 
 <!ENTITY featureConfigurator.autoSync.some "Her hesap için ayrı eşzamanlama yap">
-<!ENTITY featureConfigurator.autoSync.some.explanation "Customize your synchronization needs per account and folder.">
+<!ENTITY featureConfigurator.autoSync.some.explanation "Dizin ve hesap başına eşzamanlama ihtiyaçlarınızı özelleştirin">
 <!-- LOCALIZATION NOTE (featureConfigurator.diskSpace.currentSize): The label for the current amount of disk space your accounts are consuming. -->
 
-<!ENTITY featureConfigurator.diskSpace.currentSize "Your current total email usage:">
+<!ENTITY featureConfigurator.diskSpace.currentSize "E-postalarınızın harcadığı disk alanı:">
 <!-- LOCALIZATION NOTE (featureConfigurator.diskSpace.freeSpace): The label for the current amount of free space available on your hard disk. -->
 
-<!ENTITY featureConfigurator.diskSpace.freeSpace "﻿Boş alan:">
+<!ENTITY featureConfigurator.diskSpace.freeSpace "Boş Alan:">
 <!-- LOCALIZATION NOTE (featureConfigurator.diskSpace.requireSpace): The label for the required amount of free space to synchronize all of your IMAP accounts. -->
 
-<!ENTITY featureConfigurator.diskSpace.requireSpace "﻿Gerekli:">
+<!ENTITY featureConfigurator.diskSpace.requireSpace "Gereken:">
 <!-- LOCALIZATION NOTE (featureConfigurator.diskSpace.warning): The label shown when your estimated available hard disk space if you chose autosync would be less than 50MB or if we estimate turning on autosync would use more than 50% of your free space. -->
 
-<!ENTITY featureConfigurator.diskSpace.warning "You're running low on space!">
+<!ENTITY featureConfigurator.diskSpace.warning "Diskinizde gerekli miktarda boş alan yok!">
 
 
-<!ENTITY featureConfigurator.previousButton "Önceki">
-<!ENTITY featureConfigurator.nextButton "Sonraki">
+<!ENTITY featureConfigurator.previousButton "Geri">
+<!ENTITY featureConfigurator.nextButton "İleri">
 <!ENTITY featureConfigurator.closeButton "Kapat">
Index: thunderbird/l10n/tr/mail/chrome/messenger/folderProps.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/folderProps.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/folderProps.dtd
@@ -43,23 +43,23 @@
 <!ENTITY generalInfo.label "Genel bilgiler">
 <!ENTITY folderCharsetTab.label "Öntanımlı karakter kodlaması:">
 <!ENTITY folderCharsetTab.accesskey "e">
-<!ENTITY folderCharsetOverride.label "Öntanımlamayı bu dizindeki postalara uyumla.">
-<!ENTITY folderCharsetOverride.accesskey "a">
-<!ENTITY folderRebuildSummaryFileTip2.label "Özet dosyasını tamir et">
-<!ENTITY folderRebuildSummaryFile2.label "Dizini tamir et">
+<!ENTITY folderCharsetOverride.label "Öntanımlı değeri bu dizindeki tüm iletilere uygula (Özel ileti kodlamaları ve otomatik algılama devredışı kalacak)">
+<!ENTITY folderCharsetOverride.accesskey "A">
+<!ENTITY folderRebuildSummaryFileTip2.label "Özet dosyasını tekrar oluştur">
+<!ENTITY folderRebuildSummaryFile2.label "Dizini Onar">
 <!ENTITY folderRebuildSummaryFile2.accesskey "D">
-<!ENTITY folderRebuildSummaryFile.explanation "Sometimes the folder index (.msf) file becomes damaged and messages may appear missing or deleted messages continue showing; repairing the folder may fix these issues.">
-<!ENTITY folderIncludeInGlobalSearch.label "Include messages in this folder in Global Search results">
+<!ENTITY folderRebuildSummaryFile.explanation "Bazen dizin özet dosyası (.msf) zarar görür ve iletiler eksik görünür veya silinmiş iletiler görünmeye devam eder; dizini onarmak bu sorunları giderebilir.">
+<!ENTITY folderIncludeInGlobalSearch.label "Bu dizindeki iletileri Global Arama sonuçlarına dahil et">
 <!ENTITY folderIncludeInGlobalSearch.accesskey "G">
      
 <!ENTITY retention.label "Kayıt güvenliği">
-<!ENTITY retentionUseAccount.label "Hesap ayarlarınını kullan">
+<!ENTITY retentionUseAccount.label "Hesap ayarlarımı kullan">
 <!ENTITY retentionUseAccount.accesskey "H">
 <!ENTITY daysOld.label "günden eski ise sil">
 <!ENTITY message.label "ileti dışındakileri sil">
-<!ENTITY retentionCleanup.label "Boş yer kazanmak için eski iletileri düzenli aralıklar ile silebilirsiniz.">
-<!ENTITY retentionCleanupImap.label "Disk alanı kazanmak için sunucudaki eski özgün iletiler ve yerel kopyaları tamamen silinebilir.">
-<!ENTITY retentionCleanupPop.label "Disk alanı kazanmak için sunucudaki eski özgün iletiler tamamen silinebilir.">
+<!ENTITY retentionCleanup.label "Disk alanı kazanmak için eski iletiler kalıcı olarak silinebilir.">
+<!ENTITY retentionCleanupImap.label "Disk alanı kazanmak için hem sunucudaki eski iletiler, hem de yerel kopyaları kalıcı olarak silinebilir.">
+<!ENTITY retentionCleanupPop.label "Disk alanı kazanmak için sunucudaki eski iletiler tamamen silinebilir.">
 <!ENTITY retentionDeleteMsg.label "İleti">
 <!ENTITY retentionDeleteMsg.accesskey "t">
 <!ENTITY retentionKeepAll.label "Hiçbir iletiyi silme">
@@ -72,15 +72,15 @@
 <!ENTITY retentionApplyToFlagged.accesskey "Y">
 
 <!ENTITY folderSynchronizationTab.label "Eşzamanlama">
-<!ENTITY folderCheckForNewMessages2.label "When getting new messages for this account, always check this folder">
+<!ENTITY folderCheckForNewMessages2.label "Bu hesap için yeni iletiler alınırken bu dizini de denetle.">
 <!ENTITY folderCheckForNewMessages2.accesskey "c">
 
-<!ENTITY offlineFolder.check.label "Bu dizini çevrim dışı kullanım için seç">
+<!ENTITY offlineFolder.check.label "Bu dizini çevrimdışı kullanım için seç">
 <!ENTITY offlineFolder.check.accesskey "v">
 <!ENTITY offlineFolder.button.label "Şimdi indir">
 <!ENTITY offlineFolder.button.accesskey "d">
 
-<!ENTITY selectofflineNewsgroup.check.label "Bu haber grubunu çevrim dışı kullanımı için seç">
+<!ENTITY selectofflineNewsgroup.check.label "Bu haber grubunu çevrimdışı kullanımı için seç">
 <!ENTITY selectofflineNewsgroup.check.accesskey "u">
 <!ENTITY offlineNewsgroup.button.label "Şimdi indir">
 <!ENTITY offlineNewsgroup.button.accesskey "d">
@@ -90,13 +90,13 @@
 <!ENTITY folderProps.location.label "Konum:">
 <!ENTITY folderProps.location.accesskey "m">
 
-<!ENTITY folderSharingTab.label "Ortak kullanım">
-<!ENTITY privileges.button.label "Öncelikli haklar…">
+<!ENTITY folderSharingTab.label "Paylaşım">
+<!ENTITY privileges.button.label "Yetkiler…">
 <!ENTITY privileges.button.accesskey "h">
-<!ENTITY permissionsDesc.label "İzinleriniz:">
-<!ENTITY folderType.label "Dizin türü:">
+<!ENTITY permissionsDesc.label "Sahip olduğunuz izinleri:">
+<!ENTITY folderType.label "Dizin Türü:">
 
 <!ENTITY folderQuotaTab.label "Kota">
-<!ENTITY folderQuotaRoot.label "Kota kökü:">
+<!ENTITY folderQuotaRoot.label "Kota Kökü:">
 <!ENTITY folderQuotaUsage.label "Kullanım:">
 <!ENTITY folderQuotaStatus.label "Durum:">
Index: thunderbird/l10n/tr/mail/chrome/messenger/imapMsgs.properties
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/imapMsgs.properties
+++ thunderbird/l10n/tr/mail/chrome/messenger/imapMsgs.properties
@@ -104,7 +104,7 @@
 # Status - compacting a folder
 ## @name IMAP_STATUS_EXPUNGING_MAILBOX
 ## @loc None
-5010=Dizini sıklaştır…
+5010=Dizini sıkıştır…
 
 # Status - logging out
 ## @name IMAP_STATUS_LOGGING_OUT
Index: thunderbird/l10n/tr/mail/chrome/messenger/localMsgs.properties
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/localMsgs.properties
+++ thunderbird/l10n/tr/mail/chrome/messenger/localMsgs.properties
@@ -52,7 +52,7 @@ pop3PreviouslyEnteredPasswordIsInvalidPr
 
 ## @name MK_POP3_OUT_OF_DISK_SPACE
 ## @loc None
--321=Postaları indirmek için yeterli disk alanı yok. Eski postalarınızı silin, çöp kutusunu boşaltın, posta dizinlerini sıklaştırın ve ardından yeniden deneyin.
+-321=Postaları indirmek için yeterli disk alanı yok. Eski postalarınızı silin, çöp kutusunu boşaltın, posta dizinlerini sıkıştırın ve tekrar deneyin.
 
 # Status - parsing folder
 ## @name LOCAL_STATUS_SELECTING_MAILBOX
Index: thunderbird/l10n/tr/mail/chrome/messenger/messenger.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/messenger.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/messenger.dtd
@@ -41,20 +41,20 @@
 <!ENTITY subscribeCmd.accesskey "A">
 <!ENTITY deleteFolder.label "Dizini sil">
 <!ENTITY deleteFolder.accesskey "i">
-<!ENTITY renameFolder.label "Dinin ismini değiştir…">
+<!ENTITY renameFolder.label "Dizini yeniden adlandır…">
 <!ENTITY renameFolder.accesskey "i">
 <!ENTITY renameFolder.key "VK_F2">
 <!ENTITY compactFolders.label "Dizinleri sıkıştır">
 <!ENTITY compactFolders.accesskey "D">
 <!ENTITY emptyTrashCmd.label "Çöpü boşalt">
 <!ENTITY emptyTrashCmd.accesskey "p">
-<!ENTITY offlineMenu.label "Çevrim dışı">
+<!ENTITY offlineMenu.label "Çevrimdışı">
 <!ENTITY offlineMenu.accesskey "m">
-<!ENTITY offlineGoOfflineCmd.label "Çevrim dışı çalış">
+<!ENTITY offlineGoOfflineCmd.label "Çevrimdışı çalış">
 <!ENTITY offlineGoOfflineCmd.accesskey "v">
 <!ENTITY synchronizeOfflineCmd.label "Şimdi indir ve eşzamanla…">
 <!ENTITY synchronizeOfflineCmd.accesskey "m">
-<!ENTITY settingsOfflineCmd.label "Çevrim dışı ayarları…">
+<!ENTITY settingsOfflineCmd.label "Çevrimdışı ayarları…">
 <!ENTITY settingsOfflineCmd.accesskey "e">
 <!ENTITY downloadSelectedCmd.label "Seçili iletileri indir">
 <!ENTITY downloadSelectedCmd.accesskey "i">
@@ -63,7 +63,7 @@
 <!ENTITY printCmd.label "Yazdır…">
 <!ENTITY printCmd.accesskey "r">
 <!ENTITY printCmd.key "p">
-<!ENTITY printPreviewCmd.label "Yazdırma ön izleme">
+<!ENTITY printPreviewCmd.label "Yazdırma Önizleme">
 <!ENTITY printPreviewCmd.accesskey "Y">
 <!ENTITY printSetupCmd.label "Sayfa düzeni…">
 <!ENTITY printSetupCmd.accesskey "a">
@@ -132,29 +132,29 @@
 
 <!ENTITY folderView.label "Dizinler">
 <!ENTITY folderView.accesskey "D">
-<!ENTITY unifiedFolders.label "Unified">
+<!ENTITY unifiedFolders.label "Birleşik">
 <!ENTITY unifiedFolders.accesskey "n">
 <!ENTITY allFolders.label "Tümü">
 <!ENTITY allFolders.accesskey "m">
-<!ENTITY unreadFolders.label "Okunmayanlar">
+<!ENTITY unreadFolders.label "Okunmamışlar">
 <!ENTITY unreadFolders.accesskey "u">
-<!ENTITY favoriteFolders.label "Tercih edilenler">
+<!ENTITY favoriteFolders.label "Favoriler">
 <!ENTITY favoriteFolders.accesskey "T">
 <!ENTITY recentFolders.label "Güncel">
 <!ENTITY recentFolders.accesskey "G">
 <!-- Sort Menu -->
 
-<!ENTITY sortMenu.label "Sıralama koşulu">
+<!ENTITY sortMenu.label "Sırala">
 <!ENTITY sortMenu.accesskey "l">
 <!ENTITY sortByDateCmd.label "Tarih">
 <!ENTITY sortByDateCmd.accesskey "h">
-<!ENTITY sortByReceivedCmd.label "Alınmıa tarihi">
+<!ENTITY sortByReceivedCmd.label "Alınma Tarihi">
 <!ENTITY sortByReceivedCmd.accesskey "a">
 <!ENTITY sortByStarCmd.label "Yıldız">
 <!ENTITY sortByStarCmd.accesskey "z">
-<!ENTITY sortByAttachmentsCmd.label "Ek">
+<!ENTITY sortByAttachmentsCmd.label "Ekler">
 <!ENTITY sortByAttachmentsCmd.accesskey "E">
-<!ENTITY sortByPriorityCmd.label "Önemlilik">
+<!ENTITY sortByPriorityCmd.label "Öncelik">
 <!ENTITY sortByPriorityCmd.accesskey "n">
 <!ENTITY sortBySizeCmd.label "Boyut">
 <!ENTITY sortBySizeCmd.accesskey "t">
@@ -162,7 +162,7 @@
 <!ENTITY sortByStatusCmd.accesskey "u">
 <!ENTITY sortByTagsCmd.label "Etiket">
 <!ENTITY sortByTagsCmd.accesskey "k">
-<!ENTITY sortByJunkStatusCmd.label "Gereksiz statüsü">
+<!ENTITY sortByJunkStatusCmd.label "Gereksizlik Durumu">
 <!ENTITY sortByJunkStatusCmd.accesskey "G">
 <!ENTITY sortBySubjectCmd.label "Konu">
 <!ENTITY sortBySubjectCmd.accesskey "K">
@@ -172,7 +172,7 @@
 <!ENTITY sortByRecipientCmd.accesskey "c">
 <!ENTITY sortByUnreadCmd.label "Okunmuş">
 <!ENTITY sortByUnreadCmd.accesskey "ş">
-<!ENTITY sortByOrderReceivedCmd.label "Alma sırası">
+<!ENTITY sortByOrderReceivedCmd.label "Alma Sırası">
 <!ENTITY sortByOrderReceivedCmd.accesskey "ı">
 <!ENTITY sortAscending.label "Artarak">
 <!ENTITY sortAscending.accesskey "A">
@@ -182,7 +182,7 @@
 <!ENTITY sortThreaded.accesskey "D">
 <!ENTITY sortUnthreaded.label "Dizilimsiz">
 <!ENTITY sortUnthreaded.accesskey "D">
-<!ENTITY groupBySort.label "Sıralamaları Gruplaştır">
+<!ENTITY groupBySort.label "Sıralamayı Grupla">
 <!ENTITY groupBySort.accesskey "G">
 <!ENTITY msgsMenu.label "İletiler">
 <!ENTITY msgsMenu.accesskey "İ">
@@ -190,17 +190,17 @@
 <!ENTITY threads.accesskey "D">
 <!ENTITY allMsgsCmd.label "Tümü">
 <!ENTITY allMsgsCmd.accesskey "ü">
-<!ENTITY expandAllThreadsCmd.label "Bütün dizilimleri aç">
+<!ENTITY expandAllThreadsCmd.label "Tüm dizilimleri genişlet">
 <!ENTITY expandAllThreadsCmd.accesskey "ü">
 <!ENTITY expandAllThreadsCmd.key "*">
-<!ENTITY collapseAllThreadsCmd.label "Tüm dizilimleri kapat">
+<!ENTITY collapseAllThreadsCmd.label "Tüm dizilimleri daralt">
 <!ENTITY collapseAllThreadsCmd.accesskey "T">
 <!ENTITY collapseAllThreadsCmd.key "\">
 <!ENTITY unreadMsgsCmd.label "Okunmamış">
 <!ENTITY unreadMsgsCmd.accesskey "O">
-<!ENTITY threadsWithUnreadCmd.label "Okunmamış içeren dizilimler">
+<!ENTITY threadsWithUnreadCmd.label "Okunmamış ileti içeren dizilimler">
 <!ENTITY threadsWithUnreadCmd.accesskey "m">
-<!ENTITY watchedThreadsWithUnreadCmd.label "Takip edilen okunmamış içeren dizilimler">
+<!ENTITY watchedThreadsWithUnreadCmd.label "Okunmamış ileti içeren ve takip edilen dizilimler">
 <!ENTITY watchedThreadsWithUnreadCmd.accesskey "o">
 <!ENTITY ignoredThreadsCmd.label "Gözardı edilen dizilimler">
 <!ENTITY ignoredThreadsCmd.accesskey "l">
@@ -211,13 +211,13 @@
 <!ENTITY headersAllCmd.accesskey "ü">
 <!ENTITY headersNormalCmd.label "Normal">
 <!ENTITY headersNormalCmd.accesskey "N">
-<!ENTITY bodyMenu.label "Mesaj metni">
+<!ENTITY bodyMenu.label "Mesaj Metni">
 <!ENTITY bodyMenu.accesskey "j">
 <!ENTITY bodyAllowHTML.label "Özgün HTML">
 <!ENTITY bodyAllowHTML.accesskey "H">
 <!ENTITY bodySanitized.label "Basit HTML">
 <!ENTITY bodySanitized.accesskey "B">
-<!ENTITY bodyAsPlaintext.label "Düz metin">
+<!ENTITY bodyAsPlaintext.label "Düz Metin">
 <!ENTITY bodyAsPlaintext.accesskey "D">
 
 <!ENTITY bodyMenuFeed.label "Canlı yer imi gövdesi gösterimi">
@@ -238,14 +238,14 @@ If shift key is needed with your locale
 you can use these alternative items. Otherwise, their values should be empty.  -->
 
 
-<!ENTITY fullZoomEnlargeCmd.label "Büyüt">
+<!ENTITY fullZoomEnlargeCmd.label "Yaklaş">
 <!ENTITY fullZoomEnlargeCmd.accesskey "B">
 <!ENTITY fullZoomEnlargeCmd.commandkey "+">
 <!-- + is above this key on many keyboards -->
 <!ENTITY fullZoomEnlargeCmd.commandkey2 "=">
 <!ENTITY fullZoomEnlargeCmd.commandkey3 "">
 
-<!ENTITY fullZoomReduceCmd.label "Küçült">
+<!ENTITY fullZoomReduceCmd.label "Uzaklaş">
 <!ENTITY fullZoomReduceCmd.accesskey "K">
 <!ENTITY fullZoomReduceCmd.commandkey "-">
 <!ENTITY fullZoomReduceCmd.commandkey2 "">
@@ -255,12 +255,12 @@ you can use these alternative items. Oth
 <!ENTITY fullZoomResetCmd.commandkey "0">
 <!ENTITY fullZoomResetCmd.commandkey2 "">
 
-<!ENTITY fullZoomToggleCmd.label "Sadece metni ölçekle">
+<!ENTITY fullZoomToggleCmd.label "Sadece metni yaklaştır">
 <!ENTITY fullZoomToggleCmd.accesskey "m">
 <!ENTITY fullZoom.label "Ölçek">
 <!ENTITY fullZoom.accesskey "Ö">
 
-<!ENTITY pageSourceCmd.label "İleti kaynak metni">
+<!ENTITY pageSourceCmd.label "İleti kaynağı">
 <!ENTITY pageSourceCmd.accesskey "l">
 <!ENTITY pageSourceCmd.key "u">
 <!-- Search Menu -->
@@ -519,7 +519,7 @@ you can use these alternative items. Oth
 <!ENTITY deleteButton.tooltip "Seçilmiş iletiyi yada dizini sil">
 <!ENTITY undeleteButton.tooltip "İşaretlenmiş silinen iletileri geri al">
 <!ENTITY markButton.tooltip "İletileri işaretle">
-<!ENTITY printButton.tooltip "İletiyi	yazdır">
+<!ENTITY printButton.tooltip "İletiyi yazdır">
 <!ENTITY stopButton.tooltip "Şu anki aktarımı durdur">
 <!ENTITY junkButton.tooltip "Seçili iletileri gereksiz olarak işaretle">
 <!ENTITY notJunkButton.tooltip 'Seçili iletiyi "Gereksiz değil" olarak işaretle'>
@@ -542,13 +542,13 @@ you can use these alternative items. Oth
 <!ENTITY folderContextOpenNewWindow.accesskey "A">
 <!ENTITY folderContextOpenNewTab.label "Yeni sekmede aç">
 <!ENTITY folderContextOpenNewTab.accesskey "n">
-<!ENTITY folderContextNew.label "Yeni altdizin…">
+<!ENTITY folderContextNew.label "Yeni alt dizin…">
 <!ENTITY folderContextNew.accesskey "n">
 <!ENTITY folderContextRename.label "İsmini değiştir">
 <!ENTITY folderContextRename.accesskey "s">
 <!ENTITY folderContextRemove.label "Sil">
 <!ENTITY folderContextRemove.accesskey "S">
-<!ENTITY folderContextCompact.label "Bu dizini sıklaştır">
+<!ENTITY folderContextCompact.label "Bu dizini sıkıştır">
 <!ENTITY folderContextCompact.accesskey "B">
 <!ENTITY folderContextEmptyTrash.label "Çöpü boşalt">
 <!ENTITY folderContextEmptyTrash.accesskey "Ç">
@@ -600,14 +600,14 @@ you can use these alternative items. Oth
 <!ENTITY accountColumn.label "Hesap">
 <!ENTITY statusColumn.label "Durum">
 <!ENTITY sizeColumn.label "Boyut">
-<!ENTITY junkStatusColumn.label "Gereksiz durumu">
+<!ENTITY junkStatusColumn.label "Gereksizlik Durumu">
 <!ENTITY unreadColumn.label "Okunmamış">
 <!ENTITY totalColumn.label "Toplam">
 <!ENTITY readColumn.label "Okundu">
-<!ENTITY receivedColumn.label "Alındı tarihi">
+<!ENTITY receivedColumn.label "Alınma Tarihi">
 <!ENTITY starredColumn.label "Yıldızlı">
 <!ENTITY locationColumn.label "Konum">
-<!ENTITY idColumn.label "Alındı sırası">
+<!ENTITY idColumn.label "Alınma Sırası">
 <!ENTITY attachmentColumn.label "Ekler">
 <!-- Thread Pane Tooltips -->
 
@@ -637,7 +637,7 @@ you can use these alternative items. Oth
 <!ENTITY contextOpenNewWindow.accesskey "c">
 <!ENTITY contextOpenNewTab.label "İletiyi yeni sekmede aç">
 <!ENTITY contextOpenNewTab.accesskey "ç">
-<!ENTITY contextOpenConversation.label "İletiyi konyşma içinde aç">
+<!ENTITY contextOpenConversation.label "İletiyi konuşma içinde aç">
 <!ENTITY contextOpenConversation.accesskey "y">
 <!ENTITY contextEditAsNew.label "Yeni olarak düzenle…">
 <!ENTITY contextEditAsNew.accesskey "ü">
@@ -665,7 +665,7 @@ you can use these alternative items. Oth
 <!ENTITY contextSaveAs.accesskey "F">
 <!ENTITY contextPrint.label "Yazdır…">
 <!ENTITY contextPrint.accesskey "r">
-<!ENTITY contextPrintPreview.label "Yazdırma önizleme">
+<!ENTITY contextPrintPreview.label "Yazdırma Önizleme">
 <!ENTITY contextPrintPreview.accesskey "z">
 <!-- Thread Pane Column Picker -->
 <!-- LOCALIZATION NOTE (columnPicker.resetToInbox.label):
@@ -808,7 +808,7 @@ you can use these alternative items. Oth
 <!ENTITY copyCmd.accesskey "o">
 <!ENTITY pasteCmd.label "Yapıştır">
 <!ENTITY pasteCmd.accesskey "p">
-<!ENTITY selectAllCmd.label "Tümümü seç">
+<!ENTITY selectAllCmd.label "Tümünü seç">
 <!ENTITY selectAllCmd.accesskey "T">
 <!ENTITY copyLinkCmd.label "Bağlantı konumunu kopyala">
 <!ENTITY copyLinkCmd.accesskey "B">
Index: thunderbird/l10n/tr/mail/chrome/messenger/messenger.properties
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/messenger.properties
+++ thunderbird/l10n/tr/mail/chrome/messenger/messenger.properties
@@ -77,8 +77,8 @@ saveAttachmentFailed=Ek kaydedilemedi. L
 saveMessageFailed=Posta Kaydedilemedi. Lütfen dosya ismini kontrol edin ve tekrar deneyiniz.
 fileExists=%S aynısı var. Üstüne yazmak ister misiniz?
 
-downloadingNewsgroups=Haber Grupları çevrim dışı kullanımı için indiriliyor
-downloadingMail=Postalar çevrim dışı kullanımı için indiriliyor
+downloadingNewsgroups=Haber Grupları çevrimdışı kullanımı için indiriliyor
+downloadingMail=Postalar çevrimdışı kullanımı için indiriliyor
 sendingUnsent=Gönderilmemiş postaları gönder
 
 folderExists=Aynı isimli dizin bulundu. Lütfen değişik isim girin.
@@ -87,7 +87,7 @@ folderCreationFailed=Dizin oluşturulama
 compactingFolder=%S dizinlerini sıkıştır…
 doneCompacting=Dizinlerin sıkıştırılması tamamlandı
 autoCompactAllFoldersTitle=Dizinleri sıkıştır
-autoCompactAllFolders=Disk alanı açmak için yerel ve çevrim dışı dizinlerinizi İşlemlendirmek ister misiniz?
+autoCompactAllFolders=Disk alanı açmak için yerel ve çevrimdışı dizinlerinizi İşlemlendirmek ister misiniz?
 autoCompactAlwaysAskCheckbox=Dizinleri otomatik sıkıştırmadan önce bana hep sor
 compactNowButton=Şimdi &sıkıştır
 
@@ -101,8 +101,8 @@ alertFilterCheckbox=Beni bir daha uyarma
 compactFolderDeniedLock=Dizin '%S' işlemlenemedi çünkü başka işlem işlem halindedir. Lütfen sonra deneyin.
 compactFolderWriteFailed=Dizin '%S' işlemlenemedi çünkü dizine yazılamadı. Lütfen dizine yazma izni olduğunuzu veya yeterli disk alanı olduğunu kontrol edip  tekrar deneyiniz.
 filterFolderWriteFailed=Postalar dizinlere süzgeçlenemedi '%S' çünkü dizine yazılamadı. Yeterli disk alanı ve yazma izninizi kontrol edip tekrar deneyiniz.
-copyMsgWriteFailed=İleti '%S' dizinine taşınamadı veya kopyalanamadı, çünkü dosyaya yazma işlemi hatalı. Daha fazla boş yer yapmak için,  dosya menüsünden ilk önce 'çöpü boşalt' seçin, sonra 'Dosya sıklaştır' ve sonra tekrar deneyin.
-cantMoveMsgWOBodyOffline=Çevrim dışı çalışırken, çevrim dışı kullanımı için indirdiğiniz iletileri taşıyabilir veya kopyalayabilirsiniz. İleti penceresindeki Dosya menüsünü seçin ve çevrim dışı seçeneğini seçin ve sonra çevrim içi çalışın, ve işlemi tekrar edin.
+copyMsgWriteFailed=İletiler '%S' dizinine taşınırken/kopyalanırken hata oluştu. Diskinizde yer açmak için, Dosya menüsünden Çöpü Boşalt ve Dizinleri sıkıştır seçeneklerini seçip tekrar deneyin.
+cantMoveMsgWOBodyOffline=Çevrimdışı çalışırken, çevrim dışı kullanımı için indirdiğiniz iletileri taşıyabilir veya kopyalayabilirsiniz. İleti penceresindeki Dosya menüsünü seçin ve çevrim dışı seçeneğini seçin ve sonra çevrimiçi çalışın, ve işlemi tekrar edin.
 operationFailedFolderBusy=İşlem hatalı, çünkü dosya başka işlemler tarafından kullanılıyor. Lütfen diğer işlemlerin bitmesini bekleyin ve tekrar deneyin.
 folderRenameFailed=Dizin tekrar isimlendirilemedi. Ya isim geçerli değil yada dizin için izniniz yok.
 # LOCALIZATION NOTE(verboseFolderFormat): %1$S is folder name, %2$S is server name
@@ -306,7 +306,7 @@ mailnews.search_date_separator=
 mailnews.search_date_leading_zeros=true
 
 # offline msg 
-nocachedbodybody=Posta konu kısmı sunucudan indirilmedi. Konuyu okumak için ağa bağlanıp, çevrim dışı düğmesini dosya menüsünden seçin ve sonra çevrim içi çalış seçin.
+nocachedbodybody=Posta konu kısmı sunucudan indirilmedi. Konuyu okumak için ağa bağlanıp, çevrimdışı düğmesini dosya menüsünden seçin ve sonra çevrimiçi çalış seçin.
 
 # accountCentral
 mailnews.account_central_page.url=chrome://messenger/content/msgAccountCentral.xul
Index: thunderbird/l10n/tr/mail/chrome/messenger/msgAccountCentral.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/msgAccountCentral.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/msgAccountCentral.dtd
@@ -14,7 +14,7 @@
 <!ENTITY searchMsgsLink.label "İletilerde ara">
 <!ENTITY filtersLink.label "İleti süzgeçlerinin yönetimi">
 <!ENTITY junkSettings.label "Gereksiz posta ayarları">
-<!ENTITY offlineLink.label "Çevrim dışı ayarları">
+<!ENTITY offlineLink.label "Çevrimdışı ayarları">
 
 <!ENTITY feedsSectionHdr.label "Canlı yer imleri">
 <!ENTITY subscribeFeeds.label "Aboneliklerini denetle">
Index: thunderbird/l10n/tr/mail/chrome/messenger/msgSynchronize.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/msgSynchronize.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/msgSynchronize.dtd
@@ -2,7 +2,7 @@
 
 <!ENTITY MsgSynchronize.label "Postaları indir ve eşzamanla">
 <!ENTITY MsgSelect.label "Kapalı hat için kullanılacaklar">
-<!ENTITY MsgSyncDesc.label "Posta Dizinlerini ve Haber gruplarını çevrim dışı kullanımı için seçtiyseniz, şimdi indirebilir yada eşzamanlayabilirsiniz.Bundan başka,&quot;Seç&quot; düğmesini tıklayarak posta dizinlerini ve haber gruplarını çevrim dışı kullanımı için seçebilirsiniz.">
+<!ENTITY MsgSyncDesc.label "Posta Dizinlerini ve Haber gruplarını çevrimdışı kullanımı için seçtiyseniz, şimdi indirebilir yada eşzamanlayabilirsiniz.Bundan başka,&quot;Seç&quot; düğmesini tıklayarak posta dizinlerini ve haber gruplarını çevrim dışı kullanımı için seçebilirsiniz.">
 <!ENTITY MsgSyncDirections.label "Şunu İndir ve/veya eşzamanla:">
 <!ENTITY syncTypeMail.label "Posta mesajları">
 <!ENTITY syncTypeMail.accesskey "e">
@@ -10,10 +10,10 @@
 <!ENTITY syncTypeNews.accesskey "H">
 <!ENTITY sendMessage.label "Gönderilmemiş postaları gönder">
 <!ENTITY sendMessage.accesskey "G">
-<!ENTITY workOffline.label "İndirme ve/veya eşzamanlama bittikten sonra çevrim dışı çalış">
+<!ENTITY workOffline.label "İndirme ve/veya eşzamanlama bittikten sonra çevrimdışı çalış">
 <!ENTITY workOffline.accesskey "a">
 <!ENTITY selectButton.label "Seç…">
 <!ENTITY selectButton.accesskey "e">
-<!ENTITY MsgSelectDesc.label "Posta dizinlerini ve haber gruplarını çevrim dışı kullanımı için seç.">
+<!ENTITY MsgSelectDesc.label "Posta dizinlerini ve haber gruplarını çevrimdışı kullanımı için seç.">
 <!ENTITY MsgSelectInd.label "İndir">
 <!ENTITY MsgSelectItems.label "Dizinler ve haber grupları">
Index: thunderbird/l10n/tr/mail/chrome/messenger/news.properties
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/news.properties
+++ thunderbird/l10n/tr/mail/chrome/messenger/news.properties
@@ -48,7 +48,7 @@ cancelConfirm=Bu gönderiden vazgeçmek
 messageCancelled=Gönderi iptal edildi.
 enterUsername=Lütfen haber sunucusu bağlanmak için kullanıcı adı girin:
 enterUsernameTitle=Haber sunucusu kullanıcı adı gerekli
-saveUsername=Bu değeri kaydetmek için parola yöneticisini kullan.
+saveUsername=Bu değeri kaydetmek için parola yöneticisini kullan
 enterPassword=Lütfen haber sunucusu bağlanmak için parola girin:
 enterPasswordTitle=Haber Sunucusu Parolası Gerekli
 okButtonText=İndir
Index: thunderbird/l10n/tr/mail/chrome/messenger/offline.properties
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/offline.properties
+++ thunderbird/l10n/tr/mail/chrome/messenger/offline.properties
@@ -1,20 +1,20 @@
 # Download Messages Prompt
 downloadMessagesWindowTitle1=İletileri indir
-downloadMessagesLabel1=Çevrim dışı olmadan önce iletileri çevrim dışıyken okuyabilmek için indirmek ister misiniz?
-downloadMessagesCheckboxLabel1=Her çevrim dışı oluşumda bana sor
+downloadMessagesLabel1=Çevrimdışı olmadan önce iletileri çevrim dışıyken okuyabilmek için indirmek ister misiniz?
+downloadMessagesCheckboxLabel1=Her çevrimdışı oluşumda bana sor
 
 # Send Messages Prompt
 sendMessagesWindowTitle1=Gönderilmeyen iletiler
 sendMessagesLabel2=Gönderilmeyen iletileri şimdi göndermek ister misiniz?
-sendMessagesCheckboxLabel1=Her çevrim içi oluşumda bana sor
+sendMessagesCheckboxLabel1=Her çevrimiçi oluşumda bana sor
 
 # GetMessages While Offline Prompt 
 getMessagesOfflineWindowTitle1=İleti alma
-getMessagesOfflineLabel1=Şu an çevrim dışısınız. Yeni iletileri almak için çevrim içi olmak ister misiniz?
+getMessagesOfflineLabel1=Şu an çevrimdışısınız. Yeni iletileri almak için çevrimiçi olmak ister misiniz?
 
 # Send Messages Offline Prompt
 sendMessagesOfflineWindowTitle1=Gönderilmeyen iletiler
-sendMessagesOfflineLabel1=Şu an çevrim dışısınız. Gönderilmeyen iletileri göndermek için çevrim içi olmak ister misiniz?
+sendMessagesOfflineLabel1=Şu an çevrimdışısınız. Gönderilmeyen iletileri göndermek için çevrimiçi olmak ister misiniz?
 
-offlineTooltip=Şu an çevrim dışısınız.
-onlineTooltip=Şu an çevrim içisiniz.
+offlineTooltip=Şu an çevrimdışısınız.
+onlineTooltip=Şu an çevrimiçisiniz.
Index: thunderbird/l10n/tr/mail/chrome/messenger/offlineStartup.properties
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/offlineStartup.properties
+++ thunderbird/l10n/tr/mail/chrome/messenger/offlineStartup.properties
@@ -1,4 +1,4 @@
-title=Çevrim içi çalış
-desc=Çevrim içi olmak ister misiniz?\n\n(Çevrim dışı çalışmayı seçerseniz daha tekrar Menüden 'Çevrim içi çalış' seçeneğini seçebilirsiniz.)
-workOnline=Çevrim içi çalış
-workOffline=Çevrim dışı çalış
+title=Çevrimiçi çalış
+desc=Çevrimiçi olmak ister misiniz?\n\n(Çevrimdışı çalışmayı seçerseniz daha tekrar Menüden 'Çevrim içi çalış' seçeneğini seçebilirsiniz.)
+workOnline=Çevrimiçi çalış
+workOffline=Çevrimdışı çalış
Index: thunderbird/l10n/tr/mail/chrome/messenger/preferences/advanced.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/preferences/advanced.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/preferences/advanced.dtd
@@ -1,34 +1,34 @@
 <!ENTITY itemGeneral.label "Genel">
-<!ENTITY itemDisplay.label "Okuma ve görünüm">
+<!ENTITY itemDisplay.label "Okuma &amp; Görünüm">
 <!ENTITY itemUpdate.label "Güncelleme">
-<!ENTITY itemNetworking.label "Ağ ve disk alanı">
-<!ENTITY itemCertificates.label "Onay belgeleri">
+<!ENTITY itemNetworking.label "Ağ ve Disk Alanı">
+<!ENTITY itemCertificates.label "Sertifikalar">
 <!-- General Settings -->
 
 
-<!ENTITY enableGlodaSearch.label "Global aramayı ve endekslemeyi etkin kıl">
+<!ENTITY enableGlodaSearch.label "Global aramayı ve endekslemeyi etkinleştir">
 <!ENTITY enableGlodaSearch.accesskey "e">
-<!ENTITY systemIntegration.label "Sistem entegrasyonu">
+<!ENTITY systemIntegration.label "Sistem Entegrasyonu">
 <!ENTITY alwaysCheckDefault.label "&brandShortName; uygulamasının öntanımlı e-posta istemcisi olarak kullanıldığını her açılışta denetle">
 <!ENTITY alwaysCheckDefault.accesskey "h">
 <!ENTITY searchIntegration.label "&searchIntegration.engineName; ileti aramasına izin ver">
 <!ENTITY searchIntegration.accesskey "s">
 <!ENTITY checkNow.label "Şimdi denetle">
 <!ENTITY checkNow.accesskey "d">
-<!ENTITY configEditDesc.label "Gelişmiş ayarlar">
+<!ENTITY configEditDesc.label "Gelişmiş Ayarlar">
 <!ENTITY configEdit.label "Ayar düzenleyici">
 <!ENTITY configEdit.accesskey "y">
-<!ENTITY returnReceiptsInfo.label "&brandShortName; uygulamasının alındı onayı gönderme ayarlarını yapın">
+<!ENTITY returnReceiptsInfo.label "&brandShortName; uygulamasının alındı onayı ayarlarını yapılandırın">
 <!ENTITY showReturnReceipts.label "Alındı onayı…">
 <!ENTITY showReturnReceipts.accesskey "A">
 <!-- Display and Reading Settings -->
 
 <!ENTITY reading.caption "Okuma">
 <!ENTITY display.caption "Görünüm">
-<!ENTITY showCondensedAddresses.label "Adres defterimdeki kişilerin sadece ekran ismini göster.">
+<!ENTITY showCondensedAddresses.label "Adres defterimdeki kişilerin sadece ekran ismini göster">
 <!ENTITY showCondensedAddresses.accesskey "d">
 
-<!ENTITY autoMarkAsRead.label "İletileri okunmuş olarak otomatik işaretle">
+<!ENTITY autoMarkAsRead.label "İletileri otomatik olarak okunmuş işaretle">
 <!ENTITY autoMarkAsRead.accesskey "o">
 <!ENTITY markAsReadNoDelay.label "Görüntüleme anında">
 <!ENTITY markAsReadNoDelay.accesskey "G">
@@ -41,7 +41,7 @@
 <!ENTITY openMsgIn.label "Yeni postayı">
 <!ENTITY openMsgInNewTab.label "Yeni sekmede aç">
 <!ENTITY openMsgInNewTab.accesskey "s">
-<!ENTITY reuseExpRadio0.label "Yeni bir pencerede aç">
+<!ENTITY reuseExpRadio0.label "Yeni pencerede aç">
 <!ENTITY reuseExpRadio0.accesskey "i">
 <!ENTITY reuseExpRadio1.label "Mevcut bir iletinin içinde aç">
 <!ENTITY reuseExpRadio1.accesskey "c">
@@ -49,7 +49,7 @@
 <!ENTITY closeMsgWindowOnDelete.accesskey "c">
 <!-- Update -->
 
-<!ENTITY autoCheck.label "Periyodik olarak güncellemeleri kontrol edilecekler:">
+<!ENTITY autoCheck.label "Düzenli olarak güncellemeleri denetlenecek öğeler:">
 <!ENTITY enableAppUpdate.label "&brandShortName;">
 <!ENTITY enableAppUpdate.accesskey "d">
 <!ENTITY enableAddonsUpdate.label "Kurulu eklentiler">
@@ -69,9 +69,9 @@
 <!ENTITY showSettings.accesskey "A">
 <!ENTITY proxiesConfigure.label "&brandShortName; İnternet bağlantısını ayarla">
 <!ENTITY connectionsInfo.caption "Bağlantı">
-<!ENTITY offlineInfo.caption "Çevrim dışı">
-<!ENTITY offlineInfo.label "Çevrim dışı çalışma ayarları">
-<!ENTITY showOffline.label "Çevrim dışı…">
+<!ENTITY offlineInfo.caption "Çevrimdışı">
+<!ENTITY offlineInfo.label "Çevrimdışı çalışma ayarları">
+<!ENTITY showOffline.label "Çevrimdışı…">
 <!ENTITY showOffline.accesskey "d">
 
 <!ENTITY Diskspace "Boş disk alanı">
@@ -92,17 +92,17 @@
 <!ENTITY clearCacheNow.accesskey "t">
 <!-- Certificates -->
 
-<!ENTITY certSelection.description "Sunucu kişisel onay belgemi istediğinde:">
+<!ENTITY certSelection.description "Bir sunucu kişisel sertifikamı istediğinde:">
 <!ENTITY certs.auto "Birini otomatik olarak seç">
 <!ENTITY certs.auto.accesskey "m">
 <!ENTITY certs.ask "Her seferinde bana sor">
 <!ENTITY certs.ask.accesskey "H">
 
-<!ENTITY viewCertificates.label "Onay belgelerini göster">
+<!ENTITY viewCertificates.label "Sertifikaları Göster">
 <!ENTITY viewCertificates.accesskey "a">
-<!ENTITY viewCRLs.label "Hükümsüz sayılanları göster">
+<!ENTITY viewCRLs.label "İptal Listeleri">
 <!ENTITY viewCRLs.accesskey "e">
 <!ENTITY validation.label "Doğrulama">
 <!ENTITY validation.accesskey "D">
-<!ENTITY viewSecurityDevices.label "Güvenlik aygıtları">
+<!ENTITY viewSecurityDevices.label "Güvenlik Aygıtları">
 <!ENTITY viewSecurityDevices.accesskey "G">
Index: thunderbird/l10n/tr/mail/chrome/messenger/preferences/connection.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/preferences/connection.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/preferences/connection.dtd
@@ -4,19 +4,19 @@
 <!ENTITY proxyTitle.label "İnternet vekil sunucu ayarları">
 <!ENTITY noProxyTypeRadio.label "Vekil sunucu kullanma">
 <!ENTITY noProxyTypeRadio.accesskey "n">
-<!ENTITY systemTypeRadio.label "Sistem vekil sunucusu ayarlarını kullan">
+<!ENTITY systemTypeRadio.label "Sistemin vekil sunucu ayarlarını kullan">
 <!ENTITY systemTypeRadio.accesskey "n">
-<!ENTITY WPADTypeRadio.label "Vekil sunucu ayarlarını otomatik tanı">
+<!ENTITY WPADTypeRadio.label "Vekil sunucu ayarlarını otomatik algıla">
 <!ENTITY WPADTypeRadio.accesskey "s">
-<!ENTITY manualTypeRadio.label "Vekil sunucu kullan">
+<!ENTITY manualTypeRadio.label "Vekil sunucuyu elle belirle">
 <!ENTITY manualTypeRadio.accesskey "i">
 <!ENTITY autoTypeRadio.label "Otomatik vekil sunucu ayarlama adresi:">
 <!ENTITY autoTypeRadio.accesskey "a">
 <!ENTITY reload.label "Tekrar yükle">
 <!ENTITY reload.accesskey "T">
-<!ENTITY http.label "HTTP vekil sunucusu">
+<!ENTITY http.label "HTTP vekil sunucu:">
 <!ENTITY http.accesskey "H">
-<!ENTITY ssl.label "SSL vekil sunucusu:">
+<!ENTITY ssl.label "SSL vekil sunucu:">
 <!ENTITY ssl.accesskey "S">
 <!ENTITY socks.label "SOCKS makine:">
 <!ENTITY socks.accesskey "C">
@@ -33,5 +33,5 @@
 <!ENTITY noproxy.label "Vekil sunucu kullanma">
 <!ENTITY noproxy.accesskey "n">
 <!ENTITY noproxyExplain.label "Örnek: .linux-sevenler.org">
-<!ENTITY shareproxy.label "Vekil sunucuyu bütün protokollerde kullan">
+<!ENTITY shareproxy.label "Vekil sunucusunu bütün protokollerde kullan">
 <!ENTITY shareproxy.accesskey "V">
Index: thunderbird/l10n/tr/mail/chrome/messenger/preferences/offline.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/preferences/offline.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/preferences/offline.dtd
@@ -1,21 +1,21 @@
-<!ENTITY offlineDialog.title "Çevrim dışı çalışma ayarları">
+<!ENTITY offlineDialog.title "Çevrimdışı çalışma ayarları">
 
 <!ENTITY textStartUp "Başlangıçta bunları yap:">
 <!ENTITY radioRememberPrevState.label "Önceki bağlanma durumunu hatırla">
 <!ENTITY radioRememberPrevState.accesskey "r">
-<!ENTITY radioAskMe.label "Bana çevrim içi olunup olunmayacağını sor">
+<!ENTITY radioAskMe.label "Bana çevrimiçi olunup olunmayacağını sor">
 <!ENTITY radioAskMe.accesskey "u">
-<!ENTITY radioAlwaysOnline.label "Hep çevrim içi başla">
+<!ENTITY radioAlwaysOnline.label "Hep çevrimiçi başla">
 <!ENTITY radioAlwaysOnline.accesskey "l">
 
-<!ENTITY textGoingOnline "Çevrim içi olunca gönderilmeyen iletiler gönderilsin mi?">
+<!ENTITY textGoingOnline "Çevrimiçi olunca gönderilmeyen iletiler gönderilsin mi?">
 <!ENTITY radioAutoSend.label "Evet">
 <!ENTITY radioAutoSend.accesskey "E">
 <!ENTITY radioNotSend.label "Hayır">
 <!ENTITY radioNotSend.accesskey "H">
 <!ENTITY radioAskUnsent.label "Bana sor">
 <!ENTITY radioAskUnsent.accesskey "s">
-<!ENTITY textGoingOffline "Çevrim dışı olurken indirilmeyen iletiler indirilsin mi?">
+<!ENTITY textGoingOffline "Çevrimdışı olurken indirilmeyen iletiler indirilsin mi?">
 <!ENTITY radioAutoDownload.label "Evet">
 <!ENTITY radioAutoDownload.accesskey "E">
 <!ENTITY radioNotDownload.label "Hayır">
Index: thunderbird/l10n/tr/mail/chrome/messenger/virtualFolderProperties.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/virtualFolderProperties.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/virtualFolderProperties.dtd
@@ -10,5 +10,5 @@
 <!ENTITY chooseFoldersButton.label "Seçin…">
 <!ENTITY chooseFoldersButton.accesskey "S">
 
-<!ENTITY searchOnline.label "Çevrim içi ara ">
+<!ENTITY searchOnline.label "Çevrimiçi ara ">
 <!ENTITY searchOnline.accesskey "a">
Index: thunderbird/l10n/tr/mail/chrome/overrides/netError.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/overrides/netError.dtd
+++ thunderbird/l10n/tr/mail/chrome/overrides/netError.dtd
@@ -41,9 +41,9 @@ adres yanlış yazılmış olabilir.</li
 <!ENTITY netInterrupt.title "Bağlantı kesilmiş">
 <!ENTITY netInterrupt.longDesc "&sharedLongDesc;">
 
-<!ENTITY netOffline.title "Çevrim dışı bağlantı">
+<!ENTITY netOffline.title "Çevrimdışı bağlantı">
 <!ENTITY netOffline.longDesc "<ul> 
-<li>&brandShortName; programını Çevrim içi olarak kullanın ve yeniden deneyin.</li> 
+<li>&brandShortName; programını Çevrimiçi olarak kullanın ve yeniden deneyin.</li> 
 </ul>">
 
 <!ENTITY contentEncodingError.title "İçerik kodlama hatası">
Index: thunderbird/l10n/tr/mobile/chrome/browser.properties
===================================================================
--- thunderbird.orig/l10n/tr/mobile/chrome/browser.properties
+++ thunderbird/l10n/tr/mobile/chrome/browser.properties
@@ -115,7 +115,7 @@ typeError=Hata:
 typeWarning=Uyarı:
 
 # Offline web applications
-offlineApps.available=%S çevrim dışı kullanım için bilgisayarınıza bilgi kaydedecektir.
+offlineApps.available=%S çevrimdışı kullanım için bilgisayarınıza bilgi kaydedecektir.
 offlineApps.allow=İzin ver
 offlineApps.never=İzin verme
 offlineApps.notNow=Şimdi değil
@@ -147,5 +147,5 @@ pageactions.password.forget=Forget Passw
 pageactions.reset=Clear Site Preferences
 pageactions.geo=Konum
 pageactions.popup=Popups
-pageactions.offline-app=Çevrim dışı depolama
+pageactions.offline-app=Çevrimdışı depolama
 pageactions.password=Parola
Index: thunderbird/l10n/tr/mobile/chrome/overrides/appstrings.properties
===================================================================
--- thunderbird.orig/l10n/tr/mobile/chrome/overrides/appstrings.properties
+++ thunderbird/l10n/tr/mobile/chrome/overrides/appstrings.properties
@@ -47,7 +47,7 @@ confirmRepostPrompt=%S uygulaması bu sa
 resendButton.label=Tekrar gönder
 unknownSocketType=Tarayıcı sunucuyla nasıl bağlantı kurulacağını bilmiyor.
 netReset=Sayfa yüklenirken sunucuyla bağlantı kesildi.
-netOffline=Tarayıcı şu an çevrim dışı kipte ve internette gezinti yapamaz.
+netOffline=Tarayıcı şu an çevrimdışı kipte ve internette gezinti yapamaz.
 isprinting=Belge basılırken ya da yazdırma önizlemesindeyken değiştirilemez.
 deniedPortAccess=Bu adres taramadan başka amaçlar için kullanılan bir ağ bağlantı noktası kullanıyor. Tarayıcınız güvenliğiniz için bu bağlanma isteğini yerine getirmedi.
 proxyResolveFailure=Tarayıcı bulunamayan bir vekil sunucuya bağlanmak üzere ayarlanmış.
Index: thunderbird/l10n/tr/mobile/chrome/overrides/netError.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mobile/chrome/overrides/netError.dtd
+++ thunderbird/l10n/tr/mobile/chrome/overrides/netError.dtd
@@ -46,7 +46,7 @@ to pickup this string change. Please upd
 <li>Yeniden deneyin. &brandShortName; bağlantıyı açıp sayfayı tazelemeyi çalışacaktır.</li> 
 </ul>">
 
-<!ENTITY netOffline.title "Çevrim dışı tarama kipi">
+<!ENTITY netOffline.title "Çevrimdışı tarama kipi">
 <!ENTITY netOffline.longDesc "&netOffline2.longDesc;">
 
 <!ENTITY contentEncodingError.title "İçerik kodlama hatası">
Index: thunderbird/l10n/tr/mobile/chrome/overrides/passwordmgr.properties
===================================================================
--- thunderbird.orig/l10n/tr/mobile/chrome/overrides/passwordmgr.properties
+++ thunderbird/l10n/tr/mobile/chrome/overrides/passwordmgr.properties
@@ -36,8 +36,8 @@
 #
 # ***** END LICENSE BLOCK *****
 
-rememberValue = Bu değeri kaydetmek için parola yöneticisini kullan.
-rememberPassword = Bu parolayı hatırlamak için parola yöneticisini kullan.
+rememberValue = Bu değeri kaydetmek için parola yöneticisini kullan
+rememberPassword = Bu parolayı hatırlamak için parola yöneticisini kullan
 savePasswordTitle = Onayla
 # 1st string is product name, 2nd is the username for the login, 3rd is the
 #   login's hostname. Note that long usernames may be truncated.
Index: thunderbird/l10n/tr/other-licenses/branding/thunderbird/brand.dtd
===================================================================
--- thunderbird.orig/l10n/tr/other-licenses/branding/thunderbird/brand.dtd
+++ thunderbird/l10n/tr/other-licenses/branding/thunderbird/brand.dtd
@@ -1,5 +1,5 @@
 <!ENTITY brandShortName "Thunderbird">
 <!ENTITY brandFullName "Mozilla Thunderbird">
 <!ENTITY vendorShortName "Mozilla Thunderbird">
-<!ENTITY logoTrademark "Mozilla Thunderbird and the Thunderbird logos
-                                 are trademarks of the Mozilla Foundation.">
+<!ENTITY logoTrademark "Mozilla Thunderbird ve Thunderbird logoları
+                                 Mozilla Vakfı'nın tescilli ticarı markalarıdır.">
Index: thunderbird/l10n/tr/toolkit/chrome/global/console.dtd
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/global/console.dtd
+++ thunderbird/l10n/tr/toolkit/chrome/global/console.dtd
@@ -53,15 +53,15 @@
 <!ENTITY clear.accesskey "l">
 <!ENTITY codeEval.label "Kod:">
 <!ENTITY codeEval.accesskey "o">
-<!ENTITY evaluate.label "Değerlendirme">
+<!ENTITY evaluate.label "Çalıştır">
 <!ENTITY evaluate.accesskey "m">
 
 <!ENTITY copyCmd.label "Kopyala">
 <!ENTITY copyCmd.accesskey "K">
 <!ENTITY copyCmd.commandkey "C">
-<!ENTITY sortFirst.label "İlk > Son  sıralama düzeni">
+<!ENTITY sortFirst.label "İlkten sona doğru sırala">
 <!ENTITY sortFirst.accesskey "S">
-<!ENTITY sortLast.label "Son > İlk sıralama düzeni">
+<!ENTITY sortLast.label "Sondan ilke doğru sırala">
 <!ENTITY sortLast.accesskey "k">
 <!ENTITY closeCmd.commandkey "w">
 <!ENTITY focus1.commandkey "l">
Index: thunderbird/l10n/tr/toolkit/chrome/mozapps/downloads/downloads.properties
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/mozapps/downloads/downloads.properties
+++ thunderbird/l10n/tr/toolkit/chrome/mozapps/downloads/downloads.properties
@@ -24,8 +24,8 @@ quitCancelDownloadsAlertMsgMultiple=Eğe
 quitCancelDownloadsAlertMsgMac=Eğer şimdi çıkarsanız 1 dosyanın indirilmesinden vazgeçilecek. Gerçekten çıkmak istiyor musunuz?
 quitCancelDownloadsAlertMsgMacMultiple=Eğer şimdi çıkarsanız %S dosyanın indirilmesinden vazgeçilecek. Çıkmak istiyor musunuz?
 offlineCancelDownloadsAlertTitle=İndirmekten vazgeçmek istiyor musunuz?
-offlineCancelDownloadsAlertMsg=Eğer şimdi çevrim dışı olursanız indirilen 1 dosyadan vazgeçilecek. Çevrim dışı olmak ister misiniz?
-offlineCancelDownloadsAlertMsgMultiple=Eğer şimdi çevrim dışı olursanız indirilen %S dosyadan vazgeçilecek. Çevrim dışı olmak ister misiniz?
+offlineCancelDownloadsAlertMsg=Eğer şimdi çevrimdışı olursanız indirilen 1 dosyadan vazgeçilecek. Çevrim dışı olmak ister misiniz?
+offlineCancelDownloadsAlertMsgMultiple=Eğer şimdi çevrimdışı olursanız indirilen %S dosyadan vazgeçilecek. Çevrim dışı olmak ister misiniz?
 enterPrivateBrowsingCancelDownloadsAlertTitle=İndirmekten vazgeçilsin mi?
 enterPrivateBrowsingCancelDownloadsAlertMsg=Şimdi gizli taramaya geçerseniz 1 dosyanın indirilmesinden vazgeçilecek. Gizli taramaya geçmek istediğinizden emin misiniz?
 enterPrivateBrowsingCancelDownloadsAlertMsgMultiple=Şimdi gizli taramaya geçerseniz %S dosyanın indirilmesinden vazgeçilecek. Gizli taramaya geçmek istediğinizden emin misiniz?
@@ -36,7 +36,7 @@ cancelDownloadsOKText=Evet, indirmekten
 cancelDownloadsOKTextMultiple=Evet, %S dosyayı indirmekten vazgeç
 dontQuitButtonWin=Hayır, indirmeye devam et
 dontQuitButtonMac=Hayır, indirmeye devam et
-dontGoOfflineButton=Çevrim içi kal
+dontGoOfflineButton=Çevrimiçi kal
 dontEnterPrivateBrowsingButton=Gizli taramaya geçme
 dontLeavePrivateBrowsingButton=Gizli taramayı sürdür
 downloadsCompleteTitle=İndirme Tamamlandı
Index: thunderbird/l10n/tr/toolkit/chrome/mozapps/extensions/extensions.dtd
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/mozapps/extensions/extensions.dtd
+++ thunderbird/l10n/tr/toolkit/chrome/mozapps/extensions/extensions.dtd
@@ -9,7 +9,7 @@
 <!ENTITY cmd.close.commandKey "w">
 <!-- View labels -->
 
-<!ENTITY search.label "Eklenti edin">
+<!ENTITY search.label "Eklenti Kurun">
 <!ENTITY extensions.label "Eklentiler">
 <!ENTITY themes.label "Temalar">
 <!ENTITY locales.label "Diller">
@@ -18,10 +18,10 @@
 <!ENTITY install.label "Kurulum">
 <!-- Command Bar items -->
 
-<!ENTITY cmd.checkUpdatesAll.label "Güncellemelere bak">
+<!ENTITY cmd.checkUpdatesAll.label "Güncellemelere gözat">
 <!ENTITY cmd.checkUpdatesAll.accesskey "b">
-<!ENTITY cmd.checkUpdatesAllAddon.tooltip "Eklenti güncellemelerine bak">
-<!ENTITY cmd.checkUpdatesAllTheme.tooltip "Tema güncellemelerine bak">
+<!ENTITY cmd.checkUpdatesAllAddon.tooltip "Eklenti güncellemelerine gözat">
+<!ENTITY cmd.checkUpdatesAllTheme.tooltip "Tema güncellemelerine gözat">
 <!ENTITY cmd.checkUpdatesAllPlugin.tooltip "Eklenti güncellemelerini denetler">
 <!ENTITY cmd.installLocalFile.label "Kur…">
 <!ENTITY cmd.installLocalFile.accesskey "K">
@@ -100,7 +100,7 @@
 <!ENTITY cmd.homepage.accesskey "s">
 <!ENTITY cmd.about2.label "Eklenti hakkında">
 <!ENTITY cmd.about.accesskey "E">
-<!ENTITY cmd.checkUpdate.label "Güncellemelere bak">
+<!ENTITY cmd.checkUpdate.label "Güncellemelere gözat">
 <!ENTITY cmd.checkUpdate.accesskey "b">
 
 <!ENTITY cmd.includeUpdate.label "Güncellemeleri içersin">
@@ -126,7 +126,7 @@
 <!ENTITY getPlugins.label "Yan uygulama indir">
 
 <!ENTITY searchAddons.label "Tüm eklentileri ara">
-<!ENTITY browseAddons.label "Tüm eklentilere bak">
+<!ENTITY browseAddons.label "Tüm eklentilere gözat">
 <!ENTITY searchFailed.label "&brandShortName; eklentilere ulaşamadı">
 <!ENTITY recommendedHeader.label "Tavsiye edilenler">
 <!ENTITY recommendedThrobber.label "Tavsiye edilen eklentilere ulaşılıyor">
@@ -140,7 +140,7 @@
 <!ENTITY searchFailed.button "Tamam">
 <!ENTITY searchResultHomepage.value "Daha fazla bilgi al">
 <!ENTITY searchBox.label "Tüm eklentileri ara">
-<!ENTITY recommendedResults.label "Tavsiye edilen tüm eklentilere bak">
+<!ENTITY recommendedResults.label "Tavsiye edilen tüm eklentilere gözat">
 <!ENTITY searchResultConnecting.label "Bağlanılıyor…">
 <!ENTITY searchResultInstalling.label "Kuruluyor…">
 <!ENTITY searchResultFailed.label "Kurulum başarısız oldu">
@@ -165,7 +165,7 @@
 <!ENTITY installFinishing.label "Kuruluyor…">
 <!ENTITY installFailure.label "Kurulum başarısız.">
 
-<!ENTITY progressStatus.label "Güncellemelere bakılıyor">
+<!ENTITY progressStatus.label "Güncellemeler inceleniyor">
 
 <!ENTITY eula.title "Son kullanıcı lisans sözleşmesi">
 <!ENTITY eula.width "560px">
Index: thunderbird/l10n/tr/toolkit/chrome/mozapps/extensions/extensions.properties
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/mozapps/extensions/extensions.properties
+++ thunderbird/l10n/tr/toolkit/chrome/mozapps/extensions/extensions.properties
@@ -101,11 +101,11 @@ safeModeMsg=Tüm eklentiler güvenli kip
 disabledCompatMsg=Eklenti uyumluluk denetimi devre dışı. Uyumsuz eklentileriniz olabilir.
 disabledUpdateSecurityMsg=Eklenti güncellemesinde güvenlik denetimi devre dışı. Güncellemeler sizin için tehdit unsuru olabilir.
 noUpdatesMsg=Güncelleme bulunamadı.
-offlineUpdateMsg2=%S hâlihazırda çevrim dışı ve eklentileri güncelleyemiyor. Çevrim içi olma seçeneğine tıklayıp yeniden deneyin.
-offlineSearchMsg=%S şu an çevrim dışı ve eklenti arayamıyor. Çevrim içi olup yeniden deneyin.
+offlineUpdateMsg2=%S hâlihazırda çevrimdışı ve eklentileri güncelleyemiyor. Çevrimiçi olma seçeneğine tıklayıp yeniden deneyin.
+offlineSearchMsg=%S şu an çevrimdışı ve eklenti arayamıyor. Çevrimiçi olup yeniden deneyin.
 enableButtonLabel=Etkinleştir
 enableButtonAccesskey=n
-goOnlineButtonLabel=Çevrim içi ol
+goOnlineButtonLabel=Çevrimiçi ol
 goOnlineButtonAccesskey=o
 
 newUpdateWindowTitle=%S eklenti güncellemeleri
Index: thunderbird/l10n/tr/toolkit/chrome/mozapps/extensions/update.dtd
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/mozapps/extensions/update.dtd
+++ thunderbird/l10n/tr/toolkit/chrome/mozapps/extensions/update.dtd
@@ -1,8 +1,8 @@
 <!ENTITY updateWizard.title "&brandShortName; Güncelleme">
 
-<!ENTITY offline.title "&brandShortName; şu anda çevrim dışı çalışıyor">
-<!ENTITY offline.description "&brandShortName; uygulamasının güncellemeleri kontrol edebilmesi ve uyumlu eklentileri indirebilmesi için çevrim içi olması gereklidir.">
-<!ENTITY offline.toggleOffline.label "Çevrim içi ol.">
+<!ENTITY offline.title "&brandShortName; şu anda çevrimdışı çalışıyor">
+<!ENTITY offline.description "&brandShortName; uygulamasının güncellemeleri kontrol edebilmesi ve uyumlu eklentileri indirebilmesi için çevrimiçi olması gereklidir.">
+<!ENTITY offline.toggleOffline.label "Çevrimiçi ol.">
 <!ENTITY offline.toggleOffline.accesskey "v">
 
 <!ENTITY mismatch.win.title "Uyumsuz Eklentiler">
Index: thunderbird/l10n/tr/toolkit/chrome/mozapps/preferences/ocsp.dtd
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/mozapps/preferences/ocsp.dtd
+++ thunderbird/l10n/tr/toolkit/chrome/mozapps/preferences/ocsp.dtd
@@ -1,5 +1,5 @@
 <!ENTITY ocspDialog.title "Onaylama">
-<!ENTITY enableOCSP.label "Onay belgelerinin şu anki geçerliliğini doğrulamak için Çevrim içi onay belgesi durumu iletişim kuralı (OCSP)'nı kullan">
+<!ENTITY enableOCSP.label "Onay belgelerinin şu anki geçerliliğini doğrulamak için Çevrimiçi onay belgesi durumu iletişim kuralı (OCSP)'nı kullan">
 <!ENTITY enableOCSP.accesskey "u">
 <!ENTITY certOCSP2.label "OCSP sunucusu belirten bir onay belgesini doğrula">
 <!ENTITY certOCSP2.accesskey "d">
Index: thunderbird/l10n/tr/toolkit/chrome/mozapps/preferences/preferences.properties
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/mozapps/preferences/preferences.properties
+++ thunderbird/l10n/tr/toolkit/chrome/mozapps/preferences/preferences.properties
@@ -1,13 +1,13 @@
 #### Master Password
 
-password_not_set=(birşey ayarlanmadı)
+password_not_set=(bir şey ayarlanmadı)
 failed_pw_change=Ana parola değiştirilemiyor.
 incorrect_pw=Güncel ana parolayı girmediniz. Lütfen tekrar deneyiniz.
 pw_change_ok=Ana parola başarılı bir şekilde değiştirildi.
 pw_erased_ok=Ana parolanızı sildiniz.
 pw_not_wanted=Uyarı! Ana parola kullanmamaya karar verdiniz.
 pw_empty_warning=Kayıtlı web ve posta parolalarınız, form verileriniz ve özel anahtarlarınızın koruması şu anda kalkmıştır.
-pw_change2empty_in_fips_mode=şuanda FIPS kipindesiniz. FIPS kipinde ana parola bölümünü boş bırakamazsınız.
+pw_change2empty_in_fips_mode=Şu anda FIPS kipindesiniz. FIPS kipinde ana parola bölümünü boş bırakamazsınız.
 pw_change_success_title=Parola değiştirme başarılı
 pw_change_failed_title=Parola değiştirme başarısız
 pw_remove_button=Kaldır
Index: thunderbird/l10n/tr/toolkit/chrome/mozapps/profile/profileSelection.dtd
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/mozapps/profile/profileSelection.dtd
+++ thunderbird/l10n/tr/toolkit/chrome/mozapps/profile/profileSelection.dtd
@@ -57,7 +57,7 @@
 
 <!ENTITY pmDescription.label "&brandShortName; ayarlarınız, tercihleriniz ve kullanıcı kimliğinizdeki diğer öğelerle ilgili bilgileri depolar.">
 
-<!ENTITY offlineState.label "Çevrim dışı çalış">
+<!ENTITY offlineState.label "Çevrimdışı çalış">
 <!ENTITY offlineState.accesskey "e">
 
 <!ENTITY autoSelect.label "Başlangıçta sorma">
Index: thunderbird/l10n/tr/toolkit/chrome/mozapps/update/updates.properties
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/mozapps/update/updates.properties
+++ thunderbird/l10n/tr/toolkit/chrome/mozapps/update/updates.properties
@@ -90,13 +90,13 @@ check_error-2152398849=Başarısız (Seb
 check_error-2152398861=Bağlantı reddedildi
 check_error-2152398862=Bağlantı zaman aşımına uğradı
 # NS_ERROR_OFFLINE
-check_error-2152398864=Ağ şuanda çevrim dışı (çevrim içi olun)
+check_error-2152398864=Ağ şu anda çevrimdışı (çevrimiçi olun)
 check_error-2152398867=Port erişimine izin verilmedi
 check_error-2152398868=Veri alınamıdı (Lütfen tekrar deneyin)
 check_error-2152398878=Güncelleme sunucusu bulunamıyor (İnternet bağlantılarınızı tekrar kontrol edin veya sistem yöneticiniz ile irtibata geçip, durumu anlatın)
 check_error-2152398890=Vekil sunucu bulunamıyor (İnternet bağlantılarınızı tekrar kontrol edin veya sistem yöneticiniz ile irtibata geçip, durumu anlatın)
 # NS_ERROR_DOCUMENT_NOT_CACHED
-check_error-2152398918=Ağ çevrim dışı (çevrim içi ol)
+check_error-2152398918=Ağ çevrimdışı (çevrimiçi ol)
 check_error-2152398919=Veri alış verişinde hata oluştu (Lütfen tekrar deneyin)
 check_error-2152398920=Vekil sunucusu bağlantıyı reddetti
 check_error-2153390069=Sunucu onay belgesinin süresi doldu (Yanlışsa lütfen sistem tarih ve saatini doğru ayarlayın ya da ağ yöneticinize başvurun)
Index: thunderbird/l10n/tr/toolkit/chrome/passwordmgr/passwordmgr.properties
===================================================================
--- thunderbird.orig/l10n/tr/toolkit/chrome/passwordmgr/passwordmgr.properties
+++ thunderbird/l10n/tr/toolkit/chrome/passwordmgr/passwordmgr.properties
@@ -36,8 +36,8 @@
 #
 # ***** END LICENSE BLOCK *****
 
-rememberValue = Bu değeri kaydetmek için parola yöneticisini kullan.
-rememberPassword = Bu parolayı hatırlamak için parola yöneticisini kullan.
+rememberValue = Bu değeri kaydetmek için parola yöneticisini kullan
+rememberPassword = Bu parolayı hatırlamak için parola yöneticisini kullan
 savePasswordTitle = Onayla
 # 1st string is product name, 2nd is the username for the login, 3rd is the
 #   login's hostname. Note that long usernames may be truncated.
Index: thunderbird/l10n/tr/mail/chrome/messenger/preferences/compose.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/preferences/compose.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/preferences/compose.dtd
@@ -1,6 +1,6 @@
 <!ENTITY itemGeneral.label "Genel">
-<!ENTITY itemAutoComplete.label "Adreslendirme">
-<!ENTITY itemSpellCheck.label "İmla denetimi">
+<!ENTITY itemAutoComplete.label "Otomatik Tamamla">
+<!ENTITY itemSpellCheck.label "İmla Denetimi">
 
 <!ENTITY forwardMsg.label "İletileri">
 <!ENTITY forwardMsg.accesskey "İ">
@@ -21,9 +21,9 @@
 <!ENTITY restoreHTMLDefaults.label "Öntanımlı değerler">
 <!ENTITY restoreHTMLDefaults.accesskey "ğ">
 
-<!ENTITY spellCheck.label "Yazım hatalarını göndermeden önce kontrol et">
+<!ENTITY spellCheck.label "Yazım hatalarını göndermeden önce denetle">
 <!ENTITY spellCheck.accesskey "a">
-<!ENTITY spellCheckInline.label "Yazı esnasında imla denetimini etkinleştir">
+<!ENTITY spellCheckInline.label "Yazım esnasında imla denetimini etkinleştir">
 <!ENTITY spellCheckInline1.accesskey "k">
 <!ENTITY languagePopup.label "Diller:">
 <!ENTITY languagePopup.accessKey "D">
Index: thunderbird/l10n/tr/mail/chrome/messenger/preferences/display.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/preferences/display.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/preferences/display.dtd
@@ -3,21 +3,21 @@
 
 <!ENTITY style.label "Stil:">
 <!ENTITY style.accesskey "t">
-<!ENTITY regularStyle.label "Düzenli">
+<!ENTITY regularStyle.label "Normal">
 <!ENTITY bold.label "Kalın">
 <!ENTITY italic.label "Eğik">
-<!ENTITY boldItalic.label "Kalın ve eğik">
+<!ENTITY boldItalic.label "Kalın Eğik">
 <!ENTITY size.label "Boyut:">
 <!ENTITY size.accesskey "B">
-<!ENTITY regularSize.label "Düzenli">
-<!ENTITY bigger.label "Daha büyük">
-<!ENTITY smaller.label "Daha küçük">
+<!ENTITY regularSize.label "Normal">
+<!ENTITY bigger.label "Daha Büyük">
+<!ENTITY smaller.label "Daha Küçük">
 <!ENTITY color.label "Renk:">
 <!ENTITY color.accesskey "R">
-<!ENTITY displayWidth.label "Düz metin iletiler">
-<!ENTITY displayText.label "Tırnakla belirtilmiş düz metini görüntülerken aşağıdaki ayarları kullan:">
+<!ENTITY displayWidth.label "Düz Metin İletiler">
+<!ENTITY displayText.label "Alıntılanmış düz metin iletileri görüntülerken şu ayarları kullan:">
 <!ENTITY colors.label "Renkler:">
-<!ENTITY overrideColors.label "İletileri gösterirken bu renkleri kullan:">
+<!ENTITY overrideColors.label "İletileri gösterirken şu renkleri kullan:">
 <!ENTITY textColor.label "Metin rengi:">
 <!ENTITY textColor.accesskey "t">
 <!ENTITY backgroundColor.label "Arkaplan rengi:">
@@ -28,7 +28,7 @@
 <!ENTITY convertEmoticons.accesskey "D">
 <!-- labels -->
 
-<!ENTITY displayTagsText.label "Etiketler, iletilerde kategori oluşturmaya veya önemlilik derecesine göre sıralamanıza yarar.">
+<!ENTITY displayTagsText.label "Etiketler, iletileri sınıflandırmak ve önceliklendirmek için kullanılır.">
 <!ENTITY addTagButton.label "Ekle">
 <!ENTITY addTagButton.accesskey "E">
 <!ENTITY editTagButton.label "Düzenle">
Index: thunderbird/l10n/tr/mail/chrome/messenger/preferences/preferences.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/preferences/preferences.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/preferences/preferences.dtd
@@ -1,5 +1,5 @@
 <!ENTITY prefWindow.titleWin "Seçenekler">
-<!ENTITY prefWindow.titleGNOME "&brandShortName; tercihleri">
+<!ENTITY prefWindow.titleGNOME "&brandShortName; Tercihleri">
 <!ENTITY prefWindow.styleWindows "width: 48em; min-height: 38.5em;">
 <!ENTITY prefWindow.styleMac "width: 47em;">
 <!ENTITY prefWindow.styleGNOME "width: 47em; min-height: 38em;">
Index: thunderbird/l10n/tr/mail/chrome/messenger/newFolderDialog.dtd
===================================================================
--- thunderbird.orig/l10n/tr/mail/chrome/messenger/newFolderDialog.dtd
+++ thunderbird/l10n/tr/mail/chrome/messenger/newFolderDialog.dtd
@@ -46,5 +46,5 @@
 <!ENTITY folderRestriction2.label "Yeni dizin aşağıdakileri içerebilir:">
 <!ENTITY foldersOnly.label "Sadece dizinler">
 <!ENTITY messagesOnly.label "Sadece postalar">
-<!ENTITY accept.label "Create Folder">
+<!ENTITY accept.label "Oluştur">
 <!ENTITY accept.accesskey "r">
