Index: libffado-2.1.0/src/SConscript
===================================================================
--- libffado-2.1.0.orig/src/SConscript
+++ libffado-2.1.0/src/SConscript
@@ -294,7 +294,7 @@ if not env.GetOption( "clean" ):
 libname_versioned = "libffado.so.%s" % libenv['VERSION']
 libname_versioned_short = "libffado.so.%s" % libenv['VERSION'].split('.')[0]
 
-libenv.MergeFlags( "-Wl,-soname=%s" % libname_versioned_short )
+libenv.MergeFlags( "-Wl,-soname=%s -Wl,--as-needed" % libname_versioned_short )
 ffadolib = libenv.SharedLibrary( "ffado", source )
 
 #libenv.Install( "$libdir", ffadolib )
