Index: textbox.c
===================================================================
--- textbox.c (revision fed9b66ac8f4c6748fc6a1d1fbe8c89dff3b6dd9)
+++ textbox.c (revision 9e3bf9df5d98e9f047e457f47017a3f4d076f2dc)
@@ -176,5 +176,5 @@
     if (resultPtr) {
 	/* XXX I think this will work */
-	result = malloc(strlen(text) + (strlen(text) / width) + 2);
+	result = malloc(strlen(text) + (strlen(text) / (width - 1)) + 2);
 	*result = '\0';
     }
