diff -up kdebase-workspace-4.3.3/kwin/workspace.h.kde#171685 kdebase-workspace-4.3.3/kwin/workspace.h
--- kdebase-workspace-4.3.3/kwin/workspace.h.kde#171685	2009-06-17 15:07:38.000000000 -0500
+++ kdebase-workspace-4.3.3/kwin/workspace.h	2009-11-12 13:48:11.812385926 -0600
@@ -128,7 +128,7 @@ class Workspace : public QObject, public
         void unfakeActivity( Client* c );
         bool activateNextClient( Client* c );
         bool focusChangeEnabled()
-            { return block_focus == 0; }
+            { return block_focus < 1; }
 
         void updateColormap();
 
