commit 2ec6533494a23164ee9c2de4ab65d5fa372aff5c
Author: Lamarque V. Souza <lamarque@gmail.com>
Date:   Mon Mar 7 01:16:56 2011 -0300

    Adding missing emit.

diff --git a/solid/networkmanager-0.7/manager.cpp b/solid/networkmanager-0.7/manager.cpp
index 85e69e0..e485fd5 100644
--- a/solid/networkmanager-0.7/manager.cpp
+++ b/solid/networkmanager-0.7/manager.cpp
@@ -289,6 +289,7 @@ void NMNetworkManager::propertiesChanged(const QVariantMap &properties)
     if ( it != properties.end()) {
         d->isWirelessHardwareEnabled = it->toBool();
         kDebug(1441) << wifiHwKey << d->isWirelessHardwareEnabled;
+        emit wirelessHardwareEnabledChanged(d->isWirelessHardwareEnabled);
     }
     it = properties.find(wifiEnabledKey);
     if ( it != properties.end()) {
