diff -Nur x264-snapshot-20110615-2245-stable-old//Makefile x264-snapshot-20110615-2245-stable/Makefile
--- x264-snapshot-20110615-2245-stable-old//Makefile	2011-06-16 11:15:39.164000610 +0300
+++ x264-snapshot-20110615-2245-stable/Makefile	2011-06-16 11:18:10.650000887 +0300
@@ -148,10 +148,10 @@
 	$(if $(RANLIB), $(RANLIB) $@)
 
 $(SONAME): .depend $(OBJS) $(OBJASM) $(OBJSO)
-	$(LD)$@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(LDFLAGS)
+	$(LD)$@ $(OBJS) $(OBJASM) $(OBJSO) $(SOFLAGS) $(LDFLAGS)
 
 x264$(EXE): .depend $(OBJCLI) $(CLI_LIBX264)
-	$(LD)$@ $(OBJCLI) $(CLI_LIBX264) $(LDFLAGSCLI) $(LDFLAGS)
+	$(LD)$@ $(LDFLAGS) $(OBJCLI) $(CLI_LIBX264) $(LDFLAGSCLI)
 
 checkasm: tools/checkasm.o $(LIBX264)
 	$(LD)$@ $+ $(LDFLAGS)
