diff -Nur x264-snapshot-20100209-2245-old/configure x264-snapshot-20100209-2245/configure
--- x264-snapshot-20100209-2245-old/configure	2010-02-10 23:41:57.655692428 +0200
+++ x264-snapshot-20100209-2245/configure	2010-02-10 23:45:17.949697741 +0200
@@ -661,8 +661,8 @@
         echo 'SOFLAGS=-Wl,-h,$(SONAME)' >> config.mak
     else
         echo "SOSUFFIX=so" >> config.mak
-        echo "SONAME=libx264.so.$API" >> config.mak
-        echo 'SOFLAGS=-Wl,-soname,$(SONAME)' >> config.mak
+        echo "SONAME=libx264.so.0.$API" >> config.mak
+        echo 'SOFLAGS=-Wl,-soname,libx264.so.0' >> config.mak
     fi
     echo 'default: $(SONAME)' >> config.mak
 fi
