diff -Nur x264-snapshot-20100823-2245-old/configure x264-snapshot-20100823-2245/configure
--- x264-snapshot-20100823-2245-old/configure	2010-08-24 22:26:10.756000195 +0300
+++ x264-snapshot-20100823-2245/configure	2010-08-24 22:26:44.503999254 +0300
@@ -634,7 +634,7 @@
     CFLAGS="$CFLAGS $SWSCALE_CFLAGS"
 fi
 
-GPAC_LIBS="-lgpac_static"
+GPAC_LIBS="-lgpac"
 if [ $SYS = MINGW ]; then
     GPAC_LIBS="$GPAC_LIBS -lwinmm"
 fi
