diff -Nur v4l-utils-0.8.0-old/utils/Makefile v4l-utils-0.8.0/utils/Makefile
--- v4l-utils-0.8.0-old/utils/Makefile	2010-05-26 22:49:18.737992700 +0300
+++ v4l-utils-0.8.0/utils/Makefile	2010-05-26 22:49:40.637871743 +0300
@@ -1,4 +1,4 @@
-SUBDIRS=libv4l2util decode_tm6000 keytable rds v4l2-compliance v4l2-ctl v4l2-dbg xc3028-firmware
+SUBDIRS=libv4l2util v4l2-compliance v4l2-ctl v4l2-dbg
 
 all install: 
 	@for i in $(SUBDIRS); do \
@@ -16,7 +16,7 @@
 	else \
 		QMAKE=qmake; \
 	fi; \
-	if which $$QMAKE >/dev/null 2>&1; then \
+	if false >/dev/null 2>&1; then \
 		if $$QMAKE --version 2>&1 | grep '4\.[0-9][0-9]*\.[0-9][0-9]*' >/dev/null; then \
 			if [ ! -f qv4l2/Makefile ]; then (cd qv4l2; $$QMAKE); fi; \
 			$(MAKE) -C qv4l2 -f Makefile.install $@; \
