Index: kdeutils-4.1.80/printer-applet/printer-applet-printers.ui
===================================================================
--- kdeutils-4.1.80.orig/printer-applet/printer-applet-printers.ui
+++ kdeutils-4.1.80/printer-applet/printer-applet-printers.ui
@@ -13,7 +13,7 @@
    <string>Printer Status</string>
   </property>
   <property name="windowIcon" >
-   <iconset>/usr/share/system-config-printer/printer-128.png</iconset>
+   <iconset>printer.png</iconset>
   </property>
   <layout class="QGridLayout" >
    <property name="leftMargin" >
