Index: ark/app/ark_addtoservicemenu.desktop
===================================================================
--- ark/app/ark_addtoservicemenu.desktop.orig
+++ ark/app/ark_addtoservicemenu.desktop
@@ -2,7 +2,7 @@
 Type=Service
 ServiceTypes=KonqPopupMenu/Plugin
 MimeType=all/all;
-Actions=compressHere;compressAsZip;compressAsRar;compressAsTar;compressTo;
+Actions=compressHere;compressAsZip;compressAsTar;compressTo;
 X-KDE-Submenu=Compress
 X-KDE-Submenu[ar]=اضغط
 X-KDE-Submenu[ast]=Comprimir
@@ -170,61 +170,6 @@ Name[zh_TW]=成 ZIP 壓縮檔
 Icon=utilities-file-archiver
 Exec=ark --changetofirstpath --add --autofilename zip %F
 
-[Desktop Action compressAsRar]
-Name=As RAR Archive
-Name[ar]=كأرشيف RAR
-Name[ast]=Como archivu comprimíu RAR
-Name[ca]=Com a arxiu RAR
-Name[ca@valencia]=Com a arxiu RAR
-Name[cs]=Jako archiv RAR
-Name[da]=Som RAR-arkiv
-Name[de]=Als RAR-Archiv
-Name[el]=Ως αρχειοθήκη RAR
-Name[en_GB]=As RAR Archive
-Name[es]=Como archivo comprimido RAR
-Name[et]=RAR-arhiivina
-Name[eu]=RAR artxiboa gisa
-Name[fi]=RAR-arkistoksi
-Name[fr]=En tant qu'archive RAR
-Name[ga]=Mar Chartlann RAR
-Name[gl]=Como arquivo RAR
-Name[hr]=Kao RAR-arhiva
-Name[hu]=RAR archívumként
-Name[id]=Sebagai Arsip RAR
-Name[it]=Come archivio RAR
-Name[ja]=RAR アーカイブに
-Name[kk]=RAR архиві қылып
-Name[km]=​ជា​ប័ណ្ណសារ RAR
-Name[ko]=RAR 파일로
-Name[lt]=RAR archyvas
-Name[lv]=Kā RAR arhīvu
-Name[nb]=Som RAR-arkiv
-Name[nds]=As RAR-Archiev
-Name[nl]=Als RAR-archief
-Name[nn]=Som RAR-arkiv
-Name[pa]=RAR ਅਕਾਇਵ ਵਾਂਗ
-Name[pl]=Jako archiwum RAR
-Name[pt]=Como Pacote RAR
-Name[pt_BR]=Como arquivo RAR
-Name[ro]=Ca arhivă RAR
-Name[ru]=Как архив RAR
-Name[sk]=Do RAR archívu
-Name[sl]=Kot arhiv RAR
-Name[sr]=као РАР архиву
-Name[sr@ijekavian]=као РАР архиву
-Name[sr@ijekavianlatin]=kao RAR arhivu
-Name[sr@latin]=kao RAR arhivu
-Name[sv]=Som RAR-arkiv
-Name[th]=เป็นแฟ้มจัดเก็บบีบอัดแบบ RAR
-Name[tr]=RAR Arşivi Olarak
-Name[uk]=Як архів RAR
-Name[wa]=Come årtchive RAR
-Name[x-test]=xxAs RAR Archivexx
-Name[zh_CN]=为 RAR 归档
-Name[zh_TW]=成 RAR 壓縮檔
-Icon=utilities-file-archiver
-Exec=ark --changetofirstpath --add --autofilename rar %F
-
 [Desktop Action compressAsTar]
 Name=As ZIP/TAR Archive
 Name[ar]=كأرشيف  ZIP/TAR
