Index: ark/app/ark_addtoservicemenu.desktop
===================================================================
--- ark/app/ark_addtoservicemenu.desktop.orig
+++ ark/app/ark_addtoservicemenu.desktop
@@ -2,7 +2,7 @@
 Type=Service
 ServiceTypes=KonqPopupMenu/Plugin
 MimeType=all/all;
-Actions=compressHere;compressAsZip;compressAsRar;compressAsTar;compressTo;
+Actions=compressHere;compressAsZip;compressAsTar;compressTo;
 X-KDE-Submenu=Compress
 X-KDE-Submenu[ar]=اضغط
 X-KDE-Submenu[bg]=Компресиране
