diff -p -up grep-2.6.1/tests/Makefile.in.skip grep-2.6.1/tests/Makefile.in
--- grep-2.6.1/tests/Makefile.in.skip	2010-03-28 00:31:21.000000000 -0300
+++ grep-2.6.1/tests/Makefile.in	2010-03-28 00:31:29.000000000 -0300
@@ -766,7 +766,6 @@ TESTS = \
   case-fold-char-class				\
   case-fold-char-range				\
   case-fold-char-type				\
-  char-class-multibyte				\
   dfaexec-multibyte				\
   empty.sh					\
   ere.sh					\
