--- configure	2002-03-26 18:09:38.000000000 +0200
+++ configure.new	2006-04-06 23:35:44.000000000 +0300
@@ -6272,7 +6272,7 @@
  *)    ;;
 esac
 
-ALL_LINGUAS="cs de el eo es et fr gl hr id it ja ko nl no pl pt_BR ru sl sv"
+ALL_LINGUAS="cs de el eo es et fr gl hr id it ja ko nl no pl pt_BR ru sl sv tr"
 
         echo "$as_me:$LINENO: checking for strerror in -lcposix" >&5
 echo $ECHO_N "checking for strerror in -lcposix... $ECHO_C" >&6
