From 47622f8eb33a9327d1e625e6e89da8a47e1bd9ba Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Sat, 19 Jun 2010 11:26:04 +0400
Subject: [PATCH] Do not install VxWorks-specific docs

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 erts/doc/src/Makefile |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/erts/doc/src/Makefile b/erts/doc/src/Makefile
index be99288..4bb07c4 100644
--- a/erts/doc/src/Makefile
+++ b/erts/doc/src/Makefile
@@ -46,7 +46,6 @@ XML_REF1_FILES = epmd.xml \
 XML_REF3_FILES = \
 	driver_entry.xml \
 	erl_nif.xml \
-	erl_set_memory_block.xml \
 	erl_driver.xml \
 	erl_prim_loader.xml \
 	erlang.xml \
-- 
1.7.3.5

