Index: frei0r-plugins-1.2/Makefile.am
===================================================================
--- frei0r-plugins-1.2.orig/Makefile.am
+++ frei0r-plugins-1.2/Makefile.am
@@ -10,11 +10,11 @@
 
 ACLOCAL_AMFLAGS= -I m4
 
-SUBDIRS=src include doc
+SUBDIRS=src include
 EXTRA_DIST = Doxyfile
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = frei0r.pc
 
-docsdir = ${prefix}/share/doc/${PACKAGE}-${VERSION}
-docs_DATA = README ChangeLog TODO AUTHORS 
+docsdir = ${prefix}/share/doc/${PACKAGE}
+docs_DATA = README ChangeLog TODO AUTHORS
