diff -urNp coreutils-7.2-orig/src/copy.c coreutils-7.2/src/copy.c
--- coreutils-7.2-orig/src/copy.c	2009-04-17 15:21:26.000000000 +0200
+++ coreutils-7.2/src/copy.c	2009-04-17 15:24:17.000000000 +0200
@@ -139,10 +139,13 @@ copy_attr_error (struct error_context *c
   int err = errno;
   va_list ap;
 
-  /* use verror module to print error message */
-  va_start (ap, fmt);
-  verror (0, err, fmt, ap);
-  va_end (ap);
+  if (errno != ENOTSUP && errno != ENODATA)
+    {
+      /* use verror module to print error message */
+      va_start (ap, fmt);
+      verror (0, err, fmt, ap);
+      va_end (ap);
+    }
 }
 
 static char const *
