diff -urNp coreutils-7.4-orig/src/stty.c coreutils-7.4/src/stty.c
--- coreutils-7.4-orig/src/stty.c	2009-04-24 14:41:19.000000000 +0200
+++ coreutils-7.4/src/stty.c	2009-06-11 10:15:41.000000000 +0200
@@ -1001,7 +1001,7 @@ main (int argc, char **argv)
          spurious difference in an uninitialized portion of the structure.  */
       DECLARE_ZEROED_AGGREGATE (struct termios, new_mode);
 
-      if (tcsetattr (STDIN_FILENO, TCSADRAIN, &mode))
+      if (tcsetattr (STDIN_FILENO, TCSANOW, &mode))
         error (EXIT_FAILURE, errno, "%s", device_name);
 
       /* POSIX (according to Zlotnick's book) tcsetattr returns zero if
