Index: LVM2.2.02.73/tools/Makefile.in
===================================================================
--- LVM2.2.02.73.orig/tools/Makefile.in
+++ LVM2.2.02.73/tools/Makefile.in
@@ -104,7 +104,7 @@ ifeq ("@DMEVENTD@", "yes")
 	LVMLIBS += -ldevmapper-event
 endif
 
-LVMLIBS += -ldevmapper
+LVMLIBS += -ldevmapper -ldevmapper-event
 
 EXPORTED_HEADER = $(srcdir)/lvm2cmd.h
 EXPORTED_FN_PREFIX = lvm2
@@ -149,7 +149,7 @@ liblvm2cmd-static.a: $(top_builddir)/lib
 
 liblvm2cmd.$(LIB_SUFFIX): liblvm2cmd.a $(LDDEPS)
 	$(CC) -shared -Wl,-soname,$@.$(LIB_VERSION) \
-	$(CFLAGS) $(CLDFLAGS) -o $@ $(LVMLIBS) $(LIBS) \
+	$(CFLAGS) $(CLDFLAGS) $(LVMLIBS) $(LIBS) -o $@ \
 		@CLDWHOLEARCHIVE@ liblvm2cmd.a @CLDNOWHOLEARCHIVE@
 
 liblvm2cmd.$(LIB_SUFFIX).$(LIB_VERSION): liblvm2cmd.$(LIB_SUFFIX)
