diff -up opensc-0.11.7/src/libopensc/libopensc.pc.in.config opensc-0.11.7/src/libopensc/libopensc.pc.in
--- opensc-0.11.7/src/libopensc/libopensc.pc.in.config	2006-05-15 22:57:30.000000000 +0200
+++ opensc-0.11.7/src/libopensc/libopensc.pc.in	2009-02-27 19:05:56.000000000 +0100
@@ -6,6 +6,6 @@ includedir=@includedir@
 Name: libopensc
 Description: libopensc
 Version: @VERSION@
-Libs: -L${libdir} -lopensc -lscconf
+Libs: -L${libdir} -lopensc
 Cflags: -I${includedir}
 
diff -up opensc-0.11.7/src/libopensc/opensc-config.in.config opensc-0.11.7/src/libopensc/opensc-config.in
--- opensc-0.11.7/src/libopensc/opensc-config.in.config	2008-12-27 17:15:30.000000000 +0100
+++ opensc-0.11.7/src/libopensc/opensc-config.in	2009-02-27 19:03:11.000000000 +0100
@@ -15,10 +15,10 @@ prefix="@prefix@"
 exec_prefix="@exec_prefix@"
 exec_prefix_set=no
 
-libdir="@libdir@"
-includedir="@includedir@"
+libdir=""
+includedir=""
 opensc_cflags=""
-opensc_libs="-L${libdir} -lopensc -lscconf"
+opensc_libs="-lopensc -lscconf"
 
 usage()
 {
