--- a/src/qjackctlSetup.cpp	2009-07-30 16:38:44.000000000 +0300
+++ b/src/qjackctlSetup.cpp	2009-07-30 16:38:34.000000000 +0300
@@ -51,8 +51,8 @@
 
 	m_settings.beginGroup("/Options");
 	bStartJack               = m_settings.value("/StartJack", false).toBool();
-	bStartupScript           = m_settings.value("/StartupScript", true).toBool();
-	sStartupScriptShell      = m_settings.value("/StartupScriptShell", "artsshell -q terminate").toString();
+	bStartupScript           = m_settings.value("/StartupScript", false).toBool();
+	//sStartupScriptShell      = m_settings.value("/StartupScriptShell", "artsshell -q terminate").toString();
 	bPostStartupScript       = m_settings.value("/PostStartupScript", false).toBool();
 	sPostStartupScriptShell  = m_settings.value("/PostStartupScriptShell").toString();
 	bShutdownScript          = m_settings.value("/ShutdownScript", false).toBool();
