--- qjackctl-0.3.3/src/qjackctlSetup.cpp.orig	2009-03-08 21:43:37.000000000 +0200
+++ qjackctl-0.3.3/src/qjackctlSetup.cpp	2009-03-08 21:43:54.000000000 +0200
@@ -288,7 +288,7 @@
 
 	m_settings.beginGroup("/Settings" + sSuffix);
 	preset.sServer      = m_settings.value("/Server", "jackd").toString();
-	preset.bRealtime    = m_settings.value("/Realtime", true).toBool();
+	preset.bRealtime    = m_settings.value("/Realtime", false).toBool();
 	preset.bSoftMode    = m_settings.value("/SoftMode", false).toBool();
 	preset.bMonitor     = m_settings.value("/Monitor", false).toBool();
 	preset.bShorts      = m_settings.value("/Shorts", false).toBool();
