diff -ruN slang-2.1.3-old/src/slcurses.c slang-2.1.3/src/slcurses.c
--- slang-2.1.3-old/src/slcurses.c	2005-02-07 06:03:28.000000000 +0000
+++ slang-2.1.3/src/slcurses.c	2005-02-18 16:54:03.476533632 +0000
@@ -385,6 +385,16 @@
        SLcurses_Acs_Map[SLSMG_HLINE_CHAR] = SLSMG_HLINE_CHAR | A_ALTCHARSET;
        SLcurses_Acs_Map[SLSMG_PLUS_CHAR] = SLSMG_PLUS_CHAR | A_ALTCHARSET;
        SLcurses_Acs_Map[SLSMG_CKBRD_CHAR] = SLSMG_CKBRD_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_DIAMOND_CHAR] = SLSMG_DIAMOND_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_DEGREE_CHAR] = SLSMG_DEGREE_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_PLMINUS_CHAR] = SLSMG_PLMINUS_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_BULLET_CHAR] = SLSMG_BULLET_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_LARROW_CHAR] = SLSMG_LARROW_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_RARROW_CHAR] = SLSMG_RARROW_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_DARROW_CHAR] = SLSMG_DARROW_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_UARROW_CHAR] = SLSMG_UARROW_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_BOARD_CHAR] = SLSMG_BOARD_CHAR | A_ALTCHARSET;
+       SLcurses_Acs_Map[SLSMG_BLOCK_CHAR] = SLSMG_BLOCK_CHAR | A_ALTCHARSET;
      }
    else
      {
@@ -401,6 +411,16 @@
        SLcurses_Acs_Map[SLSMG_HLINE_CHAR] = '-';
        SLcurses_Acs_Map[SLSMG_PLUS_CHAR] = '+';
        SLcurses_Acs_Map[SLSMG_CKBRD_CHAR] = '#';
+       SLcurses_Acs_Map[SLSMG_DIAMOND_CHAR] = '+';
+       SLcurses_Acs_Map[SLSMG_DEGREE_CHAR] = '\\';
+       SLcurses_Acs_Map[SLSMG_PLMINUS_CHAR] = '#';
+       SLcurses_Acs_Map[SLSMG_BULLET_CHAR] = 'o';
+       SLcurses_Acs_Map[SLSMG_LARROW_CHAR] = '<';
+       SLcurses_Acs_Map[SLSMG_RARROW_CHAR] = '>';
+       SLcurses_Acs_Map[SLSMG_DARROW_CHAR] = 'v';
+       SLcurses_Acs_Map[SLSMG_UARROW_CHAR] = '^';
+       SLcurses_Acs_Map[SLSMG_BOARD_CHAR] = '#';
+       SLcurses_Acs_Map[SLSMG_BLOCK_CHAR] = '#';
      }
 
    return SLcurses_Stdscr;
