Index: widelands-build15-src/src/wui/interactive_base.cc
===================================================================
--- widelands-build15-src.orig/src/wui/interactive_base.cc
+++ widelands-build15-src/src/wui/interactive_base.cc
@@ -89,8 +89,8 @@ Interactive_Base::Interactive_Base
 	//  Switch to the new graphics system now, if necessary.
 	WLApplication::get()->init_graphics
 		(get_xres(), get_yres(),
-		 global_s.get_int("depth", 16),
-		 global_s.get_bool("fullscreen", false),
+		 global_s.get_int("depth", 32),
+		 global_s.get_bool("fullscreen", true),
 		 global_s.get_bool("hw_improvements", false),
 		 global_s.get_bool("double_buffer", false)
 #if HAS_OPENGL
Index: widelands-build15-src/src/constants.h
===================================================================
--- widelands-build15-src.orig/src/constants.h
+++ widelands-build15-src/src/constants.h
@@ -36,8 +36,8 @@
 #define TEXTURE_HEIGHT TEXTURE_WIDTH
 //@}
 
-#define XRES 800 ///< Fullscreen Menu Width
-#define YRES 600 ///< Fullscreen Menu Height
+#define XRES 1024 ///< Fullscreen Menu Width
+#define YRES 768 ///< Fullscreen Menu Height
 
 /// \name Fonts
 /// Font constants, defined including size
