Index: widelands-build15-src/CMakeLists.txt
===================================================================
--- widelands-build15-src.orig/CMakeLists.txt
+++ widelands-build15-src/CMakeLists.txt
@@ -39,11 +39,11 @@ if (WIN32 OR APPLE OR WL_PORTABLE)
   endif (NOT WL_INSTALL_BINDIR OR WL_INSTALL_BINDIR STREQUAL "")
 elseif (WIN32 OR APPLE OR WL_PORTABLE)
   if (NOT WL_INSTALL_PREFIX OR WL_INSTALL_PREFIX STREQUAL "")
-    set (WL_INSTALL_PREFIX "/usr/local")
+    set (WL_INSTALL_PREFIX "/usr")
   endif (NOT WL_INSTALL_PREFIX OR WL_INSTALL_PREFIX STREQUAL "")
   
   if (NOT WL_INSTALL_DATADIR OR WL_INSTALL_DATADIR STREQUAL "")
-    set (WL_INSTALL_DATADIR "share/games/widelands")
+    set (WL_INSTALL_DATADIR "share/widelands")
   endif (NOT WL_INSTALL_DATADIR OR WL_INSTALL_DATADIR STREQUAL "")
 
   if (NOT WL_INSTALL_LOCALEDIR OR WL_INSTALL_LOCALEDIR STREQUAL "")
@@ -51,23 +51,23 @@ elseif (WIN32 OR APPLE OR WL_PORTABLE)
   endif (NOT WL_INSTALL_LOCALEDIR OR WL_INSTALL_LOCALEDIR STREQUAL "")
 
   if (NOT WL_INSTALL_BINDIR OR WL_INSTALL_BINDIR STREQUAL "")
-    set (WL_INSTALL_BINDIR "games")
+    set (WL_INSTALL_BINDIR "bin")
   endif (NOT WL_INSTALL_BINDIR OR WL_INSTALL_BINDIR STREQUAL "")
 endif (WIN32 OR APPLE OR WL_PORTABLE)
 
 # General install prefix for config.h
 if (NOT WL_INSTALL_PREFIX OR WL_INSTALL_PREFIX STREQUAL "")
-	set (WL_INSTALL_PREFIX /usr/local)
+	set (WL_INSTALL_PREFIX /usr)
 endif (NOT WL_INSTALL_PREFIX OR WL_INSTALL_PREFIX STREQUAL "")
 
 # Widelands binary directory
 if (NOT WL_INSTALL_BINDIR OR WL_INSTALL_BINDIR STREQUAL "")
-	set (WL_INSTALL_BINDIR games)
+	set (WL_INSTALL_BINDIR bin)
 endif (NOT WL_INSTALL_BINDIR OR WL_INSTALL_BINDIR STREQUAL "")
 
 # Widelands data directory
 if (NOT WL_INSTALL_DATADIR OR WL_INSTALL_DATADIR STREQUAL "")
-	set (WL_INSTALL_DATADIR share/games/widelands)
+	set (WL_INSTALL_DATADIR share/widelands)
 endif (NOT WL_INSTALL_DATADIR OR WL_INSTALL_DATADIR STREQUAL "")
 
 # Widelands locale directory
