Index: util-linux-ng-2.18/configure.ac
===================================================================
--- util-linux-ng-2.18.orig/configure.ac
+++ util-linux-ng-2.18/configure.ac
@@ -70,7 +70,7 @@ AC_SUBST([usrbin_execdir])
 usrsbin_execdir='${exec_prefix}/sbin'
 AC_SUBST([usrsbin_execdir])
 
-usrlib_execdir='${exec_prefix}/'$libdirname
+usrlib_execdir=$libdir
 AC_SUBST([usrlib_execdir])
 
 
