Index: kdelibs-4.1.85/Messages.sh
===================================================================
--- kdelibs-4.1.85.orig/Messages.sh	2008-11-19 11:20:14.000000000 +0100
+++ kdelibs-4.1.85/Messages.sh	2008-12-11 18:20:59.000000000 +0100
@@ -1,5 +1,5 @@
 #! /usr/bin/env bash
-find . -type d | fgrep -v '.svn' | sed -e 's,$,/,' > dirs
+find . -type d | fgrep -v '.svn' | fgrep -v '.pc' | sed -e 's,$,/,' > dirs
 msh=`find . -name Messages.sh`
 for dir in $msh; do
   dir=`dirname $dir`
