Index: kde4libs-4.5.0b/nepomuk/ui/kratingpainter.cpp
===================================================================
--- kde4libs-4.5.0b.orig/nepomuk/ui/kratingpainter.cpp	2010-08-27 17:43:52.851442002 -0400
+++ kde4libs-4.5.0b/nepomuk/ui/kratingpainter.cpp	2010-08-27 18:46:48.571442000 -0400
@@ -28,6 +28,7 @@
 
 #include <kicon.h>
 #include <kiconeffect.h>
+#include <kiconloader.h>
 #include <kdebug.h>
 
 
@@ -198,7 +199,8 @@
     int maxHSizeOnePix = ( rect.width() - (numUsedStars-1)*usedSpacing ) / numUsedStars;
     QPixmap ratingPix = d->getPixmap( qMin( rect.height(), maxHSizeOnePix ) );
 
-    QPixmap disabledRatingPix = KIconEffect().apply( ratingPix, KIconEffect::ToGray, 1.0, QColor(), false );
+    KIconEffect *iconEffect = KIconLoader::global()->iconEffect();
+    QPixmap disabledRatingPix = iconEffect->apply( ratingPix, KIconEffect::ToGray, 1.0, QColor(), false );
     QPixmap hoverPix;
 
     // if we are disabled we become gray and more transparent
@@ -215,7 +217,7 @@
     if ( hoverRating > 0 && rating != hoverRating && d->isEnabled ) {
         numHoverStars = d->bHalfSteps ? hoverRating/2 : hoverRating;
         halfHover = d->bHalfSteps && hoverRating%2;
-        hoverPix = KIconEffect().apply( ratingPix, KIconEffect::ToGray, 0.5, QColor(), false );
+        hoverPix = iconEffect->apply( ratingPix, KIconEffect::ToGray, 0.5, QColor(), false );
     }
 
     if ( d->alignment & Qt::AlignJustify ) {
