Index: kde4libs-4.5.1/kio/bookmarks/kbookmarkmanager.cc
===================================================================
--- kde4libs-4.5.1.orig/kio/bookmarks/kbookmarkmanager.cc	2010-10-04 11:46:56.425809003 +0100
+++ kde4libs-4.5.1/kio/bookmarks/kbookmarkmanager.cc	2010-10-04 11:47:11.169809003 +0100
@@ -430,9 +430,9 @@
                              "of the error needs to be fixed as quickly as possible, "
                              "which is most likely a full hard drive.",
                          filename, file.errorString());
-        if (qApp->type() != QApplication::Tty)
-            KMessageBox::error( QApplication::activeWindow(), error );
-        else
+        //if (qApp->type() != QApplication::Tty)
+        //    KMessageBox::error( QApplication::activeWindow(), error );
+        //else
             kError() << error << endl;
     }
     hadSaveError = true;
