--- kdelibs/plasma/delegate.cpp	2009-05-06 10:14:41.000000000 +0000
+++ kdelibs/plasma/delegate.cpp	2009-05-08 12:10:34.000000000 +0000
@@ -310,7 +310,7 @@
             p.setPen(textColor);
         }
         p.setFont(subTitleFont);
-        p.drawText(subTitleRect, Qt::AlignLeft|Qt::AlignVCenter, "  " + subTitleText);
+        p.drawText(subTitleRect, Qt::AlignLeft|Qt::AlignVCenter, subTitleText);
     }
     p.end();
 
