Index: kde4libs-4.6.0/kio/kfile/kpropertiesdialog.cpp
===================================================================
--- kde4libs-4.6.0.orig/kio/kfile/kpropertiesdialog.cpp	2011-02-21 14:36:28.824597883 +0000
+++ kde4libs-4.6.0/kio/kfile/kpropertiesdialog.cpp	2011-02-21 14:36:37.224597881 +0000
@@ -526,14 +526,6 @@
         q->insertPlugin(p);
     }
 
-#ifndef Q_OS_WIN
-    if ( KAuthorized::authorizeKAction("sharefile") &&
-         KFileSharePropsPlugin::supports( m_items ) ) {
-        KPropertiesDialogPlugin *p = new KFileSharePropsPlugin(q);
-        q->insertPlugin(p);
-    }
-#endif
-
     //plugins
 
     if ( m_items.count() != 1 )
