diff -Nur qemu-kvm-0.13.0-old//vl.c qemu-kvm-0.13.0/vl.c
--- qemu-kvm-0.13.0-old//vl.c	2010-12-08 18:19:10.533000002 +0200
+++ qemu-kvm-0.13.0/vl.c	2010-12-08 18:23:18.746999997 +0200
@@ -2822,8 +2822,9 @@
             exit(1);
 #endif
 #ifdef CONFIG_KVM
-            fprintf(stderr, "Could not initialize KVM, will disable KVM support\n");
+            fprintf(stderr, "Could not initialize KVM, Do you have kvm-amd or kvm-intel modprobe'd?\nIf you want to use CPU emulation, start with -no-kvm.\n");
             kvm_allowed = 0;
+            exit(1);
 #endif
         }
     }
