Index: qemu-kvm-0.12.2/vl.c
===================================================================
--- qemu-kvm-0.12.2.orig/vl.c
+++ qemu-kvm-0.12.2/vl.c
@@ -5901,8 +5901,9 @@ int main(int argc, char **argv, char **e
             fprintf(stderr, "failed to initialize KVM\n");
             exit(1);
 #endif
-            fprintf(stderr, "Could not initialize KVM, will disable KVM support\n");
+            fprintf(stderr, "Could not initialize KVM. Do you have kvm-amd or kvm-intel modprobe'd?\nIf you want to use CPU emulation, start with -no-kvm.\n");
             kvm_allowed = 0;
+            exit(1);
         }
     }
 
