Index: qemu-kvm-0.12.1.2/configure
===================================================================
--- qemu-kvm-0.12.1.2.orig/configure
+++ qemu-kvm-0.12.1.2/configure
@@ -1975,7 +1975,7 @@ else
       prefix="/usr/local"
   fi
   mansuffix="/share/man"
-  datasuffix="/share/qemu"
+  datasuffix="/share/kvm"
   docsuffix="/share/doc/qemu"
   binsuffix="/bin"
 fi
Index: qemu-kvm-0.12.1.2/vl.c
===================================================================
--- qemu-kvm-0.12.1.2.orig/vl.c
+++ qemu-kvm-0.12.1.2/vl.c
@@ -4619,7 +4619,7 @@ static char *find_datadir(const char *ar
 /* Find a likely location for support files using the location of the binary.
    For installed binaries this will be "$bindir/../share/qemu".  When
    running from the build tree this will be "$bindir/../pc-bios".  */
-#define SHARE_SUFFIX "/share/qemu"
+#define SHARE_SUFFIX "/share/kvm"
 #define BUILD_SUFFIX "/pc-bios"
 static char *find_datadir(const char *argv0)
 {
