Index: hylafax-6.0.5/util/b64-encode.awk
===================================================================
--- hylafax-6.0.5.orig/util/b64-encode.awk
+++ hylafax-6.0.5/util/b64-encode.awk
@@ -1,4 +1,4 @@
-#!/usr/bin/awk -f
+#!/bin/awk -f
 #	$Id$
 #
 # HylaFAX Facsimile Software
Index: hylafax-6.0.5/util/qp-encode.awk
===================================================================
--- hylafax-6.0.5.orig/util/qp-encode.awk
+++ hylafax-6.0.5/util/qp-encode.awk
@@ -1,4 +1,4 @@
-#!/usr/bin/awk -f
+#!/bin/awk -f
 #	$Id$
 #
 # HylaFAX Facsimile Software
