diff -up sysstat-9.0.6.1/man/sa1.in.isag sysstat-9.0.6.1/man/sa1.in
--- sysstat-9.0.6.1/man/sa1.in.isag	2009-09-12 20:59:04.000000000 +0200
+++ sysstat-9.0.6.1/man/sa1.in	2010-08-13 14:05:40.000000000 +0200
@@ -61,7 +61,6 @@ Sebastien Godard (sysstat <at> orange.fr
 .BR sar (1),
 .BR sadc (8),
 .BR sa2 (8),
-.BR sadf (1),
-.BR isag (1)
+.BR sadf (1)
 
 .I http://pagesperso-orange.fr/sebastien.godard/
diff -up sysstat-9.0.6.1/man/sa2.in.isag sysstat-9.0.6.1/man/sa2.in
--- sysstat-9.0.6.1/man/sa2.in.isag	2009-09-12 20:59:04.000000000 +0200
+++ sysstat-9.0.6.1/man/sa2.in	2010-08-13 14:05:40.000000000 +0200
@@ -42,7 +42,6 @@ Sebastien Godard (sysstat <at> orange.fr
 .BR sar (1),
 .BR sadc (8),
 .BR sa1 (8),
-.BR sadf (1),
-.BR isag (1)
+.BR sadf (1)
 
 .I http://pagesperso-orange.fr/sebastien.godard/
diff -up sysstat-9.0.6.1/man/sadc.in.isag sysstat-9.0.6.1/man/sadc.in
--- sysstat-9.0.6.1/man/sadc.in.isag	2009-09-12 20:59:04.000000000 +0200
+++ sysstat-9.0.6.1/man/sadc.in	2010-08-13 14:05:40.000000000 +0200
@@ -201,7 +201,6 @@ Sebastien Godard (sysstat <at> orange.fr
 .BR sar (1),
 .BR sa1 (8),
 .BR sa2 (8),
-.BR sadf (1),
-.BR isag (1)
+.BR sadf (1)
 
 .I http://pagesperso-orange.fr/sebastien.godard/
diff -up sysstat-9.0.6.1/man/sadf.in.isag sysstat-9.0.6.1/man/sadf.in
--- sysstat-9.0.6.1/man/sadf.in.isag	2009-10-11 16:26:10.000000000 +0200
+++ sysstat-9.0.6.1/man/sadf.in	2010-08-13 14:05:40.000000000 +0200
@@ -179,7 +179,6 @@ Sebastien Godard (sysstat <at> orange.fr
 .BR sar (1),
 .BR sadc (8),
 .BR sa1 (8),
-.BR sa2 (8),
-.BR isag (1)
+.BR sa2 (8)
 
 .I http://pagesperso-orange.fr/sebastien.godard/
diff -up sysstat-9.0.6.1/man/sar.in.isag sysstat-9.0.6.1/man/sar.in
--- sysstat-9.0.6.1/man/sar.in.isag	2009-09-12 20:59:04.000000000 +0200
+++ sysstat-9.0.6.1/man/sar.in	2010-08-13 14:05:40.000000000 +0200
@@ -1895,8 +1895,7 @@ Sebastien Godard (sysstat <at> orange.fr
 .BR sadc (8),
 .BR sa1 (8),
 .BR sa2 (8),
-.BR sadf (1),
-.BR isag (1),
+.BR sadf (1),
 .BR pidstat (1),
 .BR mpstat (1),
 .BR iostat (1),
