Index: comar/__init__.py
===================================================================
--- comar/__init__.py	(revision 31138)
+++ comar/__init__.py	(revision 31139)
@@ -74,7 +74,7 @@
                     self.async(self.package, exception, None)
 
                 if self.quiet:
-                    met(dbus_interface="%s.%s.%s" % (self.link.interface, self.group, self.class_), ignore_reply=True, *args)
+                    met(dbus_interface="%s.%s.%s" % (self.link.interface, self.group, self.class_), *args)
                 else:
                     met(dbus_interface="%s.%s.%s" % (self.link.interface, self.group, self.class_), reply_handler=handleResult, error_handler=handleError, timeout=self.timeout, *args)
             else:
@@ -83,7 +83,7 @@
                         for package in packages:
                             obj = self.link.bus.get_object(self.link.address, "/package/%s" % package)
                             met = getattr(obj, self.method)
-                            met(dbus_interface="%s.%s.%s" % (self.link.interface, self.group, self.class_), ignore_reply=True, *args)
+                            met(dbus_interface="%s.%s.%s" % (self.link.interface, self.group, self.class_), *args)
                     else:
                         def handleResult(package):
                             def handler(*result):
