diff -ur vte-0.20.2.orig/python/Makefile.am vte-0.20.2/python/Makefile.am
--- vte-0.20.2.orig/python/Makefile.am	2009-04-18 01:05:55.000000000 +0300
+++ vte-0.20.2/python/Makefile.am	2009-06-03 00:07:10.293688330 +0300
@@ -20,8 +20,8 @@
 
 # we do this to suport building with -Wl,-z,defs which errors on
 # vtemodule.so as we cannot include $(PYTHON_LIBS) due to bug 410986.
-vtemodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex initvte $(PYTHON_LDFLAGS) $(AM_LDFLAGS)
-vtemodule_la_LIBADD = $(top_builddir)/src/libvte.la $(LIBS) $(PYGTK_LIBS) $(VTE_LIBS) $(X_LIBS)
+vtemodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex initvte -lpython2.6 $(AM_LDFLAGS)
+vtemodule_la_LIBADD = $(top_builddir)/src/libvte.la $(LIBS) -lpython2.6 $(PYGTK_LIBS) $(VTE_LIBS) $(X_LIBS)
 nodist_vtemodule_la_SOURCES = vte.c
 
 vte.c: vte.defs vte.override
