Index: kdenlive-0.7.8/src/kdenlive.desktop
===================================================================
--- kdenlive-0.7.8.orig/src/kdenlive.desktop
+++ kdenlive-0.7.8/src/kdenlive.desktop
@@ -4,11 +4,13 @@ Name=Kdenlive
 GenericName=Video Editor
 GenericName[de]=Video-Editor
 GenericName[it]=Editor Video
+GenericName[tr]=Video Düzenleyici
 GenericName[ru]=Видеоредактор
 GenericName[uk]=Редактор відео
 Comment=Nonlinear video editor for KDE
 Comment[de]=Nicht-linearer Video-Editor für KDE
 Comment[it]=Editor video non lineare per KDE
+Comment[tr]=KDE için Nonlineer video düzenleyici
 Comment[ru]=Нелинейный видеоредактор для KDE
 Comment[uk]=Нелінійний редактор відео для KDE
 Type=Application
