Index: linux-2.6.30/drivers/input/keyboard/atkbd.c
===================================================================
--- linux-2.6.30.orig/drivers/input/keyboard/atkbd.c
+++ linux-2.6.30/drivers/input/keyboard/atkbd.c
@@ -1612,6 +1612,16 @@ static struct dmi_system_id atkbd_dmi_qu
 		.callback = atkbd_setup_forced_release,
 		.driver_data = atkdb_soltech_ta12_forced_release_keys,
 	},
+	/* http://bugs.pardus.org.tr/11565 */
+	{
+		.ident = "Samsung R510/P510",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "SAMSUNG ELECTRONICS CO., LTD."),
+			DMI_MATCH(DMI_PRODUCT_NAME, "R510/P510"),
+		},
+		.callback = atkbd_setup_forced_release,
+		.driver_data = atkbd_samsung_forced_release_keys,
+	},
 	{ }
 };
 
