Index: work/src/wl/sys/wl_linux.c
===================================================================
--- work.orig/src/wl/sys/wl_linux.c
+++ work/src/wl/sys/wl_linux.c
@@ -256,7 +256,6 @@ static struct pci_device_id wl_id_table[
 	{ PCI_VENDOR_ID_BROADCOM, 0x432d, PCI_ANY_ID, PCI_ANY_ID, 0, 0, 0 },
 	{ 0 }
 };
-MODULE_DEVICE_TABLE(pci, wl_id_table);
 
 #ifdef BCMDBG
 static int msglevel = 0xdeadbeef;
