diff -urN libmicrohttpd-0.9.5/src/daemon/internal.h libmicrohttpd-0.9.5.no-abort/src/daemon/internal.h
--- libmicrohttpd-0.9.5/src/daemon/internal.h   2010-12-25 22:32:15.000000000 +0100
+++ libmicrohttpd-0.9.5.no-abort/src/daemon/internal.h  2011-01-24 21:01:13.519614064 +0100
@@ -33,7 +33,7 @@
 #include <gnutls/gnutls.h>
 #endif
 
-#define EXTRA_CHECKS MHD_YES
+#define EXTRA_CHECKS MHD_NO
 
 #define MHD_MAX(a,b) ((a)<(b)) ? (b) : (a)
 #define MHD_MIN(a,b) ((a)<(b)) ? (a) : (b)
