Index: baselayout-3.4/root/etc/passwd
===================================================================
--- baselayout-3.4.orig/root/etc/passwd
+++ baselayout-3.4/root/etc/passwd
@@ -12,7 +12,7 @@ man:x:13:15:man:/usr/share/man:/bin/fals
 postmaster:x:14:12:postmaster:/var/spool/mail:/bin/false
 cron:x:16:16:cron:/var/spool/cron:/bin/false
 ftp:x:21:21::/home/ftp:/bin/false
-sshd:x:22:22:sshd:/dev/null:/bin/false
+sshd:x:22:22:sshd:/var/empty/sshd:/bin/false
 at:x:25:25:at:/var/spool/cron/atjobs:/bin/false
 dbus:x:101:101:D-Bus:/dev/null:/bin/false
 hal:x:102:102:Hal:/dev/null:/bin/false
