--- a/root/etc/sysctl.conf	(revision 33363)
+++ b/root/etc/sysctl.conf	(revision 33364)
@@ -38,6 +38,9 @@
 #net.ipv4.conf.all.secure_redirects = 0
 #net.ipv4.conf.default.secure_redirects = 0
 
+# Enable NF_CONNTRACK_ACCT
+# net.netfilter.nf_conntrack_acct = 1
+
 # Ignore ICMP broadcasts
 net.ipv4.icmp_echo_ignore_broadcasts = 1
 
@@ -60,6 +63,6 @@
 # UDP Port for lock manager
 #fs.nfs.nlm_udpport = 0
 
-# default is 8k, Beagle runs better with 16K
-fs.inotify.max_user_watches = 16384
+# default is 8k, nepomuk runs better with 512K
+fs.inotify.max_user_watches = 524288
 
