Index: root/etc/mime.types
===================================================================
--- root/etc/mime.types	(revision 34908)
+++ root/etc/mime.types	(working copy)
@@ -2,8 +2,14 @@
 #
 #  MIME-TYPES and the extensions that represent them
 #
-#  This file is based on debian's "mime-support".
+#  The format of this file is a MIME type on the left and zero or more
+#  filename extensions on the right.  Programs using this file will map
+#  files ending with those extensions to the associated type.
 #
+#  This file is part of the "mime-support" package.  Please send email (not a
+#  bug report) to mime-support@packages.debian.org if you would like new types
+#  and/or extensions to be added.
+#
 #  The reason that all types are managed by the mime-support package instead
 #  allowing individual packages to install types in much the same way as they
 #  add entries in to the mailcap file is so these types can be referenced by
@@ -25,23 +31,27 @@
 
 application/activemessage
 application/andrew-inset			ez
+application/annodex				anx
 application/applefile
-application/atom				atom
+application/atom+xml				atom
 application/atomcat+xml				atomcat
+application/atomicmail
 application/atomserv+xml			atomsrv
-application/atomicmail
 application/batch-SMTP
+application/bbolin				lin
 application/beep+xml
 application/cals-1840
 application/cap					cap pcap
 application/commonground
 application/cu-seeme				cu
 application/cybercash
+application/davmount+xml			davmount
 application/dca-rft
 application/dec-dx
 application/docbook+xml
 application/dsptype				tsp
 application/dvcs
+application/ecmascript				es
 application/edi-consent
 application/edi-x12
 application/edifact
@@ -64,22 +74,26 @@
 application/java-archive			jar
 application/java-serialized-object		ser
 application/java-vm				class
+application/javascript				js
+application/json				json
+application/m3g					m3g
 application/mac-binhex40			hqx
 application/mac-compactpro			cpt
 application/macwriteii
 application/marc
-application/mathematica				nb
-application/mathematica-old
+application/mathematica				nb nbp
 application/ms-tnef
 application/msaccess				mdb
 application/msword				doc dot
+application/mxf					mxf
 application/news-message-id
 application/news-transmission
 application/ocsp-request
 application/ocsp-response
 application/octet-stream			bin
 application/oda					oda
-application/ogg					ogg
+application/ogg					ogx
+application/onenote				one onetoc2 onetmp onepkg
 application/parityfec
 application/pdf					pdf
 application/pgp-encrypted
@@ -92,7 +106,7 @@
 application/pkix-cert
 application/pkix-crl
 application/pkixcmp
-application/postscript				ps ai eps
+application/postscript				ps ai eps epsi epsf eps2 eps3
 application/prs.alvestrand.titrax-sheet
 application/prs.cww
 application/prs.nprend
@@ -111,6 +125,7 @@
 application/sgml
 application/sgml-open-catalog
 application/sieve
+application/sla					stl
 application/slate
 application/smil				smi smil
 application/timestamp-query
@@ -119,19 +134,19 @@
 application/whoispp-query
 application/whoispp-response
 application/wita
-application/wordperfect				wpd
-application/wordperfect5.1			wp5
 application/x400-bp
 application/xhtml+xml				xhtml xht
-application/xml					xml xsl
+application/xml					xml xsl xsd
 application/xml-dtd
 application/xml-external-parsed-entity
+application/xspf+xml				xspf
 application/zip					zip
 application/vnd.3M.Post-it-Notes
 application/vnd.accpac.simply.aso
 application/vnd.accpac.simply.imp
 application/vnd.acucobol
 application/vnd.aether.imp
+application/vnd.android.package-archive						apk
 application/vnd.anser-web-certificate-issue-initiation
 application/vnd.anser-web-funds-transfer-initiation
 application/vnd.audiograph
@@ -139,7 +154,7 @@
 application/vnd.businessobjects
 application/vnd.canon-cpdl
 application/vnd.canon-lips
-application/vnd.cinderella			cdy
+application/vnd.cinderella							cdy
 application/vnd.claymore
 application/vnd.commerce-battelle
 application/vnd.commonspace
@@ -183,8 +198,8 @@
 application/vnd.fujixerox.docuworks
 application/vnd.fujixerox.docuworks.binder
 application/vnd.fut-misnet
-application/vnd.google-earth.kml+xml		kml
-application/vnd.google-earth.kmz		kmz
+application/vnd.google-earth.kml+xml						kml
+application/vnd.google-earth.kmz						kmz
 application/vnd.grafeq
 application/vnd.groove-account
 application/vnd.groove-identity-message
@@ -245,16 +260,28 @@
 application/vnd.motorola.flexsuite.kmr
 application/vnd.motorola.flexsuite.ttc
 application/vnd.motorola.flexsuite.wem
-application/vnd.mozilla.xul+xml			xul
+application/vnd.mozilla.xul+xml							xul
 application/vnd.ms-artgalry
 application/vnd.ms-asf
-application/vnd.ms-excel			xls xlb xlt
+application/vnd.ms-excel							xls xlb xlt
+application/vnd.ms-excel.addin.macroEnabled.12					xlam
+application/vnd.ms-excel.sheet.binary.macroEnabled.12				xlsb
+application/vnd.ms-excel.sheet.macroEnabled.12					xlsm
+application/vnd.ms-excel.template.macroEnabled.12				xltm
 application/vnd.ms-lrm
-application/vnd.ms-pki.seccat			cat
-application/vnd.ms-pki.stl			stl
-application/vnd.ms-powerpoint			ppt pps
+application/vnd.ms-officetheme							thmx
+application/vnd.ms-pki.seccat							cat
+#application/vnd.ms-pki.stl							stl
+application/vnd.ms-powerpoint							ppt pps
+application/vnd.ms-powerpoint.addin.macroEnabled.12				ppam
+application/vnd.ms-powerpoint.presentation.macroEnabled.12			pptm
+application/vnd.ms-powerpoint.slide.macroEnabled.12				sldm
+application/vnd.ms-powerpoint.slideshow.macroEnabled.12				ppsm
+application/vnd.ms-powerpoint.template.macroEnabled.12				potm
 application/vnd.ms-project
 application/vnd.ms-tnef
+application/vnd.ms-word.document.macroEnabled.12				docm
+application/vnd.ms-word.template.macroEnabled.12				dotm
 application/vnd.ms-works
 application/vnd.mseq
 application/vnd.msign
@@ -267,20 +294,30 @@
 application/vnd.novadigm.EDM
 application/vnd.novadigm.EDX
 application/vnd.novadigm.EXT
-application/vnd.oasis.opendocument.chart			odc
-application/vnd.oasis.opendocument.database			odb
-application/vnd.oasis.opendocument.formula			odf
-application/vnd.oasis.opendocument.graphics			odg
-application/vnd.oasis.opendocument.graphics-template		otg
-application/vnd.oasis.opendocument.image			odi
-application/vnd.oasis.opendocument.presentation			odp
-application/vnd.oasis.opendocument.presentation-template	otp
-application/vnd.oasis.opendocument.spreadsheet			ods
-application/vnd.oasis.opendocument.spreadsheet-template		ots
-application/vnd.oasis.opendocument.text				odt
-application/vnd.oasis.opendocument.text-master			odm
-application/vnd.oasis.opendocument.text-template		ott
-application/vnd.oasis.opendocument.text-web			oth
+application/vnd.oasis.opendocument.chart					odc
+application/vnd.oasis.opendocument.database					odb
+application/vnd.oasis.opendocument.formula					odf
+application/vnd.oasis.opendocument.graphics					odg
+application/vnd.oasis.opendocument.graphics-template				otg
+application/vnd.oasis.opendocument.image					odi
+application/vnd.oasis.opendocument.presentation					odp
+application/vnd.oasis.opendocument.presentation-template			otp
+application/vnd.oasis.opendocument.spreadsheet					ods
+application/vnd.oasis.opendocument.spreadsheet-template				ots
+application/vnd.oasis.opendocument.text						odt
+application/vnd.oasis.opendocument.text-master					odm
+application/vnd.oasis.opendocument.text-template				ott
+application/vnd.oasis.opendocument.text-web					oth
+application/vnd.openxmlformats-officedocument.presentationml.presentation	pptx
+application/vnd.openxmlformats-officedocument.presentationml.slide		sldx
+application/vnd.openxmlformats-officedocument.presentationml.slideshow		ppsx
+application/vnd.openxmlformats-officedocument.presentationml.template		potx
+application/vnd.openxmlformats-officedocument.spreadsheetml.sheet		xlsx
+application/vnd.openxmlformats-officedocument.spreadsheetml.sheet		xlsx
+application/vnd.openxmlformats-officedocument.spreadsheetml.template		xltx
+application/vnd.openxmlformats-officedocument.spreadsheetml.template		xltx
+application/vnd.openxmlformats-officedocument.wordprocessingml.document		docx
+application/vnd.openxmlformats-officedocument.wordprocessingml.template		dotx
 application/vnd.osa.netdeploy
 application/vnd.palm
 application/vnd.pg.format
@@ -296,38 +333,38 @@
 application/vnd.pvi.ptid1
 application/vnd.pwg-xhtml-print+xml
 application/vnd.rapid
-application/vnd.rim.cod				cod
+application/vnd.rim.cod								cod
 application/vnd.s3sms
 application/vnd.seemail
 application/vnd.shana.informed.formdata
 application/vnd.shana.informed.formtemplate
 application/vnd.shana.informed.interchange
 application/vnd.shana.informed.package
-application/vnd.smaf				mmf
+application/vnd.smaf								mmf
 application/vnd.sss-cod
 application/vnd.sss-dtf
 application/vnd.sss-ntf
-application/vnd.stardivision.calc		sdc
-application/vnd.stardivision.chart		sds
-application/vnd.stardivision.draw		sda
-application/vnd.stardivision.impress		sdd
-application/vnd.stardivision.math		sdf
-application/vnd.stardivision.writer		sdw
-application/vnd.stardivision.writer-global	sgl
+application/vnd.stardivision.calc						sdc
+application/vnd.stardivision.chart						sds
+application/vnd.stardivision.draw						sda
+application/vnd.stardivision.impress						sdd
+application/vnd.stardivision.math						sdf
+application/vnd.stardivision.writer						sdw
+application/vnd.stardivision.writer-global					sgl
 application/vnd.street-stream
-application/vnd.sun.xml.calc			sxc
-application/vnd.sun.xml.calc.template		stc
-application/vnd.sun.xml.draw			sxd
-application/vnd.sun.xml.draw.template		std
-application/vnd.sun.xml.impress			sxi
-application/vnd.sun.xml.impress.template	sti
-application/vnd.sun.xml.math			sxm
-application/vnd.sun.xml.writer			sxw
-application/vnd.sun.xml.writer.global		sxg
-application/vnd.sun.xml.writer.template		stw
+application/vnd.sun.xml.calc							sxc
+application/vnd.sun.xml.calc.template						stc
+application/vnd.sun.xml.draw							sxd
+application/vnd.sun.xml.draw.template						std
+application/vnd.sun.xml.impress							sxi
+application/vnd.sun.xml.impress.template					sti
+application/vnd.sun.xml.math							sxm
+application/vnd.sun.xml.writer							sxw
+application/vnd.sun.xml.writer.global						sxg
+application/vnd.sun.xml.writer.template						stw
 application/vnd.svd
 application/vnd.swiftview-ics
-application/vnd.symbian.install			sis
+application/vnd.symbian.install							sis
 application/vnd.triscape.mxs
 application/vnd.trueapp
 application/vnd.truedoc
@@ -349,14 +386,16 @@
 application/vnd.vcx
 application/vnd.vectorworks
 application/vnd.vidsoft.vidconference
-application/vnd.visio				vsd
+application/vnd.visio								vsd
 application/vnd.vividence.scriptfile
 application/vnd.wap.sic
 application/vnd.wap.slc
-application/vnd.wap.wbxml			wbxml
-application/vnd.wap.wmlc			wmlc
-application/vnd.wap.wmlscriptc			wmlsc
+application/vnd.wap.wbxml							wbxml
+application/vnd.wap.wmlc							wmlc
+application/vnd.wap.wmlscriptc							wmlsc
 application/vnd.webturbo
+application/vnd.wordperfect							wpd
+application/vnd.wordperfect5.1							wp5
 application/vnd.wrq-hp3000-labelled
 application/vnd.wt.stf
 application/vnd.xara
@@ -371,9 +410,10 @@
 application/x-cab				cab
 application/x-cbr				cbr
 application/x-cbz				cbz
-application/x-cdf				cdf
+application/x-cdf				cdf cda
 application/x-cdlink				vcd
 application/x-chess-pgn				pgn
+application/x-comsol				mph
 application/x-core
 application/x-cpio				cpio
 application/x-csh				csh
@@ -382,30 +422,33 @@
 application/x-dms				dms
 application/x-doom				wad
 application/x-dvi				dvi
-application/x-httpd-eruby			rhtml
 application/x-executable
-application/x-flac				flac
 application/x-font				pfa pfb gsf pcf pcf.Z
 application/x-freemind				mm
 application/x-futuresplash			spl
+application/x-ganttproject			gan
 application/x-gnumeric				gnumeric
 application/x-go-sgf				sgf
 application/x-graphing-calculator		gcf
-application/x-gtar				gtar tgz taz
+application/x-gtar				gtar
+application/x-gtar-compressed			tgz taz
 application/x-hdf				hdf
+application/x-httpd-eruby			rhtml
 application/x-httpd-php				phtml pht php
 application/x-httpd-php-source			phps
 application/x-httpd-php3			php3
 application/x-httpd-php3-preprocessed		php3p
 application/x-httpd-php4			php4
+application/x-httpd-php5			php5
 application/x-ica				ica
+application/x-info				info
 application/x-internet-signup			ins isp
 application/x-iphone				iii
 application/x-iso9660-image			iso
+application/x-jam				jam
 application/x-java-applet
 application/x-java-bean
 application/x-java-jnlp-file			jnlp
-application/x-javascript			js
 application/x-jmol				jmz
 application/x-kchart				chrt
 application/x-kdelnk
@@ -421,12 +464,13 @@
 application/x-lzx				lzx
 application/x-maker				frm maker frame fm fb book fbdoc
 application/x-mif				mif
+application/x-mpegURL				m3u8
 application/x-ms-wmd				wmd
 application/x-ms-wmz				wmz
 application/x-msdos-program			com exe bat dll
 application/x-msi				msi
 application/x-netcdf				nc
-application/x-ns-proxy-autoconfig		pac
+application/x-ns-proxy-autoconfig		pac dat
 application/x-nwc				nwc
 application/x-object				o
 application/x-oz-application			oza
@@ -434,13 +478,19 @@
 application/x-pkcs7-crl				crl
 application/x-python-code			pyc pyo
 application/x-pisi			pisi
+application/x-qgis				qgs shp shx
 application/x-quicktimeplayer			qtl
+application/x-rdp				rdp
 application/x-redhat-package-manager		rpm
+application/x-ruby				rb
 application/x-rx
+application/x-scilab				sci sce
 application/x-sh				sh
 application/x-shar				shar
 application/x-shellscript
 application/x-shockwave-flash			swf swfl
+application/x-silverlight			scr
+application/x-sql				sql
 application/x-stuffit				sit sitx
 application/x-sv4cpio				sv4cpio
 application/x-sv4crc				sv4crc
@@ -465,7 +515,14 @@
 
 audio/32kadpcm
 audio/3gpp
+audio/amr					amr
+audio/amr-wb					awb
+audio/amr					amr
+audio/amr-wb					awb
+audio/annodex					axa
 audio/basic					au snd
+audio/csound					csd orc sco
+audio/flac					flac
 audio/g.722.1
 audio/l16
 audio/midi					mid midi kar
@@ -473,6 +530,7 @@
 audio/mpa-robust
 audio/mpeg					mpga mpega mp2 mp3 m4a
 audio/mpegurl					m3u
+audio/ogg					oga ogg spx
 audio/parityfec
 audio/prs.sid					sid
 audio/telephone-event
@@ -545,7 +603,7 @@
 chemical/x-mol2					mol2
 chemical/x-molconn-Z				b
 chemical/x-mopac-graph				gpt
-chemical/x-mopac-input				mop mopcrt mpc dat zmt
+chemical/x-mopac-input				mop mopcrt mpc zmt
 chemical/x-mopac-out				moo
 chemical/x-mopac-vib				mvb
 chemical/x-ncbi-asn1				asn
@@ -586,15 +644,20 @@
 image/vnd.svf
 image/vnd.wap.wbmp				wbmp
 image/vnd.xiff
+image/x-canon-cr2				cr2
+image/x-canon-crw				crw
 image/x-cmu-raster				ras
 image/x-coreldraw				cdr
 image/x-coreldrawpattern			pat
 image/x-coreldrawtemplate			cdt
 image/x-corelphotopaint				cpt
+image/x-epson-erf				erf
 image/x-icon					ico
 image/x-jg					art
 image/x-jng					jng
 image/x-ms-bmp					bmp
+image/x-nikon-nef				nef
+image/x-olympus-orf				orf
 image/x-photoshop				psd
 image/x-portable-anymap				pnm
 image/x-portable-bitmap				pbm
@@ -631,6 +694,9 @@
 model/vnd.mts
 model/vnd.vtu
 model/vrml					wrl vrml
+model/x3d+vrml					x3dv
+model/x3d+xml					x3d
+model/x3d+binary				x3db
 
 multipart/alternative
 multipart/appledouble
@@ -646,6 +712,7 @@
 multipart/signed
 multipart/voice-message
 
+text/cache-manifest				manifest
 text/calendar					ics icz
 text/css					css
 text/csv					csv
@@ -657,14 +724,14 @@
 text/iuls					uls
 text/mathml					mml
 text/parityfec
-text/plain					asc txt text pot
+text/plain					asc txt text pot brf
 text/prs.lines.tag
 text/rfc822-headers
 text/richtext					rtx
 text/rtf
 text/scriptlet					sct wsc
 text/t140
-text/texmacs					tm ts
+text/texmacs					tm
 text/tab-separated-values			tsv
 text/uri-list
 text/vnd.abc
@@ -705,8 +772,10 @@
 text/x-pcs-gcd					gcd
 text/x-perl					pl pm
 text/x-python					py
+text/x-scala					scala
 text/x-server-parsed-html
 text/x-setext					etx
+text/x-sfv					sfv
 text/x-sh					sh
 text/x-tcl					tcl tk
 text/x-tex					tex ltx sty cls
@@ -714,16 +783,20 @@
 text/x-vcard					vcf
 
 video/3gpp					3gp
+video/annodex					axv
 video/dl					dl
 video/dv					dif dv
 video/fli					fli
 video/gl					gl
 video/mpeg					mpeg mpg mpe
+video/MP2T					ts
 video/mp4					mp4
 video/quicktime					qt mov
 video/mp4v-es
+video/ogg					ogv
 video/parityfec
 video/pointer
+video/webm					webm
 video/vnd.fvt
 video/vnd.motorola.video
 video/vnd.motorola.videop
@@ -731,6 +804,7 @@
 video/vnd.mts
 video/vnd.nokia.interleaved-multimedia
 video/vnd.vivo
+video/x-flv					flv
 video/x-la-asf					lsf lsx
 video/x-mng					mng
 video/x-ms-asf					asf asx
@@ -740,6 +814,7 @@
 video/x-ms-wvx					wvx
 video/x-msvideo					avi
 video/x-sgi-movie				movie
+video/x-matroska				mpv mkv
 
 x-conference/x-cooltalk				ice
 
