Index: kdebindings-4.4.90/cmake/modules/FindQwt5.cmake
===================================================================
--- kdebindings-4.4.90.orig/cmake/modules/FindQwt5.cmake
+++ kdebindings-4.4.90/cmake/modules/FindQwt5.cmake
@@ -41,7 +41,7 @@ IF( QT4_FOUND )
 		STRING(REGEX REPLACE ".*#define[\\t\\ ]+QWT_VERSION_STR[\\t\\ ]+\"([0-9]+\\.[0-9]+\\.[0-9]+)\".*" "\\1" Qwt_VERSION "${QWT_GLOBAL_H}")
 
 		# Find Qwt5 library linked to Qt4
-		FIND_LIBRARY( Qwt5_Qt4_TENTATIVE_LIBRARY NAMES qwt5-qt4 qwt-qt4 qwt5 qwt PATHS /usr/local/qwt/lib /usr/local/lib /usr/lib )
+		FIND_LIBRARY( Qwt5_Qt4_TENTATIVE_LIBRARY NAMES qwt5-qt4 qwt-qt4 qwt5 qwt PATHS /usr/local/qwt/lib /usr/local/lib /usr/lib ${QT_LIBRARY_DIR})
 		IF( UNIX AND NOT CYGWIN)
 			IF( Qwt5_Qt4_TENTATIVE_LIBRARY )
 				#MESSAGE("Qwt5_Qt4_TENTATIVE_LIBRARY = ${Qwt5_Qt4_TENTATIVE_LIBRARY}")
