--- pykdeuic4	2009-12-31 15:34:46.000000000 +0200
+++ python/pykde4/tools/pykdeuic4/pykdeuic4.py	2010-09-08 18:29:18.061000017 +0300
@@ -33,13 +33,23 @@
 from PyKDE4 import kdecore
 from PyKDE4 import kdeui
 """
+
+import re
+_esc_regex = re.compile(r"(\"|\'|\\)")
+def _escape(text):
+    # This escapes any escaped single or double quote or backslash.
+    x = _esc_regex.sub(r"\\\1", text)
+
+    # This replaces any '\n' with an escaped version and a real line break.
+    return re.sub(r'\n', r'\\n"\n"', x)
+
 # Override how messages are translated.
 original_i18n_string = qtproxies.i18n_string
 class kde_i18n_string(qtproxies.i18n_string):
     def __init__(self,string):
         original_i18n_string.__init__(self,string)
     def __str__(self):
-        return "kdecore.i18n(\"%s\")" % (qtproxies.escape(self.string),)
+        return "kdecore.i18n(\"%s\")" % (_escape(self.string),)
 qtproxies.i18n_string = kde_i18n_string
 
 def kdeFilter():
@@ -71,7 +81,7 @@
     indenter.indentwidth = indent
     comp = compiler.UICompiler()
     comp.factory._cwFilters.append(kdeFilter())
-    winfo = comp.compileUi(uifile, output)
+    winfo = comp.compileUi(uifile, output, False)
 
     if exe:
         output.write("""
