--- Makefile.am	2006-08-01 17:07:09.000000000 +0200
+++ Makefile.am.new	2006-08-01 17:07:19.000000000 +0200
@@ -8,26 +8,6 @@
 pkgconfigdir=$(libdir)/pkgconfig
 pkgconfig_DATA = libwbxml2.pc
 
-wbxmldocdir = ${prefix}/share/doc/wbxml2-${VERSION}
-wbxmldoc_DATA = \
-	AUTHORS\
-	bootstrap\
-	BUGS\
-	ChangeLog\
-	COPYING\
-	GNU-LGPL\
-	INSTALL\
-	NEWS\
-	README\
-	References\
-	THANKS\
-	TODO\
-	doxygen.h
-
-install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(wbxmldocdir)/manual
-	cp -Rp doc/* $(DESTDIR)$(wbxmldocdir)/manual
-
 dist-bz2: distdir
 	sed -e "s/tar.gz/tar.bz2/g" $(distdir)/wbxml2.spec > $(distdir)/wbxml2.spec.aux
 	mv $(distdir)/wbxml2.spec.aux $(distdir)/wbxml2.spec
