Index: gcc-4.3.3/gcc/config/linux.h
===================================================================
--- gcc-4.3.3.orig/gcc/config/linux.h
+++ gcc-4.3.3/gcc/config/linux.h
@@ -112,9 +112,9 @@ along with GCC; see the file COPYING3.  
    GLIBC_DYNAMIC_LINKER must be defined for each target using them, or
    GLIBC_DYNAMIC_LINKER32 and GLIBC_DYNAMIC_LINKER64 for targets
    supporting both 32-bit and 64-bit compilation.  */
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib32/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER "/usr/lib/uClibc/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER32 "/usr/lib32/uClibc/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 "/usr/lib/uClibc/lib/ld64-uClibc.so.0"
 #define LINUX_DYNAMIC_LINKER \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER, UCLIBC_DYNAMIC_LINKER)
 #define LINUX_DYNAMIC_LINKER32 \
