diff -Nur gcc-4.5-20100715-old/gcc/common.opt gcc-4.5-20100715/gcc/common.opt
--- gcc-4.5-20100715-old/gcc/common.opt	2010-07-21 15:14:03.129594959 +0300
+++ gcc-4.5-20100715/gcc/common.opt	2010-07-21 15:14:33.161722007 +0300
@@ -1135,6 +1135,10 @@
 Common C ObjC C++ ObjC++ Report Var(flag_show_column) Init(1)
 Show column numbers in diagnostics, when available.  Default on
 
+fshow-error-count
+Common Report Var(flag_show_error_count) Init(1)
+Show total number of error and warning messages issued at the end. Default on
+
 fsignaling-nans
 Common Report Var(flag_signaling_nans) Optimization
 Disable optimizations observable by IEEE signaling NaNs
diff -Nur gcc-4.5-20100715-old/gcc/testsuite/lib/gcc.exp gcc-4.5-20100715/gcc/testsuite/lib/gcc.exp
--- gcc-4.5-20100715-old/gcc/testsuite/lib/gcc.exp	2010-07-21 15:14:03.028594641 +0300
+++ gcc-4.5-20100715/gcc/testsuite/lib/gcc.exp	2010-07-21 15:17:35.997729861 +0300
@@ -149,6 +149,7 @@
 	set options [concat "{additional_flags=$TOOL_OPTIONS}" $options]
     }
     lappend options "timeout=[timeout_value]"
+    lappend options "additional_flags=-fno-show-error-count"
     lappend options "compiler=$GCC_UNDER_TEST"
     set options [dg-additional-files-options $options $source]
     return [target_compile $source $dest $type $options]
diff -Nur gcc-4.5-20100715-old/gcc/testsuite/lib/g++.exp gcc-4.5-20100715/gcc/testsuite/lib/g++.exp
--- gcc-4.5-20100715-old/gcc/testsuite/lib/g++.exp	2010-07-21 15:14:03.028594641 +0300
+++ gcc-4.5-20100715/gcc/testsuite/lib/g++.exp	2010-07-21 15:19:52.548629490 +0300
@@ -289,6 +289,7 @@
     }
 
     lappend options "additional_flags=[libio_include_flags]"
+    lappend options "additional_flags=-fno-show-error-count"
     lappend options "compiler=$GXX_UNDER_TEST"
     lappend options "timeout=[timeout_value]"
 
diff -Nur gcc-4.5-20100715-old/gcc/testsuite/lib/gfortran.exp gcc-4.5-20100715/gcc/testsuite/lib/gfortran.exp
--- gcc-4.5-20100715-old/gcc/testsuite/lib/gfortran.exp	2010-07-21 15:14:03.028594641 +0300
+++ gcc-4.5-20100715/gcc/testsuite/lib/gfortran.exp	2010-07-21 15:14:33.165722517 +0300
@@ -214,6 +214,7 @@
 
     lappend options "compiler=$GFORTRAN_UNDER_TEST"
     lappend options "timeout=[timeout_value]"
+    lappend options "additional_flags=-fno-show-error-count"
 
     set options [concat "$ALWAYS_GFORTRANFLAGS" $options]
     set options [dg-additional-files-options $options $source]
diff -Nur gcc-4.5-20100715-old/gcc/testsuite/lib/obj-c++.exp gcc-4.5-20100715/gcc/testsuite/lib/obj-c++.exp
--- gcc-4.5-20100715-old/gcc/testsuite/lib/obj-c++.exp	2010-07-21 15:14:03.028594641 +0300
+++ gcc-4.5-20100715/gcc/testsuite/lib/obj-c++.exp	2010-07-21 15:14:33.166722365 +0300
@@ -360,6 +360,7 @@
     lappend options "additional_flags=[libio_include_flags]"
     lappend options "compiler=$OBJCXX_UNDER_TEST";
     lappend options "timeout=[timeout_value]"
+    lappend options "additional_flags=-fno-show-error-count"
 
     set options [concat $gpp_compile_options $options]
 
diff -Nur gcc-4.5-20100715-old/gcc/testsuite/lib/objc.exp gcc-4.5-20100715/gcc/testsuite/lib/objc.exp
--- gcc-4.5-20100715-old/gcc/testsuite/lib/objc.exp	2010-07-21 15:14:03.028594641 +0300
+++ gcc-4.5-20100715/gcc/testsuite/lib/objc.exp	2010-07-21 15:14:33.167722493 +0300
@@ -246,6 +246,7 @@
     }
     lappend options "compiler=$OBJC_UNDER_TEST"
     lappend options "timeout=[timeout_value]"
+    lappend options "additional_flags=-fno-show-error-count"
 
     set_ld_library_path_env_vars
 
diff -Nur gcc-4.5-20100715-old/gcc/toplev.c gcc-4.5-20100715/gcc/toplev.c
--- gcc-4.5-20100715-old/gcc/toplev.c	2010-07-21 15:14:02.583595403 +0300
+++ gcc-4.5-20100715/gcc/toplev.c	2010-07-21 15:14:33.168722621 +0300
@@ -2386,6 +2386,12 @@
 
   /* Language-specific end of compilation actions.  */
   lang_hooks.finish ();
+
+
+  /* Print error / warning counts.  */
+  if ( flag_show_error_count && (errorcount || warningcount ))
+    fprintf (stderr, "\n*** %d errors, %d warnings\n",
+             errorcount, warningcount);
 }
 
 /* Initialize the compiler, and compile the input file.  */
diff -Nur gcc-4.5-20100715-old/libffi/testsuite/lib/libffi-dg.exp gcc-4.5-20100715/libffi/testsuite/lib/libffi-dg.exp
--- gcc-4.5-20100715-old/libffi/testsuite/lib/libffi-dg.exp	2010-07-21 15:14:02.409594708 +0300
+++ gcc-4.5-20100715/libffi/testsuite/lib/libffi-dg.exp	2010-07-21 15:14:33.169718279 +0300
@@ -208,6 +208,7 @@
     }
 
     lappend options "libs= -lffi"
+    lappend options "additional_flags=-fno-show-error-count"
 
     verbose "options: $options"
     return [target_compile $source $dest $type $options]
diff -Nur gcc-4.5-20100715-old/libgomp/testsuite/lib/libgomp.exp gcc-4.5-20100715/libgomp/testsuite/lib/libgomp.exp
--- gcc-4.5-20100715-old/libgomp/testsuite/lib/libgomp.exp	2010-07-21 15:14:02.200594576 +0300
+++ gcc-4.5-20100715/libgomp/testsuite/lib/libgomp.exp	2010-07-21 15:14:33.171724959 +0300
@@ -197,6 +197,7 @@
     lappend options "additional_flags=[libio_include_flags]"
     lappend options "timeout=[timeout_value]"
     lappend options "compiler=$GCC_UNDER_TEST"
+    lappend options "additional_flags=-fno-show-error-count"
 
     set options [concat $libgomp_compile_options $options]
 
diff -Nur gcc-4.5-20100715-old/libmudflap/testsuite/lib/libmudflap.exp gcc-4.5-20100715/libmudflap/testsuite/lib/libmudflap.exp
--- gcc-4.5-20100715-old/libmudflap/testsuite/lib/libmudflap.exp	2010-07-21 15:14:01.667719596 +0300
+++ gcc-4.5-20100715/libmudflap/testsuite/lib/libmudflap.exp	2010-07-21 15:14:33.172719220 +0300
@@ -220,6 +220,7 @@
     # Picks up the freshly-built testsuite library corresponding to the
     # multilib under test.
     lappend options "ldflags=-L${blddir}/testsuite"
+    lappend options "additional_flags=-fno-show-error-count"
 
     return [target_compile $source $dest $type $options]
 }
diff -Nur gcc-4.5-20100715-old/libstdc++-v3/testsuite/lib/libstdc++.exp gcc-4.5-20100715/libstdc++-v3/testsuite/lib/libstdc++.exp
--- gcc-4.5-20100715-old/libstdc++-v3/testsuite/lib/libstdc++.exp	2010-07-21 15:14:02.260594690 +0300
+++ gcc-4.5-20100715/libstdc++-v3/testsuite/lib/libstdc++.exp	2010-07-21 15:14:33.172719220 +0300
@@ -442,6 +442,7 @@
 
     lappend options "compiler=$cxx_final"
     lappend options "timeout=[timeout_value]"
+    lappend options "additional_flags=-fno-show-error-count"
 
     return [target_compile $source $dest $type $options]
 }
