diff -Nur gcc-4.5-20100715-old/gcc/config/i386/linux64.h gcc-4.5-20100715/gcc/config/i386/linux64.h
--- gcc-4.5-20100715-old/gcc/config/i386/linux64.h	2010-07-21 15:14:02.437594650 +0300
+++ gcc-4.5-20100715/gcc/config/i386/linux64.h	2010-07-21 15:23:55.742761305 +0300
@@ -58,8 +58,8 @@
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux-x86-64.so.2"
 
 #if TARGET_64BIT_DEFAULT
 #define SPEC_32 "m32"
diff -Nur gcc-4.5-20100715-old/gcc/config/i386/t-linux64 gcc-4.5-20100715/gcc/config/i386/t-linux64
--- gcc-4.5-20100715-old/gcc/config/i386/t-linux64	2010-07-21 15:14:02.437594650 +0300
+++ gcc-4.5-20100715/gcc/config/i386/t-linux64	2010-07-21 15:23:55.743759198 +0300
@@ -25,7 +25,7 @@
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32 
-MULTILIB_OSDIRNAMES = ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
+MULTILIB_OSDIRNAMES = ../lib $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
diff -Nur gcc-4.5-20100715-old/gcc/config/linux.h gcc-4.5-20100715/gcc/config/linux.h
--- gcc-4.5-20100715-old/gcc/config/linux.h	2010-07-21 15:14:02.420594715 +0300
+++ gcc-4.5-20100715/gcc/config/linux.h	2010-07-21 15:23:55.744722450 +0300
@@ -118,7 +118,7 @@
    GLIBC_DYNAMIC_LINKER32 and GLIBC_DYNAMIC_LINKER64 for targets
    supporting both 32-bit and 64-bit compilation.  */
 #define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
 #define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
 #define LINUX_DYNAMIC_LINKER \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER, UCLIBC_DYNAMIC_LINKER)
