Index: cpupowerutils-cpupowerutils/Makefile
===================================================================
--- cpupowerutils-cpupowerutils.orig/Makefile
+++ cpupowerutils-cpupowerutils/Makefile
@@ -191,7 +191,7 @@ $(UTIL_OBJS): $(UTIL_HEADERS)
 	$(QUIET) $(CC) $(CFLAGS) $(CPPFLAGS) -I./lib -I ./utils -o $@ -c $*.c
 
 cpupower: $(UTIL_OBJS) libcpufreq
-	$(QUIET) $(CC) $(CFLAGS) $(LDFLAGS) -lcpufreq -lrt -lpci -L. -o $@ $(UTIL_OBJS)
+	$(QUIET) $(CC) $(CFLAGS) $(LDFLAGS) $(UTIL_OBJS) -lcpufreq -lrt -lpci -L. -o $@
 	$(STRIPCMD) $@
 
 po/$(PACKAGE).pot: $(UTIL_SRC)
