Index: choqok-0.9.85/cmake/modules/FindQtOAuth.cmake
===================================================================
--- choqok-0.9.85.orig/cmake/modules/FindQtOAuth.cmake
+++ choqok-0.9.85/cmake/modules/FindQtOAuth.cmake
@@ -18,7 +18,7 @@ if (QTOAUTH_INCLUDE_DIR AND QTOAUTH_LIBR
 else (QTOAUTH_INCLUDE_DIR AND QTOAUTH_LIBRARY)
   if (NOT WIN32)
     find_package(PkgConfig)
-    pkg_check_modules(PC_QTOAUTH QtOAuth)
+    pkg_check_modules(PC_QTOAUTH qoauth)
   endif(NOT WIN32)
 
   find_path(QTOAUTH_INCLUDE_DIR QtOAuth/interface.h
