Index: choqok-0.9.55/libchoqok/ui/microblogwidget.cpp
===================================================================
--- choqok-0.9.55.orig/libchoqok/ui/microblogwidget.cpp
+++ choqok-0.9.55/libchoqok/ui/microblogwidget.cpp
@@ -304,6 +304,8 @@ void MicroBlogWidget::errorPost(Choqok::
 QLayout * MicroBlogWidget::createToolbar()
 {
     d->toolbar = new QHBoxLayout;
+    d->toolbar->setContentsMargins(10, -1, 10, -1);
+
     KPushButton *btnActions = new KPushButton(i18n("Actions"), this);
 
     QLabel *lblLatestUpdate = new QLabel( i18n("Latest update:"), this);
