--- OpenJPEG_v1_3/libopenjpeg/opj_malloc.h	2007-12-21 04:19:01.000000000 -0600
+++ trunk/libopenjpeg/opj_malloc.h	2007-10-18 07:26:11.000000000 -0500
@@ -75,9 +75,6 @@
 #else /* Not WIN32 */
 	#if defined(__sun)
 			#define HAVE_MEMALIGN
-		#elif defined(__GNUC__)
-			#define HAVE_MEMALIGN
-			#include <malloc.h>		
 		/* Linux x86_64 and OSX always align allocations to 16 bytes */
 		#elif !defined(__amd64__) && !defined(__APPLE__)	
 			/* FIXME: Yes, this is a big assumption */
@@ -85,8 +82,6 @@
 	#endif
 #endif
 
-
-
 #define opj_aligned_malloc(size) malloc(size)
 #define opj_aligned_free(m) free(m)
 
