diff -urN -x '*.orig' -x '*.rej' -x '*~' -x '.*' OpenJPEG.orig/codec/convert.c OpenJPEG.patched/codec/convert.c
--- OpenJPEG.orig/codec/convert.c	2007-07-17 11:19:41.000000000 -0500
+++ OpenJPEG.patched/codec/convert.c	2007-07-17 19:40:02.000000000 -0500
@@ -33,7 +33,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include "openjpeg.h"
-#include "../libs/libtiff/tiffio.h"
+#include "tiffio.h"
 #include "convert.h"
 
 /*
