diff -urN -x '*.orig' -x '*.rej' -x '*~' -x '.*' OpenJPEG.orig/libopenjpeg/openjpeg.h OpenJPEG.patched/libopenjpeg/openjpeg.h
--- OpenJPEG.orig/libopenjpeg/openjpeg.h	2007-11-14 05:34:48.000000000 -0600
+++ OpenJPEG.patched/libopenjpeg/openjpeg.h	2007-11-16 20:41:53.000000000 -0600
@@ -61,7 +61,7 @@
 #endif /* !OPJ_STATIC || !WIN32 */
 
 #ifndef __cplusplus
-#if defined(HAVE_STDBOOL_H)
+#if 1
 /*
 The C language implementation does correctly provide the standard header
 file "stdbool.h".
