diff -Naur patches-orig//dev300/apply patches/dev300/apply
--- patches-orig//dev300/apply	2010-10-25 19:06:25.123000280 +0300
+++ patches/dev300/apply	2010-10-25 19:06:45.098000437 +0300
@@ -2114,8 +2114,6 @@
 fields-nested-set.diff, n#634478, cbosdo
 fields-double-click.diff, n#639288, cbosdo
 
-fields-table-formula.diff, n#631912, cbosdo
-
 # Backport from cbosdo06 CWS
 sw-table-frame-uno-fix.diff, i#112564, cbosdo
 
