diff -Naur patches-orig//dev300/apply patches/dev300/apply
--- patches-orig//dev300/apply	2010-10-02 18:30:33.118000598 +0300
+++ patches/dev300/apply	2010-10-02 18:30:52.468000598 +0300
@@ -22,7 +22,7 @@
 
 LinuxCommon : Common, Defaults, FontConfigTemporaryHacks, \
 	      FedoraLinuxOnlyFixes, LinuxOnly, msaccess, OpenGLTransitions, \
-	      CliMonoCommon, Mono, MonoComp, AddressBooks, QuickStarter, KDE4, Netbook, \
+	      CliMonoCommon, Mono, AddressBooks, QuickStarter, KDE4, Netbook, \
 	      GlobalNSPluginSupport, SystemBitsWithSTLport
 MacOSXCommon : Common, Defaults
 Win32Common : Common, Defaults, Win32Only, CliMonoCommon, CliComponent, NotDebian
@@ -1099,8 +1099,6 @@
 mono-climaker-bridgetest.diff
 # Mono component support of course requires Mono
 # same caveat applied too e.g. experimental
-[ MonoComp ]
-mono-component-support.diff
 
 [ VBAObjects ]
 SectionOwner => noelpwer
