Index: player-3.0.2/cmake/internal/SetupDirectories.cmake
===================================================================
--- player-3.0.2.orig/cmake/internal/SetupDirectories.cmake
+++ player-3.0.2/cmake/internal/SetupDirectories.cmake
@@ -11,11 +11,5 @@ MESSAGE (STATUS "Player will be installe
 STRING (TOLOWER ${PROJECT_NAME} projectNameLower)
 SET (PLAYER_INCLUDE_INSTALL_DIR "include/${projectNameLower}-${PLAYER_MAJOR_VERSION}.${PLAYER_MINOR_VERSION}")
 
-IF (PLAYER_PROC_64BIT)
-    SET (PLAYER_LIBRARY_INSTALL_DIR "lib64")
-    MESSAGE (STATUS "Player libraries will be in lib64")
-ELSE (PLAYER_PROC_64BIT)
-    SET (PLAYER_LIBRARY_INSTALL_DIR "lib")
-    MESSAGE (STATUS "Player libraries will be in lib")
-ENDIF (PLAYER_PROC_64BIT)
-
+SET (PLAYER_LIBRARY_INSTALL_DIR "lib")
+MESSAGE (STATUS "Player libraries will be in lib")
