Index: kdebase-runtime-4.4.80/knetattach/knetattach.cpp
===================================================================
--- kdebase-runtime-4.4.80.orig/knetattach/knetattach.cpp	2010-05-24 14:48:29.349438001 -0400
+++ kdebase-runtime-4.4.80/knetattach/knetattach.cpp	2010-05-24 14:48:29.369458953 -0400
@@ -48,7 +48,7 @@
     connect(_useEncryption, SIGNAL(toggled(bool)), this, SLOT(updatePort(bool)));
     connect(_createIcon, SIGNAL(toggled(bool)), this, SLOT(updateFinishButtonText(bool)));
     connect( this, SIGNAL(helpRequested()), this, SLOT( slotHelpClicked() ) );
-    setWindowIcon(KIcon("knetattach"));
+    setWindowIcon(KIcon("folder-new"));
     setOption(HaveHelpButton, true);
     button(FinishButton)->setText(i18n("Save && C&onnect"));
     //setResizeMode(Fixed); FIXME: make the wizard fixed-geometry
Index: kdebase-runtime-4.4.80/knetattach/knetattach.desktop
===================================================================
--- kdebase-runtime-4.4.80.orig/knetattach/knetattach.desktop	2010-05-24 14:48:29.329460140 -0400
+++ kdebase-runtime-4.4.80/knetattach/knetattach.desktop	2010-05-24 14:48:29.369458953 -0400
@@ -173,7 +173,7 @@
 GenericName[zh_TW]=網路資料夾精靈
 Exec=knetattach
 InitialPreference=6
-Icon=knetattach
+Icon=folder-new
 Type=Application
 Terminal=false
 X-KDE-StartupNotify=true
