Index: kdebase-runtime-4.4.80/knetattach/knetattach.desktop
===================================================================
--- kdebase-runtime-4.4.80.orig/knetattach/knetattach.desktop	2010-04-15 04:58:48.000000000 -0400
+++ kdebase-runtime-4.4.80/knetattach/knetattach.desktop	2010-05-24 14:48:49.076938176 -0400
@@ -179,3 +179,4 @@
 X-KDE-StartupNotify=true
 Categories=Qt;KDE;Network;RemoteAccess;
 X-DocPath=knetattach/index.html
+NoDisplay=true
