Index: openshot-1.2.2/openshot/windows/preferences.py
===================================================================
--- openshot-1.2.2.orig/openshot/windows/preferences.py
+++ openshot-1.2.2/openshot/windows/preferences.py
@@ -227,7 +227,7 @@ class Settings:
 	#values will be overwritten with the file values.
 	general = {
 		"imported_image_length" : "7",
-		"default_theme" : "blue_glass",
+		"default_theme" : "simple",
 		"default_profile" : "DV NTSC",
 		"project_file_type" : "ascii",
 		"max_history_size" : "20",
