From dac817a3972dd03d48d34ebc59738490dd140da5 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Thu, 30 Dec 2010 14:43:28 +0200
Subject: [PATCH 1/2] Change firstboot option value

---
 yali-bin |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/yali-bin b/yali-bin
index 46d8bb2..78633b3 100755
--- a/yali-bin
+++ b/yali-bin
@@ -170,12 +170,12 @@ def load_options(argv=None):
     parser.add_option("--system", dest="install_type",
                       action="store_const", const= 1, default=0,
                       help="start system installation")
-    parser.add_option("--oem", dest="install_type",
-                      action="store_const", const=2, default=0,
-                      help="start Yali with oem mode")
     parser.add_option("--firstboot", dest="install_type",
-                      action="store_const", const=3, default=0,
+                      action="store_const", const=2, default=0,
                       help="start with first boot options")
+    parser.add_option("--oem", dest="install_type",
+                      action="store_const", const=3, default=0,
+                      help="start Yali with oem mode")
     parser.add_option("--rescue", dest="install_type",
                       action="store_const", const=4, default=0,
                       help="start Yali with rescue mode")
-- 
1.7.3.4

