From 1ecf5c813804ecbdf8e13e5484858e7f9b6f7916 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Wed, 1 Dec 2010 21:48:46 +0200
Subject: [PATCH 3/4] yali.gui.lvm_gui:Fixes setting same mountpoint more than one

BUG:FIXED:14892
---
 yali/gui/lvm_gui.py |   30 +++++++++++++++++++++---------
 1 files changed, 21 insertions(+), 9 deletions(-)

diff --git a/yali/gui/lvm_gui.py b/yali/gui/lvm_gui.py
index bd55c3b..a6d2bd6 100644
--- a/yali/gui/lvm_gui.py
+++ b/yali/gui/lvm_gui.py
@@ -604,11 +604,14 @@ class LogicalVolumeEditor:
 
             targetSize = None
             migrate = None
+            format = None
 
             widget = self.dialog.content
 
+            format = self.origrequest.format
+
             mountpoint = str(widget.mountpointMenu.currentText())
-            if widget.mountpointMenu.isEditable() and mountpoint:
+            if mountpoint and widget.mountpointMenu.isEditable():
                 msg = sanityCheckMountPoint(mountpoint)
                 if msg:
                     self.intf.messageWindow(_("Mount Point Error"),
@@ -616,30 +619,39 @@ class LogicalVolumeEditor:
                                             customIcon="error")
                     continue
 
+            if not self.origrequest.exists:
+                format_type = str(widget.filesystemMenu.currentText())
+            else:
+                format_type = str(widget.formatCombo.currentText())
+
+            format_class = formats.getFormat(format_type)
+            if mountpoint and format_class.mountable:
                 used = False
-                for (mp, dev) in self.parent.parent.storage.mountpoints.iteritems():
-                    if (dev.type != "lvmlv" or dev.vg.id != self.origrequest.vg.id) and \
-                        mp == mountpoint and \
-                        self.origrequest.name in [parent.name for parent in dev.parents]:
-                        used = True
-                        break
+
+                current_mountpoint = getattr(format, "mountpoint", None)
 
                 for lv in self.parent.parent.lvs.values():
                     format = lv["format"]
-                    if not format.mountable or mountpoint and \
-                        format.mountpoint == mountpoint:
+                    if not format.mountable or current_mountpoint and \
+                        format.mountpoint == current_mountpoint:
                         continue
 
                     if format.mountpoint == mountpoint:
                         used =True
                         break
 
+                for (mp, dev) in self.parent.parent.storage.mountpoints.iteritems():
+                    if (dev.type != "lvmlv" or dev.vg.id != self.origrequest.vg.id) and mp == mountpoint:
+                        used = True
+                        break
+
                 if used:
                     self.intf.messageWindow(_("Mount point in use"),
                                             _("The mount point \"%s\" is in "
                                               "use. Please pick another.") %
                                             (mountpoint,),
                                             customIcon="error")
+                    continue
 
 
             name = str(widget.name.text())
-- 
1.7.2.2

