From e7d30547cbce1a6b7163daae97b16aebe7967040 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Wed, 20 Oct 2010 23:28:30 +0300
Subject: [PATCH 3/3] Redundat value assignment

---
 yali/installer.py |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/yali/installer.py b/yali/installer.py
index f12051f..3741f59 100644
--- a/yali/installer.py
+++ b/yali/installer.py
@@ -155,7 +155,6 @@ class Yali:
 
         self.install_type = install_type
         self.checkCDStop = True
-        self._inconsistentLVMAnswers = {}
 
     def getPlugin(self, p):
         try:
-- 
1.7.2.2

