From 24f0983dcc0ca8a389aa55aa85511d676a5ddae0 Mon Sep 17 00:00:00 2001
From: Mete Alpaslan <mete@pardus.org.tr>
Date: Wed, 1 Dec 2010 22:07:44 +0200
Subject: [PATCH 4/4] yali.util:Add LC_ALL=C to subprocess call

---
 yali-bin     |    1 +
 yali/util.py |    6 +++---
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/yali-bin b/yali-bin
index 294e232..a1ae56a 100755
--- a/yali-bin
+++ b/yali-bin
@@ -97,6 +97,7 @@ def setup_environment():
 
     """
     import locale
+    #FIXME:Why do we need this?
     locale.setlocale(locale.LC_ALL, "")
     os.environ['LC_NUMERIC'] = 'C'
     os.environ["LVM_SUPPRESS_FD_WARNINGS"] = "1"
diff --git a/yali/util.py b/yali/util.py
index a2bf151..4f2f75c 100644
--- a/yali/util.py
+++ b/yali/util.py
@@ -122,11 +122,11 @@ def get_edd_dict(devices):
 
 def run_batch(cmd, argv):
     """Run command and report return value and output."""
-    ctx.logger.info(_('Running %s') % "".join(cmd))
-    cmd = "%s %s" % (cmd, ' '.join(argv))
+    ctx.logger.info('Running %s' % "".join(cmd))
+    cmd = "LC_ALL=C %s %s" % (cmd, ' '.join(argv))
     p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     out, err = p.communicate()
-    ctx.logger.debug(_('return value for "%(command)s" is %(return)s') % {"command":cmd, "return":p.returncode})
+    ctx.logger.debug('return value for "%(command)s" is %(return)s' % {"command":cmd, "return":p.returncode})
     return (p.returncode, out, err)
 
 
-- 
1.7.2.2

