diff --git a/yali/gui/Ui/autopartwidget.ui b/yali/gui/Ui/autopartwidget.ui
index ffd4bb6..b4870f5 100644
--- a/yali/gui/Ui/autopartwidget.ui
+++ b/yali/gui/Ui/autopartwidget.ui
@@ -195,6 +195,9 @@ p, li { white-space: pre-wrap; }
           </property>
           <item row="0" column="0">
            <widget class="QRadioButton" name="shrinkCurrent">
+            <property name="enabled">
+             <bool>false</bool>
+            </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
               <horstretch>0</horstretch>
@@ -208,6 +211,9 @@ p, li { white-space: pre-wrap; }
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="label_8">
+            <property name="enabled">
+             <bool>false</bool>
+            </property>
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
@@ -245,6 +251,9 @@ p, li { white-space: pre-wrap; }
         </item>
         <item>
          <widget class="QLabel" name="label_3">
+          <property name="enabled">
+           <bool>false</bool>
+          </property>
           <property name="text">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
 &lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
diff --git a/yali/gui/lvm_gui.py b/yali/gui/lvm_gui.py
index 8d91612..ebff24a 100644
--- a/yali/gui/lvm_gui.py
+++ b/yali/gui/lvm_gui.py
@@ -77,7 +77,8 @@ class LVMEditor(object):
             operations = []
             if not rc:
                 if self.isNew:
-                    del self.lvs[self.origrequest.name]
+                    if self.lvs.has_key(self.origrequest.name):
+                        del self.lvs[self.origrequest.name]
                 self.destroy()
                 return []
 
@@ -646,13 +647,10 @@ class VolumeGroupWidget(QtGui.QWidget):
         (total, used, free) = self.computeSpace()
         if free <= 0:
             self.parent.intf.messageWindow(_("No free space"),
-                                    _("There is no room left in the "
-                                      "volume group to create new logical "
-                                      "volumes. "
-                                      "To add a logical volume you must "
-                                      "reduce the size of one or more of "
-                                      "the currently existing "
-                                      "logical volumes"), customIcon="error")
+                                    _("There is no room left in the volume group to create new\n"
+                                      "logical volumes. To add a logical volume you must reduce\n"
+                                      "the size of one or more of the currently existing logical\n"
+                                      "volumes"), customIcon="error")
             return
 
         tempvg = self.tmpVolumeGroup
@@ -735,7 +733,8 @@ class LogicalVolumeEditor:
             rc = self.dialog.exec_()
             if not rc:
                 if self.isNew:
-                    del self.parent.parent.lvs[self.origrequest.name]
+                    if self.parent.parent.lvs.has_key(self.origrequest.name):
+                        del self.parent.parent.lvs[self.origrequest.name]
                 self.destroy()
                 return None
 
diff --git a/yali/postinstall.py b/yali/postinstall.py
index f60b8c9..13fc22c 100644
--- a/yali/postinstall.py
+++ b/yali/postinstall.py
@@ -266,10 +266,10 @@ def writeInitramfsConf(parameters=[]):
     rootDevice = ctx.storage.storageset.rootDevice
     parameters.append("root=%s" % rootDevice.fstabSpec)
 
-    swapDevice = ctx.storage.storageset.swapDevices[0]
+    swapDevices = ctx.storage.storageset.swapDevices
 
-    if swapDevice:
-        parameters.append("resume=%s" % swapDevice.path)
+    if swapDevices:
+        parameters.append("resume=%s" % swapDevices[0].path)
 
     parameters = " ".join(parameters)
     ctx.logger.debug("Configuring initramfs.conf file with parameters:%s" % parameters)
