Index: yali/gui/ScrInstall.py
===================================================================
--- yali/gui/ScrInstall.py	(revision 31663)
+++ yali/gui/ScrInstall.py	(working copy)
@@ -361,6 +361,7 @@
             self.lastPackage = keywords['package'].name
             qevent.setData(data)
             objectSender(qevent)
+        QtGui.QApplication.processEvents()
 
     def display_progress(self, operation, percent, info, **keywords):
         pass
