diff --git a/yali/localeutils.py b/yali/localeutils.py
index 3b6b0e1..7c22f50 100644
--- a/yali/localeutils.py
+++ b/yali/localeutils.py
@@ -31,7 +31,7 @@ def setKeymap(keymap, variant=None):
     if variant:
         ad = "-variant %s" % variant
     else:
-        variant = ""
+        variant = "\"\""
     yali.util.run_batch("setxkbmap", ["-layout", keymap, ad])
     yali.util.run_batch("hav", ["call", "zorg", "Xorg.Display", "setKeymap", keymap, variant])
 
