diff --git a/yali/gui/YaliDialog.py b/yali/gui/YaliDialog.py
index ec433ea..5e65d6a 100644
--- a/yali/gui/YaliDialog.py
+++ b/yali/gui/YaliDialog.py
@@ -184,8 +184,10 @@ class MessageWindow:
         elif customIcon == "info":
             icon = "info"
 
+        text = "<qt>%s</qt>" % text.replace("\n", " ")
         self.msgBox.setText(text)
         if detailed:
+            longText = "<qt>%s</qt>" % longText.replace("\n", " ")
             self.msgBox.setDetailedText(unicode(longText))
 
         if self.doCustom:
