diff --git a/yali/gui/ScrSummary.py b/yali/gui/ScrSummary.py
index edd678f..da35b6b 100644
--- a/yali/gui/ScrSummary.py
+++ b/yali/gui/ScrSummary.py
@@ -21,7 +21,7 @@ from PyQt4.QtCore import *
 
 import yali.util
 import yali.context as ctx
-from yali.installdata import YALI_PLUGIN, YALI_INSTALL, YALI_DVDINSTALL
+from yali.installdata import YALI_PLUGIN, YALI_INSTALL, YALI_DVDINSTALL, methodInstallAutomatic, defaultKernel, paeKernel, rtKernel
 from yali.gui.ScreenWidget import ScreenWidget
 from yali.gui.YaliDialog import QuestionDialog
 from yali.gui.Ui.summarywidget import Ui_SummaryWidget
diff --git a/yali/storage/bootloader.py b/yali/storage/bootloader.py
index 4a13ae2..a8d8d91 100644
--- a/yali/storage/bootloader.py
+++ b/yali/storage/bootloader.py
@@ -90,7 +90,7 @@ def get_disk_partition(device):
 grub_conf = """\
 default 0
 timeout 10
-gfxmenu /boot/grub/message
+gfxmenu %(bootpath)sgrub/message
 background 10333C
 
 title %(release)s
