Index: yali-2.4.0_20101121/start-yali
===================================================================
--- yali-2.4.0_20101121.orig/start-yali
+++ yali-2.4.0_20101121/start-yali
@@ -4,4 +4,15 @@ HOME=/root
 USER=root
 export HOME USER
 
-exec /usr/bin/xinit /usr/bin/yali-bin -- vt7 -nolisten tcp -br -dpi 96
+ROOT_WINDOW_BG=-br
+
+if test -x /bin/plymouth && /bin/plymouth --ping; then
+    /bin/plymouth deactivate
+    if /bin/plymouth --has-active-vt; then
+        ROOT_WINDOW_BG=-nr
+    else
+        /bin/plymouth quit
+    fi
+fi
+
+exec /usr/bin/xinit /usr/bin/yali-bin -- vt7 -nolisten tcp -dpi 96 -novtswitch $ROOT_WINDOW_BG
Index: yali-2.4.0_20101121/yali-bin
===================================================================
--- yali-2.4.0_20101121.orig/yali-bin
+++ yali-2.4.0_20101121/yali-bin
@@ -196,6 +196,9 @@ def main():
         print "%s must be run as root." % sys.argv[0]
         sys.exit(1)
 
+    # Plymount running
+    if os.path.exists("/bin/plymouth") and not os.system("/bin/plymouth --ping"):
+        os.system("/bin/plymouth quit --retain-splash")
 
     ctx.logger.debug("Yali has been started")
     load_modules()
