diff --git a/yali-bin b/yali-bin
index 2af73b3..2f07304 100755
--- a/yali-bin
+++ b/yali-bin
@@ -39,6 +39,7 @@ def main():
 
 if __name__ == "__main__":
     locale.setlocale(locale.LC_ALL, "")
+    os.system("modprobe dm-mod")
     yali.init_logging("/var/log")
     main()
 
diff --git a/yali/gui/ScrPartitionManual.py b/yali/gui/ScrPartitionManual.py
index afbcd08..531d7d2 100644
--- a/yali/gui/ScrPartitionManual.py
+++ b/yali/gui/ScrPartitionManual.py
@@ -65,7 +65,7 @@ about disk partitioning.
         self.connect(self.ui.editButton, SIGNAL("clicked()"),self.editDevice)
         self.connect(self.ui.deleteButton, SIGNAL("clicked()"),self.deleteDevice)
         self.connect(self.ui.resetButton, SIGNAL("clicked()"),self.reset)
-        self.connect(self.ui.deviceTree, SIGNAL("itemActivated(QTreeWidgetItem *, int)"), self.activateButtons)
+        self.connect(self.ui.deviceTree, SIGNAL("itemClicked(QTreeWidgetItem *, int)"), self.activateButtons)
 
     def shown(self):
         checkForSwapNoMatch(ctx.yali, self.storage)
@@ -91,7 +91,7 @@ about disk partitioning.
             ctx.mainScreen.disableNext()
 
     def activateButtons(self, item, index):
-        if item.device is not None and isinstance(item.device, Device):
+        if item and item.device is not None and isinstance(item.device, Device):
             self.ui.editButton.setEnabled(True)
             self.ui.deleteButton.setEnabled(True)
         else:
