diff --git a/yali/constants.py b/yali/constants.py
index c11fac6..cdd70f6 100644
--- a/yali/constants.py
+++ b/yali/constants.py
@@ -89,8 +89,8 @@ consts.cd_repo_name = "pardus-cd"
 consts.cd_repo_uri = join(consts.source_dir, "repo/pisi-index.xml.bz2")
 
 # pardus repository
-consts.pardus_repo_name = "pardus-2009"
-consts.pardus_repo_uri = "http://packages.pardus.org.tr/pardus-2009/pisi-index.xml.bz2"
+consts.pardus_repo_name = "pardus"
+consts.pardus_repo_uri = "@REPO_URI@"
 consts.pardus_release_path = "etc/pardus-release"
 
 # min root partition size
