Index: agent/policykitlistener.cpp
===================================================================
--- agent/policykitlistener.cpp.orig
+++ agent/policykitlistener.cpp
@@ -23,6 +23,7 @@
 
 #include <KDebug>
 #include <KWindowSystem>
+#include <KApplication>
 
 #include <PolkitQt1/Agent/Listener>
 #include <PolkitQt1/Agent/Session>
@@ -49,6 +50,8 @@ void PolicyKitListener::initiateAuthenti
         QList<PolkitQt1::Identity *> identities,
         PolkitQt1::Agent::AsyncResult* result)
 {
+    kapp->updateUserTimestamp( 0 ); // HACK make sure it gets focus when shown
+
     kDebug() << "Initiating authentication";
 
     if (m_inProgress) {
