--- polkit-kde-1-0.95.1/agent/AuthDialog.cpp.gcc45	2010-08-04 11:46:42.000000000 +0200
+++ polkit-kde-1-0.95.1/agent/AuthDialog.cpp	2010-08-04 11:46:45.000000000 +0200
@@ -169,7 +169,7 @@
         foreach(PolkitQt1::Identity *identity, identities) {
             // First check to see if the user is valid
             qDebug() << "User: " << identity;
-            KUser user = KUser::KUser(identity->toString().remove("unix-user:"));
+            KUser user = KUser(identity->toString().remove("unix-user:"));
             if (!user.isValid()) {
                 kWarning() << "User invalid: " << user.loginName();
                 continue;
