Index: polkit-kde-1-0.95.1/agent/AuthDialog.cpp
===================================================================
--- polkit-kde-1-0.95.1.orig/agent/AuthDialog.cpp
+++ polkit-kde-1-0.95.1/agent/AuthDialog.cpp
@@ -26,6 +26,7 @@
 #include <QtCore/QProcess>
 #include <QtGui/QPainter>
 #include <QtGui/QStandardItemModel>
+#include <QTimer>
 
 #include <KDebug>
 
@@ -92,6 +93,8 @@ AuthDialog::AuthDialog(const QString &ac
                 this, SLOT(on_userCB_currentIndexChanged(int)));
 
         createUserCB(identities);
+        QTimer::singleShot(0, this, SLOT(selectUser()));
+
     } else {
         userCB->setCurrentIndex(0);
         QStandardItem *item = new QStandardItem("");
@@ -104,6 +107,25 @@ AuthDialog::~AuthDialog()
 {
 }
 
+void AuthDialog::selectUser()
+{
+    char *userName = getenv("USER");
+    int currentUserIndex = -1;
+
+    if (userName)
+        currentUserIndex = userCB->findData(qVariantFromValue<QString> (QString("unix-user:") + userName), Qt::UserRole);
+
+    if (currentUserIndex == -1)
+    {
+        qDebug() << "Current user not found and will not be selected in combobox.";
+    }
+    else
+    {
+        qDebug() << "Current user is " << QString(userName) << ", and will be selected.";
+        userCB->setCurrentIndex(currentUserIndex);
+    }
+}
+
 void AuthDialog::accept()
 {
     // Do nothing, do not close the dialog. This is needed so that the dialog stays
Index: polkit-kde-1-0.95.1/agent/AuthDialog.h
===================================================================
--- polkit-kde-1-0.95.1.orig/agent/AuthDialog.h
+++ polkit-kde-1-0.95.1/agent/AuthDialog.h
@@ -62,6 +62,7 @@ public slots:
 
 private slots:
     void on_userCB_currentIndexChanged(int index);
+    void selectUser();
 
 private:
     QStandardItemModel *m_userModelSIM;
