Index: patches/dev300/apply
===================================================================
--- patches/dev300/apply.orig
+++ patches/dev300/apply
@@ -22,7 +22,7 @@ Common : BuildBits, TemporaryHacks, Fixe
 
 LinuxCommon : Common, Defaults, FontConfigTemporaryHacks, \
 	      FedoraLinuxOnlyFixes, LinuxOnly, msaccess, OpenGLTransitions, \
-	      CliMonoCommon, Mono, MonoComp, AddressBooks, QuickStarter, KDE4, Netbook, \
+	      CliMonoCommon, Mono, AddressBooks, QuickStarter, KDE4, Netbook, \
 	      GlobalNSPluginSupport
 MacOSXCommon : Common, Defaults
 Win32Common : Common, Defaults, Win32Only, CliMonoCommon, CliComponent, NotDebian
@@ -1067,8 +1067,6 @@ mono-climaker-config.diff
 mono-climaker-bridgetest.diff
 # Mono component support of course requires Mono
 # same caveat applied too e.g. experimental
-[ MonoComp ]
-mono-component-support.diff
 
 [ VBAObjects ]
 SectionOwner => noelpwer
