Index: libreoffice-bootstrap-3.4.1.3/configure.in
===================================================================
--- libreoffice-bootstrap-3.4.1.3.orig/configure.in
+++ libreoffice-bootstrap-3.4.1.3/configure.in
@@ -4276,7 +4276,7 @@ int main(int argc, char **argv) {
     DB_LIB=
     for dbver in '' -5.1 5.1 -5.0 5.0 -5 5 -4.8 4.8 -4.7 4.7 -4 4; do
         AC_CHECK_LIB(db$dbver, dbopen, [ DB_LIB="db$dbver"; DB_CPPLIB="db_cxx$dbver"; break; ] ,
-            AC_CHECK_LIB(db$dbver, __db185_open, [ DB_LIB="db$dbver"; DB_CPPLIB="db_cxx$dbver"; break; ]
+            AC_CHECK_LIB(db$dbver, __db_open_4008, [ DB_LIB="db$dbver"; DB_CPPLIB="db_cxx$dbver"; break; ]
             )
         )
     done
