Index: libreoffice-bootstrap-3.4.1.3/configure.in
===================================================================
--- libreoffice-bootstrap-3.4.1.3.orig/configure.in
+++ libreoffice-bootstrap-3.4.1.3/configure.in
@@ -6849,7 +6849,7 @@ if test "$test_kde4" = "yes" -a "$ENABLE
    qt_incdirs="$QT4INC /usr/include/qt4 /usr/include $x_includes"
    qt_libdirs="$QT4LIB /usr/lib $x_libraries"
 
-   kde_incdirs="/usr/include $x_includes"
+   kde_incdirs="/usr/include /usr/include/kde4 $x_includes"
    kde_libdirs="/usr/lib $x_libraries"
 
    if test "$host_cpu" = "x86_64" ; then
