Index: bin/unpack
===================================================================
--- bin/unpack	(revision 11255)
+++ bin/unpack	(working copy)
@@ -393,6 +393,13 @@
     ( bzip2 -dc $SRCDIR/$OOO_EXTRA_ARTWORK | $GNUTAR xpf - ) || exit 1;
 fi
 
+# Pardus only stuff
+tar xjf $SRCDIR/extras_tr.tar.bz2
+mkdir -p $OOBUILDDIR/solver/680/unxlngi6.pro/pck/
+cp $SRCDIR/openclipart_tr.zip $OOBUILDDIR/solver/680/unxlngi6.pro/pck/
+cp $SRCDIR/openclipart_en-US.zip $OOBUILDDIR/solver/680/unxlngi6.pro/pck/
+
+
 for i in $OOO_LANGS; do
 	if test -f $SRCDIR/extras_$i.tar.gz; then
 		echo "Extracting extra $i extras"
