Index: libreoffice-bootstrap-3.4.1.3/officecfg/registry/data/org/openoffice/Office/Common.xcu
===================================================================
--- libreoffice-bootstrap-3.4.1.3.orig/officecfg/registry/data/org/openoffice/Office/Common.xcu
+++ libreoffice-bootstrap-3.4.1.3/officecfg/registry/data/org/openoffice/Office/Common.xcu
@@ -655,8 +655,7 @@
   </node>
   <node oor:name="ExternalMailer" install:module="unixdesktop">
     <prop oor:name="Program">
-      <value oor:external=
-          "com.sun.star.configuration.backend.DesktopBackend ExternalMailer"/>
+        <value>/usr/bin/xdg-email</value>
     </prop>
   </node>
   <node oor:name="Font">
Index: libreoffice-bootstrap-3.4.1.3/src/libreoffice-libs-core-3.4.1.3/shell/source/unix/misc/senddoc.sh
===================================================================
--- libreoffice-bootstrap-3.4.1.3.orig/shell/source/unix/misc/senddoc.sh
+++ libreoffice-bootstrap-3.4.1.3/shell/source/unix/misc/senddoc.sh
@@ -208,7 +208,7 @@ case `basename "$MAILER" | sed 's/-.*$//
         rm -f $BODY
         ;;
         
-    evolution)
+    evolution | xdg )
     
         while [ "$1" != "" ]; do
             case $1 in
