Index: libreoffice-bootstrap-3.4.2.3/bridges/source/cpp_uno/shared/vtablefactory.cxx
===================================================================
--- libreoffice-bootstrap-3.4.2.3.orig/bridges/source/cpp_uno/shared/vtablefactory.cxx
+++ libreoffice-bootstrap-3.4.2.3/bridges/source/cpp_uno/shared/vtablefactory.cxx
@@ -258,8 +258,8 @@ bool VtableFactory::createBlock(Block &b
     osl::Security aSecurity;
     rtl::OUString strDirectory;
     rtl::OUString strURLDirectory;
-    if (aSecurity.getHomeDir(strURLDirectory))
-        osl::File::getSystemPathFromFileURL(strURLDirectory, strDirectory);
+    //if (aSecurity.getHomeDir(strURLDirectory))
+    //    osl::File::getSystemPathFromFileURL(strURLDirectory, strDirectory);
 
     for (int i = strDirectory.getLength() == 0 ? 1 : 0; i < 2; ++i)
     {
