diff -p -up tzdata/tzdata2010c/backward.orig tzdata/tzdata2010c/backward
--- tzdata/tzdata2010c/backward.orig	2010-01-20 15:25:02.000000000 -0200
+++ tzdata/tzdata2010c/backward	2010-03-04 16:39:01.263660599 -0300
@@ -114,3 +114,18 @@ Link	Etc/UTC			UTC
 Link	Etc/UTC			Universal
 Link	Europe/Moscow		W-SU
 Link	Etc/UTC			Zulu
+
+# links requested by users (to access timezones by the name of main cities)
+# pablo@mandriva.com
+# Canada/Saskatchewan 
+Link	America/Regina	America/Saskatoon
+# Canada/Mountain (Alberta)
+Link	America/Edmonton	America/Calgary
+# Canada/Atlantic (New Brunswick)
+Link	America/Halifax		America/Fredericton
+# America/Eastern (Ontario)
+Link	America/Montreal	America/Ontario
+# PRC
+Link	Asia/Shanghai		Asia/Beijing
+# South Africa
+Link	Africa/Johannesburg	Africa/Pretoria
