diff -up tzdata2011b/europe\~ tzdata2011b/europe
--- tzdata2011d/europe~	2010-12-16 18:24:20.000000000 +0100
+++ tzdata2011d/europe	2011-03-04 00:11:25.000000000 +0100
@@ -2131,7 +2131,7 @@ Zone Asia/Novokuznetsk	 5:48:48 -	NMT	19
 			 7:00	Russia	KRA%sT	1991 Mar 31 2:00s
 			 6:00	Russia	KRA%sT	1992 Jan 19 2:00s
 			 7:00	Russia	KRA%sT	2010 Mar 28 2:00s
-			 6:00	Russia	NOV%sT # Novosibirsk/Novokuznetsk Time
+			 6:00	Russia	OMS%sT # Novosibirsk/Novokuznetsk Time
 
 #
 # From Oscar van Vlijmen (2001-08-25): [This region consists of]

Diff finished.  Fri Mar  4 00:11:35 2011
