diff -p -up tzdata/tzdata2010l/backward.orig tzdata/tzdata2010l/backward
--- tzdata/tzdata2010l/backward.orig	2010-08-16 07:57:07.000000000 -0300
+++ tzdata/tzdata2010l/backward	2010-08-16 18:14:54.653273397 -0300
@@ -116,3 +116,18 @@ Link	Etc/UTC			UTC
 Link	Etc/UTC			Universal
 Link	Europe/Moscow		W-SU
 Link	Etc/UTC			Zulu
+
+# links requested by users (to access timezones by the name of main cities)
+# pablo@mandriva.com
+# Canada/Saskatchewan 
+Link	America/Regina	America/Saskatoon
+# Canada/Mountain (Alberta)
+Link	America/Edmonton	America/Calgary
+# Canada/Atlantic (New Brunswick)
+Link	America/Halifax		America/Fredericton
+# America/Eastern (Ontario)
+Link	America/Montreal	America/Ontario
+# PRC
+Link	Asia/Shanghai		Asia/Beijing
+# South Africa
+Link	Africa/Johannesburg	Africa/Pretoria
