commit 7a5921d092674b92c85c0d2c76e5f912784ec169
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Wed Apr 28 18:43:04 2010 +0200

    [G45] Fix build without proper libdrm updates.

diff --git a/i965_drv_video/i965_drv_video.c b/i965_drv_video/i965_drv_video.c
index b8d622d..9482b18 100644
--- a/i965_drv_video/i965_drv_video.c
+++ b/i965_drv_video/i965_drv_video.c
@@ -121,9 +121,11 @@ i965_QueryConfigProfiles(VADriverContextP ctx,
 
     profile_list[i++] = VAProfileMPEG2Simple;
     profile_list[i++] = VAProfileMPEG2Main;
+#ifndef DISABLE_H264_DECODER
     profile_list[i++] = VAProfileH264Baseline;
     profile_list[i++] = VAProfileH264Main;
     profile_list[i++] = VAProfileH264High;
+#endif
 
     /* If the assert fails then I965_MAX_PROFILES needs to be bigger */
     assert(i <= I965_MAX_PROFILES);
@@ -147,12 +149,14 @@ i965_QueryConfigEntrypoints(VADriverContextP ctx,
         entrypoint_list[0] = VAEntrypointVLD;
         break;
 
+#ifndef DISABLE_H264_DECODER
     case VAProfileH264Baseline:
     case VAProfileH264Main:
     case VAProfileH264High:
         *num_entrypoints = 1;
         entrypoint_list[0] = VAEntrypointVLD;
         break;
+#endif
 
     default:
         vaStatus = VA_STATUS_ERROR_UNSUPPORTED_PROFILE;
@@ -249,6 +253,7 @@ i965_CreateConfig(VADriverContextP ctx,
         }
         break;
 
+#ifndef DISABLE_H264_DECODER
     case VAProfileH264Baseline:
     case VAProfileH264Main:
     case VAProfileH264High:
@@ -257,8 +262,8 @@ i965_CreateConfig(VADriverContextP ctx,
         } else {
             vaStatus = VA_STATUS_ERROR_UNSUPPORTED_ENTRYPOINT;
         }
-
         break;
+#endif
 
     default:
         vaStatus = VA_STATUS_ERROR_UNSUPPORTED_PROFILE;
@@ -720,11 +725,13 @@ i965_CreateContext(VADriverContextP ctx,
     }
 
     switch (obj_config->profile) {
+#ifndef DISABLE_H264_DECODER
     case VAProfileH264Baseline:
     case VAProfileH264Main:
     case VAProfileH264High:
         render_state->interleaved_uv = 1;
         break;
+#endif
     default:
         render_state->interleaved_uv = 0;
     }
@@ -967,11 +974,13 @@ i965_BeginPicture(VADriverContextP ctx,
         vaStatus = VA_STATUS_SUCCESS;
         break;
 
+#ifndef DISABLE_H264_DECODER
     case VAProfileH264Baseline:
     case VAProfileH264Main:
     case VAProfileH264High:
         vaStatus = VA_STATUS_SUCCESS;
         break;
+#endif
 
     default:
         assert(0);
diff --git a/i965_drv_video/intel_batchbuffer.c b/i965_drv_video/intel_batchbuffer.c
index abe548e..c592b52 100644
--- a/i965_drv_video/intel_batchbuffer.c
+++ b/i965_drv_video/intel_batchbuffer.c
@@ -33,6 +33,17 @@
 
 #include "intel_batchbuffer.h"
 
+#ifdef DISABLE_H264_DECODER
+/* XXX: also add a run-time check for exec2 support */
+#define drm_intel_bo_mrb_exec local_drm_intel_bo_mrb_exec
+static int drm_intel_bo_mrb_exec(drm_intel_bo *bo, int used,
+                                 drm_clip_rect_t * cliprects, int num_cliprects,
+                                 int DR4, int ring_flag)
+{
+    return drm_intel_bo_exec(bo, used, cliprects, num_cliprects, DR4);
+}
+#endif
+
 static void 
 intel_batchbuffer_reset(struct intel_batchbuffer *batch)
 {
diff --git a/i965_drv_video/intel_driver.h b/i965_drv_video/intel_driver.h
index ffa8cad..1dccaf8 100644
--- a/i965_drv_video/intel_driver.h
+++ b/i965_drv_video/intel_driver.h
@@ -17,6 +17,23 @@
 #define INLINE
 #endif
 
+#ifndef I915_EXEC_RENDER
+#ifdef  ON_RENDER_RING
+#define I915_EXEC_RENDER  ON_RENDER_RING
+#else
+#define I915_EXEC_RENDER  (1 << 0)
+#define DISABLE_H264_DECODER
+#endif
+#endif
+#ifndef I915_EXEC_BSD
+#ifdef  ON_BSD_RING
+#define I915_EXEC_BSD     ON_BSD_RING
+#else
+#define I915_EXEC_BSD     (1 << 1)
+#define DISABLE_H264_DECODER
+#endif
+#endif
+
 #define BATCH_SIZE      0x10000
 #define BATCH_RESERVED  0x10
 
