Index: eggdbus-0.6/configure.ac
===================================================================
--- eggdbus-0.6.orig/configure.ac
+++ eggdbus-0.6/configure.ac
@@ -32,8 +32,6 @@ AC_PATH_PROG([XSLTPROC], [xsltproc])
 fi
 AM_CONDITIONAL(MAN_PAGES_ENABLED, test x$enable_man_pages = xyes)
 
-GTK_DOC_CHECK([1.3])
-
 #### gcc warning flags
 
 if test "x$GCC" = "xyes"; then
Index: eggdbus-0.6/docs/eggdbus/Makefile.am
===================================================================
--- eggdbus-0.6.orig/docs/eggdbus/Makefile.am
+++ eggdbus-0.6/docs/eggdbus/Makefile.am
@@ -87,13 +87,7 @@ HTML_IMAGES = 				\
 # Extra options to supply to gtkdoc-fixref
 FIXXREF_OPTIONS=
 
-if ENABLE_GTK_DOC
-include $(top_srcdir)/gtk-doc.make
-else
-CLEANFILES =
-endif
-
-CLEANFILES +=   *~				\
+CLEANFILES =   *~				\
 		eggdbus-scan.*			\
 		eggdbus.*			\
 		*.bak				\
Index: eggdbus-0.6/docs/tests/Makefile.am
===================================================================
--- eggdbus-0.6.orig/docs/tests/Makefile.am
+++ eggdbus-0.6/docs/tests/Makefile.am
@@ -65,13 +65,7 @@ HTML_IMAGES = 				\
 # Extra options to supply to gtkdoc-fixref
 FIXXREF_OPTIONS=
 
-if ENABLE_GTK_DOC
-include $(top_srcdir)/gtk-doc.make
-else
-CLEANFILES =
-endif
-
-CLEANFILES +=   *~				\
+CLEANFILES =   *~				\
 		tests-scan.*			\
 		tests.*				\
 		*.bak				\
