Index: MultiSrc.c
===================================================================
--- MultiSrc.c.orig
+++ MultiSrc.c
@@ -77,6 +77,7 @@ in this Software without prior written a
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <wchar.h>
 
 /****************************************************************
  *
Index: Panner.c
===================================================================
--- Panner.c.orig
+++ Panner.c
@@ -36,7 +36,7 @@ in this Software without prior written a
 #include <X11/Xmu/Misc.h>		/* for Min */
 #include <X11/Xmu/Drawing.h>
 #include <ctype.h>			/* for isascii() etc. */
-#include <math.h>			/* for atof() */
+#include <stdlib.h>			/* for atof() */
 
 extern Bool XmuDistinguishablePixels(); /* not defined in any Xmu headers */
 
Index: Text.c
===================================================================
--- Text.c.orig
+++ Text.c
@@ -75,6 +75,7 @@ SOFTWARE.
 #include <X11/Xfuncs.h>
 #include <ctype.h>		/* for isprint() */
 
+#include <wchar.h>
 #include "XawAlloc.h"
 
 #ifndef MAX_LEN_CT
Index: TextAction.c
===================================================================
--- TextAction.c.orig
+++ TextAction.c
@@ -45,7 +45,9 @@ in this Software without prior written a
 #include "XawI18n.h"
 #endif
 #include <stdio.h>
+#include <stdlib.h>
 #include <ctype.h>
+#include <wchar.h>
 
 #define SrcScan                XawTextSourceScan
 #define FindDist               XawTextSinkFindDistance
Index: TextPop.c
===================================================================
--- TextPop.c.orig
+++ TextPop.c
@@ -65,6 +65,7 @@ in this Software without prior written a
 #include <stdio.h>
 #include <X11/Xos.h>		/* for O_RDONLY */
 #include <errno.h>
+#include <wchar.h>
 
 #include "XawAlloc.h"
 
Index: TextSrc.c
===================================================================
--- TextSrc.c.orig
+++ TextSrc.c
@@ -49,6 +49,7 @@ in this Software without prior written a
 #endif
 #include <stdio.h>
 #include <ctype.h>
+#include <wchar.h>
 
 /****************************************************************
  *
Index: ThreeD.c
===================================================================
--- ThreeD.c.orig
+++ ThreeD.c
@@ -39,6 +39,7 @@ SOFTWARE.
 #include <X11/Xaw3d/XawInit.h>
 #include <X11/Xaw3d/ThreeDP.h>
 #include <X11/Xosdefs.h>
+#include <X11/Xmu/CharSet.h>
 
 /* Initialization of defaults */
 
Index: Tip.c
===================================================================
--- Tip.c.orig
+++ Tip.c
@@ -41,6 +41,7 @@
 #include <X11/Xmu/Converters.h>
 #include <X11/Xaw3d/TipP.h>
 #include <X11/Xaw3d/XawInit.h>
+#include <stdlib.h>
 
 #define	TIP_EVENT_MASK (ButtonPressMask	  |	\
 			ButtonReleaseMask |	\
Index: XawI18n.c
===================================================================
--- XawI18n.c.orig
+++ XawI18n.c
@@ -55,6 +55,7 @@ in this Software without prior written a
 
 */
 
+#include <stdlib.h>
 #include <X11/IntrinsicP.h>
 #include "XawI18n.h"
 
Index: XawIm.c
===================================================================
--- XawIm.c.orig
+++ XawIm.c
@@ -59,6 +59,7 @@ in this Software without prior written a
 #include <X11/Xos.h>
 #include <X11/Xfuncs.h>
 #include <X11/ShellP.h>
+#include <X11/ResourceI.h>
 #include <X11/Xaw3d/TextP.h>
 #include <X11/Xaw3d/MultiSrc.h>
 #include <X11/Xaw3d/MultiSinkP.h>
Index: XawI18n.h
===================================================================
--- XawI18n.h.orig
+++ XawI18n.h
@@ -72,7 +72,4 @@ extern wchar_t _Xaw_atowc (
 
 #ifndef HAS_ISW_FUNCS
 #include <ctype.h>
-#ifndef iswspace
-#define iswspace(c) (isascii(c) && isspace(toascii(c)))
-#endif
 #endif
