---
 stage2/builtins.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/stage2/builtins.c
+++ b/stage2/builtins.c
@@ -4259,7 +4259,7 @@ terminal_func (char *arg, int flags)
 
   /* If multiple terminals are specified, wait until the user pushes any
      key on one of the terminals.  */
-  if (term_bitmap & ~(1 << default_term))
+  if ((term_bitmap & ~(1 << default_term)) && !(saved_entryno & 0x4000))
     {
       int time1, time2 = -1;
 
