--- jpeg/configure
+++ jpeg/configure
@@ -1529,7 +1529,8 @@
 
 if test "x$LTSHARED" != xno  -o  "x$LTSTATIC" != xno; then
   USELIBTOOL="yes"
-  LIBTOOL="./libtool"
+  LIBTOOL="./libtool-wrap"
+  chmod a+rx libtool-wrap
   O="lo"
   A="la"
   LN='$(LIBTOOL) --mode=link $(CC)'
@@ -1559,7 +1559,7 @@
   if test "x$LTSTATIC" = xno; then
     disable_static="--disable-static"
   fi
-  $srcdir/ltconfig $disable_shared $disable_static $srcdir/ltmain.sh
+  $srcdir/ltconfig $disable_shared $disable_static $srcdir/ltmain.sh $CHOST
 fi
 
 # Select memory manager depending on user input.
--- jpeg/libtool-wrap
+++ jpeg/libtool-wrap
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+export PATH=${PATH}:/usr/bin
+
+doit=
+for x in glibtool libtool ; do
+	if type -p ${x} > /dev/null ; then
+		doit=${x}
+		break
+	fi
+done
+
+if [ -z "${doit}" ] ; then
+	echo "Unable to locate libtool :("
+	exit 1
+fi
+
+exec ${doit} "$@"
--- jpeg/ltconfig
+++ jpeg/ltconfig
@@ -553,7 +553,9 @@
     # On HP-UX, both CC and GCC only warn that PIC is supported... then they
     # create non-PIC objects.  So, if there were any warnings, we assume that
     # PIC is not supported.
+    # Make sure we only test warnings on HP-UX (pic_flag == +Z) or we can
+    # easily break Linux builds http://bugs.gentoo.org/70947
-    if test -s conftest.err; then
+    if test -s conftest.err -a "$pic_flag" = "+Z"; then
       echo "$ac_t"no 1>&6
       can_build_shared=no
       pic_flag=
@@ -1210,7 +1210,6 @@
   else
     # Only the GNU ld.so supports shared libraries on MkLinux.
     case "$host_cpu" in
-    powerpc*) dynamic_linker=no ;;
     *) dynamic_linker='Linux ld.so' ;;
     esac
   fi
