diff -up linux-2.6.31.noarch/drivers/net/wireless/prism54/islpci_hotplug.c.orig linux-2.6.31.noarch/drivers/net/wireless/prism54/islpci_hotplug.c
--- linux-2.6.31.noarch/drivers/net/wireless/prism54/islpci_hotplug.c.orig	2009-11-02 13:53:48.000000000 -0500
+++ linux-2.6.31.noarch/drivers/net/wireless/prism54/islpci_hotplug.c	2009-11-02 13:53:44.000000000 -0500
@@ -72,8 +72,10 @@ static const struct pci_device_id prism5
 	{0,0,0,0,0,0,0}
 };
 
+#if 0 /* Favor p54pci instead... */
 /* register the device with the Hotplug facilities of the kernel */
 MODULE_DEVICE_TABLE(pci, prism54_id_tbl);
+#endif
 
 static int prism54_probe(struct pci_dev *, const struct pci_device_id *);
 static void prism54_remove(struct pci_dev *);
