diff -Nur psyco-1.6-old/test/life-psyco.py psyco-1.6/test/life-psyco.py
--- psyco-1.6-old/test/life-psyco.py	2008-05-30 15:02:32.000000000 +0300
+++ psyco-1.6/test/life-psyco.py	2008-05-30 15:02:43.000000000 +0300
@@ -127,11 +127,11 @@
 
     def makeRandom(self):
 	"Fill the board with a random pattern"
-	import whrandom
+	import random
 	self.state={}
 	for i in range(0, self.X): 
             for j in range(0, self.Y):
-		if whrandom.random()*10>5.0: self.set(j,i)
+		if random.random()*10>5.0: self.set(j,i)
 
 
 import psyco
diff -Nur psyco-1.6-old/test/life.py psyco-1.6/test/life.py
--- psyco-1.6-old/test/life.py	2008-05-30 15:02:32.000000000 +0300
+++ psyco-1.6/test/life.py	2008-05-30 15:02:43.000000000 +0300
@@ -121,11 +121,11 @@
 
     def makeRandom(self):
 	"Fill the board with a random pattern"
-	import whrandom
+	import random
 	self.state={}
 	for i in range(0, self.X): 
             for j in range(0, self.Y):
-		if whrandom.random()*10>5.0: self.set(j,i)
+		if random.random()*10>5.0: self.set(j,i)
 
 
 def erase_menu(stdscr, menu_y):
