Index: kvm-kmod-2.6.35.6/configure
===================================================================
--- kvm-kmod-2.6.35.6.orig/configure
+++ kvm-kmod-2.6.35.6/configure
@@ -89,7 +89,7 @@ arch=${arch%%-*}
 kerneldir=$(cd $kerneldir; pwd)
 
 # see if we have split build and source directories
-if [ ! -d "$kerneldir/Kbuild" ]; then
+if [ ! -f "$kerneldir/Kbuild" ]; then
     kernelsourcedir=$kerneldir/source
     if [ ! -L "$kernelsourcedir" ]; then
         kernelsourcedir=${kerneldir%/build*}/source
