diff -up ghostscript-8.71/base/gdevtifs.h.tiff-default-strip-size ghostscript-8.71/base/gdevtifs.h
--- ghostscript-8.71/base/gdevtifs.h.tiff-default-strip-size	2010-02-04 17:47:57.000000000 +0000
+++ ghostscript-8.71/base/gdevtifs.h	2010-03-16 09:58:29.269600052 +0000
@@ -50,7 +50,7 @@ int tiff_print_page(gx_device_printer *d
  * Sets the compression tag for TIFF and updates the rows_per_strip tag to
  * reflect max_strip_size under the new compression scheme.
  */
-#define TIFF_DEFAULT_STRIP_SIZE 8192
+#define TIFF_DEFAULT_STRIP_SIZE 1048576
 
 int tiff_set_compression(gx_device_printer *pdev,
 			 TIFF *tif,
