diff -Nuar gpicview-0.2.1.orig/gpicview.desktop.in gpicview-0.2.1/gpicview.desktop.in
--- gpicview-0.2.1.orig/gpicview.desktop.in	2011-06-16 08:40:21.729000115 +0300
+++ gpicview-0.2.1/gpicview.desktop.in	2011-06-16 08:50:30.069000116 +0300
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Encoding=UTF-8
-Categories=Application;Graphics;Utility;Core;GTK;Viewer;RasterGraphics;2DGraphics;Photography;
+Categories=Application;Graphics;Core;GTK;Viewer;RasterGraphics;2DGraphics;Photography;
 Exec=gpicview %f
 Icon=gpicview
 StartupNotify=true
@@ -13,6 +13,7 @@
 Name[fr]=Visionneur d'images
 Name[ja]=画像ビューワ
 Name[pl]=Przeglądarka obrazów
+Name[tr]=Resim Görüntüleyici
 Name[zh_CN]=图片查看器
 Name[zh_TW]=圖片檢視器
 GenericName=Image Viewer
@@ -21,6 +22,7 @@
 GenericName[fr]=Visionneur d'images
 GenericName[ja]=画像ビューワ
 GenericName[pl]=Przeglądarka obrazów
+GenericName[tr]=Resim Görüntüleyici
 GenericName[zh_CN]=图片查看器
 GenericName[zh_TW]=圖片檢視器
 Comment=View your images easily
@@ -29,5 +31,6 @@
 Comment[fr]=Visionner des images facilement
 Comment[ja]=簡単に画像を見ることができます
 Comment[pl]=Wyświetla obrazy
+Comment[tr]=Görsellerinizi kolayca görüntüleyin
 Comment[zh_CN]=轻松查看数码相片和图片
-Comment[zh_TW]=輕鬆檢視數位相片和圖片
\ Dosya sonunda yenisatır yok.
+Comment[zh_TW]=輕鬆檢視數位相片和圖片
