Index: src/config.c
===================================================================
--- src/config.c	(revision 36793)
+++ src/config.c	(working copy)
@@ -38,6 +38,9 @@
 //! D-Bus socket file
 char *config_server_address = DBUS_SERVER_ADDRESS;
 
+//! D-Bus unique address
+char *config_unique_address;
+
 //! D-Bus service name
 char *config_service_name = DBUS_SERVICE_NAME;
 
Index: src/loop.c
===================================================================
--- src/loop.c	(revision 36793)
+++ src/loop.c	(working copy)
@@ -196,6 +196,10 @@
 
     PyObject *py_args = pydbus_import(bus_msg);
 
+    if (!dbus_message_has_destination(bus_msg, config_unique_address)) {
+        return DBUS_HANDLER_RESULT_HANDLED;
+    }
+
     switch (dbus_message_get_type(bus_msg)) {
         case DBUS_MESSAGE_TYPE_METHOD_CALL:
             if (my_proc.nr_children > MAX_PROC) {
@@ -320,6 +324,7 @@
         return -1;
     }
 
+    config_unique_address = dbus_bus_get_unique_name(bus_conn);
     log_debug("Connected to D-Bus, unique id is %s\n", dbus_bus_get_unique_name(bus_conn));
 
     // Request a name (example: com.server.test)
Index: include/config.h
===================================================================
--- include/config.h	(revision 36793)
+++ include/config.h	(working copy)
@@ -64,6 +64,7 @@
 #endif
 
 extern char *config_server_address;
+extern char *config_unique_address;
 extern char *config_service_name;
 extern char *config_interface;
 extern char *config_dir_data;
