Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt      (revision 31368)
+++ CMakeLists.txt      (working copy)
@@ -4,6 +4,9 @@
 # CMake 2.6 required
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
 
+# Python is required
+FIND_PACKAGE(PythonLibs)
+
 # Application name
 SET (APP_NAME "comar")
 SET (APP_SUFFIX "3")
@@ -44,12 +47,12 @@
              src/utils.c)
 
 # Set standard libraries
-SET (LIBS python2.6
+SET (LIBS ${PYTHON_LIBRARIES}
           dbus-1)
 
 # Include standard headers
 INCLUDE_DIRECTORIES (include/
-                     /usr/include/python2.6
+                     ${PYTHON_INCLUDE_DIRS}
                      /usr/include/dbus-1.0
                      /usr/lib/dbus-1.0/include)
 

