--- docs/gftp.desktop.orig	2008-08-07 18:03:14.000000000 +0300
+++ docs/gftp.desktop	2008-08-07 18:03:21.000000000 +0300
@@ -3,6 +3,7 @@
 Name=gFTP
 Comment=Download and upload files using multiple file transfer protocols
 Comment[fr]=Télécharge des fichiers en utilisant le protocole FTP
+Comment[tr]=Dosyaları çoklu dosya aktarım protokolü ile indirin ve yükleyin
 GenericName=gFTP
 Exec=gftp %u
 Terminal=false
