diff -Naur src.orig/bspline_test.cpp src/bspline_test.cpp
--- src.orig/bspline_test.cpp	2008-04-29 13:40:52.000000000 +0300
+++ src/bspline_test.cpp	2008-04-29 13:52:01.000000000 +0300
@@ -1,6 +1,8 @@
 // some test code for the 2d bsplines!
 #include "bspline.h"
 #include <fstream>
+#include <cstdlib>
+
 using namespace std;
 
 double rnd() { return double(rand())/RAND_MAX; }
diff -Naur src.orig/convoy.h src/convoy.h
--- src.orig/convoy.h	2008-04-29 13:40:52.000000000 +0300
+++ src/convoy.h	2008-04-29 13:42:48.000000000 +0300
@@ -27,6 +27,8 @@
 #include "vector2.h"
 #include <new>
 #include <list>
+#include <memory>
+
 class ship;
 
 ///\brief Grouping of ships and other objects with central control.
diff -Naur src.orig/date.cpp src/date.cpp
--- src.orig/date.cpp	2008-04-29 13:40:52.000000000 +0300
+++ src/date.cpp	2008-04-29 13:44:19.000000000 +0300
@@ -21,6 +21,8 @@
 // subsim (C)+(W) Markus Petermann and Thorsten Jordan. SEE LICENSE
 
 #include <iomanip>
+#include <cstdlib>
+
 #include "date.h"
 #include "texts.h"
 
diff -Naur src.orig/ocean_wave_generator.h src/ocean_wave_generator.h
--- src.orig/ocean_wave_generator.h	2008-04-29 13:40:52.000000000 +0300
+++ src/ocean_wave_generator.h	2008-04-29 13:50:16.000000000 +0300
@@ -29,6 +29,7 @@
 #include "environment.h"
 #include <complex>
 #include <vector>
+#include <cstdlib>
 
 // use float fftw (faster) or double (default) ?
 #ifdef WITH_FLOAT_FFTW
diff -Naur src.orig/ptrlist.h src/ptrlist.h
--- src.orig/ptrlist.h	2008-04-29 13:40:52.000000000 +0300
+++ src/ptrlist.h	2008-04-29 13:41:29.000000000 +0300
@@ -25,6 +25,7 @@
 
 #include <list>
 #include <stdexcept>
+#include <memory>
 
 // same as std::list regarding the interface (partly), but handles pointers.
 template <class T>
diff -Naur src.orig/ptrvector.h src/ptrvector.h
--- src.orig/ptrvector.h	2008-04-29 13:40:52.000000000 +0300
+++ src/ptrvector.h	2008-04-29 13:41:46.000000000 +0300
@@ -25,6 +25,7 @@
 
 #include <vector>
 #include <stdexcept>
+#include <memory>
 
 // same as std::vector regarding the interface, but handles pointers.
 template <class T>
diff -Naur src.orig/widget.cpp src/widget.cpp
--- src.orig/widget.cpp	2008-04-29 13:40:52.000000000 +0300
+++ src/widget.cpp	2008-04-29 13:48:54.000000000 +0300
@@ -32,6 +32,8 @@
 #include "datadirs.h"
 #include <set>
 #include <sstream>
+#include <algorithm>
+
 using std::vector;
 using std::list;
 using std::string;
