diff -ur sysvinit-2.86.orig/src/init.c sysvinit-2.86/src/init.c
--- sysvinit-2.86.orig/src/init.c	2006-06-08 03:07:42.000000000 +0300
+++ sysvinit-2.86/src/init.c	2006-06-08 03:09:23.000000000 +0300
@@ -2222,7 +2222,7 @@
 			break;
 	}
 	if (loglevel > 0) {
-		initlog(L_VB, "Entering runlevel: %c", runlevel);
+		initlog(L_VB, "%c", runlevel);
 		write_utmp_wtmp("runlevel", "~~", runlevel + 256 * oldlevel, RUN_LVL, "~");
 		thislevel = runlevel;
 		prevlevel = oldlevel;
