diff -Nur htop-0.8.3/RichString.c htop-0.8.3.new/RichString.c
--- htop-0.8.3/RichString.c	2009-06-02 21:18:19.000000000 +0200
+++ htop-0.8.3.new/RichString.c	2009-07-07 16:15:34.831453590 +0200
@@ -12,11 +12,7 @@
 
 #include "debug.h"
 #include <assert.h>
-#ifdef HAVE_LIBNCURSESW
-#include <ncursesw/curses.h>
-#else
 #include <curses.h>
-#endif
 
 #define RICHSTRING_MAXLEN 300
 
diff -Nur htop-0.8.3/RichString.h htop-0.8.3.new/RichString.h
--- htop-0.8.3/RichString.h	2009-06-02 21:18:34.000000000 +0200
+++ htop-0.8.3.new/RichString.h	2009-07-07 16:15:39.445452438 +0200
@@ -15,11 +15,7 @@
 
 #include "debug.h"
 #include <assert.h>
-#ifdef HAVE_LIBNCURSESW
-#include <ncursesw/curses.h>
-#else
 #include <curses.h>
-#endif
 
 #define RICHSTRING_MAXLEN 300
 
