Index: pisi-2.4_beta2/pisi.xml.in
===================================================================
--- /dev/null
+++ pisi-2.4_beta2/pisi.xml.in
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
+  <mime-type type="application/x-pisi">
+    <_comment>PiSi package</_comment>
+    <glob pattern="*.pisi"/>
+  </mime-type>
+</mime-info>
