Index: pisi-2.2.14/pisi/actionsapi/kde.py
===================================================================
--- pisi-2.2.14.orig/pisi/actionsapi/kde.py
+++ pisi-2.2.14/pisi/actionsapi/kde.py
@@ -55,6 +55,7 @@ def configure(parameters = ''):
                 --enable-mitshm \
                 --with-xinerama \
                 --with-qt-dir=%s \
+                --without-arts \
                 --enable-mt \
                 --with-qt-libraries=%s/lib \
                 --disable-dependency-tracking \
