Index: index.py
===================================================================
--- pisi/index.py	(revision 36495)
+++ pisi/index.py	(revision 36496)
@@ -112,8 +112,8 @@
         for pkg in util.filter_latest_packages(packages):
             pkg_name = util.parse_package_name(os.path.basename(pkg))[0]
             if pkg_name.endswith(ctx.const.debug_name_suffix):
-                pkg_name = util.remove_suffix(pkg_name,
-                                              ctx.const.debug_name_suffix)
+                pkg_name = util.remove_suffix(ctx.const.debug_name_suffix,
+                                              pkg_name)
             if pkg_name not in obsoletes_list:
                 ctx.ui.info(_('Adding %s to package index') % pkg)
                 self.add_package(pkg, deltas, repo_uri)
