diff -Nur pisi-2.4_alpha1-old//pisi/actionsapi/texlivemodules.py pisi-2.4_alpha1/pisi/actionsapi/texlivemodules.py
--- pisi-2.4_alpha1-old//pisi/actionsapi/texlivemodules.py	2010-10-13 11:36:00.046000435 +0300
+++ pisi-2.4_alpha1/pisi/actionsapi/texlivemodules.py	2010-10-13 11:36:06.666000435 +0300
@@ -84,7 +84,6 @@
     if handleConfigFiles():
         raise Installing, _('Handle config files failed')
 
-
 def createSymlinksFormat2Engines():
     '''Create symlinks from format to engines'''
     for formatfile in ls("%s/texmf/fmtutil/format*.cnf" % get.curDIR()):
@@ -157,7 +156,6 @@
                     domove("/usr/share/texmf/%s/%s" % (dirname,configFile), "/etc/texmf/%s.d" % dirname)
                     dosym("/etc/texmf/%s.d/%s" % (dirname, configFile), "/usr/share/texmf/%s/%s" %(dirname, configFile))
 
-
 def addFormat(parameters):
     '''Add format files'''
     if not os.path.isdir("%s/texmf/fmtutil/" % get.curDIR()):
@@ -178,12 +176,13 @@
             para_dict[pair[0]] = pair[1]
             if pair[0] == "patterns" and pair[1] == '':
                 para_dict["patterns"] = '-'     # Specified in the texlive-module.eclass
+            elif not pair[0] == 'patterns':
+                para_dict["patterns"] = '-'
 
     cnf_file = open('%s/texmf/fmtutil/format.%s.cnf' % (get.curDIR(),get.srcNAME()), 'a')
     cnf_file.write("%s\t%s\t%s\t%s\n" % (para_dict["name"], para_dict["engine"], para_dict["patterns"], para_dict["options"]))
     cnf_file.close()
 
-
 def moveSources():
     reloc = "texmf-dist"
 
@@ -199,7 +198,6 @@
                     os.system("mkdir -p %s/%s" % (reloc,dirname))
                 shutil.move("%s" % path , "%s/%s" % (reloc,dirname))
 
-
 def buildFormatFiles():
     '''Build format files'''
     if os.path.isdir("%s/texmf/fmtutil/" % get.curDIR()):
@@ -228,7 +226,6 @@
         language_dat.write("=%s\n" % para_dict["synonyms"])
         language_dat.close()
 
-
 def addLanguageDef(parameter):
     '''Create language.*.def files'''
     parameter = parameter.split()
@@ -252,8 +249,6 @@
         language_def.write("\\addlanguage{%s}{%s}{}{%s}{%s}\n" % (para_dict["synonyms"], para_dict["file"],  para_dict["lefthyphenmin"],  para_dict["righthyphenmin"]))
         language_def.close()
 
-
-
 def generateConfigFiles():
     '''Generate config files'''
     for tlpobjfile in ls("%s/tlpkg/tlpobj/*" % get.curDIR()):
