--- kdebase-workspace/plasma/generic/applets/digital-clock/clock.cpp	2009-03-17 23:20:04.000000000 +0000
+++ kdebase-workspace/plasma/generic/applets/digital-clock/clock.cpp	2009-03-17 23:30:34.000000000 +0000
@@ -79,7 +79,7 @@
 
     kDebug() << "showTimezone:" << m_showTimezone;
 
-    m_showDate = cg.readEntry("showDate", false);
+    m_showDate = cg.readEntry("showDate", true);
     m_showYear = cg.readEntry("showYear", false);
     m_showDay = cg.readEntry("showDay", false);
 
