Index: kdebase-workspace-4.6.1/powerdevil/daemon/backends/upower/backlighthelper.cpp
===================================================================
--- kdebase-workspace-4.6.1.orig/powerdevil/daemon/backends/upower/backlighthelper.cpp
+++ kdebase-workspace-4.6.1/powerdevil/daemon/backends/upower/backlighthelper.cpp
@@ -36,7 +36,7 @@ void BacklightHelper::init()
     QStringList interfaces;
     interfaces << "nv_backlight" << "asus_laptop" << "toshiba"
                << "eeepc" << "thinkpad_screen" << "acpi_video0"
-               << "mbp_backlight" << "acpi_video1"
+               << "mbp_backlight" << "acpi_video1" << "apple_backlight"
                << "fujitsu-laptop" << "samsung"
                << "nvidia_backlight" << "dell_backlight" << "sony";
 
