Index: plasma/desktop/shell/desktopcorona.cpp
===================================================================
--- plasma/desktop/shell/desktopcorona.cpp.orig
+++ plasma/desktop/shell/desktopcorona.cpp
@@ -436,6 +436,8 @@ void DesktopCorona::loadDefaultLayout()
         applet->setGlobalShortcut(KShortcut("Alt+F1"));
     }
 
+    loadDefaultApplet("showdesktop", panel);
+    loadDefaultApplet("notifier", panel);
     loadDefaultApplet("pager", panel);
     loadDefaultApplet("tasks", panel);
     Plasma::Applet *sysTray = loadDefaultApplet("systemtray", panel);
@@ -447,7 +448,9 @@ void DesktopCorona::loadDefaultLayout()
         }
     }
 
+    loadDefaultApplet("nm-applet", panel);
     loadDefaultApplet("digital-clock", panel);
+    loadDefaultApplet("lockout", panel);
     emit containmentAdded(panel);
 
     QTimer::singleShot(1000, this, SLOT(saveDefaultSetup()));
