Index: kdebase-workspace-4.2.4/systemsettings/systemsettings.desktop
===================================================================
--- kdebase-workspace-4.2.4.orig/systemsettings/app/systemsettings.desktop
+++ kdebase-workspace-4.2.4/systemsettings/app/systemsettings.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=systemsettings -caption "%c" %i
+Exec=systemsettings %i
 Icon=preferences-system
 Type=Application
 X-DocPath=systemsettings/index.html
