Index: kdebase-workspace-4.6.2/systemsettings/categories/settings-account-details.desktop
===================================================================
--- kdebase-workspace-4.6.2.orig/systemsettings/categories/settings-account-details.desktop
+++ kdebase-workspace-4.6.2/systemsettings/categories/settings-account-details.desktop
@@ -59,7 +59,7 @@ Name[sr@latin]=Detalji naloga
 Name[sv]=Kontoinformation
 Name[tg]=Тафсилоти ҳисоб
 Name[th]=รายละเอียดบัญชีผู้ใช้
-Name[tr]=Hesap Detayları>
+Name[tr]=Hesap Detayları
 Name[uk]=Параметри облікового запису
 Name[wa]=Detays do conte
 Name[x-test]=xxAccount Detailsxx
