Index: kdebase-workspace-4.3.1/plasma/applets/kickoff/core/favoritesmodel.cpp
===================================================================
--- kdebase-workspace-4.3.1.orig/plasma/desktop/applets/kickoff/core/favoritesmodel.cpp
+++ kdebase-workspace-4.3.1/plasma/desktop/applets/kickoff/core/favoritesmodel.cpp
@@ -105,12 +105,12 @@ public:
     static QList<QString> defaultFavorites()
     {
         QList<QString> applications;
-        applications << "konqbrowser" << "kmail" << "systemsettings" << "dolphin";
+        applications << "mozillafirefox" << "Kontact" << "kopete" << "systemsettings" << "dolphin" << "amarok";
 
         QList<QString> desktopFiles;
 
         foreach (const QString& application, applications) {
-            KService::Ptr service = KService::serviceByStorageId("kde4-" + application + ".desktop");
+            KService::Ptr service = KService::serviceByStorageId(application + ".desktop");
             if (service) {
                 desktopFiles << service->entryPath();
             }
