Index: kdm/config.def
===================================================================
--- kdm/config.def.orig
+++ kdm/config.def
@@ -1243,10 +1243,10 @@ Type: enum
  None/BO_NONE: no boot manager
  Grub/BO_GRUB: Grub boot manager
  Lilo/BO_LILO: Lilo boot manager (Linux on i386 &amp; x86-64 only)
-Default: None
+Default: Grub
 User: core
 User: greeter
-Instance: #Grub
+Instance: Grub
 Merge: kdm:UseLilo(P_UseLilo)
 Comment: &
 Description:
@@ -1332,7 +1332,7 @@ Description:
 
 Key: ServerTimeout
 Type: int
-Default: 15
+Default: 60
 User: core
 Instance: #:*/
 Comment:
@@ -1448,7 +1448,7 @@ Description:
 
 Key: TerminateServer
 Type: bool
-Default: false
+Default: true
 User: core
 Instance: #:*/!
 Merge: xdm
@@ -1549,7 +1549,7 @@ Description:
 Key: Resources
 # XXX strictly speaking this is supposed to be a string list, i think.
 Type: string
-Default: ""
+Default: "/etc/X11/xdm/Xresources"
 User: core
 Instance: #*/""
 Update: cp_resources
@@ -2158,7 +2158,7 @@ Description:
 # This needs to come _in front_ of the font settings to be effective!
 Key: AntiAliasing
 Type: bool
-Default: false
+Default: true
 User: greeter
 Instance: #*/!
 Comment: &
@@ -2167,30 +2167,30 @@ Description:
 
 Key: GreetFont
 Type: string
-Default: "Serif,20,5,0,50,0"
-CDefault: "Serif,20,bold"
+Default: "DejaVu Sans,20,5,0,50,0"
+CDefault: "DejaVu Sans,20,bold"
 User: greeter:font
-Instance: #*/"Serif,20,5,0,50,0"
+Instance: #*/"DejaVu Sans,20,5,0,50,0"
 Comment: &
 Description:
  The font for the greeter headline.
 
 Key: StdFont
 Type: string
-Default: "Sans Serif,10,5,0,50,0"
-CDefault: "Sans Serif,10"
+Default: "DejaVu Sans,10,5,0,50,0"
+CDefault: "DejaVu Sans,10"
 User: greeter(normalFont):font
-Instance: #*/"Sans Serif,10,5,0,50,0"
+Instance: #*/"DejaVu Sans,10,5,0,50,0"
 Comment: &
 Description:
  The normal font used in the greeter.
 
 Key: FailFont
 Type: string
-Default: "Sans Serif,10,5,0,75,0"
-CDefault: "Sans Serif,10,bold"
+Default: "DejaVu Sans,10,5,0,75,0"
+CDefault: "DejaVu Sans,10,bold"
 User: greeter:font
-Instance: #*/"Sans Serif,10,5,0,75,0"
+Instance: #*/"DejaVu Sans,10,5,0,75,0"
 Comment: &
 Description:
  The font used for the <quote>Login Failed</quote> message.
@@ -2331,9 +2331,9 @@ Type: enum
  PreferAdmin/FACE_PREFER_ADMIN: prefer &lt;<option>FaceDir</option>&gt;, fallback on $<envar>HOME</envar>
  PreferUser/FACE_PREFER_USER: ... and the other way round
  UserOnly/FACE_USER_ONLY: from the user's <filename>$<envar>HOME</envar>/.face[.icon]</filename>
-Default: AdminOnly
+Default: PreferUser
 User: greeter
-Instance: #*/PreferUser
+Instance: */PreferUser
 Comment:
  Specify, where the users' pictures should be taken from.
 Description:
@@ -2659,7 +2659,7 @@ Description:
 
 Key: UseTheme
 Type: bool
-Default: false
+Default: true
 User: greeter
 Instance: */true
 Comment: &
