Index: plasma/desktop/applets/tasks/tasksConfig.ui
===================================================================
--- plasma/desktop/applets/tasks/tasksConfig.ui	(Revision 1166313)
+++ plasma/desktop/applets/tasks/tasksConfig.ui	(Arbeitskopie)
@@ -6,11 +6,14 @@
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>444</width>
-    <height>347</height>
+    <width>462</width>
+    <height>363</height>
    </rect>
   </property>
   <layout class="QFormLayout" name="formLayout">
+   <property name="fieldGrowthPolicy">
+    <enum>QFormLayout::ExpandingFieldsGrow</enum>
+   </property>
    <item row="0" column="0">
     <widget class="QLabel" name="label_5">
      <property name="sizePolicy">
@@ -130,26 +133,7 @@
      </item>
     </layout>
    </item>
-   <item row="5" column="0">
-    <widget class="QLabel" name="label_6">
-     <property name="sizePolicy">
-      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
-       <horstretch>0</horstretch>
-       <verstretch>0</verstretch>
-      </sizepolicy>
-     </property>
-     <property name="font">
-      <font>
-       <weight>75</weight>
-       <bold>true</bold>
-      </font>
-     </property>
-     <property name="text">
-      <string>Grouping and Sorting</string>
-     </property>
-    </widget>
-   </item>
-   <item row="6" column="0">
+   <item row="7" column="0">
     <widget class="QLabel" name="label">
      <property name="text">
       <string>Grouping:</string>
@@ -162,7 +146,7 @@
      </property>
     </widget>
    </item>
-   <item row="6" column="1">
+   <item row="7" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QComboBox" name="groupingStrategy">
@@ -189,7 +173,7 @@
      </item>
     </layout>
    </item>
-   <item row="7" column="0">
+   <item row="8" column="0">
     <widget class="QLabel" name="label_10">
      <property name="text">
       <string>Only when the taskbar is full</string>
@@ -202,7 +186,7 @@
      </property>
     </widget>
    </item>
-   <item row="7" column="1">
+   <item row="8" column="1">
     <widget class="QCheckBox" name="groupWhenFull">
      <property name="enabled">
       <bool>false</bool>
@@ -215,7 +199,7 @@
      </property>
     </widget>
    </item>
-   <item row="8" column="0">
+   <item row="10" column="0">
     <widget class="QLabel" name="label_2">
      <property name="text">
       <string>Sorting:</string>
@@ -228,7 +212,7 @@
      </property>
     </widget>
    </item>
-   <item row="8" column="1">
+   <item row="10" column="1">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QComboBox" name="sortingStrategy">
@@ -255,7 +239,7 @@
      </item>
     </layout>
    </item>
-   <item row="9" column="0">
+   <item row="11" column="0">
     <widget class="QLabel" name="label_4">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
@@ -274,7 +258,7 @@
      </property>
     </widget>
    </item>
-   <item row="10" column="0">
+   <item row="12" column="0">
     <widget class="QLabel" name="label_11">
      <property name="text">
       <string>Only show tasks from the current screen</string>
@@ -287,14 +271,14 @@
      </property>
     </widget>
    </item>
-   <item row="10" column="1">
+   <item row="12" column="1">
     <widget class="QCheckBox" name="showOnlyCurrentScreen">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
-   <item row="11" column="0">
+   <item row="13" column="0">
     <widget class="QLabel" name="label_12">
      <property name="text">
       <string>Only show tasks from the current desktop</string>
@@ -307,14 +291,14 @@
      </property>
     </widget>
    </item>
-   <item row="11" column="1">
+   <item row="13" column="1">
     <widget class="QCheckBox" name="showOnlyCurrentDesktop">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
-   <item row="12" column="0">
+   <item row="14" column="0">
     <widget class="QLabel" name="label_13">
      <property name="text">
       <string>Only show tasks that are minimized</string>
@@ -327,14 +311,14 @@
      </property>
     </widget>
    </item>
-   <item row="12" column="1">
+   <item row="14" column="1">
     <widget class="QCheckBox" name="showOnlyMinimized">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
-   <item row="13" column="1">
+   <item row="15" column="1">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
@@ -347,6 +331,35 @@
      </property>
     </spacer>
    </item>
+   <item row="6" column="0">
+    <widget class="QLabel" name="label_6">
+     <property name="sizePolicy">
+      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
+       <horstretch>0</horstretch>
+       <verstretch>0</verstretch>
+      </sizepolicy>
+     </property>
+     <property name="font">
+      <font>
+       <weight>75</weight>
+       <bold>true</bold>
+      </font>
+     </property>
+     <property name="text">
+      <string>Grouping and Sorting</string>
+     </property>
+    </widget>
+   </item>
+   <item row="5" column="0">
+    <widget class="QLabel" name="showIconLabel">
+     <property name="text">
+      <string>Show Icons only</string>
+     </property>
+    </widget>
+   </item>
+   <item row="5" column="1">
+    <widget class="QCheckBox" name="showIconOnly"/>
+   </item>
   </layout>
  </widget>
  <customwidgets>
Index: plasma/desktop/applets/tasks/tasks.cpp
===================================================================
--- plasma/desktop/applets/tasks/tasks.cpp	(Revision 1166313)
+++ plasma/desktop/applets/tasks/tasks.cpp	(Arbeitskopie)
@@ -132,6 +132,7 @@
     m_groupManager->setSortingStrategy( static_cast<TaskManager::GroupManager::TaskSortingStrategy>(cg.readEntry("sortingStrategy", static_cast<int>(TaskManager::GroupManager::AlphaSorting))));
     m_rootGroupItem->setMaxRows( cg.readEntry("maxRows", 2));
     m_rootGroupItem->setForceRows( cg.readEntry("forceRows", false));
+    m_rootGroupItem->setShowIconOnly( cg.readEntry("showIconOnly", false) );
 
     emit settingsChanged();
 }
@@ -252,9 +253,11 @@
 
     m_ui.showTooltip->setChecked(m_showTooltip);
     m_ui.highlightWindows->setChecked(m_highlightWindows);
+    m_ui.showIconOnly->setChecked(m_rootGroupItem->showIconOnly());
     m_ui.showOnlyCurrentDesktop->setChecked(m_groupManager->showOnlyCurrentDesktop());
     m_ui.showOnlyCurrentScreen->setChecked(m_groupManager->showOnlyCurrentScreen());
     m_ui.showOnlyMinimized->setChecked(m_groupManager->showOnlyMinimized());
+
     m_ui.fillRows->setChecked(m_rootGroupItem->forceRows());
 
     m_ui.groupingStrategy->addItem(i18n("Do Not Group"),QVariant(TaskManager::GroupManager::NoGrouping));
@@ -378,6 +381,13 @@
         cg.writeEntry("showTooltip", m_showTooltip);
         changed = true;
     }
+    
+    if (m_rootGroupItem->showIconOnly()!= (m_ui.showIconOnly->checkState() == Qt::Checked)) {
+        m_rootGroupItem->setShowIconOnly(!m_rootGroupItem->showIconOnly());
+        KConfigGroup cg = config();
+        cg.writeEntry("showIconOnly", m_showTooltip);
+        changed = true;
+    }
 
     if (m_highlightWindows != (m_ui.highlightWindows->checkState() == Qt::Checked)) {
         m_highlightWindows = !m_highlightWindows;
Index: plasma/desktop/applets/tasks/taskitemlayout.h
===================================================================
--- plasma/desktop/applets/tasks/taskitemlayout.h	(Revision 1166313)
+++ plasma/desktop/applets/tasks/taskitemlayout.h	(Arbeitskopie)
@@ -59,6 +59,9 @@
     void setMaximumRows(int);
     /** force the layout to use maximumRows setting and fill rows before columns */
     void setForceRows(bool);
+    /**  */
+    void setShowIconOnly(bool);
+    
 
     /** the size including expanded groups*/
     int size();
@@ -72,6 +75,8 @@
     int maximumRows();
     /** Returns the preferred number of columns calculated on base of AbstractGroupableItem::basicPreferredSize()*/
     int preferredColumns();
+    
+    bool showIconOnly();
 
     /** Populates the actual QGraphicsGridLayout with items*/
     void layoutItems();
@@ -88,6 +93,7 @@
 
 private:
     bool m_hasSpacer;
+    bool m_showIconOnly;
     QGraphicsWidget *m_spacer;
 
     TaskGroupItem *m_groupItem;
Index: plasma/desktop/applets/tasks/taskitemlayout.cpp
===================================================================
--- plasma/desktop/applets/tasks/taskitemlayout.cpp	(Revision 1166313)
+++ plasma/desktop/applets/tasks/taskitemlayout.cpp	(Arbeitskopie)
@@ -38,6 +38,7 @@
 TaskItemLayout::TaskItemLayout(TaskGroupItem *parent, Tasks *applet)
     : QGraphicsGridLayout(0),
       m_hasSpacer(false),
+      m_showIconOnly(false),
       m_spacer(0),
       m_groupItem(parent),
       m_rowSize(1),
@@ -101,6 +102,7 @@
         return;
     }
 
+    item->setShowIconOnly(m_showIconOnly);
     item->show();
     //kDebug() << "end";
 }
@@ -221,6 +223,11 @@
     return maxRows;
 }
 
+bool TaskItemLayout::showIconOnly()
+{
+    return m_showIconOnly;
+}
+
 //returns a reasonable amount of columns
 int TaskItemLayout::preferredColumns()
 {
@@ -442,6 +449,14 @@
     m_forceRows = forceRows;
 }
 
+void TaskItemLayout::setShowIconOnly(bool show)
+{
+    m_showIconOnly = show;
+    foreach (AbstractTaskItem *item, m_groupItem->members()) {
+        item->setShowIconOnly(show);
+    }
+}
+
 int TaskItemLayout::insertionIndexAt(const QPointF &pos)
 {
     int insertIndex = -1;
Index: plasma/desktop/applets/tasks/abstracttaskitem.h
===================================================================
--- plasma/desktop/applets/tasks/abstracttaskitem.h	(Revision 1166313)
+++ plasma/desktop/applets/tasks/abstracttaskitem.h	(Arbeitskopie)
@@ -131,6 +131,9 @@
     QSize basicPreferredSize() const;
     void setPreferredOffscreenSize();
     void setPreferredOnscreenSize();
+    
+    virtual void setShowIconOnly(bool show);
+    bool showIconOnly();
 
     //TODO: to be removed when we have proper animated layouts
     QPointF animationPos() const;
@@ -253,6 +256,7 @@
     QPropertyAnimation *m_backgroundFadeAnim;
     WId m_lastViewId;
 
+    bool m_showIconOnly : 1; // show only the icon in the panel
     bool m_showText : 1;
     bool m_layoutAnimationLock : 1;
     bool m_firstGeometryUpdate : 1;
Index: plasma/desktop/applets/tasks/abstracttaskitem.cpp
===================================================================
--- plasma/desktop/applets/tasks/abstracttaskitem.cpp	(Revision 1166313)
+++ plasma/desktop/applets/tasks/abstracttaskitem.cpp	(Arbeitskopie)
@@ -104,7 +104,7 @@
 QSize AbstractTaskItem::basicPreferredSize() const
 {
     QFontMetrics fm(KGlobalSettings::taskbarFont());
-    QSize mSize = fm.size(0, "M");
+    QSize mSize = !m_showIconOnly ? fm.size(0, "M") : QSize(0,0);
     const int iconsize = KIconLoader::SizeSmall;
 
     //the 4 should be the default spacing between layout items, is there a way to fetch it without hardcoding?
@@ -149,6 +149,16 @@
     setPreferredSize(basicPreferredSize());
 }
 
+void AbstractTaskItem::setShowIconOnly(bool show)
+{
+    m_showIconOnly = show;
+}
+
+bool AbstractTaskItem::showIconOnly()
+{
+    return m_showIconOnly;
+}
+
 AbstractTaskItem::~AbstractTaskItem()
 {
     stopWindowHoverEffect();
@@ -548,7 +558,7 @@
     // check to see if there is enough room!
     QFontMetrics fm(font());
     const int minimumWidth = left + 8 + IconTextSpacing + right;
-    m_showText = (size().width() >= fm.width("M") * 6 + minimumWidth);
+    m_showText = size().width() >= fm.width("M") * 6 + minimumWidth;
 }
 
 void AbstractTaskItem::resizeEvent(QGraphicsSceneResizeEvent *event)
Index: plasma/desktop/applets/tasks/taskgroupitem.h
===================================================================
--- plasma/desktop/applets/tasks/taskgroupitem.h	(Revision 1166313)
+++ plasma/desktop/applets/tasks/taskgroupitem.h	(Arbeitskopie)
@@ -82,6 +82,8 @@
 
     void setForceRows(bool);
     bool forceRows();
+    
+    virtual void setShowIconOnly(bool show);
 
     /*
     *Why the split group works.
Index: plasma/desktop/applets/tasks/taskgroupitem.cpp
===================================================================
--- plasma/desktop/applets/tasks/taskgroupitem.cpp	(Revision 1166313)
+++ plasma/desktop/applets/tasks/taskgroupitem.cpp	(Arbeitskopie)
@@ -866,6 +866,7 @@
         m_tasksLayout->setMaximumRows(m_maximumRows);
         m_tasksLayout->setForceRows(m_forceRows);
         m_tasksLayout->setOrientation(m_applet->formFactor());
+	m_tasksLayout->setShowIconOnly(showIconOnly());
     }
 
     return m_tasksLayout;
@@ -1335,6 +1336,14 @@
     }
 }
 
+void TaskGroupItem::setShowIconOnly(bool show)
+{
+    AbstractTaskItem::setShowIconOnly(show);
+    if (m_tasksLayout) {
+        m_tasksLayout->setShowIconOnly(show);
+    }
+}
+
 int TaskGroupItem::optimumCapacity()
 {
     if (m_tasksLayout) {
