Index: kdebase-workspace-4.5.5/startkde.cmake
===================================================================
--- kdebase-workspace-4.5.5.orig/startkde.cmake
+++ kdebase-workspace-4.5.5/startkde.cmake
@@ -106,7 +106,7 @@ mkdir -m 700 -p $kdehome
 mkdir -m 700 -p $kdehome/share
 mkdir -m 700 -p $kdehome/share/config
 cat >$kdehome/share/config/startupconfigkeys <<EOF
-kcminputrc Mouse cursorTheme 'Oxygen_Black'
+kcminputrc Mouse cursorTheme 'Oxygen_White'
 kcminputrc Mouse cursorSize ''
 ksplashrc KSplash Theme Default
 ksplashrc KSplash Engine KSplashX
