diff -Nur slmodem-2.9.11-20100718-old//drivers/amrmo_init.c slmodem-2.9.11-20100718/drivers/amrmo_init.c
--- slmodem-2.9.11-20100718-old//drivers/amrmo_init.c	2010-12-22 19:34:39.307000037 +0200
+++ slmodem-2.9.11-20100718/drivers/amrmo_init.c	2010-12-22 19:44:33.507000445 +0200
@@ -439,11 +439,11 @@
 
 
 
-static int amrmo_ioctl(struct inode *inode, struct file *file, unsigned int cmd, unsigned long parg)
+static long amrmo_unlocked_ioctl(struct file *file, unsigned int cmd, unsigned long parg)
 {
 	struct amrmo_struct *amrmo = (struct amrmo_struct *)file->private_data;
         unsigned long flags;
-	int ret;
+	long ret;
 	unsigned stat;
 	unsigned long arg=0;
 
@@ -519,7 +519,7 @@
         .read =    amrmo_read,
         .write =   amrmo_write,
         .poll =    amrmo_poll,
-        .ioctl =   amrmo_ioctl,
+        .unlocked_ioctl =   amrmo_unlocked_ioctl,
         .open =    amrmo_open,
         .release = amrmo_release,
 };
