diff -p -up slmodem-2.9.11-20080817/drivers/amrmo_init.c.orig slmodem-2.9.11-20080817/drivers/amrmo_init.c
--- slmodem-2.9.11-20080817/drivers/amrmo_init.c.orig	2009-11-06 15:47:34.000000000 -0500
+++ slmodem-2.9.11-20080817/drivers/amrmo_init.c	2009-11-06 15:48:36.000000000 -0500
@@ -245,7 +245,9 @@ static const char *card_names[] = {
 
 
 static struct pci_device_id amrmo_pci_tbl [] __devinitdata = {
-	{PCI_VENDOR_ID_SILICON, PCI_DEVICE_ID_SL1900,	   /* SiliconInstr SL1543:3052 */
+	{PCI_VENDOR_ID_SILICON, PCI_DEVICE_ID_SL1900,      /* SiliconInstr SL1543:3052 SL1900 */
+	 0x1543, 0x3000, 0, 0, SL1900_CARD},
+	{PCI_VENDOR_ID_SILICON, PCI_DEVICE_ID_SL1900,	   /* SiliconInstr SL1543:3052 SL1543 */
 	 PCI_ANY_ID, PCI_ANY_ID, 0, 0, SL1543_CARD},
 	{PCI_VENDOR_ID_MOTOROLA, PCI_DEVICE_ID_SL1900,     /* Motorola 1057:3052 */
 	 PCI_ANY_ID, PCI_ANY_ID, 0, 0, SL1900_CARD},
