Index: plymouth-0.8.3_20100929/src/ply-splash-core.pc.in
===================================================================
--- plymouth-0.8.3_20100929.orig/src/ply-splash-core.pc.in
+++ plymouth-0.8.3_20100929/src/ply-splash-core.pc.in
@@ -10,5 +10,5 @@ policydir=@PLYMOUTH_POLICY_DIR@
 Name: Plymouth
 Description: Utility Library for Boot Splash Plugins
 Version: @VERSION@
-Libs: -L${libdir} -lply -lply-splash-core
+Libs: -L${libdir} -lplybootsplash -lply -ldl
 Cflags: -I${includedir}/plymouth-1/ply -I${includedir}/plymouth-1/ply-splash-core
