Index: plymouth-0.8.3_20100929/scripts/plymouth-populate-initrd.in
===================================================================
--- plymouth-0.8.3_20100929.orig/scripts/plymouth-populate-initrd.in
+++ plymouth-0.8.3_20100929/scripts/plymouth-populate-initrd.in
@@ -73,7 +73,7 @@ set_verbose $verbose || :
 
 mkdir -p ${INITRDDIR}${PLYMOUTH_DATADIR}/plymouth/themes
 inst /sbin/plymouthd $INITRDDIR /bin/plymouthd
-inst /bin/plymouth $INITRDDIR
+inst /bin/plymouth $INITRDDIR /bin/plymouth
 inst ${PLYMOUTH_DATADIR}/plymouth/themes/text/text.plymouth $INITRDDIR
 inst ${PLYMOUTH_PLUGIN_PATH}/text.so $INITRDDIR
 inst ${PLYMOUTH_DATADIR}/plymouth/themes/details/details.plymouth $INITRDDIR
