From 8531fcfcc5261231b44a12c0a6d329b3cc4280fa Mon Sep 17 00:00:00 2001
From: unknown author <cooker@mandrivalinux.org>
Date: Tue, 25 May 2010 16:15:41 +0000
Subject: [PATCH] retain vt

---
 src/main.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

Index: plymouth-0.8.3_20100929/src/main.c
===================================================================
--- plymouth-0.8.3_20100929.orig/src/main.c
+++ plymouth-0.8.3_20100929/src/main.c
@@ -112,6 +112,7 @@ typedef struct
   uint32_t should_retain_splash : 1;
   uint32_t is_inactive : 1;
   uint32_t should_force_details : 1;
+  uint32_t should_retain_vt: 1;
 
   char *kernel_console_tty;
   char *override_splash_path;
@@ -805,6 +806,7 @@ static void
 on_show_splash (state_t *state)
 {
   bool has_display;
+/*  state->should_retain_vt = false;*/
 
   if (state->is_inactive)
     {
@@ -871,7 +873,7 @@ quit_splash (state_t *state)
 
   if (state->terminal != NULL)
     {
-      if (!state->should_retain_splash)
+      if (!state->should_retain_splash && !state->should_retain_vt)
         {
           ply_trace ("Not retaining splash, so deallocating VT");
           ply_terminal_deactivate_vt (state->terminal);
@@ -914,6 +916,7 @@ on_hide_splash (state_t *state)
     return;
 
   ply_trace ("hiding boot splash");
+  state->should_retain_vt = true;
   dump_details_and_quit_splash (state);
 }
 
