diff -Naur perl-5.12.1.orig/cpan/Module-Build/lib/Module/Build/Base.pm perl-5.12.1/cpan/Module-Build/lib/Module/Build/Base.pm
--- perl-5.12.1.orig/cpan/Module-Build/lib/Module/Build/Base.pm	2010-08-22 07:01:55.027018924 +0300
+++ perl-5.12.1/cpan/Module-Build/lib/Module/Build/Base.pm	2010-08-22 07:00:23.053000479 +0300
@@ -903,7 +903,7 @@
 __PACKAGE__->add_property(mymetafile => 'MYMETA.yml');
 __PACKAGE__->add_property(recurse_into => []);
 __PACKAGE__->add_property(use_rcfile => 1);
-__PACKAGE__->add_property(create_packlist => 1);
+__PACKAGE__->add_property(create_packlist => 0);
 __PACKAGE__->add_property(allow_mb_mismatch => 0);
 __PACKAGE__->add_property(config => undef);
 __PACKAGE__->add_property(test_file_exts => ['.t']);
@@ -2810,7 +2810,7 @@
   }
 }
 
-sub process_pod_files { shift()->process_files_by_extension(shift()) }
+#sub process_pod_files { shift()->process_files_by_extension(shift()) }
 sub process_pm_files  { shift()->process_files_by_extension(shift()) }
 
 sub process_script_files {
