--- dist/Pod-Perldoc/lib/Pod/Perldoc.pm.orig	2010-08-20 10:00:58.372001181 +0300
+++ dist/Pod-Perldoc/lib/Pod/Perldoc.pm	2010-08-20 10:01:42.657998642 +0300
@@ -655,7 +655,7 @@
 
     $self->options_sanity;
 
-    $self->opt_n("nroff") unless $self->opt_n;
+    $self->opt_n("nroff -c") unless $self->opt_n;
     $self->add_formatter_option( '__nroffer' => $self->opt_n );
 
     # Get language from PERLDOC_POD2 environment variable
