--- ./lib/Pod/Perldoc.pm.orig	2009-01-14 14:57:51.000000000 +0000
+++ ./lib/Pod/Perldoc.pm	2009-01-14 14:58:15.000000000 +0000
@@ -648,7 +648,7 @@ sub options_processing {
 
     $self->options_sanity;
 
-    $self->opt_n("nroff") unless $self->opt_n;
+    $self->opt_n("nroff -c") unless $self->opt_n;
     $self->add_formatter_option( '__nroffer' => $self->opt_n );
 
     # Adjust for using translation packages
