Index: perl-5.10.1/lib/Module/Build/t/compat.t
===================================================================
--- perl-5.10.1.orig/lib/Module/Build/t/compat.t
+++ perl-5.10.1/lib/Module/Build/t/compat.t
@@ -7,6 +7,8 @@ use File::Spec;
 use IO::File;
 use Config;
 
+plan skip_all => "Skip for sandbox violation";
+
 # Don't let our own verbosity/test_file get mixed up with our subprocess's
 my @makefile_keys = qw(TEST_VERBOSE HARNESS_VERBOSE TEST_FILES MAKEFLAGS);
 local  @ENV{@makefile_keys};
