diff -Nru ant-1.7.1.orig/src/main/org/apache/tools/ant/taskdefs/optional/Rpm.java ant-1.7.1/src/main/org/apache/tools/ant/taskdefs/optional/Rpm.java
--- ant-1.7.1.orig/src/main/org/apache/tools/ant/taskdefs/optional/Rpm.java	2008-11-10 10:50:29.000000000 -0600
+++ ant-1.7.1/src/main/org/apache/tools/ant/taskdefs/optional/Rpm.java	2008-11-10 10:50:49.000000000 -0600
@@ -120,7 +120,7 @@
                                 : rpmBuildCommand);
         if (topDir != null) {
             toExecute.createArgument().setValue("--define");
-            toExecute.createArgument().setValue("_topdir" + topDir);
+            toExecute.createArgument().setValue("_topdir " + topDir);
         }
 
         toExecute.createArgument().setLine(command);
