Index: pardus/netutils.py
===================================================================
--- pardus/netutils.py	(revision 25189)
+++ pardus/netutils.py	(revision 25215)
@@ -231,7 +231,7 @@
                 time.sleep(0.2)
                 tt -= 0.2
 
-        return subprocess.call(self.autoCmd)
+        return os.system(" ".join(self.autoCmd))
 
     def stopAuto(self):
         # dhcpcd does not create a pid file until it gets 
