Index: koffice-2.0.1/tools/kthesaurus/KThesaurus.desktop
===================================================================
--- koffice-2.0.1.orig/tools/kthesaurus/KThesaurus.desktop
+++ koffice-2.0.1/tools/kthesaurus/KThesaurus.desktop
@@ -13,7 +13,7 @@ Name[ta]=கேதிசோர்ஸ்
 Name[x-test]=xxKThesaurusxx
 Name[zh_TW]=KDE 同義字
 Type=Application
-Icon=kthesaurus
+Icon=accessories-dictionary
 Exec=kthesaurus
 GenericName=Related Words
 GenericName[af]=Verwanter Woorde
