Index: koffice-2.3.3/cmake/modules/FindIconv.cmake
===================================================================
--- koffice-2.3.3.orig/cmake/modules/FindIconv.cmake	2011-03-16 14:50:07.281514916 +0000
+++ koffice-2.3.3/cmake/modules/FindIconv.cmake	2011-03-16 14:50:38.731514908 +0000
@@ -15,8 +15,8 @@
 
 FIND_PATH(ICONV_INCLUDE_DIR iconv.h) 
  
-FIND_LIBRARY(ICONV_LIBRARIES NAMES iconv libiconv libiconv-2 c)
- 
+SET(ICONV_LIBRARIES c) 
+
 IF(ICONV_INCLUDE_DIR AND ICONV_LIBRARIES) 
    SET(ICONV_FOUND TRUE) 
 ENDIF(ICONV_INCLUDE_DIR AND ICONV_LIBRARIES) 
