Origin: vendor
Bug: <url in upstream bugtracker>
Bug-Ubuntu: https://launchpad.net/bugs/707794
Forwarded: no
Author: Jonathan Riddell <jr@ubuntu.com>
Reviewed-By: Scott Kitterman <scott@kitterman.com>
Last-Update: <2011-03-28>

--- koffice-2.3.3.orig/krita/ui/CMakeLists.txt
+++ koffice-2.3.3/krita/ui/CMakeLists.txt
@@ -214,7 +214,7 @@ kde4_add_ui_files(kritaui_LIB_SRCS
 
 kde4_add_library(kritaui SHARED ${kritaui_LIB_SRCS} )
 
-target_link_libraries(kritaui ${KDE4_THREADWEAVER_LIBRARIES} kritaimage komain kowidgets ${PNG_LIBRARIES} ${EXIV2_LIBRARIES})
+target_link_libraries(kritaui ${KDE4_THREADWEAVER_LIBRARIES} kritaimage komain kowidgets ${PNG_LIBRARIES} ${EXIV2_LIBRARIES} -lX11)
 
 if(HAVE_OPENGL)
     target_link_libraries(kritaui ${OPENGL_LIBRARIES} ${QT_QTOPENGL_LIBRARY} ${OPENEXR_LIBRARIES})
