Index: linux-2.6.37/drivers/staging/brcm80211/sys/wl_mac80211.c
===================================================================
--- linux-2.6.37.orig/drivers/staging/brcm80211/sys/wl_mac80211.c
+++ linux-2.6.37/drivers/staging/brcm80211/sys/wl_mac80211.c
@@ -161,7 +161,7 @@ static struct pci_device_id wl_id_table[
 	{0}
 };
 
-MODULE_DEVICE_TABLE(pci, wl_id_table);
+/*MODULE_DEVICE_TABLE(pci, wl_id_table);*/
 static void wl_remove(struct pci_dev *pdev);
 #endif				/* !BCMSDIO */
 
