Index: linux-2.6.37/arch/x86/mm/init_32.c
===================================================================
--- linux-2.6.37.orig/arch/x86/mm/init_32.c
+++ linux-2.6.37/arch/x86/mm/init_32.c
@@ -226,6 +226,10 @@ page_table_range_init(unsigned long star
 
 static inline int is_kernel_text(unsigned long addr)
 {
+#if defined(CONFIG_X86_32) && defined(CONFIG_ACPI_SLEEP)
+	if (addr >= PAGE_OFFSET && addr <= (PAGE_OFFSET+(1<<20)))
+		return 1;
+#endif
 	if (addr >= PAGE_OFFSET && addr <= (unsigned long)__init_end)
 		return 1;
 	return 0;
Index: linux-2.6.37/arch/x86/mm/pageattr.c
===================================================================
--- linux-2.6.37.orig/arch/x86/mm/pageattr.c
+++ linux-2.6.37/arch/x86/mm/pageattr.c
@@ -315,6 +315,10 @@ static inline pgprot_t static_protection
 			pgprot_val(forbidden) |= _PAGE_RW;
 	}
 #endif
+#if defined(CONFIG_X86_32) && defined(CONFIG_ACPI_SLEEP)
+	if (within(address, PAGE_OFFSET, PAGE_OFFSET+(1<<20)))
+		pgprot_val(forbidden) |= _PAGE_NX;
+#endif
 
 	prot = __pgprot(pgprot_val(prot) & ~pgprot_val(forbidden));
 
