Index: linux-2.6.37/drivers/net/wireless/rtl8192se/rtl_core.c
===================================================================
--- linux-2.6.37.orig/drivers/net/wireless/rtl8192se/rtl_core.c
+++ linux-2.6.37/drivers/net/wireless/rtl8192se/rtl_core.c
@@ -5562,7 +5562,6 @@ static int __devinit rtl8192_pci_probe(s
 #endif
 
 	netif_carrier_off(dev);
-	netif_stop_queue(dev);
 
 	register_netdev(dev);
 	RT_TRACE(COMP_INIT, "dev name: %s\n",dev->name);
