Index: linux-2.6.37/drivers/staging/rtl8192e/r8192E_core.c
===================================================================
--- linux-2.6.37.orig/drivers/staging/rtl8192e/r8192E_core.c
+++ linux-2.6.37/drivers/staging/rtl8192e/r8192E_core.c
@@ -6008,7 +6008,6 @@ static int __devinit rtl8192_pci_probe(s
 	}
 
 	netif_carrier_off(dev);
-	netif_stop_queue(dev);
 
 	register_netdev(dev);
 	RT_TRACE(COMP_INIT, "dev name=======> %s\n",dev->name);
