diff -ur grub-0.97.orig/util/grub-install.in grub-0.97/util/grub-install.in
--- grub-0.97.orig/util/grub-install.in	2009-01-12 21:47:56.000000000 -0500
+++ grub-0.97/util/grub-install.in	2009-01-12 21:48:16.000000000 -0500
@@ -102,7 +102,7 @@
     linux*)
 	tmp_disk=`echo "$1" | grep -v '/mapper/control$' |
 		grep -v '/mapper/[[:alnum:]]\+-[[:alnum:]]\+$' | uniq |
-		sed -e 's%\([sh]d[a-z]\)[0-9]*$%\1%' \
+		sed -e 's%\([shv]d[a-z]\)[0-9]*$%\1%' \
 				  -e 's%\(d[0-9]*\)p[0-9]*$%\1%' \
 				  -e 's%\(fd[0-9]*\)$%\1%' \
 				  -e 's%/part[0-9]*$%/disc%' \
@@ -112,7 +112,7 @@
 	  -e 's%\(/mapper/[[:alpha:]]\+_[[:alpha:]]\+\)[[:digit:]]\+$%\1%'`
 	tmp_part=`echo "$1" | grep -v '/mapper/control$' |
 		grep -v '/mapper/[[:alnum:]]\+-[[:alnum:]]\+$' | uniq |
-		sed -e 's%.*/[sh]d[a-z]\([0-9]*\)$%\1%' \
+		sed -e 's%.*/[shv]d[a-z]\([0-9]*\)$%\1%' \
 				  -e 's%.*d[0-9]*p%%' \
 				  -e 's%.*/fd[0-9]*$%%' \
 				  -e 's%.*/floppy/[0-9]*$%%' \
