Index: xdg-utils-1.0.2/scripts/xdg-open
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-open
+++ xdg-utils-1.0.2/scripts/xdg-open
@@ -303,7 +303,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
     fi
 }
 
Index: xdg-utils-1.0.2/scripts/xdg-desktop-icon
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-desktop-icon
+++ xdg-utils-1.0.2/scripts/xdg-desktop-icon
@@ -404,7 +404,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
     fi
 }
 
Index: xdg-utils-1.0.2/scripts/xdg-desktop-menu
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-desktop-menu
+++ xdg-utils-1.0.2/scripts/xdg-desktop-menu
@@ -600,7 +600,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
     fi
 }
 
Index: xdg-utils-1.0.2/scripts/xdg-email
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-email
+++ xdg-utils-1.0.2/scripts/xdg-email
@@ -356,7 +356,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
     fi
 }
 
Index: xdg-utils-1.0.2/scripts/xdg-icon-resource
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-icon-resource
+++ xdg-utils-1.0.2/scripts/xdg-icon-resource
@@ -396,7 +396,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
     fi
 }
 
Index: xdg-utils-1.0.2/scripts/xdg-mime
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-mime
+++ xdg-utils-1.0.2/scripts/xdg-mime
@@ -425,7 +425,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
     fi
 }
 
Index: xdg-utils-1.0.2/scripts/xdg-screensaver
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-screensaver
+++ xdg-utils-1.0.2/scripts/xdg-screensaver
@@ -333,7 +333,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
     fi
 }
 
Index: xdg-utils-1.0.2/scripts/xdg-su
===================================================================
--- xdg-utils-1.0.2.orig/scripts/xdg-su
+++ xdg-utils-1.0.2/scripts/xdg-su
@@ -300,7 +300,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif xprop -root | grep -qw XFCE_DESKTOP_WINDOW 2>&1; then DE=xfce;
     fi
 }
 
