--- Makefile.PL.org	    2006-11-07 10:08:43.000000000 +0200
+++ Makefile.PL	        2006-11-07 10:46:19.000000000 +0200
@@ -42,7 +42,7 @@
         $script .= <<"INSTALL";
 
 install_sax_pureperl :
-\t\@\$(PERL) -MXML::SAX -e "XML::SAX->add_parser(q(XML::SAX::PurePerl))->save_parsers()"
+PERL5LIB=\$(INSTALLSITELIB)\t\@\$(PERL) -MXML::SAX -e "XML::SAX->add_parser(q(XML::SAX::PurePerl))->save_parsers()"
 
 INSTALL
 
