Index: kdepim-runtime-4.4.93/kcm/kcm_akonadi.desktop
===================================================================
--- kdepim-runtime-4.4.93.orig/kcm/kcm_akonadi.desktop
+++ kdepim-runtime-4.4.93/kcm/kcm_akonadi.desktop
@@ -8,9 +8,10 @@ X-KDE-ModuleType=Library
 X-KDE-Library=kcm_akonadi
 X-KDE-FactoryName=kcm_akonadi
 X-KDE-HasReadOnlyMode=false
-X-KDE-ParentApp=
+X-KDE-ParentApp=kcontrol
 X-KDE-ParentComponents=
 X-KDE-CfgDlgHierarchy=
+X-KDE-System-Settings-Parent-Category=personal-information
 
 Name=Akonadi Configuration
 Name[ar]=ضبط اكوندا
Index: kdepim-runtime-4.4.93/kcm/kcm_akonadi_resources.desktop
===================================================================
--- kdepim-runtime-4.4.93.orig/kcm/kcm_akonadi_resources.desktop
+++ kdepim-runtime-4.4.93/kcm/kcm_akonadi_resources.desktop
@@ -8,10 +8,9 @@ X-KDE-ModuleType=Library
 X-KDE-Library=kcm_akonadi_resources
 X-KDE-FactoryName=kcm_akonadi_resources
 X-KDE-HasReadOnlyMode=false
-X-KDE-ParentApp=kcontrol
+X-KDE-ParentApp=
 X-KDE-ParentComponents=
 X-KDE-CfgDlgHierarchy=
-X-KDE-System-Settings-Parent-Category=personal-information
 
 Name=Akonadi Resources Configuration
 Name[ar]=ضبط موارد اكوندا
