diff -ur kdepim-runtime-4.4.0/kcm/kcm_akonadi.desktop kdepim-runtime-4.4.0-show-akonadi-kcm/kcm/kcm_akonadi.desktop
--- kdepim-runtime-4.4.0/kcm/kcm_akonadi.desktop	2010-02-04 10:33:11.000000000 +0100
+++ kdepim-runtime-4.4.0-show-akonadi-kcm/kcm/kcm_akonadi.desktop	2010-02-15 17:23:05.000000000 +0100
@@ -8,9 +8,10 @@
 X-KDE-Library=kcm_akonadi
 X-KDE-FactoryName=kcm_akonadi
 X-KDE-HasReadOnlyMode=false
-X-KDE-ParentApp=
+X-KDE-ParentApp=kcontrol
 X-KDE-ParentComponents=
 X-KDE-CfgDlgHierarchy=
+X-KDE-System-Settings-Parent-Category=advanced-user-settings
 
 Name=Akonadi Configuration
 Name[ar]=ضبط اكوندا
