--- configure.ac~	2009-05-11 19:18:45.606313205 +0300
+++ configure.ac	2009-05-11 19:18:45.606313205 +0300
@@ -59,8 +59,6 @@
         CFLAGS="$CFLAGS -g -Wall -Werror"
 fi
 
-GTK_DOC_CHECK([1.0])
-
 AC_OUTPUT([
 Makefile
 libgssdp/Makefile
--- doc/Makefile.am~	2009-05-11 19:20:51.774317196 +0300
+++ doc/Makefile.am	2009-05-11 19:20:51.780320796 +0300
@@ -74,8 +74,8 @@
 GTKDOC_LIBS=$(top_builddir)/libgssdp/libgssdp-1.0.la $(LIBGSSDP_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
+# include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += version.xml.in
+EXTRA_DIST = version.xml.in
