Index: libnotify-0.4.5/docs/reference/Makefile.am
===================================================================
--- libnotify-0.4.5.orig/docs/reference/Makefile.am
+++ libnotify-0.4.5/docs/reference/Makefile.am
@@ -73,9 +73,6 @@ expand_content_files=
 INCLUDES=-I$(top_srcdir) -I$(top_builddir) $(PACKAGE_CFLAGS)
 GTKDOC_LIBS=$(PACKAGE_LIBS) $(top_builddir)/libnotify/libnotify.la
 
-# This includes the standard gtk-doc make rules, copied by gtkdocize.
-include $(top_srcdir)/gtk-doc.make
-
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += 
+EXTRA_DIST = 
Index: libnotify-0.4.5/configure.ac
===================================================================
--- libnotify-0.4.5.orig/configure.ac
+++ libnotify-0.4.5/configure.ac
@@ -112,12 +112,6 @@ AC_DEFINE_UNQUOTED(DBUS_MICRO_VER, $DBUS
 AC_DEFINE_UNQUOTED(DBUS_API_SUBJECT_TO_CHANGE, ,DBUS API is subject to change)
 
 
-dnl ################################################################
-dnl # Set up gtk-doc
-dnl ################################################################
-GTK_DOC_CHECK([1.4])
-
-
 dnl # Use wall if we have GCC
 if test "x$GCC" = "xyes"; then
 	CFLAGS="$CFLAGS -Wall"
