
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=181963

Index: 2008-devel/fs/fat/inode.c
===================================================================
--- 2008-devel.orig/fs/fat/inode.c
+++ 2008-devel/fs/fat/inode.c
@@ -945,7 +945,8 @@ static int parse_options(char *options, 
 		opts->shortname = 0;
 	opts->name_check = 'n';
 	opts->quiet = opts->showexec = opts->sys_immutable = opts->dotsOK =  0;
-	opts->utf8 = opts->unicode_xlate = 0;
+	opts->utf8 = 1;
+	opts->unicode_xlate = 0;
 	opts->numtail = 1;
 	opts->usefree = opts->nocase = 0;
 	*debug = 0;
