--- supertuxkart-0.6.2a/data/Makefile.in.orig	2010-12-20 13:54:28.832999419 +0200
+++ supertuxkart-0.6.2a/data/Makefile.in	2010-12-20 13:59:20.374000037 +0200
@@ -253,7 +253,6 @@
  $(shell find $(srcdir) -name "*.cup") \
  $(shell find $(srcdir) -maxdepth 1 -name "*.challenge") \
  $(shell find $(srcdir) -name "*.collectable") \
- $(shell find $(srcdir) -name "*.projectile") \
  CREDITS
 
 all: all-recursive
