diff -Nur supertuxkart-0.6.2-old/configure.ac supertuxkart-0.6.2/configure.ac
--- supertuxkart-0.6.2-old/configure.ac	2009-09-13 21:53:58.832873986 +0300
+++ supertuxkart-0.6.2/configure.ac	2009-09-13 21:56:20.175873322 +0300
@@ -357,7 +357,6 @@
   ;;
 esac
 SUMMARY="$SUMMARY\nUsing enet network support."
-ENETTREE="src/enet"
 
 # Check for gettext
 CHECK_GETTEXT(external)
@@ -394,7 +393,6 @@
 AC_SUBST(glut_LIBS)
 AC_SUBST(sdl_LIBS)
 AC_SUBST(BULLETTREE)
-AC_SUBST(ENETTREE)
 
 dnl ================
 dnl Create makefiles
@@ -418,7 +416,6 @@
  src/bullet/Makefile                \
  src/bullet/src/Makefile            \
  src/bullet/Demos/OpenGL/Makefile   \
- src/enet/Makefile                  \
 ])
 AC_OUTPUT
 echo -e $SUMMARY
