diff -Nur supertuxkart-0.6.2-old/src/Makefile.am supertuxkart-0.6.2/src/Makefile.am
--- supertuxkart-0.6.2-old/src/Makefile.am	2009-09-13 21:53:58.835873879 +0300
+++ supertuxkart-0.6.2/src/Makefile.am	2009-09-13 21:55:54.686623156 +0300
@@ -2,10 +2,9 @@
 
 SUBDIRS = ide
 
-bindir=$(prefix)/games
 bin_PROGRAMS = supertuxkart
 
-AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/games/$(PACKAGE)/\""  \
+AM_CPPFLAGS = -DSUPERTUXKART_DATADIR="\"$(datadir)/$(PACKAGE)/\""  \
  -I$(srcdir)/bullet/src/ -I$(srcdir)/enet/include/
 
 supertuxkart_SOURCES = \
