diff -Nur supertuxkart-0.7-old//configure.ac supertuxkart-0.7/configure.ac
--- supertuxkart-0.7-old//configure.ac	2011-03-11 14:21:12.443001414 +0200
+++ supertuxkart-0.7/configure.ac	2011-03-11 14:21:46.973001444 +0200
@@ -233,7 +233,7 @@
 
 case "${host}" in
 *-*-linux* )
-  irrlicht_LIBS="-L/$with_irrlicht/lib/Linux -L/$with_irrlicht/lib -lIrrlicht"
+  irrlicht_LIBS="-L/$with_irrlicht/lib/Linux -L/$with_irrlicht/lib -lIrrlicht -lIrrXML"
   ;;
 *darwin*|*macosx*)
   irrlicht_LIBS="-L/$with_irrlicht/lib/ -lIrrlicht -framework Cocoa"
