diff -Nur supertuxkart-0.6.2-old/data/fonts/Makefile.am supertuxkart-0.6.2/data/fonts/Makefile.am
--- supertuxkart-0.6.2-old/data/fonts/Makefile.am	2009-09-13 21:53:58.876873099 +0300
+++ supertuxkart-0.6.2/data/fonts/Makefile.am	2009-09-13 21:55:20.932624036 +0300
@@ -1,5 +1,5 @@
 # data/fonts/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/fonts
+pkgdatadir = $(datadir)/$(PACKAGE)/data/fonts
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.txf")
diff -Nur supertuxkart-0.6.2-old/data/grandprix/Makefile.am supertuxkart-0.6.2/data/grandprix/Makefile.am
--- supertuxkart-0.6.2-old/data/grandprix/Makefile.am	2009-09-13 21:54:07.913872745 +0300
+++ supertuxkart-0.6.2/data/grandprix/Makefile.am	2009-09-13 21:55:20.933623815 +0300
@@ -1,6 +1,6 @@
 # data/music
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/grandprix
+pkgdatadir = $(datadir)/$(PACKAGE)/data/grandprix
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.grandprix")  \
                     $(shell find $(srcdir) -name "*.challenge")
diff -Nur supertuxkart-0.6.2-old/data/items/Makefile.am supertuxkart-0.6.2/data/items/Makefile.am
--- supertuxkart-0.6.2-old/data/items/Makefile.am	2009-09-13 21:54:07.918872614 +0300
+++ supertuxkart-0.6.2/data/items/Makefile.am	2009-09-13 21:55:20.934623104 +0300
@@ -1,6 +1,6 @@
 # data/herrings/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/items
+pkgdatadir = $(datadir)/$(PACKAGE)/data/items
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.ac") License.txt
 
diff -Nur supertuxkart-0.6.2-old/data/karts/Makefile.am supertuxkart-0.6.2/data/karts/Makefile.am
--- supertuxkart-0.6.2-old/data/karts/Makefile.am	2009-09-13 21:53:59.190872893 +0300
+++ supertuxkart-0.6.2/data/karts/Makefile.am	2009-09-13 21:55:20.933623815 +0300
@@ -1,6 +1,6 @@
 # data/karts
 
-pkgdatadir= $(datadir)/games/$(PACKAGE)/data/karts
+pkgdatadir= $(datadir)/$(PACKAGE)/data/karts
 
 nobase_dist_pkgdata_DATA = $(shell find $(srcdir) -name "materials.dat") \
                            $(shell find $(srcdir) -name "*.rgb") \
diff -Nur supertuxkart-0.6.2-old/data/Makefile.am supertuxkart-0.6.2/data/Makefile.am
--- supertuxkart-0.6.2-old/data/Makefile.am	2009-09-13 21:53:58.876873099 +0300
+++ supertuxkart-0.6.2/data/Makefile.am	2009-09-13 21:55:20.932624036 +0300
@@ -1,7 +1,7 @@
 # data/
 
 SUBDIRS    = fonts items karts models music po sfx textures tracks grandprix
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data
+pkgdatadir = $(datadir)/$(PACKAGE)/data
 
 dist_pkgdata_DATA = \
  $(shell find $(srcdir) -name "*.data") \
@@ -13,16 +13,7 @@
  $(shell find $(srcdir) -name "*.projectile") \
  CREDITS
 
-desktopdir = $(prefix)/share/applications
-desktop_DATA = supertuxkart.desktop
 
-icondir = $(prefix)/share/pixmaps
-dist_icon_DATA = supertuxkart_32.xpm supertuxkart_64.xpm
 
 
-# The desktop file needs the absolute path to the binary
 # since e.g. /usr/games might not be in the standard path
-EXTRA_DIST = supertuxkart_desktop.template
-CLEANFILES = supertuxkart.desktop
-supertuxkart.desktop: Makefile supertuxkart_desktop.template
-	sed 's#PREFIX#$(prefix)#' $(srcdir)/supertuxkart_desktop.template >supertuxkart.desktop
diff -Nur supertuxkart-0.6.2-old/data/models/Makefile.am supertuxkart-0.6.2/data/models/Makefile.am
--- supertuxkart-0.6.2-old/data/models/Makefile.am	2009-09-13 21:53:59.185872535 +0300
+++ supertuxkart-0.6.2/data/models/Makefile.am	2009-09-13 21:55:20.932624036 +0300
@@ -1,6 +1,6 @@
 # data/models/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/models
+pkgdatadir = $(datadir)/$(PACKAGE)/data/models
 
 dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.ac") License.txt
 
diff -Nur supertuxkart-0.6.2-old/data/music/Makefile.am supertuxkart-0.6.2/data/music/Makefile.am
--- supertuxkart-0.6.2-old/data/music/Makefile.am	2009-09-13 21:54:03.691872929 +0300
+++ supertuxkart-0.6.2/data/music/Makefile.am	2009-09-13 21:55:20.933623815 +0300
@@ -1,6 +1,6 @@
 # data/music
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/music
+pkgdatadir = $(datadir)/$(PACKAGE)/data/music
 
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.ogg")  \
                     $(shell find $(srcdir) -name "*.music") \
diff -Nur supertuxkart-0.6.2-old/data/po/Makefile.am supertuxkart-0.6.2/data/po/Makefile.am
--- supertuxkart-0.6.2-old/data/po/Makefile.am	2009-09-13 21:54:07.914872663 +0300
+++ supertuxkart-0.6.2/data/po/Makefile.am	2009-09-13 21:55:20.933623815 +0300
@@ -1,6 +1,6 @@
 # data/po/
 
-pkgdatadir = $(datadir)/games/@PACKAGE@/data/po
+pkgdatadir = $(datadir)/@PACKAGE@/data/po
 
 nobase_pkgdata_DATA = $(shell find $(srcdir) -name "*.po") \
                       $(shell find $(srcdir) -name "*.mo") \
diff -Nur supertuxkart-0.6.2-old/data/sfx/Makefile.am supertuxkart-0.6.2/data/sfx/Makefile.am
--- supertuxkart-0.6.2-old/data/sfx/Makefile.am	2009-09-13 21:54:07.919872881 +0300
+++ supertuxkart-0.6.2/data/sfx/Makefile.am	2009-09-13 21:55:20.934623104 +0300
@@ -1,6 +1,6 @@
 # sfx/
 
-pkgdatadir = $(datadir)/games/$(PACKAGE)/data/sfx
+pkgdatadir = $(datadir)/$(PACKAGE)/data/sfx
 
 nobase_dist_pkgdata_DATA =  $(shell find $(srcdir) -name "*.wav") \
                             $(shell find $(srcdir) -name "README")\
diff -Nur supertuxkart-0.6.2-old/data/textures/Makefile.am supertuxkart-0.6.2/data/textures/Makefile.am
--- supertuxkart-0.6.2-old/data/textures/Makefile.am	2009-09-13 21:54:05.338872806 +0300
+++ supertuxkart-0.6.2/data/textures/Makefile.am	2009-09-13 21:55:20.933623815 +0300
@@ -1,6 +1,6 @@
 # data/textures
 
-pkgdatadir   = $(datadir)/games/$(PACKAGE)/data/textures
+pkgdatadir   = $(datadir)/$(PACKAGE)/data/textures
 dist_pkgdata_DATA = $(shell find $(srcdir) -name "*.rgb") materials.dat License.txt
 
 
diff -Nur supertuxkart-0.6.2-old/data/tracks/Makefile.am supertuxkart-0.6.2/data/tracks/Makefile.am
--- supertuxkart-0.6.2-old/data/tracks/Makefile.am	2009-09-13 21:53:58.916873518 +0300
+++ supertuxkart-0.6.2/data/tracks/Makefile.am	2009-09-13 21:55:20.932624036 +0300
@@ -1,6 +1,6 @@
 # data/tracks
 
-pkgdatadir= $(datadir)/games/$(PACKAGE)/data/tracks
+pkgdatadir= $(datadir)/$(PACKAGE)/data/tracks
 nobase_dist_pkgdata_DATA = $(shell find $(srcdir) -name "readme.txt")      \
                            $(shell find $(srcdir) -name "*.track")         \
                            $(shell find $(srcdir) -name "*.drv?")          \
