Index: pavucontrol-0.9.8/src/pavucontrol.glade
===================================================================
--- pavucontrol-0.9.8.orig/src/pavucontrol.glade
+++ pavucontrol-0.9.8/src/pavucontrol.glade
@@ -571,7 +571,7 @@ Monitors</property>
                             <child>
                               <widget class="GtkImage" id="image18">
                                 <property name="visible">True</property>
-                                <property name="icon_name">stock_lock</property>
+                                <property name="icon_name">object-locked</property>
                                 <property name="icon-size">1</property>
                               </widget>
                             </child>
