Index: gcc/Makefile.in
===================================================================
*** gcc/Makefile.in.orig	2010-11-05 14:40:33.000000000 +0100
--- gcc/Makefile.in	2010-11-05 14:41:19.000000000 +0100
*************** TARGET_LIBGCC2_CFLAGS =
*** 659,665 ****
  CRTSTUFF_CFLAGS = -O2 $(GCC_CFLAGS) $(INCLUDES) $(MULTILIB_CFLAGS) -g0 \
    -finhibit-size-directive -fno-inline -fno-exceptions \
    -fno-zero-initialized-in-bss -fno-toplevel-reorder -fno-tree-vectorize \
!   -fno-stack-protector \
    $(INHIBIT_LIBC_CFLAGS)
  
  # Additional sources to handle exceptions; overridden by targets as needed.
--- 659,665 ----
  CRTSTUFF_CFLAGS = -O2 $(GCC_CFLAGS) $(INCLUDES) $(MULTILIB_CFLAGS) -g0 \
    -finhibit-size-directive -fno-inline -fno-exceptions \
    -fno-zero-initialized-in-bss -fno-toplevel-reorder -fno-tree-vectorize \
!   -fno-stack-protector -fno-unwind-tables -fno-asynchronous-unwind-tables \
    $(INHIBIT_LIBC_CFLAGS)
  
  # Additional sources to handle exceptions; overridden by targets as needed.
