Index: gcc/Makefile.in
===================================================================
*** gcc/Makefile.in.orig	2009-11-24 10:58:41.000000000 +0100
--- gcc/Makefile.in	2009-11-24 10:59:06.000000000 +0100
*************** TARGET_LIBGCC2_CFLAGS =
*** 639,644 ****
--- 639,645 ----
  CRTSTUFF_CFLAGS = -O2 $(GCC_CFLAGS) $(INCLUDES) $(MULTILIB_CFLAGS) -g0 \
    -finhibit-size-directive -fno-inline -fno-exceptions \
    -fno-zero-initialized-in-bss -fno-toplevel-reorder -fno-tree-vectorize \
+   -fno-unwind-tables -fno-asynchronous-unwind-tables \
    $(INHIBIT_LIBC_CFLAGS)
  
  # Additional sources to handle exceptions; overridden by targets as needed.
