Index: libjava/configure
===================================================================
--- libjava/configure.orig	2009-11-03 12:00:58.000000000 +0100
+++ libjava/configure	2009-11-18 14:58:52.000000000 +0100
@@ -3482,6 +3482,26 @@ else
 fi
 
 
+# Default to --enable-libjava-multilib
+# Check whether --enable-libjava-multilib or --disable-libjava-multilib was given.
+if test "${enable_libjava_multilib+set}" = set; then
+  enableval="$enable_libjava_multilib"
+  case "${enableval}" in
+  yes) multilib=yes ;;
+  no)  multilib=no ;;
+  *)   { { echo "$as_me:$LINENO: error: bad value ${enableval} for libjava-multilib option" >&5
+echo "$as_me: error: bad value ${enableval} for libjava-multilib option" >&2;}
+   { (exit 1); exit 1; }; } ;;
+ esac
+else
+ multilib=yes
+fi;
+if test "$multilib" = no; then
+# Reset also --enable-multilib state, as that is what is looked at
+# by config-ml.in
+ ac_configure_args="$ac_configure_args --disable-multilib"
+fi
+
 # It may not be safe to run linking tests in AC_PROG_CC/AC_PROG_CXX.
 
 
