--- libgpod-0.7.2-orig/tools/Makefile.am	2009-04-15 23:18:36.000000000 +0200
+++ libgpod-0.7.2/tools/Makefile.am	2009-04-28 12:35:43.000000000 +0200
@@ -1,7 +1,6 @@
 if HAVE_SGUTILS
 
 bin_PROGRAMS=ipod-read-sysinfo-extended
-noinst_PROGRAMS=ipod-time-sync
 
 ipod_read_sysinfo_extended_SOURCES =	\
 	read-sysinfoextended-sgutils.c	\
@@ -12,6 +11,7 @@
 if HAVE_HAL
 haldir = $(HALCALLOUTSDIR)
 hal_PROGRAMS = libgpod-callout
+noinst_PROGRAMS=ipod-time-sync
 
 libgpod_callout_SOURCES =		\
 	hal-callout.c 		\
