diff --git a/tools/hal-callout.c b/tools/hal-callout.c
index e7313fe..4d6044b 100644
--- a/tools/hal-callout.c
+++ b/tools/hal-callout.c
@@ -52,7 +52,7 @@ static char *mount_ipod (const char *dev_path)
         if (fstype == NULL) {
                 return NULL;
         }
-        filename = g_build_filename (g_get_tmp_dir (), "ipodXXXXXX", NULL);
+        filename = g_build_filename ("/var/run/hald", "ipodXXXXXX", NULL);
         if (filename == NULL) {
                 return NULL;
         }
 
