Index: gecko-mediaplayer-0.9.9.2/configure.in
===================================================================
--- gecko-mediaplayer-0.9.9.2.orig/configure.in
+++ gecko-mediaplayer-0.9.9.2/configure.in
@@ -35,8 +35,8 @@ AC_ARG_WITH([xulrunner-sdk],
         [Path to xulrunner SDK for build --with-xulrunner-sdk=/path/to/sdk])],
         [
 	XULRUNNER_SDK_PREFIX=$with_xulrunner_sdk
-	GECKO_CFLAGS="-I$XULRUNNER_SDK_PREFIX/include -I$XULRUNNER_SDK_PREFIX/include/plugin -I$XULRUNNER_SDK_PREFIX/include/nspr -I$XULRUNNER_SDK_PREFIX/include/xpcom -I$XULRUNNER_SDK_PREFIX/include/string -I$XULRUNNER_SDK_PREFIX/include/java"
-	GECKO_LIBS="-L$XULRUNNER_SDK_PREFIX/lib -lxpcomglue_s -lxul -lxpcom -lplds4 -lplc4 -lnspr4 -lpthread -ldl"
+	GECKO_CFLAGS=`$PKG_CONFIG --cflags libxul`
+	GECKO_LIBS=`$PKG_CONFIG --libs libxul`
 	AC_ARG_ENABLE(new-libxul,
 		AC_HELP_STRING([--enable-new-libxul],[libxul/xulrunner-sdk is 1.9.2 or higher]),
 		[usenewlibxul=$enableval],[usenewlibxul=no])
