diff -up ncurses-5.7/misc/ncurses-config.in.config ncurses-5.7/misc/ncurses-config.in
--- ncurses-5.7/misc/ncurses-config.in.config	2010-07-13 16:55:47.000000000 +0200
+++ ncurses-5.7/misc/ncurses-config.in	2010-07-13 16:59:07.000000000 +0200
@@ -35,7 +35,6 @@ exec_prefix="@exec_prefix@"
 
 bindir="@bindir@"
 includedir="@includedir@"
-libdir="@libdir@"
 datadir="@datadir@"
 mandir="@mandir@"
 
@@ -85,11 +84,11 @@ ENDECHO
 	--libs)
 		if test @TINFO_NAME@ = @LIB_NAME@ ; then
 		sed -e 's,^[ ]*,,' -e 's, [ ]*, ,g' -e 's,[ ]*$,,' <<-ENDECHO
-			-L$libdir @EXTRA_LDFLAGS@ -l${THIS} @LIBS@
+			@EXTRA_LDFLAGS@ -l${THIS} @LIBS@
 ENDECHO
 		else
 		sed -e 's,^[ ]*,,' -e 's, [ ]*, ,g' -e 's,[ ]*$,,' <<-ENDECHO
-			-L$libdir @EXTRA_LDFLAGS@ -l${THIS} -l${TINFO_LIB} @LIBS@
+			@EXTRA_LDFLAGS@ -l${THIS} -l${TINFO_LIB} @LIBS@
 ENDECHO
 		fi
 		;;
@@ -114,7 +113,6 @@ ENDECHO
 		echo "${includedir}"
 		;;
 	--libdir)
-		echo "${libdir}"
 		;;
 	--mandir)
 		echo "${mandir}"
