Index: module-init-tools-3.8/configure.ac
===================================================================
--- module-init-tools-3.8.orig/configure.ac
+++ module-init-tools-3.8/configure.ac
@@ -29,14 +29,6 @@ fi])
 AC_PROG_CC
 AC_PROG_RANLIB
 
-AC_CHECK_PROGS(DOCBOOKTOMAN, docbook-to-man docbook2man, [no],)
-if test x"$DOCBOOKTOMAN" = xno
-then
-	AC_MSG_WARN([docbook2man not found])
-	# fail with a meaningfull error if $DOCBOOKTOMAN called by the makefile
-	DOCBOOKTOMAN=docbook2man
-fi
- 
 # Delay adding the zlib_flags until after AC_PROG_CC, so we can distinguish
 # between a broken cc and a working cc but missing libz.a.
 LDADD="$LDADD $zlib_flags"
