Index: module-init-tools-3.12/modprobe.c
===================================================================
--- module-init-tools-3.12.orig/modprobe.c
+++ module-init-tools-3.12/modprobe.c
@@ -1129,6 +1129,8 @@ static void parse_toplevel_config(const
 		warn("Deprecated config file /etc/modprobe.conf, "
 		      "all config files belong into /etc/modprobe.d/.\n");
 
+	parse_config_file("/dev/.modprobe.initramfs.conf", conf, dump_only, removing);
+
 	/* default config */
 	parse_config_scan("/etc/modprobe.d", conf, dump_only, removing);
 }
