Index: squashfs-tools_4.0_20100816/Makefile
===================================================================
--- squashfs-tools_4.0_20100816.orig/Makefile
+++ squashfs-tools_4.0_20100816/Makefile
@@ -29,8 +29,8 @@ MKSQUASHFS_OBJS = mksquashfs.o read_fs.o
 UNSQUASHFS_OBJS = unsquashfs.o unsquash-1.o unsquash-2.o unsquash-3.o \
 	unsquash-4.o swap.o compressor.o gzip_wrapper.o
 
-CFLAGS = $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
-	-D_GNU_SOURCE -DCOMP_DEFAULT=\"$(COMP_DEFAULT)\"  -O2 -Wall
+CFLAGS = $(RPM_OPT_FLAGS) $(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
+	-D_GNU_SOURCE -DCOMP_DEFAULT=\"$(COMP_DEFAULT)\"
 
 LIBS =
 ifdef LZMA_SUPPORT
