--- a/src/oui.c	13 Jan 2007 17:48:12 -0000	1.2
+++ b/src/oui.c	25 Jan 2008 12:16:58 -0000
@@ -38,7 +38,7 @@
 
 /* http://standards.ieee.org/regauth/oui/oui.txt */
 
-#define OUIFILE "/var/lib/misc/oui.txt"
+#define OUIFILE "/usr/share/misc/oui.txt"
 
 char *ouitocomp(const char *oui)
 {
@@ -46,14 +46,9 @@ char *ouitocomp(const char *oui)
 	char *str, *map, *off, *end;
 	int fd;
 
-	fd = open("oui.txt", O_RDONLY);
+	fd = open(OUIFILE, O_RDONLY);
 	if (fd < 0) {
-		fd = open(OUIFILE, O_RDONLY);
-		if (fd < 0) {
-			fd = open("/usr/share/misc/oui.txt", O_RDONLY);
-			if (fd < 0)
-				return NULL;
-		}
+		return NULL;
 	}
 
 	if (fstat(fd, &st) < 0) {
