Index: kdenetwork-4.5.5/filesharing/advanced/kcm_sambaconf/kcmsambaconf.desktop
===================================================================
--- kdenetwork-4.5.5.orig/filesharing/advanced/kcm_sambaconf/kcmsambaconf.desktop
+++ kdenetwork-4.5.5/filesharing/advanced/kcm_sambaconf/kcmsambaconf.desktop
@@ -13,7 +13,7 @@ X-KDE-SubstituteUID=false
 X-KDE-HasReadOnlyMode=true
 X-KDE-RootOnly=true
 X-KDE-ParentApp=kcontrol
-X-KDE-System-Settings-Parent-Category=sharing
+X-KDE-System-Settings-Parent-Category=network-and-connectivity
 Categories=Qt;KDE;X-KDE-settings-network;Settings;
 OnlyShowIn=KDE;
 
