Index: synaptiks-0.5.0_pre1182832/applet/plasma-applet-synaptiks.desktop
===================================================================
--- synaptiks-0.5.0_pre1182832.orig/applet/plasma-applet-synaptiks.desktop
+++ synaptiks-0.5.0_pre1182832/applet/plasma-applet-synaptiks.desktop
@@ -10,8 +10,10 @@ Name[pt]=synaptiks
 Name[pt_BR]=synaptiks
 Name[sv]=Synaptiks
 Name[uk]=synaptiks
+Name[tr]=synaptiks
 Name[x-test]=xxsynaptiksxx
 Comment=Touchpad management applet
+Comment[tr]=Dokunmatik fare yönetimi programcığı
 Comment[da]=Applet til håndtering af touchpad
 Comment[en_GB]=Touchpad management applet
 Comment[es]=Miniaplicación de gestión del touchpad
Index: synaptiks-0.5.0_pre1182832/daemon/synaptiks.desktop
===================================================================
--- synaptiks-0.5.0_pre1182832.orig/daemon/synaptiks.desktop
+++ synaptiks-0.5.0_pre1182832/daemon/synaptiks.desktop
@@ -19,9 +19,11 @@ Name[pt]=synaptiks
 Name[pt_BR]=synaptiks
 Name[sv]=Synaptiks
 Name[uk]=synaptiks
+Name[uk]=synaptiks
 Name[x-test]=xxsynaptiksxx
 
 Comment=Touchpad control daemon
+Comment[tr]=Dokunmatik fare kontrol servisi
 Comment[da]=En kontroldæmon til touchpad
 Comment[en_GB]=Touchpad control dæmon
 Comment[es]=Demonio de control del touchpad
Index: synaptiks-0.5.0_pre1182832/kcmodule/synaptiks.desktop
===================================================================
--- synaptiks-0.5.0_pre1182832.orig/kcmodule/synaptiks.desktop
+++ synaptiks-0.5.0_pre1182832/kcmodule/synaptiks.desktop
@@ -13,6 +13,7 @@ X-KDE-System-Settings-Parent-Category-V2
 X-KDE-Weight=65
 
 Name=Touchpad
+Name[tr]=Dokunmatik fare
 Name[da]=Touchpad
 Name[en_GB]=Touchpad
 Name[es]=Touchpad
@@ -25,6 +26,7 @@ Name[uk]=Сенсорний пристр
 Name[x-test]=xxTouchpadxx
 
 Comment=Touchpad settings and management
+Comment[tr]=Dokunmatik fare yönetimi ve ayarları
 Comment[da]=Håndtering og indstilling af touchpad
 Comment[en_GB]=Touchpad settings and management
 Comment[es]=Gestión y configuración del touchpad
