Index: synaptiks-0.5.0_pre1182832/daemon/synaptiksdaemon.cpp
===================================================================
--- synaptiks-0.5.0_pre1182832.orig/daemon/synaptiksdaemon.cpp
+++ synaptiks-0.5.0_pre1182832/daemon/synaptiksdaemon.cpp
@@ -121,6 +121,7 @@ SynaptiksDaemon::SynaptiksDaemon(QObject
         // we did not yet know, that touchpad configuration is not supported
         // on this system, so show an error message to the user
         if (!d->config->touchpadNotSupported()) {
+            /*
             if (!Touchpad::isSupported()) {
                 this->notifyError(
                     i18nc("@info daemon error notification",
@@ -132,6 +133,7 @@ SynaptiksDaemon::SynaptiksDaemon(QObject
                           "<warning>No configurable touchpad was "
                           "found.</warning>"));
             }
+            */
             // remember, that touchpad configuration is not supported to
             // avoid needless repetitions of the above error messages
             d->config->setTouchpadNotSupported(true);
