Index: src/thread.c
===================================================================
--- src/thread.c	(revision 1741)
+++ src/thread.c	(working copy)
@@ -445,6 +445,7 @@
             send_command("get_property sub_demux\n", TRUE);
         }
 
+        if (strncmp(vm, "gl2", strlen(vm)) != 0 ) {
         cmd = g_strdup_printf("brightness %i\n", idledata->brightness);
         send_command(cmd, TRUE);
         g_free(cmd);
@@ -460,6 +461,7 @@
         cmd = g_strdup_printf("hue %i\n", idledata->hue);
         send_command(cmd, TRUE);
         g_free(cmd);
+        }
 
         cmd = g_strdup_printf("saturation %i\n", idledata->saturation);
         send_command(cmd, TRUE);
