From 30da860bc64ac237fcd92013b0ca7d400289225e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ozan=20=C3=87a=C4=9Flayan?= <ozan@pardus.org.tr>
Date: Mon, 8 Nov 2010 11:26:51 +0100
Subject: [PATCH 121/122] keyfile: remove unused D-Bus calls from test-keyfile.c

---
 .../plugins/keyfile/tests/test-keyfile.c           |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/system-settings/plugins/keyfile/tests/test-keyfile.c b/system-settings/plugins/keyfile/tests/test-keyfile.c
index 7bf63ef..05131c8 100644
--- a/system-settings/plugins/keyfile/tests/test-keyfile.c
+++ b/system-settings/plugins/keyfile/tests/test-keyfile.c
@@ -27,8 +27,6 @@
 #include <arpa/inet.h>
 #include <sys/socket.h>
 
-#include <dbus/dbus-glib.h>
-
 #include <nm-utils.h>
 #include <nm-setting-connection.h>
 #include <nm-setting-wired.h>
@@ -1989,11 +1987,9 @@ test_write_gsm_connection (void)
 int main (int argc, char **argv)
 {
 	GError *error = NULL;
-	DBusGConnection *bus;
 	char *base;
 
 	g_type_init ();
-	bus = dbus_g_bus_get (DBUS_BUS_SESSION, NULL);
 
 	if (!nm_utils_init (&error))
 		FAIL ("nm-utils-init", "failed to initialize libnm-util: %s", error->message);
@@ -2022,7 +2018,6 @@ int main (int argc, char **argv)
 	base = g_path_get_basename (argv[0]);
 	fprintf (stdout, "%s: SUCCESS\n", base);
 	g_free (base);
-	dbus_g_connection_unref (bus);
 	return 0;
 }
 
-- 
1.7.3.4

