Index: modules/libpref/src/init/all.js
===================================================================
RCS file: /cvsroot/mozilla/modules/libpref/src/init/all.js,v
retrieving revision 3.712
diff -u -r3.712 all.js
--- modules/libpref/src/init/all.js	5 Dec 2007 21:16:33 -0000	3.712
+++ modules/libpref/src/init/all.js	19 Dec 2007 08:07:06 -0000
@@ -822,7 +822,7 @@
 pref("intl.charset.default",                "chrome://navigator-platform/locale/navigator.properties");
 pref("intl.content.langcode",               "chrome://communicator-region/locale/region.properties");
 pref("intl.ellipsis",                       "chrome://global-platform/locale/intl.properties");
-pref("intl.locale.matchOS",                 false);
+pref("intl.locale.matchOS",                 true);
 // fallback charset list for Unicode conversion (converting from Unicode)
 // currently used for mail send only to handle symbol characters (e.g Euro, trademark, smartquotes)
 // for ISO-8859-1
