--- QTfrontend/gameuiconfig.cpp.orig	2010-04-17 09:58:16.486610157 +0300
+++ QTfrontend/gameuiconfig.cpp	2010-04-17 09:58:45.743609644 +0300
@@ -43,8 +43,8 @@
 
     int t = Form->ui.pageOptions->CBResolution->findText(value("video/resolution").toString());
     Form->ui.pageOptions->CBResolution->setCurrentIndex((t < 0) ? 0 : t);
-    Form->ui.pageOptions->CBFullscreen->setChecked(value("video/fullscreen", false).toBool());
-    bool ffscr=value("frontend/fullscreen", false).toBool();
+    Form->ui.pageOptions->CBFullscreen->setChecked(value("video/fullscreen", true).toBool());
+    bool ffscr=value("frontend/fullscreen", true).toBool();
     Form->ui.pageOptions->CBFrontendFullscreen->setChecked(ffscr);
 
     Form->ui.pageOptions->CBReduceQuality->setChecked(value("video/reducequality", false).toBool());
