diff -up cups-1.4.6/backend/runloop.c.usb-buffer-size cups-1.4.6/backend/runloop.c
--- cups-1.4.6/backend/runloop.c.usb-buffer-size	2011-01-10 15:37:56.885677732 +0000
+++ cups-1.4.6/backend/runloop.c	2011-01-10 15:37:59.499623140 +0000
@@ -45,7 +45,7 @@ backendDrainOutput(int print_fd,	/* I - 
   fd_set	input;			/* Input set for reading */
   ssize_t	print_bytes,		/* Print bytes read */
 		bytes;			/* Bytes written */
-  char		print_buffer[8192],	/* Print data buffer */
+  char		print_buffer[512],	/* Print data buffer */
 		*print_ptr;		/* Pointer into print data buffer */
   struct timeval timeout;		/* Timeout for read... */
 
@@ -160,7 +160,7 @@ backendRunLoop(
 		bytes;			/* Bytes written */
   int		paperout;		/* "Paper out" status */
   int		offline;		/* "Off-line" status */
-  char		print_buffer[8192],	/* Print data buffer */
+  char		print_buffer[512],	/* Print data buffer */
 		*print_ptr,		/* Pointer into print data buffer */
 		bc_buffer[1024];	/* Back-channel data buffer */
   struct timeval timeout;		/* Timeout for select() */
diff -up cups-1.4.6/backend/usb.c.usb-buffer-size cups-1.4.6/backend/usb.c
