diff -up cups-1.4b2/backend/serial.c.serial cups-1.4b2/backend/serial.c
--- cups-1.4b2/backend/serial.c.serial	2008-12-16 16:25:02.000000000 +0000
+++ cups-1.4b2/backend/serial.c	2008-12-16 16:25:21.000000000 +0000
@@ -74,6 +74,7 @@
 #endif /* __APPLE__ */
 
 #if defined(__linux) && defined(TIOCGSERIAL)
+#  include <linux/types.h>
 #  include <linux/serial.h>
 #  include <linux/ioctl.h>
 #endif /* __linux && TIOCGSERIAL */
