diff -up cups-1.4.7/config.h.in.avahi-1-config cups-1.4.7/config.h.in
--- cups-1.4.7/config.h.in.avahi-1-config	2011-01-10 07:39:37.000000000 +0000
+++ cups-1.4.7/config.h.in	2011-06-28 12:53:16.860630055 +0100
@@ -345,6 +345,13 @@
 
 
 /*
+ * Do we have Avahi for DNS Service Discovery?
+ */
+
+#undef HAVE_AVAHI
+
+
+/*
  * Do we have <sys/ioctl.h>?
  */
 
diff -up cups-1.4.7/config-scripts/cups-dnssd.m4.avahi-1-config cups-1.4.7/config-scripts/cups-dnssd.m4
--- cups-1.4.7/config-scripts/cups-dnssd.m4.avahi-1-config	2009-08-28 23:54:34.000000000 +0100
+++ cups-1.4.7/config-scripts/cups-dnssd.m4	2011-06-28 12:53:16.860630055 +0100
@@ -27,6 +27,21 @@ AC_ARG_WITH(dnssd-includes, [  --with-dn
 DNSSDLIBS=""
 DNSSD_BACKEND=""
 
+AC_ARG_ENABLE(avahi, [  --enable-avahi          turn on DNS Service Discovery support, default=no],
+	      [if test x$enable_avahi = xyes; then
+		       AC_MSG_CHECKING(for Avahi)
+		       if $PKGCONFIG --exists avahi-client; then
+			       AC_MSG_RESULT(yes)
+			       CFLAGS="$CFLAGS `$PKGCONFIG --cflags avahi-client`"
+			       DNSSDLIBS="`$PKGCONFIG --libs avahi-client`"
+			       DNSSD_BACKEND="dnssd"
+			       AC_DEFINE(HAVE_AVAHI)
+			       enable_dnssd=no
+		       else
+			       AC_MSG_RESULT(no)
+		       fi
+	       fi])
+
 if test x$enable_dnssd != xno; then
 	AC_CHECK_HEADER(dns_sd.h, [
 		case "$uname" in
