diff -up cups-1.3.5/cups/ppd.c~ cups-1.3.5/cups/ppd.c
--- cups-1.3.5/cups/ppd.c~	2007-11-30 19:29:50.000000000 +0000
+++ cups-1.3.5/cups/ppd.c	2008-01-09 12:08:48.000000000 +0000
@@ -2801,7 +2801,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	*lineptr++ = ch;
 	col ++;
 
-	if (col > (PPD_MAX_LINE - 1))
+	if (col > (PPD_MAX_LINE - 1) && cg->ppd_conform == PPD_CONFORM_STRICT)
 	{
 	 /*
           * Line is too long...
@@ -2868,7 +2868,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	{
 	  col ++;
 
-	  if (col > (PPD_MAX_LINE - 1))
+	  if (col > (PPD_MAX_LINE - 1) && cg->ppd_conform == PPD_CONFORM_STRICT)
 	  {
 	   /*
             * Line is too long...
@@ -2931,7 +2931,7 @@ ppd_read(cups_file_t    *fp,		/* I - Fil
 	{
 	  col ++;
 
-	  if (col > (PPD_MAX_LINE - 1))
+	  if (col > (PPD_MAX_LINE - 1) && cg->ppd_conform == PPD_CONFORM_STRICT)
 	  {
 	   /*
             * Line is too long...
