--- src/libmdb/data.c
+++ src/libmdb/data.c	2007/04/10 11:19:10
@@ -231,7 +231,7 @@
 					col->col_type, len);
 			}
 			strcpy(col->bind_ptr, str);
-			g_free(str);
+			//g_free(str);
 		}
 		ret = strlen(col->bind_ptr);
 		if (col->len_ptr) {
@@ -679,7 +679,7 @@
 			fprintf(stderr, "Warning: incorrect memo length\n");
 		}
 		mdb_unicode2ascii(mdb, tmp, tmpoff, text, MDB_BIND_SIZE);
-		g_free(tmp);
+		//g_free(tmp);
 		return text;
 	} else {
 		fprintf(stderr, "Unhandled memo field flags = %02x\n", memo_len >> 24);
