diff -Naur liblastfm-0.3.0.orig/admin/lastfm.h.rb liblastfm-0.3.0/admin/lastfm.h.rb
--- liblastfm-0.3.0.orig/admin/lastfm.h.rb	2010-05-15 13:53:35.334111813 +0300
+++ liblastfm-0.3.0/admin/lastfm.h.rb	2010-05-15 13:55:03.416237174 +0300
@@ -1,6 +1,6 @@
 #!/usr/bin/ruby
 
 f = File.new(ARGV[0], "w")
-Dir["_include/lastfm/*"].each do |h| 
+Dir["_include/lastfm/*"].sort.each do |h| 
 	f.write %Q{#include "lastfm/#{File.basename h}"\n}
 end
