Index: mxcl-liblastfm-1c739eb/admin/lastfm.h.rb
===================================================================
--- mxcl-liblastfm-1c739eb.orig/admin/lastfm.h.rb
+++ mxcl-liblastfm-1c739eb/admin/lastfm.h.rb
@@ -1,5 +1,5 @@
 #!/usr/bin/ruby
 f = File.new(ARGV[0], "w")
-Dir["_include/lastfm/*"].each do |h| 
+Dir["_include/lastfm/*"].sort.each do |h| 
 	f.write %Q{#include "lastfm/#{File.basename h}"\n}
-end
\ No newline at end of file
+end
