--- ntop-4.0.1/Makefile.am	2010-10-22 09:07:57.755355672 +0300
+++ ntop-4.0.1/Makefile.am	2010-10-22 09:08:11.255105600 +0300
@@ -336,7 +336,7 @@
 install-selinux-policy:
 	chcon -t textrel_shlib_t $(DESTDIR)$(libdir)/*ntop*.so
 
-install-data-local: ntop.txt ntop.html faq.html dnetter
+install-data-local: ntop.txt ntop.html faq.html
 	@$(top_srcdir)/mkinstalldirs $(DESTDIR)/$(libdir)/plugins;
 	@if ! test -d $(DESTDIR)/$(datadir)/ntop; then \
 	  $(top_srcdir)/mkinstalldirs $(DESTDIR)/$(datadir)/ntop; \
