Index: linphone-3.2.1/share/linphone_applet.desktop
===================================================================
--- linphone-3.2.1.orig/share/linphone_applet.desktop
+++ linphone-3.2.1/share/linphone_applet.desktop
@@ -1,9 +1,12 @@
 [Desktop Entry]
 Name=Linphone Applet
+GenericName=VoIP Applet
+GenericName[tr]=VoIP Programcığı
 Comment=Linphone is a web-phone
 Comment[fr]=Linphone est un web-phone.
 Comment[de]=Linphone ist ein web-phone.
+Comment[tr]=Linphone bir ağ telefonudur.
 Type=Application
 Exec=linphone --applet --activate-goad-server=linphone_applet
 Icon=linphone/linphone2.png
-Terminal=0
\ Dosya sonunda yenisatır yok.
+Terminal=0
Index: linphone-3.2.1/share/linphone.desktop
===================================================================
--- linphone-3.2.1.orig/share/linphone.desktop
+++ linphone-3.2.1/share/linphone.desktop
@@ -1,10 +1,13 @@
 [Desktop Entry]
 Name=Linphone
+GenericName=VoIP application
+GenericName[tr]=VoIP uygulaması
 Comment=Linphone is a web-phone
 Comment[fr]=Linphone est un web-phone.
 Comment[de]=Linphone ist ein web-phone.
+Comment[tr]=Linphone bir ağ telefonudur.
 Type=Application
 Exec=linphone-3
 Icon=linphone/linphone2.png
 Terminal=false
-Categories=Application;Network;
\ Dosya sonunda yenisatır yok.
+Categories=Application;Network;
