diff --git a/configure.in b/configure.in
index 3cb7bf7..307679b 100644
--- a/configure.in
+++ b/configure.in
@@ -275,7 +275,6 @@ AC_ARG_ENABLE(strict,
 
 if test "$GCC$strictness" = "yesyes" ; then
 	STRICT_OPTIONS="-Wall -Wp,-D_FORTIFY_SOURCE=2"
-	STRICT_OPTIONS="$STRICT_OPTIONS -Werror"
 	CFLAGS="$CFLAGS -fno-strict-aliasing"
 fi
 
diff --git a/mediastreamer2/configure.ac b/mediastreamer2/configure.ac
index f663b56..18c1e30 100644
--- a/mediastreamer2/configure.ac
+++ b/mediastreamer2/configure.ac
@@ -109,10 +109,6 @@ if test x$ac_cv_c_bigendian = xyes ; then
 	CFLAGS="$CFLAGS -D_BIGENDIAN "
 fi
 
-if test $GCC = yes && test $wall_werror = yes;  then
-	CFLAGS="$CFLAGS -Werror "
-fi
-
 macosx_found=no
 
 dnl add thread flags
