--- mediastreamer2/tests/mediastream.c.str	2009-01-30 14:29:25.000000000 +0100
+++ mediastreamer2/tests/mediastream.c	2009-04-27 04:53:49.000000000 +0200
@@ -154,7 +154,7 @@
 	vs.width=MS_VIDEO_SIZE_CIF_W;
 	vs.height=MS_VIDEO_SIZE_CIF_H;
 	if (argc<4) {
-		printf(usage);
+		printf("%s", usage);
 		return -1;
 	}
 	for (i=1;i<argc;i++){
@@ -164,7 +164,7 @@
 		}else if (strcmp(argv[i],"--remote")==0){
 			i++;
 			if (!parse_addr(argv[i],ip,sizeof(ip),&remoteport)) {
-				printf(usage);
+				printf("%s", usage);
 				return -1;
 			}
 			printf("Remote addr: ip=%s port=%i\n",ip,remoteport);
