diff -up celestia-1.6.0/src/celengine/mesh.cpp~ celestia-1.6.0/src/celengine/mesh.cpp
--- celestia-1.6.0/src/celengine/mesh.cpp~	2010-12-08 12:25:06.000000000 +0100
+++ celestia-1.6.0/src/celengine/mesh.cpp	2010-12-08 12:25:23.684401003 +0100
@@ -7,13 +7,13 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <cassert>
+#include <iostream>
+#include <algorithm>
 #include "mesh.h"
 #include "rendcontext.h"
 #include "gl.h"
 #include "glext.h"
-#include <cassert>
-#include <iostream>
-#include <algorithm>
 
 using namespace std;
 
diff -up celestia-1.6.0/src/celengine/model.cpp~ celestia-1.6.0/src/celengine/model.cpp
--- celestia-1.6.0/src/celengine/model.cpp~	2009-06-22 17:44:08.000000000 +0200
+++ celestia-1.6.0/src/celengine/model.cpp	2010-12-08 12:25:46.562401008 +0100
@@ -7,11 +7,11 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
-#include "model.h"
-#include "rendcontext.h"
 #include <cassert>
 #include <functional>
 #include <algorithm>
+#include "model.h"
+#include "rendcontext.h"
 
 using namespace std;
 
diff -up celestia-1.6.0/src/celestia/celx_object.cpp~ celestia-1.6.0/src/celestia/celx_object.cpp
--- celestia-1.6.0/src/celestia/celx_object.cpp~	2009-06-22 17:44:22.000000000 +0200
+++ celestia-1.6.0/src/celestia/celx_object.cpp	2010-12-08 12:33:02.902401126 +0100
@@ -773,7 +773,7 @@ static int object_mark(lua_State* l)
         markAlpha = 1.0f;
     
     Color markColorAlpha(0.0f, 1.0f, 0.0f, 0.9f);
-    markColorAlpha = Color::Color(markColor, markAlpha);
+    markColorAlpha = Color(markColor, markAlpha);
     
     const char* markLabel = celx.safeGetString(6, WrongType, "Fifth argument to object:mark must be a string");
     if (markLabel == NULL)
