Index: kdeedu-4.3.5/parley/plugins/wiktionary/mwclient/__init__.py
===================================================================
--- kdeedu-4.3.5.orig/parley/plugins/wiktionary/mwclient/__init__.py
+++ kdeedu-4.3.5/parley/plugins/wiktionary/mwclient/__init__.py
@@ -26,8 +26,8 @@
 import sys as _sys
 import os as _os
 _path = _os.path.dirname(__file__)
-_sys.path.append(_os.path.abspath(_path))
+_sys.path.insert(0, _os.path.abspath(_path))
 
 from errors import *
 from client import Site, __ver__
-import ex
\ Dosya sonunda yenisatır yok.
+import ex
