--- kstars/kstars/kstars.kcfg.orig	2009-12-04 11:40:20.698608243 +0200
+++ kstars/kstars/kstars.kcfg	2009-12-04 11:40:46.897516151 +0200
@@ -835,7 +835,7 @@
                 <entry name="XplanetPath" type="String">
 			<label>Path to xplanet binary</label>
 			<whatsthis>Xplanet binary path</whatsthis>
-			<default></default>
+            <default>/usr/bin/xplanet</default>
 		</entry>
 		<entry name="XplanetTitle" type="String">
 			<label>Window title</label>
