diff -up logrotate-3.7.7/config.c.curdir logrotate-3.7.7/config.c
--- logrotate-3.7.7/config.c.curdir	2008-05-09 09:28:59.000000000 +0200
+++ logrotate-3.7.7/config.c	2008-11-20 15:35:05.000000000 +0100
@@ -316,11 +316,6 @@ static int readConfigPath(const char *pa
 	DIR *dirp;
 
 	here = open(".", O_RDONLY);
-	if (here < 0) {
-	    message(MESS_ERROR, "cannot open current directory: %s\n",
-		    strerror(errno));
-	    return 1;
-	}
 
 	if ((dirp = opendir(path)) == NULL) {
 	    message(MESS_ERROR, "cannot open directory %s: %s\n", path,
