Index: everything-websearch-0.0.1_49539/src/e_mod_main.c
===================================================================
--- everything-websearch-0.0.1_49539.orig/src/e_mod_main.c
+++ everything-websearch-0.0.1_49539/src/e_mod_main.c
@@ -1453,7 +1453,7 @@ _conf_new(void)
    IFMODCFGEND;
 
    IFMODCFG(0x009d);
-   _conf->translate = eina_stringshare_add("en|de");
+   _conf->translate = eina_stringshare_add("en|tr");
    IFMODCFGEND;
 
    IFMODCFG(0x00dd);
