Index: everything-websearch-0.0.1_49539/src/e_mod_main.c
===================================================================
--- everything-websearch-0.0.1_49539.orig/src/e_mod_main.c
+++ everything-websearch-0.0.1_49539/src/e_mod_main.c
@@ -1448,7 +1448,7 @@ _conf_new(void)
 
    /* setup defaults */
    IFMODCFG(0x008d);
-   _conf->browser = eina_stringshare_add("firefox");
+   _conf->browser = eina_stringshare_add("xdg-open");
    _conf->lang = eina_stringshare_add("en");
    IFMODCFGEND;
 
