diff -Naur autotrace-0.31.1.orig/configure.in autotrace-0.31.1/configure.in
--- autotrace-0.31.1.orig/configure.in	2002-11-08 18:49:23.000000000 +0100
+++ autotrace-0.31.1/configure.in	2004-11-07 12:59:31.750331496 +0100
@@ -79,11 +79,11 @@
 			| sed -e 's/^[0-9]*.[0-9]*.\([0-9]*\)$/\1/'`
 			changequote([,])dnl
 			MAGICK_MAJOR_VERSION_CHECK=`expr $MAGICK_MAJOR_VERSION \>= 5`
-			MAGICK_MINOR_VERSION_CHECK=`expr $MAGICK_MINOR_VERSION \>= 2`
+			MAGICK_MINOR_VERSION_CHECK=`expr $MAGICK_MINOR_VERSION \<= 2`
 			dnl I don't consider ImageMagick 6.x.
 			if test "$MAGICK_VERSION" = "5.2.0" \
 			   || test "$MAGICK_MAJOR_VERSION_CHECK" = 0 \
-			   || test "$MAGICK_MINOR_VERSION_CHECK" = 0 ; then
+			   && test "$MAGICK_MINOR_VERSION_CHECK" = 0 ; then
 			   AC_MSG_RESULT(no($MAGICK_VERSION))
 		   	   AC_MSG_WARN([*** Version 5.2.1 or higher is required.             ***])
 			   AC_MSG_WARN([*** ImageMagick input handler is disabled.           ***])
