diff -Nur etherape-0.9.9-old/etherape.desktop.in etherape-0.9.9/etherape.desktop.in
--- etherape-0.9.9-old/etherape.desktop.in	2010-09-11 17:51:08.631208758 +0300
+++ etherape-0.9.9/etherape.desktop.in	2010-09-11 17:52:41.460093970 +0300
@@ -2,10 +2,12 @@
 Version=1.0
 Encoding=UTF-8
 Name=EtherApe
+Name[tr]=EtherApe
 Comment=Graphical Network Monitor
 Comment[es]=Monitor Gráfico de Red
-Exec=etherape
+Comment[tr]=Grafiksel Ağ İzleyici
+Exec=xdg-su -c "etherape"
 Terminal=false
 Type=Application
-Icon=etherape.png
-Categories=GNOME;System;Network;
+Icon=etherape
+Categories=System;Monitor;Network;
