--- shadow-4.1.0.orig/src/login.c	2007-12-09 01:27:30.000000000 +0200
+++ shadow-4.1.0/src/login.c	2008-03-12 11:04:50.000000000 +0200
@@ -667,7 +667,7 @@
 			    failent_user = "UNKNOWN";
 			  }
 
-			  if (retcode == PAM_MAXTRIES || failcount >= retries) {
+			  if (retcode == PAM_MAXTRIES || retcode != PAM_SUCCESS && failcount >= retries) {
 			    SYSLOG ((LOG_NOTICE,
 				    "TOO MANY LOGIN TRIES (%d)%s FOR `%s'",
 				    failcount, fromhost, failent_user));
