From 4cd751bcc48798f19b089f6cff58bfca2f18fe36 Mon Sep 17 00:00:00 2001
From: Brian Paul <brianp@vmware.com>
Date: Tue, 14 Sep 2010 11:01:03 -0600
Subject: [PATCH] glx: add const qualifiers to __indirect_glMultiDrawArraysEXT()

---
 src/glx/indirect_vertex_array.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/glx/indirect_vertex_array.c b/src/glx/indirect_vertex_array.c
index 372618d..115f755 100644
--- a/src/glx/indirect_vertex_array.c
+++ b/src/glx/indirect_vertex_array.c
@@ -1034,8 +1034,8 @@ __indirect_glDrawRangeElements(GLenum mode, GLuint start, GLuint end,
 
 
 void
-__indirect_glMultiDrawArraysEXT(GLenum mode, GLint * first, GLsizei * count,
-                                GLsizei primcount)
+__indirect_glMultiDrawArraysEXT(GLenum mode, const GLint *first,
+                                const GLsizei *count, GLsizei primcount)
 {
    struct glx_context *gc = __glXGetCurrentContext();
    const __GLXattribute *state =
-- 
1.7.2.2

