commit 37eda947bd8181a73ad0fffc88e66c25ddd69f28
Author: Rick W. Chen <stuffcorpse@archlinux.us>
Date:   Thu Mar 17 18:38:19 2011 +1300

    Fix plugin loading and bump_plugin_version.rb script

diff --git a/release_scripts/bump_plugin_version.rb b/release_scripts/bump_plugin_version.rb
index 9f0c71a..d2194a2 100755
diff --git a/src/PluginManager.cpp b/src/PluginManager.cpp
index ff52646..1e34510 100644
--- a/src/PluginManager.cpp
+++ b/src/PluginManager.cpp
@@ -32,7 +32,7 @@
 #include <QFile>
 #include <QMetaEnum>
 
-const int Plugins::PluginManager::s_pluginFrameworkVersion = 60;
+const int Plugins::PluginManager::s_pluginFrameworkVersion = 61;
 Plugins::PluginManager* Plugins::PluginManager::s_instance = 0;
 
 Plugins::PluginManager*
