--- src/amarok_append.desktop.orig	2009-09-21 13:49:01.000000000 +0300
+++ src/amarok_append.desktop	2009-09-21 13:50:50.000000000 +0300
@@ -167,7 +167,8 @@
 Name[zh_CN]=追加并播放
 Name[zh_TW]=添加並播放
 Icon=amarok
-Exec=qdbus org.kde.amarok /PlayerList playMedia %U
+#Exec=qdbus org.kde.amarok /PlayerList playMedia %U
+Exec=amarok -a -p %U
 
 [Desktop Action queueTrack]
 Name=Queue Track
@@ -224,4 +225,4 @@
 Name[zh_CN]=音轨排队
 Name[zh_TW]=佇列曲目
 Icon=amarok
-Exec=amarok --queue %U
+Exec=amarok --queue -a -p %U
