Index: kdebase-workspace-4.6.1/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
===================================================================
--- kdebase-workspace-4.6.1.orig/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
+++ kdebase-workspace-4.6.1/plasma/desktop/shell/data/layouts/org.kde.plasma-desktop.defaultPanel/contents/layout.js
@@ -5,7 +5,8 @@ if (panelIds.length == 1) {
 }
 
 panel.height = 30
-panel.addWidget("launcher")
+var launcher = panel.addWidget("launcher")
+launcher.globalShortcut = "Alt+F1";
 panel.addWidget("showdesktop")
 panel.addWidget("pager")
 panel.addWidget("tasks")
