Index: Makefile.in
===================================================================
--- Makefile.in.orig
+++ Makefile.in
@@ -737,7 +737,7 @@ EXTRA_DIST = COPYING ChangeLog TODO conf
 DISTCLEANFILES = tool-versions.txt
 
 # "tests" is the gnulib unit test dir.
-SUBDIRS = gnulib tests build-aux lib find xargs locate doc po m4
+SUBDIRS = gnulib tests build-aux lib find xargs doc po m4
 ACLOCAL_AMFLAGS = -I gnulib/m4 -I m4
 TESTFILE_SUFFIXES = .exp .xo .xe .xi
 all: config.h
