--- diffutils-2.8.1/src/analyze.c.badc	2004-01-08 10:05:59.094598972 +0000
+++ diffutils-2.8.1/src/analyze.c	2004-01-08 10:06:22.294841900 +0000
@@ -983,7 +983,7 @@
 	changes = briefly_report (changes, cmp->file);
       else
 	{
-	  if (changes | !no_diff_means_no_output)
+	  if (changes || !no_diff_means_no_output)
 	    {
 	      /* Record info for starting up output,
 		 to be used if and when we have some output to print.  */
