--- script/scons
+++ script/scons
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python
 #
 # SCons - a Software Constructor
 #
--- script/scons-time
+++ script/scons-time
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python
 #
 # scons-time - run SCons timings and collect statistics
 #
--- script/sconsign
+++ script/sconsign
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/bin/python
 #
 # SCons - a Software Constructor
 #
