diff -aup eric4-4.3.0/eric/UI/UserInterface.py.orig eric4-4.3.0/eric/UI/UserInterface.py
--- eric4-4.3.0/eric/UI/UserInterface.py.orig	2009-02-09 15:38:51.000000000 +0100
+++ eric4-4.3.0/eric/UI/UserInterface.py	2009-02-09 15:42:47.000000000 +0100
@@ -1583,6 +1583,7 @@ class UserInterface(KQMainWindow):
             """<b>Check for Updates...</b>"""
             """<p>Checks the internet for updates of eric4.</p>"""
                              ))
+        self.checkUpdateAct.setVisible(False)
         self.connect(self.checkUpdateAct, SIGNAL('triggered()'), self.performVersionCheck)
         self.actions.append(self.checkUpdateAct)
     
@@ -1596,6 +1597,7 @@ class UserInterface(KQMainWindow):
             """<p>Shows the eric4 versions available for download """
             """from the internet.</p>"""
                              ))
+        self.showVersionsAct.setVisible(False)
         self.connect(self.showVersionsAct, SIGNAL('triggered()'), 
                 self.showAvailableVersionsInfo)
         self.actions.append(self.showVersionsAct)
