diff -Naur libcdio-0.78.2-orig/include/cdio++/iso9660.hpp libcdio-0.78.2/include/cdio++/iso9660.hpp
--- libcdio-0.78.2-orig/include/cdio++/iso9660.hpp	2006-03-07 15:04:01.000000000 -0600
+++ libcdio-0.78.2/include/cdio++/iso9660.hpp	2007-11-18 19:41:03.000000000 -0600
@@ -30,6 +30,7 @@
 #include <cdio/iso9660.h>
 #include <cdio++/cdio.hpp>
 #include <string.h>
+#include <cstdlib>
 #include <vector>		// vector class library
 using namespace std;
 
