--- coreutils-6.7/src/dircolors.hin
+++ coreutils-6.7/src/dircolors.hin
@@ -5,6 +5,9 @@
 
 # The keywords COLOR, OPTIONS, and EIGHTBIT (honored by the
 # slackware version of dircolors) are recognized but ignored.
+
+# You can copy this file to .dir_colors in your $HOME directory to override
+# the system defaults.
 
 # Below, there should be one TERM entry for each termtype that is colorizable
 TERM Eterm
@@ -66,7 +66,8 @@
 DOOR 01;35	# door
 BLK 40;33;01	# block device driver
 CHR 40;33;01	# character device driver
-ORPHAN 40;31;01 # symlink to nonexistent file, or non-stat'able file
+ORPHAN 01;05;37;41  # orphaned symlinks
+MISSING 01;05;37;41 # ... and the files they point to
 SETUID 37;41	# file that is setuid (u+s)
 SETGID 30;43	# file that is setgid (g+s)
 STICKY_OTHER_WRITABLE 30;42 # dir that is sticky and other-writable (+t,o+w)
@@ -125,6 +154,16 @@
 .xwd 01;35
 .yuv 01;35
 
+# Document files
+.pdf 00;32
+.ps 00;32
+.txt 00;32
+.patch 00;32
+.diff 00;32
+.log 00;32
+.tex 00;32
+.doc 00;32
+
 # audio formats
 .aac 00;36
 .au 00;36
