diff -Nur coreutils-7.6-old/configure.ac coreutils-7.6/configure.ac
--- coreutils-7.6-old/configure.ac	2009-09-15 20:13:31.407290457 +0300
+++ coreutils-7.6/configure.ac	2009-09-15 20:14:13.774041835 +0300
@@ -33,7 +33,7 @@
 AC_CONFIG_HEADERS([lib/config.h:lib/config.hin])
 
 AM_INIT_AUTOMAKE([1.11 dist-xz color-tests parallel-tests])
-AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
+#AM_SILENT_RULES([yes]) # make --enable-silent-rules the default.
 
 AC_PROG_CC_STDC
 AM_PROG_CC_C_O
