diff -Nur coreutils-8.7-old/src/dircolors.hin coreutils-8.7/src/dircolors.hin
--- coreutils-8.7-old/src/dircolors.hin	2011-07-11 12:23:55.290623111 +0300
+++ coreutils-8.7/src/dircolors.hin	2011-07-11 12:22:43.804682205 +0300
@@ -202,23 +202,23 @@
 .ogx 01;35
 
 # audio formats (cyan)
-.aac 01;36
-.au 01;36
-.flac 01;36
-.mid 01;36
-.midi 01;36
-.mka 01;36
-.mp3 01;36
-.mpc 01;36
-.ogg 01;36
-.ra 01;36
-.wav 01;36
+.aac 00;36
+.au 00;36
+.flac 00;36
+.mid 00;36
+.midi 00;36
+.mka 00;36
+.mp3 00;36
+.mpc 00;36
+.ogg 00;36
+.ra 00;36
+.wav 00;36
 
 # http://wiki.xiph.org/index.php/MIME_Types_and_File_Extensions
-.axa 01;36
-.oga 01;36
-.spx 01;36
-.xspf 01;36
+.axa 00;36
+.oga 00;36
+.spx 00;36
+.xspf 00;36
 
 # document formats
 .pdf 00;32
