diff -ru wordwarvi-0.26.orig/wwviaudio.c wordwarvi-0.26/wwviaudio.c
--- wwviaudio.c	2009-08-09 10:33:02.000000000 -0400
+++ wwviaudio.c	2010-02-08 17:45:07.127666854 -0500
@@ -124,7 +124,7 @@
 	strncpy(filebuf, filename, PATH_MAX);
 	rc = stat(filebuf, &statbuf);
 	if (rc != 0) {
-		snprintf(filebuf, PATH_MAX, "%s", filename);
+		snprintf(filebuf, PATH_MAX, "/usr/share/wordwarvi/%s", filename);
 		rc = stat(filebuf, &statbuf);
 		if (rc != 0) {
 			fprintf(stderr, "stat('%s') failed.\n", filebuf);
