diff -up ./gettext-tools/gnulib-tests/Makefile.gnulib.rpathFix~ ./gettext-tools/gnulib-tests/Makefile.gnulib
--- ./gettext-tools/gnulib-tests/Makefile.gnulib.rpathFix~	2007-10-28 01:39:18.000000000 +1000
+++ ./gettext-tools/gnulib-tests/Makefile.gnulib	2008-08-28 16:43:18.000000000 +1000
@@ -16,7 +16,7 @@ ACLOCAL_AMFLAGS = -I ../gnulib-m4
 
 SUBDIRS =
 TESTS =
-TESTS_ENVIRONMENT =
+TESTS_ENVIRONMENT = LD_LIBRARY_PATH='../intl/.libs:../src/.libs:../.libs:../../../../../lib:../../../../../lib64'
 noinst_PROGRAMS =
 check_PROGRAMS =
 noinst_HEADERS =
diff -up ./gettext-tools/src/Makefile.am.rpathFix~ ./gettext-tools/src/Makefile.am
--- ./gettext-tools/src/Makefile.am.rpathFix~	2007-10-08 05:37:38.000000000 +1000
+++ ./gettext-tools/src/Makefile.am	2008-08-28 17:16:45.000000000 +1000
@@ -62,6 +62,7 @@ projectsdir = $(pkgdatadir)/projects
 pkglibdir = $(libdir)/gettext
 
 AM_CPPFLAGS = \
+  -I../../../../../../usr/include \
   -I. -I$(srcdir) \
   -I.. -I$(top_srcdir) \
   -I$(top_srcdir)/libgrep \
@@ -82,7 +83,7 @@ LDADD = ../gnulib-lib/libgettextlib.la @
 
 SED = sed
 YACC = @YACC@ -d
-GCJ = @GCJ@
+GCJ = gcj
 GCJFLAGS = @GCJFLAGS@
 JAR = @JAR@
 JAVACOMP = $(SHELL) ../javacomp.sh
diff -up ./gettext-tools/tests/Makefile.am.rpathFix~ ./gettext-tools/tests/Makefile.am
--- ./gettext-tools/tests/Makefile.am.rpathFix~	2007-10-21 07:54:40.000000000 +1000
+++ ./gettext-tools/tests/Makefile.am	2008-08-28 16:43:18.000000000 +1000
@@ -181,6 +181,7 @@ TESTS_ENVIRONMENT = top_srcdir=$(top_src
 		    LOCALE_JA='@LOCALE_JA@' \
 		    host_os='@host_os@' \
 		    CONFIG_SHELL='$(SHELL)' \
+		    LD_LIBRARY_PATH='../intl/.libs:../src/.libs:../.libs:../../../../../lib:../../../../../lib64' \
 		    $(SHELL)
 
 xg-c-1.ok.po: $(top_srcdir)/src/xgettext.c $(top_srcdir)/src/msgfmt.c
diff -up ./m4/libtool.m4.rpathFix~ ./m4/libtool.m4
--- ./m4/libtool.m4.rpathFix~	2007-10-27 10:46:10.000000000 +1000
+++ ./m4/libtool.m4	2008-08-28 16:43:18.000000000 +1000
@@ -1616,7 +1616,7 @@ linux* | k*bsd*-gnu)
   # This implies no fast_install, which is unacceptable.
   # Some rework will be needed to allow for fast_install
   # before this can be enabled.
-  hardcode_into_libs=yes
+  #hardcode_into_libs=yes
 
   # Append ld.so.conf contents to the search path
   if test -f /etc/ld.so.conf; then
@@ -2872,7 +2872,8 @@ if test "$GXX" = yes; then
     _LT_AC_TAGVAR(archive_cmds, $1)='$CC -shared -nostdlib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags ${wl}-soname $wl$soname -o $lib'
     _LT_AC_TAGVAR(archive_expsym_cmds, $1)='$CC -shared -nostdlib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags ${wl}-soname $wl$soname ${wl}-retain-symbols-file $wl$export_symbols -o $lib'
 
-    _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}--rpath ${wl}$libdir'
+#    _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}--rpath ${wl}$libdir'
+    _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
     _LT_AC_TAGVAR(export_dynamic_flag_spec, $1)='${wl}--export-dynamic'
 
     # If archive_cmds runs LD, not CC, wlarc should be empty
@@ -3340,7 +3341,8 @@ case $host_os in
 	# dependencies.
 	output_verbose_link_cmd='templist=`$CC $CFLAGS -v conftest.$objext -o libconftest$shared_ext 2>&1 | grep "ld"`; rm -f libconftest$shared_ext; list=""; for z in $templist; do case $z in conftest.$objext) list="$list $z";; *.$objext);; *) list="$list $z";;esac; done; echo $list'
 
-	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}--rpath,$libdir'
+#	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}--rpath,$libdir'
+	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
 	_LT_AC_TAGVAR(export_dynamic_flag_spec, $1)='${wl}--export-dynamic'
 
 	# Archives containing C++ object files must be created using
@@ -3368,7 +3370,8 @@ case $host_os in
 	  ;;
 	esac
 	_LT_AC_TAGVAR(archive_cmds_need_lc, $1)=no
-	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-rpath,$libdir'
+#	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-rpath,$libdir'
+	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
 	_LT_AC_TAGVAR(export_dynamic_flag_spec, $1)='${wl}--export-dynamic'
 	_LT_AC_TAGVAR(whole_archive_flag_spec, $1)='${wl}--whole-archive$convenience ${wl}--no-whole-archive'
 	;;
@@ -3377,7 +3380,8 @@ case $host_os in
 	_LT_AC_TAGVAR(archive_cmds, $1)='$CC -shared $pic_flag $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags ${wl}-soname ${wl}$soname -o $lib'
   	_LT_AC_TAGVAR(archive_expsym_cmds, $1)='$CC -shared $pic_flag $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags ${wl}-soname ${wl}$soname ${wl}-retain-symbols-file ${wl}$export_symbols -o $lib'
 
-	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}--rpath ${wl}$libdir'
+#	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}--rpath ${wl}$libdir'
+	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
 	_LT_AC_TAGVAR(export_dynamic_flag_spec, $1)='${wl}--export-dynamic'
 	_LT_AC_TAGVAR(whole_archive_flag_spec, $1)='${wl}--whole-archive`for conv in $convenience\"\"; do test  -n \"$conv\" && new_convenience=\"$new_convenience,$conv\"; done; $echo \"$new_convenience\"` ${wl}--no-whole-archive'
         ;;
@@ -3387,7 +3391,8 @@ case $host_os in
 	_LT_AC_TAGVAR(archive_expsym_cmds, $1)='$CC -shared $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags ${wl}-soname $wl$soname  -o $lib ${wl}-retain-symbols-file $wl$export_symbols'
 
 	runpath_var=LD_RUN_PATH
-	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-rpath $libdir'
+#	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-rpath $libdir'
+	_LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
 	_LT_AC_TAGVAR(hardcode_libdir_separator, $1)=:
 
 	# Commands to make compiler produce verbose output that lists
@@ -3584,8 +3589,9 @@ case $host_os in
 	  _LT_AC_TAGVAR(allow_undefined_flag, $1)=' ${wl}-expect_unresolved ${wl}\*'
 	 _LT_AC_TAGVAR(archive_cmds, $1)='$CC -shared -nostdlib ${allow_undefined_flag} $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags ${wl}-msym ${wl}-soname ${wl}$soname `test -n "$verstring" && echo ${wl}-set_version ${wl}$verstring` ${wl}-update_registry ${wl}${output_objdir}/so_locations -o $lib'
 
-	  _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-rpath ${wl}$libdir'
-	  _LT_AC_TAGVAR(hardcode_libdir_separator, $1)=:
+#	  _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}-rpath ${wl}$libdir'
+	 _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
+	 _LT_AC_TAGVAR(hardcode_libdir_separator, $1)=:
 
 	  # Commands to make compiler produce verbose output that lists
 	  # what "hidden" libraries, object files and flags are used when
@@ -5582,7 +5588,8 @@ ifelse([$1],[CXX],[
     # are reset later if shared libraries are not supported. Putting them
     # here allows them to be overridden if necessary.
     runpath_var=LD_RUN_PATH
-    _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}--rpath ${wl}$libdir'
+#    _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='${wl}--rpath ${wl}$libdir'
+    _LT_AC_TAGVAR(hardcode_libdir_flag_spec, $1)='-L$libdir'
     _LT_AC_TAGVAR(export_dynamic_flag_spec, $1)='${wl}--export-dynamic'
     # ancient GNU ld didn't support --whole-archive et. al.
     if $LD --help 2>&1 | grep 'no-whole-archive' > /dev/null; then
