Index: xf86-video-intel-2.12.902/src/intel_display.c
===================================================================
--- xf86-video-intel-2.12.902.orig/src/intel_display.c
+++ xf86-video-intel-2.12.902/src/intel_display.c
@@ -719,6 +719,11 @@ intel_output_mode_valid(xf86OutputPtr ou
 			return MODE_PANEL;
 	}
 
+	/* I don't actually believe any VGA is this good */
+	if (intel_output->mode_output->connector_type == DRM_MODE_CONNECTOR_VGA)
+		if (pModes->Clock > 250000)
+			return MODE_CLOCK_HIGH;
+
 	return MODE_OK;
 }
 
