Index: xf86-video-intel-2.12.902/src/intel_display.c
===================================================================
--- xf86-video-intel-2.12.902.orig/src/intel_display.c
+++ xf86-video-intel-2.12.902/src/intel_display.c
@@ -1519,7 +1519,7 @@ Bool intel_mode_pre_init(ScrnInfoPtr scr
 	intel_screen_private *intel = intel_get_screen_private(scrn);
 	struct drm_i915_getparam gp;
 	struct intel_mode *mode;
-	unsigned int i;
+	unsigned int i, lvds = -1;
 	int has_flipping;
 
 	mode = calloc(1, sizeof *mode);
@@ -1546,9 +1546,22 @@ Bool intel_mode_pre_init(ScrnInfoPtr scr
 			     mode->mode_res->max_height);
 	for (i = 0; i < mode->mode_res->count_crtcs; i++)
 		intel_crtc_init(scrn, mode, i);
+ 
+        /* do LVDS first */
+        for (i = 0; i < mode->mode_res->count_connectors; i++) {
+                drmModeConnectorPtr koutput =
+                    drmModeGetConnector(mode->fd,
+                                        mode->mode_res->connectors[i]);
+                if (koutput->connector_type == DRM_MODE_CONNECTOR_LVDS) {
+                        intel_output_init(scrn, mode, i);
+                        lvds = i;
+                }
+                drmModeFreeConnector(koutput);
+        }
 
 	for (i = 0; i < mode->mode_res->count_connectors; i++)
-		intel_output_init(scrn, mode, i);
+                if (i != lvds)
+		        intel_output_init(scrn, mode, i);
 
 	xf86InitialConfiguration(scrn, TRUE);
 
