Index: xf86-video-intel-2.12.902/src/intel_display.c
===================================================================
--- xf86-video-intel-2.12.902.orig/src/intel_display.c
+++ xf86-video-intel-2.12.902/src/intel_display.c
@@ -1207,8 +1207,19 @@ intel_output_get_property(xf86OutputPtr
 	return FALSE;
 }
 
+#ifdef RANDR_GET_CRTC_INTERFACE
+static xf86CrtcPtr
+drmmode_get_crtc(xf86OutputPtr output)
+{
+    return output->crtc;
+}
+#endif
+
 static const xf86OutputFuncsRec intel_output_funcs = {
 	.create_resources = intel_output_create_resources,
+#ifdef RANDR_GET_CRTC_INTERFACE
+	.get_crtc = drmmode_get_crtc,
+#endif
 #ifdef RANDR_12_INTERFACE
 	.set_property = intel_output_set_property,
 	.get_property = intel_output_get_property,
