diff -Nuar cake.orig/console/cake cake/console/cake
--- cake.orig/console/cake	2010-12-05 17:00:48.583099797 +0200
+++ cake/console/cake	2010-12-05 17:12:49.545100483 +0200
@@ -18,7 +18,7 @@
 # @license			MIT License (http://www.opensource.org/licenses/mit-license.php)
 #
 ################################################################################
-LIB=${0/%cake/}
+LIB=/usr/share/php5/cakephp/cake/console/
 APP=`pwd`
 
 exec php -q ${LIB}cake.php -working "${APP}" "$@"
