Index: e/src/bin/e_startup.c
===================================================================
--- e.orig/src/bin/e_startup.c
+++ e/src/bin/e_startup.c
@@ -7,6 +7,7 @@
 
 /* local subsystem functions */
 static void _e_startup(void);
+static void _e_autostart_xdg(const char *dir);
 static void _e_startup_next_cb(void *data);
 static Eina_Bool _e_startup_event_cb(void *data, int ev_type, void *ev);
 
@@ -20,9 +21,10 @@ EAPI void
 e_startup(E_Startup_Mode mode)
 {
    char buf[PATH_MAX];
-
+   char xdg_path[PATH_MAX] = "/etc/xdg/autostart";
    if (mode == E_STARTUP_START)
      {
+	_e_autostart_xdg(xdg_path);
 	e_user_dir_concat_static(buf, "applications/startup/.order");
 	if (!ecore_file_exists(buf))
 	  e_prefix_data_concat_static(buf, "data/applications/startup/.order");
@@ -69,6 +71,69 @@ _e_startup(void)
 }
 
 static void
+_e_autostart_xdg(const char *dir)
+{
+    Efreet_Desktop *desktop = NULL;
+    Eina_List *files;
+    char *only_show_in_val;
+    char *file;
+    /*
+    Enlightenment is not a recognized DE for FDo, thus instead of searching for
+    Enlightenment value in desktop entries, just ignore KDE to ease our work.
+
+    http://standards.freedesktop.org/menu-spec/latest/apb.html
+    */
+    const char *desktop_environment = "KDE";
+
+    files = ecore_file_ls(dir);
+    if(!files) return;
+
+    EINA_LIST_FREE(files, file)
+    {
+        unsigned int kde_found = 0;
+        unsigned int de_count = 0;
+        char file_path[PATH_MAX];
+
+        snprintf(file_path, sizeof(file_path), "%s/%s", dir, file);
+
+        if (!ecore_file_exists(file_path) || ecore_file_is_dir(file_path))
+        {
+            free(file);
+        }
+
+        desktop = efreet_desktop_get(file_path);
+
+        /* skip if the desktop file is not valid */
+        if (desktop == NULL) continue;
+
+        de_count = eina_list_count(desktop->only_show_in);
+
+        if ((!desktop->only_show_in ) && (!desktop->not_show_in))
+        {
+            e_exec(NULL, desktop, NULL, NULL, NULL);
+        }
+
+        /*
+        Ignore NotShowIn entry, handle OnlyShowIn entries only. We dont
+        want KDE for now, other DE's are just fine with us :)
+        */
+        EINA_LIST_FREE(desktop->only_show_in, only_show_in_val)
+        {
+            if (!strcmp(only_show_in_val, desktop_environment))
+            {
+                kde_found = 1;
+            }
+        }
+        /* check the length of the list to prevent multiple execution of the same desktop file */
+        if ((de_count > 0) && (kde_found == 0))
+        {
+            e_exec(NULL, desktop, NULL, NULL, NULL);
+        }
+    }
+}
+
+
+static void
 _e_startup_next_cb(void *data __UNUSED__)
 {
    _e_startup();
