diff -up akonadi-1.1.1/server/src/storage/mysql-global.conf.mysql_conf akonadi-1.1.1/server/src/storage/mysql-global.conf
--- akonadi-1.1.1/server/src/storage/mysql-global.conf.mysql_conf	2009-01-21 12:29:03.000000000 -0600
+++ akonadi-1.1.1/server/src/storage/mysql-global.conf	2009-02-20 11:18:57.000000000 -0600
@@ -49,6 +49,6 @@ innodb_additional_mem_pool_size=1M
 # SELECT sum(data_length) as bla, sum(index_length) as blub FROM information_schema.tables WHERE table_schema not in ("mysql", "information_schema");
 innodb_buffer_pool_size=80M
 # size of average write burst, keep Innob_log_waits small, keep Innodb_buffer_pool_wait_free small (see show global status like "inno%", show global variables)
-innodb_log_file_size=64M
+innodb_log_file_size=8M
 innodb_flush_log_at_trx_commit=2
 
