diff -up tk8.5.2/unix/Makefile.in tk8.5.2/unix/Makefile.in
--- tk8.5.2/unix/Makefile.in	2008-03-28 19:46:36.000000000 +0100
+++ tk8.5.2/unix/Makefile.in	2008-05-19 14:45:52.000000000 +0200
@@ -713,7 +713,7 @@ install-binaries: $(TK_LIB_FILE) $(TK_ST
 	    fi
 	@echo "Installing $(LIB_FILE) to $(LIB_INSTALL_DIR)/"
 	@@INSTALL_LIB@
-	@chmod 555 "$(LIB_INSTALL_DIR)"/$(LIB_FILE)
+	@chmod 755 "$(LIB_INSTALL_DIR)"/$(LIB_FILE)
 	@if test "$(TK_BUILD_EXP_FILE)" != ""; then \
 	    echo "Installing $(TK_EXP_FILE) to $(LIB_INSTALL_DIR)/"; \
 	    $(INSTALL_DATA) $(TK_BUILD_EXP_FILE) \
