Index: kdeadmin-4.6.2/system-config-printer-kde/system-config-printer-kde.py
===================================================================
--- kdeadmin-4.6.2.orig/system-config-printer-kde/system-config-printer-kde.py
+++ kdeadmin-4.6.2/system-config-printer-kde/system-config-printer-kde.py
@@ -478,8 +478,8 @@ class GUI(QWidget):
     def on_tvMainList_changed(self, new, old):
         """about to change, offer to save"""
         if self.changed:
-            answer = KMessageBox.questionYesNo(self, "Do you want to save changes?", "Save Changes", KStandardGuiItem.save(), KStandardGuiItem.discard() )
-    
+            answer = KMessageBox.questionYesNo(self, i18n("Do you want to save changes?"), i18n("Save Changes"), KStandardGuiItem.save(), KStandardGuiItem.discard() )
+
             if answer == KMessageBox.Yes: #save is equivalent to yes
                 self.printer_properties_response()
             elif answer == KMessageBox.No: #discard is equivalent to no
@@ -4181,7 +4181,7 @@ class NewPrinterGUI(QDialog):
                               i18n("Printer '%1' requires the '%2' program but "
                                 "it is not currently installed.  Please "
                                 "install it before using this printer.", name, (exes + pkgs)[0]))
-                KMessageBox.error(self, error_text, "Missing Driver")
+                KMessageBox.error(self, error_text, i18n("Missing Driver"))
 
             """
             if pkg and response == gtk.RESPONSE_OK:
