--- a/system-config-printer-kde/system-config-printer-kde.py	(revision 1231563)
+++ b/system-config-printer-kde/system-config-printer-kde.py	(revision 1231563)
@@ -478,8 +478,8 @@
     def on_tvMainList_changed(self, new, old):
         """about to change, offer to save"""
         if self.changed:
-            answer = KMessageBox.questionYesNo(self, "Do you want to save changes?", "Save Changes", KStandardGuiItem.save(), KStandardGuiItem.discard() )
-    
+            answer = KMessageBox.questionYesNo(self, i18n("Do you want to save changes?"), i18n("Save Changes"), KStandardGuiItem.save(), KStandardGuiItem.discard() )
+
             if answer == KMessageBox.Yes: #save is equivalent to yes
                 self.printer_properties_response()
             elif answer == KMessageBox.No: #discard is equivalent to no
