Index: kdeadmin-4.4.2/CMakeLists.txt
===================================================================
--- kdeadmin-4.4.2.orig/CMakeLists.txt
+++ kdeadmin-4.4.2/CMakeLists.txt
@@ -12,16 +12,11 @@ include_directories (${QDBUS_INCLUDE_DIR
 
 macro_optional_add_subdirectory(strigi-analyzer)
 if ( UNIX )
-  macro_optional_add_subdirectory(kuser)
   macro_optional_add_subdirectory(kcron)
   macro_optional_add_subdirectory(ksystemlog)
   macro_optional_add_subdirectory(system-config-printer-kde)
 endif( UNIX )
 
-if(Q_WS_X11)
-   macro_optional_add_subdirectory(knetworkconf)
-endif(Q_WS_X11)
-
 macro_optional_add_subdirectory(doc)
 
 macro_display_feature_log()
Index: kdeadmin-4.4.2/doc/CMakeLists.txt
===================================================================
--- kdeadmin-4.4.2.orig/doc/CMakeLists.txt
+++ kdeadmin-4.4.2/doc/CMakeLists.txt
@@ -1,8 +1,5 @@
 
 if ( UNIX )
-  if(BUILD_kuser)
-    add_subdirectory(kuser)
-  endif()
   if(BUILD_kcron)
     add_subdirectory(kcron)
   endif()
@@ -10,8 +7,3 @@ if ( UNIX )
     add_subdirectory(ksystemlog)
   endif()
 endif( UNIX )
-
-if(Q_WS_X11)
-   add_subdirectory(kcontrol)
-endif(Q_WS_X11)
-
