Index: kdeadmin-4.5.2/system-config-printer-kde/options.py
===================================================================
--- kdeadmin-4.5.2.orig/system-config-printer-kde/options.py
+++ kdeadmin-4.5.2/system-config-printer-kde/options.py
@@ -398,7 +398,7 @@ class OptionText(Option):
         Option.__init__(self, name, value, supported, on_change)
 
         self.selector = QLineEdit()
-        self.selector.setText(value)
+        self.selector.setText(str(value))
         self.selector.connect(self.selector, SIGNAL("textChanged(QString)"), self.changed)
 
     def get_current_value(self):
