--- branches/KDE/4.6/kdeadmin/system-config-printer-kde/system-config-printer-kde.py	2011/02/14 15:50:13	1220673
+++ branches/KDE/4.6/kdeadmin/system-config-printer-kde/system-config-printer-kde.py	2011/02/14 15:51:42	1220674
@@ -1666,7 +1666,7 @@
         self.server_side_options[name] = option
         if name in self.changed: # was deleted before
             option.is_new = False
-        self.changed.add(option)
+        #self.changed.add(option) #disable this, not necessary and was causing http://bugs.kde.org/254672
         self.setDataButtonState()
         if is_new:
             ##option.selector.grab_focus ()
