--- system-config-printer-kde/system-config-printer-kde.py	2009-06-28 13:07:55.190195137 -0400
+++ system-config-printer-kde/system-config-printer-kde.py	2009-06-28 13:07:43.335195179 -0400
@@ -1088,6 +1088,7 @@
         self.ui.chkPEnabled.setChecked(printer.enabled)
         self.ui.chkPAccepting.setChecked(not printer.rejecting)
         self.ui.chkPShared.setChecked(printer.is_shared)
+        """
         try:
             if printer.is_shared:
                 flag = cups.CUPS_SERVER_SHARE_PRINTERS
@@ -1100,6 +1101,7 @@
                 self.ui.lblNotPublished.hide()
         except:
             self.ui.lblNotPublished.hide()
+        """
 
         # Job sheets
         self.ui.cmbPStartBanner.setEnabled(editable)
--- system-config-printer-kde/system-config-printer.ui	2009-06-28 13:07:55.285219320 -0400
+++ system-config-printer-kde/system-config-printer.ui	2009-06-28 13:07:43.336195219 -0400
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>826</width>
-    <height>509</height>
+    <height>519</height>
    </rect>
   </property>
   <property name="windowIcon">
@@ -546,13 +546,6 @@
                 </widget>
                </item>
                <item>
-                <widget class="QLabel" name="lblNotPublished">
-                 <property name="text">
-                  <string comment="KDE::DoNotExtract" >TextLabel</string>
-                 </property>
-                </widget>
-               </item>
-               <item>
                 <spacer name="horizontalSpacer_10">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
@@ -906,8 +899,8 @@
                  <rect>
                   <x>0</x>
                   <y>0</y>
-                  <width>509</width>
-                  <height>1026</height>
+                  <width>562</width>
+                  <height>1188</height>
                  </rect>
                 </property>
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
