Index: kdeadmin-4.5.5/system-config-printer-kde/system-config-printer-kde.py
===================================================================
--- kdeadmin-4.5.5.orig/system-config-printer-kde/system-config-printer-kde.py
+++ kdeadmin-4.5.5/system-config-printer-kde/system-config-printer-kde.py
@@ -374,7 +374,7 @@ class GUI(QWidget):
         #catch the currentItemChanged signal. This should stop the
         # 'Do you want to save settings' pop-up at startup
         self.ui.mainlist.blockSignals(True)
-        self.ui.mainlist.setCurrentItem(self.settingsItem)
+        self.ui.mainlist.setCurrentItem(self.newPrinterItem)
         self.ui.mainlist.blockSignals(False) #unblock the signal
         
         #FIXME hide labels until implemented
@@ -603,7 +603,7 @@ class GUI(QWidget):
             if not more_entries: break
         """
         self.ui.mainlist.clear()
-        QTreeWidgetItem(self.ui.mainlist, [i18n("New Printer"), 'New'])
+        self.newPrinterItem = QTreeWidgetItem(self.ui.mainlist, [i18n("New Printer"), 'New'])
         self.settingsItem = QTreeWidgetItem(self.ui.mainlist, [i18n("Server Settings"), 'Settings'])
         if serverPageShowing:
             self.settingsItem.setSelected(True)
