--- Lib/site.py	2008-09-15 08:59:31.000000000 +0300
+++ Lib/site.py	2008-09-15 09:00:19.000000000 +0300
@@ -269,6 +269,7 @@
                                         "python" + sys.version[:3],
                                         "site-packages"))
             sitedirs.append(os.path.join(prefix, "lib", "site-python"))
+            sitedirs.insert(0, os.path.join(prefix, "lib", "pardus"))
         else:
             sitedirs.append(prefix)
             sitedirs.append(os.path.join(prefix, "lib", "site-packages"))
