Index: Python-2.6.2/configure.in
===================================================================
--- Python-2.6.2.orig/configure.in
+++ Python-2.6.2/configure.in
@@ -1158,7 +1158,7 @@ then
 AC_MSG_CHECKING(whether $CC accepts -pthread)
 AC_CACHE_VAL(ac_cv_thread,
 [ac_save_cc="$CC"
-CC="$CC -pthread"
+CC="$CC"
 AC_TRY_RUN([
 #include <pthread.h>
 
@@ -1197,7 +1197,7 @@ then
   ac_cv_cxx_thread=yes
 elif test "$ac_cv_pthread" = "yes"
 then 
-  CXX="$CXX -pthread"
+  CXX="$CXX"
   ac_cv_cxx_thread=yes
 fi
 
@@ -1452,7 +1452,7 @@ then CC="$CC -Kpthread"
 elif test "$ac_cv_kthread" = "yes"
 then CC="$CC -Kthread"
 elif test "$ac_cv_pthread" = "yes"
-then CC="$CC -pthread"
+then CC="$CC"
 fi
 AC_MSG_CHECKING(for pthread_t)
 have_pthread_t=no
@@ -1996,10 +1996,6 @@ then
     THREADOBJ="Python/thread.o"
 elif test "$ac_cv_pthread" = "yes"
 then
-    CC="$CC -pthread"
-    if test "$ac_cv_cxx_thread" = "yes"; then
-        CXX="$CXX -pthread"
-    fi
     AC_DEFINE(WITH_THREAD)
     posix_threads=yes
     THREADOBJ="Python/thread.o"
