diff -Nur Python-2.6.2-old/Lib/ssl.py Python-2.6.2/Lib/ssl.py
--- Python-2.6.2-old/Lib/ssl.py	2009-09-11 14:44:24.309795283 +0300
+++ Python-2.6.2/Lib/ssl.py	2009-09-11 14:45:11.868670103 +0300
@@ -112,12 +112,7 @@
                                         keyfile, certfile,
                                         cert_reqs, ssl_version, ca_certs)
             if do_handshake_on_connect:
-                timeout = self.gettimeout()
-                try:
-                    self.settimeout(None)
-                    self.do_handshake()
-                finally:
-                    self.settimeout(timeout)
+                self.do_handshake()
         self.keyfile = keyfile
         self.certfile = certfile
         self.cert_reqs = cert_reqs
