--- CMakeLists.txt~	2009-09-30 20:21:05.000000000 -0400
+++ CMakeLists.txt	2009-10-05 08:35:20.000000000 -0400
@@ -341,6 +341,7 @@
         endif()
         if(WITH_V4L)
             CHECK_MODULE(libv4l1 HAVE_LIBV4L)
+	    CHECK_MODULE(libv4l2 HAVE_LIBV4L)
             CHECK_INCLUDE_FILE(linux/videodev.h HAVE_CAMV4L)
             CHECK_INCLUDE_FILE(linux/videodev2.h HAVE_CAMV4L2)
         else()
