Index: plymouth-0.8.3_20101018/themes/Makefile.am
===================================================================
--- plymouth-0.8.3_20101018.orig/themes/Makefile.am
+++ plymouth-0.8.3_20101018/themes/Makefile.am
@@ -1,2 +1,2 @@
-SUBDIRS = spinfinity fade-in text details solar glow script
+SUBDIRS = spinfinity fade-in text details solar glow script pardus
 MAINTAINERCLEANFILES = Makefile.in
Index: plymouth-0.8.3_20101018/src/plymouthd.conf
===================================================================
--- plymouth-0.8.3_20101018.orig/src/plymouthd.conf
+++ plymouth-0.8.3_20101018/src/plymouthd.conf
@@ -1,3 +1,3 @@
 # Administrator customizations go in this file
 #[Daemon]
-#Theme=fade-in
+#Theme=pardus
Index: plymouth-0.8.3_20101018/src/plymouthd.defaults
===================================================================
--- plymouth-0.8.3_20101018.orig/src/plymouthd.defaults
+++ plymouth-0.8.3_20101018/src/plymouthd.defaults
@@ -1,4 +1,4 @@
 # Distribution defaults. Changes to this file will get overwritten during
 # upgrades.
 [Daemon]
-Theme=fade-in
+Theme=pardus
Index: plymouth-0.8.3_20101018/configure.ac
===================================================================
--- plymouth-0.8.3_20101018.orig/configure.ac
+++ plymouth-0.8.3_20101018/configure.ac
@@ -308,6 +308,7 @@ AC_CONFIG_FILES([Makefile
            themes/solar/Makefile
            themes/glow/Makefile
            themes/script/Makefile
+           themes/pardus/Makefile
            images/Makefile
 	   scripts/plymouth-generate-initrd
 	   scripts/plymouth-populate-initrd
