Index: plymouth-0.8.3_20100929/src/libply-splash-graphics/ply-image.c
===================================================================
--- plymouth-0.8.3_20100929.orig/src/libply-splash-graphics/ply-image.c
+++ plymouth-0.8.3_20100929/src/libply-splash-graphics/ply-image.c
@@ -151,7 +151,7 @@ ply_image_load (ply_image_t *image)
     png_set_palette_to_rgb (png);
 
   if ((color_type == PNG_COLOR_TYPE_GRAY) && (bits_per_pixel < 8))
-    png_set_gray_1_2_4_to_8 (png);
+    png_set_expand_gray_1_2_4_to_8 (png);
 
   if (png_get_valid (png, info, PNG_INFO_tRNS))
     png_set_tRNS_to_alpha (png);
