diff -uNr SimGear.orig/simgear/nasal/naref.h SimGear/simgear/nasal/naref.h
--- SimGear.orig/simgear/nasal/naref.h	2009-06-29 00:20:22.000000000 +0200
+++ SimGear/simgear/nasal/naref.h	2010-01-31 12:12:42.362757128 +0100
@@ -15,7 +15,7 @@
       defined(_M_X64) 
 # define NASAL_LE
 #elif defined(__sparc) || defined(__ppc__) || defined(__PPC) || \
-      defined(__mips) || defined(__ARMEB__)
+      defined(__mips) || defined(__ARMEB__) || defined(__powerpc__) || defined(__powerpc64__) || defined(__alpha__) || defined(__s390__)
 # define NASAL_BE
 #else
 # error Unrecognized CPU architecture
