Index: libs/service/networkinterfaceactivatableprovider.cpp
===================================================================
--- libs/service/networkinterfaceactivatableprovider.cpp.orig
+++ libs/service/networkinterfaceactivatableprovider.cpp
@@ -113,7 +113,7 @@ bool NetworkInterfaceActivatableProvider
              || (connType == Knm::Connection::Wireless && ifaceType == Solid::Control::NetworkInterface::Ieee80211)
              || (connType == Knm::Connection::Gsm && ifaceType == Solid::Control::NetworkInterface::Gsm)
              || (connType == Knm::Connection::Cdma && ifaceType == Solid::Control::NetworkInterface::Cdma)
-             || (connType == Knm::Connection::Pppoe && ifaceType == Solid::Control::NetworkInterface::Serial)
+             || ( (connType == Knm::Connection::Pppoe && ifaceType == Solid::Control::NetworkInterface::Serial) || (connType == Knm::Connection::Pppoe && ifaceType == Solid::Control::NetworkInterface::Ieee8023) )
 #ifdef NM_0_8
              || (connType == Knm::Connection::Bluetooth && ifaceType == Solid::Control::NetworkInterface::Bluetooth)
 #endif
