Index: libs/service/networkinterfaceactivatableprovider.cpp
===================================================================
--- libs/service/networkinterfaceactivatableprovider.cpp	(revision 1201933)
+++ libs/service/networkinterfaceactivatableprovider.cpp	(working copy)
@@ -113,7 +113,7 @@
              || (connType == Knm::Connection::Wireless && ifaceType == Solid::Control::NetworkInterface::Ieee80211)
              || (connType == Knm::Connection::Gsm && ifaceType == Solid::Control::NetworkInterface::Gsm)
              || (connType == Knm::Connection::Cdma && ifaceType == Solid::Control::NetworkInterface::Cdma)
-             || (connType == Knm::Connection::Pppoe && ifaceType == Solid::Control::NetworkInterface::Serial)
+             || ( (connType == Knm::Connection::Pppoe && ifaceType == Solid::Control::NetworkInterface::Serial) || (connType == Knm::Connection::Pppoe && ifaceType == Solid::Control::NetworkInterface::Ieee8023) )
              );
 }
 
