Index: networkmanagement/applet/plasma-applet-networkmanagement.desktop
===================================================================
--- networkmanagement.orig/applet/plasma-applet-networkmanagement.desktop
+++ networkmanagement/applet/plasma-applet-networkmanagement.desktop
@@ -15,6 +15,7 @@ Name[pt_BR]=Gerenciamento de redes
 Name[sk]=Správa siete
 Name[sv]=Nätverkshantering
 Name[th]=การจัดการเครือข่าย
+Name[tr]=Ağ Yönetimi
 Name[uk]=Керування мережею
 Name[x-test]=xxNetwork Managementxx
 Comment=Network status and control utility
@@ -31,11 +32,12 @@ Comment[pt_BR]=Utilitário de controle e
 Comment[sk]=Nástroj na riadenie a stav siete
 Comment[sv]=Verktyg för nätverksstatus och kontroll
 Comment[th]=โปรแกรมสำหรับดูสถานะและควบคุมเครือข่าย
+Comment[tr]=Ağ izleme ve denetim aracı
 Comment[uk]=Інструмент стеження за станом і керування мережею
 Comment[x-test]=xxNetwork status and control utilityxx
 Type=Service
 X-KDE-ServiceTypes=Plasma/Applet
-Icon=networkmanager
+Icon=knetworkmanager
 
 X-KDE-Library=plasma_applet_networkmanagement
 X-KDE-PluginInfo-Author=Will Stephenson
Index: networkmanagement/settings/config/kcm_networkmanagement.desktop
===================================================================
--- networkmanagement.orig/settings/config/kcm_networkmanagement.desktop
+++ networkmanagement/settings/config/kcm_networkmanagement.desktop
@@ -14,6 +14,7 @@ Name[pt_BR]=Conexões de rede
 Name[sk]=Sieťové pripojenia
 Name[sv]=Nätverksanslutningar
 Name[th]=การเชื่อมต่อเครือข่าย
+Name[tr]=Ağ Bağlantıları
 Name[uk]=Мережеві з'єднання
 Name[x-test]=xxNetwork Connectionsxx
 Comment=Add, Edit and Remove network connections
@@ -30,11 +31,12 @@ Comment[pt_BR]=Adiciona, edita e remove 
 Comment[sk]=Pridanie, úprava a odstránenie sieťových pripojení
 Comment[sv]=Lägg till, redigera och ta bort nätverksanslutningar
 Comment[th]=เพิ่ม แก้ไข และลบการเชื่อมต่อเครือข่าย
+Comment[tr]=Ağ Bağlantılarını Ekleyin, Değiştirin ve Kaldırın
 Comment[uk]=Додати, змінити або вилучити мережеві з’єднання
 Comment[x-test]=xxAdd, Edit and Remove network connectionsxx
 Type=Service
 Exec=kcmshell4 kcm_networkmanagement
-Icon=networkmanager
+Icon=knetworkmanager
 X-KDE-ServiceTypes=KCModule
 X-KDE-Library=kcm_networkmanagement
 X-KDE-FactoryName=kcm_networkmanagement
Index: networkmanagement/settings/config/kcm_networkmanagement_tray.desktop
===================================================================
--- networkmanagement.orig/settings/config/kcm_networkmanagement_tray.desktop
+++ networkmanagement/settings/config/kcm_networkmanagement_tray.desktop
@@ -15,6 +15,7 @@ Name[pt_BR]=Outra
 Name[sk]=Ostatné
 Name[sv]=Övriga
 Name[th]=อื่น ๆ
+Name[tr]=Diğer
 Name[uk]=Інше
 Name[x-test]=xxOtherxx
 Comment=Configure the Network Management tool
@@ -31,6 +32,7 @@ Comment[pt_BR]=Configurar a ferramenta d
 Comment[sk]=Nastavenie nástroja pre správu siete
 Comment[sv]=Anpassa nätverkshanteringsverktyget
 Comment[th]=เครื่องมือตั้งค่าการจัดการเครือข่าย
+Comment[tr]=Ağ Yönetim Aracı Yapılandırması
 Comment[uk]=Налаштувати інструмент керування мережею
 Comment[x-test]=xxConfigure the Network Management toolxx
 Type=Service
