Index: networkmanagement/applet/plasma-applet-networkmanagement.desktop
===================================================================
--- networkmanagement.orig/applet/plasma-applet-networkmanagement.desktop
+++ networkmanagement/applet/plasma-applet-networkmanagement.desktop
@@ -19,6 +19,7 @@ Name[sk]=Správa siete
 Name[sl]=Omrežno upravljanje
 Name[sv]=Nätverkshantering
 Name[th]=การจัดการเครือข่าย
+Name[tr]=Ağ Yönetimi
 Name[uk]=Керування мережею
 Name[x-test]=xxNetwork Managementxx
 Name[zh_TW]=網路管理
@@ -40,6 +41,7 @@ Comment[sk]=Nástroj na riadenie a stav
 Comment[sl]=Orodje za pregled omrežnega stanja in nadzor
 Comment[sv]=Verktyg för nätverksstatus och kontroll
 Comment[th]=โปรแกรมสำหรับดูสถานะและควบคุมเครือข่าย
+Comment[tr]=Ağ izleme ve denetim aracı
 Comment[uk]=Інструмент стеження за станом і керування мережею
 Comment[x-test]=xxNetwork status and control utilityxx
 Comment[zh_TW]=網路狀態與控制工具
Index: networkmanagement/settings/config/kcm_networkmanagement.desktop
===================================================================
--- networkmanagement.orig/settings/config/kcm_networkmanagement.desktop
+++ networkmanagement/settings/config/kcm_networkmanagement.desktop
@@ -18,6 +18,7 @@ Name[sk]=Sieťové pripojenia
 Name[sl]=Omrežne povezave
 Name[sv]=Nätverksanslutningar
 Name[th]=การเชื่อมต่อเครือข่าย
+Name[tr]=Ağ Bağlantıları
 Name[uk]=Мережеві з'єднання
 Name[x-test]=xxNetwork Connectionsxx
 Name[zh_TW]=網路連線
@@ -39,6 +40,7 @@ Comment[sk]=Pridanie, úprava a odstrán
 Comment[sl]=Dodajte, urejajte in odstranite omrežne povezave
 Comment[sv]=Lägg till, redigera och ta bort nätverksanslutningar
 Comment[th]=เพิ่ม แก้ไข และลบการเชื่อมต่อเครือข่าย
+Comment[tr]=Ağ Bağlantılarını Ekleyin, Değiştirin ve Kaldırın
 Comment[uk]=Додати, змінити або вилучити мережеві з’єднання
 Comment[x-test]=xxAdd, Edit and Remove network connectionsxx
 Comment[zh_TW]=新增、編輯與移除網路連線
Index: networkmanagement/settings/config/kcm_networkmanagement_tray.desktop
===================================================================
--- networkmanagement.orig/settings/config/kcm_networkmanagement_tray.desktop
+++ networkmanagement/settings/config/kcm_networkmanagement_tray.desktop
@@ -19,6 +19,7 @@ Name[sk]=Ostatné
 Name[sl]=Drugo
 Name[sv]=Övriga
 Name[th]=อื่น ๆ
+Name[tr]=Diğer
 Name[uk]=Інше
 Name[x-test]=xxOtherxx
 Name[zh_TW]=其他
@@ -39,6 +40,7 @@ Comment[sk]=Nastavenie nástroja pre spr
 Comment[sl]=Nastavite orodje za omrežno upravljanje
 Comment[sv]=Anpassa nätverkshanteringsverktyget
 Comment[th]=เครื่องมือตั้งค่าการจัดการเครือข่าย
+Comment[tr]=Ağ Yönetim Aracı Yapılandırması
 Comment[uk]=Налаштувати інструмент керування мережею
 Comment[x-test]=xxConfigure the Network Management toolxx
 Comment[zh_TW]=設定網路管理工具
