Index: settings/config/manageconnectionwidget.cpp
===================================================================
--- settings/config/manageconnectionwidget.cpp	(revision 1198436)
+++ settings/config/manageconnectionwidget.cpp	(working copy)
@@ -155,6 +155,7 @@
     mConnEditUi.listVpn->clear();
     mConnEditUi.listPppoe->clear();
 
+    KNetworkManagerServicePrefs::self()->readConfig();
     QStringList connectionIds = KNetworkManagerServicePrefs::self()->connections();
     QList<QTreeWidgetItem *> wiredItems, wirelessItems, cellularItems, vpnItems, pppoeItems;
     foreach (const QString &connectionId, connectionIds) {
