diff --git a/settings/config/manageconnectionwidget.cpp b/settings/config/manageconnectionwidget.cpp
index a68990d..9593b24 100644
--- a/settings/config/manageconnectionwidget.cpp
+++ b/settings/config/manageconnectionwidget.cpp
@@ -896,14 +896,14 @@ void ManageConnectionWidget::tabChanged(int index)
                 mVpnMenu->addAction(vpnAction);
             }
             connect(mVpnMenu, SIGNAL(triggered(QAction*)), SLOT(connectionTypeMenuTriggered(QAction*)));
+            connect(mConnEditUi.buttonSetVpn->importButton(),SIGNAL(clicked()),SLOT(importClicked()));
+            connect(mConnEditUi.buttonSetVpn->exportButton(),SIGNAL(clicked()),SLOT(exportClicked()));
             mConnEditUi.buttonSetVpn->addButton()->setMenu(mVpnMenu);
         }
         mConnEditUi.buttonSetVpn->addButton()->setEnabled(!mVpnMenu->isEmpty());
         mConnEditUi.buttonSetVpn->importButton()->setEnabled(!mVpnMenu->isEmpty());
         mConnEditUi.buttonSetVpn->importButton()->setVisible(!mVpnMenu->isEmpty());
         mConnEditUi.buttonSetVpn->exportButton()->setVisible(!mVpnMenu->isEmpty());
-        connect(mConnEditUi.buttonSetVpn->importButton(),SIGNAL(clicked()),SLOT(importClicked()));
-        connect(mConnEditUi.buttonSetVpn->exportButton(),SIGNAL(clicked()),SLOT(exportClicked()));
     }
 }
 
