Index: networkmanagement/CMakeLists.txt
===================================================================
--- networkmanagement.orig/CMakeLists.txt
+++ networkmanagement/CMakeLists.txt
@@ -52,7 +52,6 @@ add_subdirectory(tests)
 add_subdirectory(settings)
 add_subdirectory(vpnplugins)
 add_subdirectory(solidcontrolfuture)
-add_subdirectory(monolithic)
 add_subdirectory(kded)
 
 install( FILES networkmanagement.notifyrc DESTINATION ${DATA_INSTALL_DIR}/networkmanagement )
