Index: libs/ui/security/wirelesssecuritysettingwidget.cpp
===================================================================
--- libs/ui/security/wirelesssecuritysettingwidget.cpp	(revision 1202063)
+++ libs/ui/security/wirelesssecuritysettingwidget.cpp	(revision 1202064)
@@ -308,7 +308,7 @@
     else if (d->ui.cboType->currentIndex() == d->dynamicWep.first) {
         d->setting8021x->setEnabled(true);
         d->settingSecurity->setSecurityType(Knm::WirelessSecuritySetting::EnumSecurityType::DynamicWep); // FIXME
-        d->settingSecurity->setKeymgmt(Knm::WirelessSecuritySetting::EnumKeymgmt::WPAEAP);
+        d->settingSecurity->setKeymgmt(Knm::WirelessSecuritySetting::EnumKeymgmt::Ieee8021x);
     }
     else if (d->ui.cboType->currentIndex() == d->wpaPsk.first) {
         d->setting8021x->setEnabled(false);
