diff --git a/vpnplugins/openvpn/openvpnwidget.cpp b/vpnplugins/openvpn/openvpnwidget.cpp
index 94dd33a..624cb05 100644
--- a/vpnplugins/openvpn/openvpnwidget.cpp
+++ b/vpnplugins/openvpn/openvpnwidget.cpp
@@ -242,7 +242,6 @@ void OpenVpnSettingWidget::writeConfig()
         if (!d->ui.passUserName->text().isEmpty()) {
             data.insert( NM_OPENVPN_KEY_USERNAME, d->ui.passUserName->text());
         }
-        d->setting->setUserName(d->ui.passUserName->text());
         // password
         secretData.insert(QLatin1String( NM_OPENVPN_KEY_PASSWORD ), d->ui.passPassword->text());
         // ca
@@ -254,7 +253,6 @@ void OpenVpnSettingWidget::writeConfig()
         if (!d->ui.x509PassUsername->text().isEmpty()) {
             data.insert(NM_OPENVPN_KEY_USERNAME, d->ui.x509PassUsername->text());
         }
-        d->setting->setUserName(d->ui.x509PassUsername->text());
         // ca
         data.insert(NM_OPENVPN_KEY_CA, d->ui.x509PassCaFile->url().path().toUtf8());
         // cert
