Index: kmail/messagelistview/core/manager.cpp
===================================================================
--- kmail/messagelistview/core/manager.cpp.orig
+++ kmail/messagelistview/core/manager.cpp
@@ -337,7 +337,7 @@ void Manager::createDefaultAggregations(
                 "and all the threads that have been active today."
             ),
           Aggregation::GroupByDate,
-          Aggregation::ExpandRecentGroups,
+          Aggregation::AlwaysExpandGroups,
           Aggregation::PerfectReferencesAndSubject,
           Aggregation::MostRecentMessage,
           Aggregation::ExpandThreadsWithUnreadOrImportantMessages,
@@ -353,7 +353,7 @@ void Manager::createDefaultAggregations(
                 "So for example, in \"Today\" you will simply find all the messages arrived today."
             ),
           Aggregation::GroupByDate,
-          Aggregation::ExpandRecentGroups,
+          Aggregation::AlwaysExpandGroups,
           Aggregation::NoThreading,
           Aggregation::MostRecentMessage,
           Aggregation::NeverExpandThreads,
