Index: akregator/src/akregator_part.cpp
===================================================================
--- akregator/src/akregator_part.cpp.orig
+++ akregator/src/akregator_part.cpp
@@ -182,7 +182,7 @@ Part::Part( QWidget *parentWidget, QObje
 
     initFonts();
 
-    QString useragent = QString( "Akregator/%1; syndication" ).arg( AKREGATOR_VERSION );
+    QString useragent = QString( "Akregator/%1; syndication Pardus" ).arg( AKREGATOR_VERSION );
 
     if( !Settings::customUserAgent().isEmpty() )
         useragent = Settings::customUserAgent();
