Index: kdepim-4.4.2/kmail/kmreaderwin.cpp
===================================================================
--- kdepim-4.4.2.orig/kmail/kmreaderwin.cpp
+++ kdepim-4.4.2/kmail/kmreaderwin.cpp
@@ -1360,7 +1360,7 @@ void KMReaderWin::displaySplashPage( con
   QString fontSize = QString::number( pointsToPixel( mCSSHelper->bodyFont().pointSize() ) );
   QString appTitle = i18n("KMail");
   QString catchPhrase = ""; //not enough space for a catch phrase at default window size i18n("Part of the Kontact Suite");
-  QString quickDescription = i18n("The email client for the K Desktop Environment.");
+  QString quickDescription = "";//i18n("The email client for the K Desktop Environment.");
   mViewer->write(content.arg(fontSize).arg(appTitle).arg(catchPhrase).arg(quickDescription).arg(info));
   mViewer->end();
 }
