--- kdebase-4.2.0/apps/dolphin/src/dolphinviewcontainer.cpp	2009/04/15 17:13:23	954342
+++ kdebase-4.2.0.new/apps/dolphin/src/dolphinviewcontainer.cpp	2009/04/15 17:14:30	954343
@@ -402,6 +402,10 @@
             const QString browser = config.readEntry("BrowserApplication");
             if (!browser.isEmpty()) {
                 app = browser;
+                if (app.startsWith('!')) {
+                    // a literal command has been configured, remove the '!' prefix
+                    app = app.mid(1);
+                }
             }
         } else {
             showErrorMessage(i18nc("@info:status",
