Index: kdebase-4.4.2/apps/konsole/src/ViewManager.cpp
===================================================================
--- kdebase-4.4.2.orig/apps/konsole/src/ViewManager.cpp	2010-04-16 11:43:42.909426295 -0400
+++ kdebase-4.4.2/apps/konsole/src/ViewManager.cpp	2010-04-16 11:44:21.748371342 -0400
@@ -955,6 +955,14 @@
         _viewSplitter->activeContainer()->setActiveView(display);
         display->setFocus(Qt::OtherFocusReason);
     }
+    if (ids.isEmpty()) // Session file is unusable, start default Profile
+    {
+        Profile::Ptr profile = SessionManager::instance()->defaultProfile();
+        Session* session = SessionManager::instance()->createSession(profile);
+        createView(session);
+        if (!session->isRunning())
+            session->run();
+    }
 }
 
 uint qHash(QPointer<TerminalDisplay> display)
