Index: knights-2.2.0/src/knights.desktop
===================================================================
--- knights-2.2.0.orig/src/knights.desktop
+++ knights-2.2.0/src/knights.desktop
@@ -13,6 +13,7 @@ Name[pt]=Knights
 Name[pt_BR]=Knights
 Name[ru]=Knights
 Name[sv]=Knights
+Name[tr]=Knights
 Name[uk]=Knights
 Name[x-test]=xxKnightsxx
 Name[zh_TW]=Knights
@@ -35,6 +36,7 @@ GenericName[pt]=Jogo de xadrez
 GenericName[pt_BR]=Jogo de Xadrez
 GenericName[ru]=Игра в шахматы
 GenericName[sv]=Schackspel
+GenericName[tr]=Satranç
 GenericName[uk]=Гра у шахи
 GenericName[x-test]=xxChess gamexx
 GenericName[zh_TW]=西洋棋遊戲
