Index: knights/src/knights.desktop
===================================================================
--- knights.orig/src/knights.desktop
+++ knights/src/knights.desktop
@@ -31,6 +31,7 @@ GenericName[pt]=Jogo de xadrez
 GenericName[pt_BR]=Jogo de Xadrez
 GenericName[ru]=Игра в шахматы
 GenericName[sv]=Schackspel
+GenericName[tr]=Satranç
 GenericName[uk]=Гра у шахи
 GenericName[x-test]=xxChess gamexx
 GenericName[zh_TW]=西洋棋遊戲
