Index: wine-1.2-rc5/programs/winecfg/audio.c
===================================================================
--- wine-1.2-rc5.orig/programs/winecfg/audio.c
+++ wine-1.2-rc5/programs/winecfg/audio.c
@@ -90,8 +90,8 @@ typedef struct
 } AUDIO_DRIVER;
 
 static AUDIO_DRIVER sAudioDrivers[] = {
-  {IDS_DRIVER_PULSE,     "pulse"},
   {IDS_DRIVER_ALSA,      "alsa"},
+  {IDS_DRIVER_PULSE,     "pulse"},
   {IDS_DRIVER_OSS,       "oss"},
   {IDS_DRIVER_COREAUDIO, "coreaudio"},
   {IDS_DRIVER_JACK,      "jack"},
