diff -Nur ffmpeg-old/configure ffmpeg/configure
--- ffmpeg-old/configure	2011-01-05 15:56:16.359978539 +0200
+++ ffmpeg/configure	2011-01-05 15:57:40.545042535 +0200
@@ -2171,6 +2171,9 @@
         # targets that do support conditional mov (cmov)
         i686|pentiumpro|pentium[23]|pentium-m|athlon|athlon-tbird|athlon-4|athlon-[mx]p|athlon64|k8|opteron|athlon-fx|core2|amdfam10|barcelona|atom)
             cpuflags="-march=$cpu"
+            if test "$cpu" = "atom"; then
+                cpuflags="-march=i686"
+            fi
             enable cmov
             enable fast_cmov
         ;;
