diff -Nur ffmpeg-old/configure ffmpeg/configure
--- ffmpeg-old/configure	2008-05-08 11:28:31.000000000 +0300
+++ ffmpeg/configure	2008-05-08 11:28:54.000000000 +0300
@@ -1353,9 +1353,6 @@
     die_gpl_disabled "The software scaler"     swscale
 fi
 
-if ! enabled nonfree && enabled_any libamr_nb libamr_wb; then
-    die "libamr is nonfree and --enable-nonfree is not specified."
-fi
 
 check_deps $ARCH_EXT_LIST
 
