diff -Nur ffmpeg-old/libavcodec/wmadec.c ffmpeg/libavcodec/wmadec.c
--- ffmpeg-old/libavcodec/wmadec.c	2010-08-25 14:00:03.630202740 +0300
+++ ffmpeg/libavcodec/wmadec.c	2010-08-25 14:00:23.131206409 +0300
@@ -807,7 +807,7 @@
         const float *output[MAX_CHANNELS];
         for (ch = 0; ch < MAX_CHANNELS; ch++)
             output[ch] = s->frame_out[ch];
-        s->dsp.float_to_int16_interleave(samples, output, n, incr);
+        s->dsp.float_to_int16_interleave(samples, (const float **)output, n, incr);
         for(ch = 0; ch < incr; ch++) {
             /* prepare for next block */
             memmove(&s->frame_out[ch][0], &s->frame_out[ch][n], n * sizeof(float));
