diff -Nur ffmpeg-old/libavcodec/wmadec.c ffmpeg/libavcodec/wmadec.c
--- ffmpeg-old/libavcodec/wmadec.c	2010-08-27 22:23:01.684720869 +0300
+++ ffmpeg/libavcodec/wmadec.c	2010-08-27 22:23:31.733848837 +0300
@@ -790,7 +790,7 @@
     /* convert frame to integer */
     n = s->frame_len;
     incr = s->nb_channels;
-    if (s->dsp.float_to_int16_interleave == ff_float_to_int16_interleave_c) {
+    if (1) {
         for(ch = 0; ch < s->nb_channels; ch++) {
             ptr = samples + ch;
             iptr = s->frame_out[ch];
