diff -Nur ffmpeg-old/cmdutils.c ffmpeg/cmdutils.c
--- ffmpeg-old/cmdutils.c	2010-08-19 20:15:24.263523289 +0300
+++ ffmpeg/cmdutils.c	2010-08-19 20:17:30.380528388 +0300
@@ -350,11 +350,13 @@
 {
     fprintf(stderr, "%s version " FFMPEG_VERSION ", Copyright (c) %d-%d the FFmpeg developers\n",
             program_name, program_birth_year, this_year);
+    /*
     fprintf(stderr, "  built on %s %s with %s %s\n",
             __DATE__, __TIME__, CC_TYPE, CC_VERSION);
     fprintf(stderr, "  configuration: " FFMPEG_CONFIGURATION "\n");
     print_all_libs_info(stderr, INDENT|SHOW_CONFIG);
     print_all_libs_info(stderr, INDENT|SHOW_VERSION);
+    */
 }
 
 void show_version(void) {
@@ -365,7 +367,7 @@
 void show_license(void)
 {
     printf(
-#if CONFIG_NONFREE
+#if 0
     "This version of %s has nonfree parts compiled in.\n"
     "Therefore it is not legally redistributable.\n",
     program_name
