diff -Nur ffmpeg-old/configure ffmpeg/configure
--- ffmpeg-old/configure	2009-05-26 14:51:56.745777358 +0300
+++ ffmpeg/configure	2009-05-26 14:56:44.921402593 +0300
@@ -2013,6 +2013,9 @@
 if enabled pthreads; then
     if check_func pthread_create; then
         :
+    elif check_func pthread_create -lpthread; then
+        add_cflags -lpthread
+        add_extralibs -lpthread
     elif check_func pthread_create -pthread; then
         add_cflags -pthread
         add_extralibs -pthread
