diff -Nur ffmpeg-old/libavcodec/wma.c ffmpeg/libavcodec/wma.c
--- ffmpeg-old/libavcodec/wma.c	2009-07-15 10:08:29.375599338 +0300
+++ ffmpeg/libavcodec/wma.c	2009-07-15 10:08:52.112349210 +0300
@@ -509,7 +509,7 @@
     }
     /** NOTE: EOB can be omitted */
     if (offset > num_coefs) {
-        av_log(avctx, AV_LOG_ERROR, "overflow in spectral RLE, ignoring\n");
+        //av_log(avctx, AV_LOG_ERROR, "overflow in spectral RLE, ignoring\n");
         return -1;
     }
 
