diff -Nur ffmpeg-old/libavutil/cpu.c ffmpeg/libavutil/cpu.c
--- ffmpeg-old/libavutil/cpu.c	2010-09-11 20:41:27.623083649 +0300
+++ ffmpeg/libavutil/cpu.c	2010-09-11 20:44:06.247093525 +0300
@@ -18,10 +18,12 @@
 
 #include "cpu.h"
 #include "config.h"
+#include "log.h"
 
 int av_get_cpu_flags(void)
 {
     static int flags, checked;
+    static int caps_printed;
 
     if (checked)
         return flags;
@@ -31,19 +33,15 @@
     if (ARCH_X86) flags = ff_get_cpu_flags_x86();
 
     checked = 1;
-    return flags;
-}
-
-#ifdef TEST
 
-#undef printf
+    /* print cpu flags once. we don't touch indentation for easier maintenance */
+    if(!caps_printed) {
+        caps_printed = 1;
 
-int main(void)
-{
     int cpu_flags = av_get_cpu_flags();
 
-    printf("cpu_flags = 0x%08X\n", cpu_flags);
-    printf("cpu_flags = %s%s%s%s%s%s%s%s%s%s%s%s\n",
+    //av_log(NULL, AV_LOG_INFO, "CPU Flags detected: 0x%08X\n", cpu_flags);
+    av_log(NULL, AV_LOG_INFO, "CPU Flags detected: %s%s%s%s%s%s%s%s%s%s%s%s\n",
 #if   ARCH_ARM
            cpu_flags & AV_CPU_FLAG_IWMMXT   ? "IWMMXT "     : "",
 #elif ARCH_PPC
@@ -62,7 +60,7 @@
            cpu_flags & AV_CPU_FLAG_3DNOW    ? "3DNow "      : "",
            cpu_flags & AV_CPU_FLAG_3DNOWEXT ? "3DNowExt "   : "");
 #endif
-    return 0;
+    }
+    return flags;
 }
 
-#endif
