Index: configure
===================================================================
--- configure	(revision 23430)
+++ configure	(revision 23429)
@@ -2707,8 +2707,8 @@
     enabled libvpx_encoder && { check_lib2 "vpx/vpx_encoder.h vpx/vp8cx.h" vpx_codec_enc_init_ver -lvpx ||
                                 die "ERROR: libvpx encoder version must be >=0.9.1"; } }
 enabled libx264    && require  libx264 x264.h x264_encoder_encode -lx264 &&
-                      { check_cpp_condition x264.h "X264_BUILD >= 98" ||
-                        die "ERROR: libx264 version must be >= 0.98."; }
+                      { check_cpp_condition x264.h "X264_BUILD >= 90" ||
+                        die "ERROR: libx264 version must be >= 0.90."; }
 enabled libxavs    && require  libxavs xavs.h xavs_encoder_encode -lxavs
 enabled libxvid    && require  libxvid xvid.h xvid_global -lxvidcore
 enabled mlib       && require  mediaLib mlib_types.h mlib_VectorSub_S16_U8_Mod -lmlib
Index: libavcodec/libx264.c
===================================================================
--- libavcodec/libx264.c	(revision 23430)
+++ libavcodec/libx264.c	(revision 23429)
@@ -89,7 +89,6 @@
     int nnal, i;
     x264_picture_t pic_out;
 
-    x264_picture_init( &x4->pic );
     x4->pic.img.i_csp   = X264_CSP_I420;
     x4->pic.img.i_plane = 3;
 
