diff --git a/runtime/nsdpoll_ptcp.c b/runtime/nsdpoll_ptcp.c
index 5100670..bc374c6 100644
--- a/runtime/nsdpoll_ptcp.c
+++ b/runtime/nsdpoll_ptcp.c
@@ -42,7 +42,6 @@
 #include "nspoll.h"
 #include "nsd_ptcp.h"
 #include "nsdpoll_ptcp.h"
-#include "unlimited_select.h"
 
 /* static data */
 DEFobjStaticHelpers
diff --git a/runtime/unlimited_select.h b/runtime/unlimited_select.h
index 32dadc0..3fa7eb0 100644
--- a/runtime/unlimited_select.h
+++ b/runtime/unlimited_select.h
@@ -23,6 +23,7 @@
  */
 
 #ifndef	UNLIMITED_SELECT_H_INCLUDED
+#define UNLIMITED_SELECT_H_INCLUDED
 
 #include <string.h>
 #include <stdlib.h>
