Index: chromium-10.0.648.82/chrome/browser/download/download_extensions.cc
===================================================================
--- chromium-10.0.648.82.orig/chrome/browser/download/download_extensions.cc
+++ chromium-10.0.648.82/chrome/browser/download/download_extensions.cc
@@ -181,6 +181,7 @@ static const struct Executables {
   { "exe", AllowOnUserGesture },
   { "ksh", AllowOnUserGesture },
   { "rpm", AllowOnUserGesture },
+  { "pisi", AllowOnUserGesture },
   { "sh", AllowOnUserGesture },
   { "tcsh", AllowOnUserGesture },
 #endif
