Index: chromium-13.0.782.24/chrome/browser/download/download_extensions.cc
===================================================================
--- chromium-13.0.782.24.orig/chrome/browser/download/download_extensions.cc
+++ chromium-13.0.782.24/chrome/browser/download/download_extensions.cc
@@ -182,6 +182,7 @@ static const struct Executables {
   { "exe", AllowOnUserGesture },
   { "ksh", AllowOnUserGesture },
   { "rpm", AllowOnUserGesture },
+  { "pisi", AllowOnUserGesture },
   { "sh", AllowOnUserGesture },
   { "tcsh", AllowOnUserGesture },
 #endif
