--- bitlbee-1.2.3/doc/bitlbee.xinetd.orig	2008-09-17 13:47:55.000000000 +0300
+++ bitlbee-1.2.3/doc/bitlbee.xinetd	2008-09-22 14:03:25.000000000 +0300
@@ -10,15 +10,16 @@
 	wait            = no
 	
 	## You most likely want to change these two
-	user            = nobody
-	server          = /usr/local/sbin/bitlbee
+	user            = bitlbee
+	server          = /usr/sbin/bitlbee
 	
 	## You might want to limit access to localhost only:
-	# bind            = 127.0.0.1
+	bind            = 127.0.0.1
 	
 	## Thanks a lot to friedman@splode.com for telling us about the type
 	## argument, so now this file can be used without having to edit
 	## /etc/services too.
 	type            = UNLISTED
 	port            = 6667
+	disable         = yes
 }
