Index: smplayer-0.6.9_20100211/smplayer_enqueue.desktop
===================================================================
--- smplayer-0.6.9_20100211.orig/smplayer_enqueue.desktop
+++ smplayer-0.6.9_20100211/smplayer_enqueue.desktop
@@ -13,6 +13,7 @@ Name[nl]=In wachtrij van SMPlayer
 Name[pt]=Em fila no SMPlayer
 Name[ru]=Добавить в очередь SMPlayer
 Name[uk]=Додати до черги SMPlayer
+Name[tr]=SMPlayer kuyruğuna ekle
 Comment=Add file(s) to the SMPlayer playlist
 Comment[cs]=Přidá soubor(y) do přehrávací fronty SMPlayeru
 Comment[es]=Añade ficheros a la lista de reproducción de SMPlayer
@@ -23,6 +24,7 @@ Comment[nl]=Bestanden toevoegen aan de a
 Comment[pt]=Adicionar ficheiro(s) à lista de reprodução do SMPlayer
 Comment[ru]=Добавить файл(ы) в конец списка воспроизведения SMPlayer
 Comment[uk]=Додати файл(и) до переліку програвання SMPlayer
+Comment[tr]=SMPlayer oynatma listesine dosya ekle
 Type=Application
 X-KDE-StartupNotify=false
 NoDisplay=true
