diff -Naur LibVNCServer-0.9.7/libvncclient/rfbproto.c LibVNCServer-0.9.7-fix-format-errors/libvncclient/rfbproto.c
--- LibVNCServer-0.9.7/libvncclient/rfbproto.c	2008-02-04 18:58:11.000000000 +0100
+++ LibVNCServer-0.9.7-fix-format-errors/libvncclient/rfbproto.c	2009-02-05 19:28:40.000000000 +0100
@@ -72,7 +72,7 @@
 
     time(&log_clock);
     strftime(buf, 255, "%d/%m/%Y %X ", localtime(&log_clock));
-    fprintf(stderr,buf);
+    fprintf(stderr,"%s",buf);
 
     vfprintf(stderr, format, args);
     fflush(stderr);
diff -Naur LibVNCServer-0.9.7/libvncserver/main.c LibVNCServer-0.9.7-fix-format-errors/libvncserver/main.c
--- LibVNCServer-0.9.7/libvncserver/main.c	2007-03-17 01:13:13.000000000 +0100
+++ LibVNCServer-0.9.7-fix-format-errors/libvncserver/main.c	2009-02-05 19:26:12.000000000 +0100
@@ -242,7 +242,7 @@
 
     time(&log_clock);
     strftime(buf, 255, "%d/%m/%Y %X ", localtime(&log_clock));
-    fprintf(stderr,buf);
+    fprintf(stderr,"%s",buf);
 
     vfprintf(stderr, format, args);
     fflush(stderr);
diff -Naur LibVNCServer-0.9.7/test/encodingstest.c LibVNCServer-0.9.7-fix-format-errors/test/encodingstest.c
--- LibVNCServer-0.9.7/test/encodingstest.c	2005-05-24 11:59:31.000000000 +0200
+++ LibVNCServer-0.9.7-fix-format-errors/test/encodingstest.c	2009-02-05 19:30:42.000000000 +0100
@@ -277,7 +277,7 @@
 
 	time(&log_clock);
 	strftime(buf, 255, "%d/%m/%Y %X (client) ", localtime(&log_clock));
-	fprintf(stderr,buf);
+	fprintf(stderr,"%s",buf);
 
 	vfprintf(stderr, format, args);
 	fflush(stderr);
