--- Makefile.in.orig	2008-02-26 19:56:15.000000000 +0100
+++ Makefile.in	2008-02-26 19:56:25.000000000 +0100
@@ -708,8 +708,6 @@
 	rm -f $(DESTDIR)${sysconfdir}/partimaged/partimaged.csr ;\
 	chmod 600 $(DESTDIR)${sysconfdir}/partimaged/partimaged.key || true;\
 	chmod 600 $(DESTDIR)${sysconfdir}/partimaged/partimaged.cert || true;\
-	chown partimag:root $(DESTDIR)${sysconfdir}/partimaged/partimaged.key || true;\
-	chown partimag:root $(DESTDIR)${sysconfdir}/partimaged/partimaged.cert || true;\
 	else \
 	echo "SSL disabled, no certificate will be generated." ;\
 	fi
@@ -741,7 +739,6 @@
 	echo "#sample # user 'sample' is allowed to connect partimaged" >> \
            $(DESTDIR)${sysconfdir}/partimaged/partimagedusers ;\
 	chmod 600 $(DESTDIR)${sysconfdir}/partimaged/partimagedusers || true;\
-	chown partimag:root $(DESTDIR)${sysconfdir}/partimaged/partimagedusers || true ;\
 	fi 
 
 distclean-local:
--- Makefile.am.orig	2008-02-26 19:55:53.000000000 +0100
+++ Makefile.am	2008-02-26 19:56:11.000000000 +0100
@@ -21,8 +21,6 @@
 	rm -f $(DESTDIR)${sysconfdir}/partimaged/partimaged.csr ;\
 	chmod 600 $(DESTDIR)${sysconfdir}/partimaged/partimaged.key || true;\
 	chmod 600 $(DESTDIR)${sysconfdir}/partimaged/partimaged.cert || true;\
-	chown partimag:root $(DESTDIR)${sysconfdir}/partimaged/partimaged.key || true;\
-	chown partimag:root $(DESTDIR)${sysconfdir}/partimaged/partimaged.cert || true;\
 	else \
 	echo "SSL disabled, no certificate will be generated." ;\
 	fi
@@ -55,7 +53,6 @@
 	echo "#sample # user 'sample' is allowed to connect partimaged" >> \
            $(DESTDIR)${sysconfdir}/partimaged/partimagedusers ;\
 	chmod 600 $(DESTDIR)${sysconfdir}/partimaged/partimagedusers || true;\
-	chown partimag:root $(DESTDIR)${sysconfdir}/partimaged/partimagedusers || true ;\
 	fi 
 
 distclean-local:
