--- Makefile.in.orig	2008-02-26 19:59:54.000000000 +0100
+++ Makefile.in	2008-02-26 20:00:16.000000000 +0100
@@ -732,7 +732,7 @@
           $(DESTDIR)${sysconfdir}/partimaged/partimagedusers ;\
 	echo -n "#add only users allowed to " >>  \
            $(DESTDIR)${sysconfdir}/partimaged/partimagedusers ;\
-	echo "connect partimaged" >> ${sysconfdir}/partimaged/partimagedusers ;\
+	echo "connect partimaged" >> $(DESTDIR)${sysconfdir}/partimaged/partimagedusers ;\
 	echo "# (only one login per line)" >> \
            $(DESTDIR)${sysconfdir}/partimaged/partimagedusers ;\
 	echo "" >> $(DESTDIR)${sysconfdir}/partimaged/partimagedusers ;\
