diff -Nuar src/server/partimaged.cpp src/server/partimaged.cpp
--- src/server/partimaged.cpp	2008-07-07 00:07:53.000000000 +0300
+++ src/server/partimaged.cpp	2008-07-07 00:08:25.000000000 +0300
@@ -22,7 +22,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 
 #include "net.h"
 #include "netserver.h"
diff -Nuar src/server/partimaged-main.cpp src/server/partimaged-main.cpp
--- src/server/partimaged-main.cpp	2008-02-03 23:57:53.000000000 +0200
+++ src/server/partimaged-main.cpp	2008-07-07 00:04:41.000000000 +0300
@@ -27,7 +27,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <signal.h>
diff -Nuar src/shared/net.h src/shared/net.h
--- src/shared/net.h	2008-07-07 00:08:47.000000000 +0300
+++ src/shared/net.h	2008-07-07 00:09:06.000000000 +0300
@@ -28,7 +28,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>
-#include <iostream.h>
+#include <iostream>
 #include <pthread.h>
 
 #include "pathnames.h"
