diff -up hplip-3.10.9/ui4/setupdialog.py.emit-SIGNAL hplip-3.10.9/ui4/setupdialog.py
--- hplip-3.10.9/ui4/setupdialog.py.emit-SIGNAL	2010-11-10 18:42:13.000000000 +0100
+++ hplip-3.10.9/ui4/setupdialog.py	2010-11-11 11:59:47.000000000 +0100
@@ -868,7 +868,7 @@ class SetupDialog(QDialog, Ui_Dialog):
                 self.printer_name_ok = True
 
                 self.FaxNameLineEdit.emit(SIGNAL("textChanged(const QString &)"),
-                            (self.FaxNameLineEdit.text(),))
+                                          self.FaxNameLineEdit.text())
 
         self.setIndicators()
         self.setAddPrinterButton()
@@ -900,7 +900,7 @@ class SetupDialog(QDialog, Ui_Dialog):
                 self.fax_name_ok = True
 
                 self.PrinterNameLineEdit.emit(SIGNAL("textChanged(const QString&)"),
-                            (self.PrinterNameLineEdit.text(),))
+                                              self.PrinterNameLineEdit.text())
 
         self.setIndicators()
         self.setAddPrinterButton()
