diff -up hplip-3.10.9/prnt/cupsext/cupsext.c~ hplip-3.10.9/prnt/cupsext/cupsext.c
--- hplip-3.10.9/prnt/cupsext/cupsext.c~	2010-11-12 10:35:59.316957426 +0000
+++ hplip-3.10.9/prnt/cupsext/cupsext.c	2010-11-12 15:58:27.467044126 +0000
@@ -510,6 +510,7 @@ PyObject * addPrinter( PyObject * self, 
         strcpy( info, name );
 
     /* Connect to the HTTP server */
+    cupsSetUser ("root");
     if ( ( http = httpConnectEncrypt( cupsServer(), ippPort(), cupsEncryption() ) ) == NULL )
     {
         r = 0;
