diff -Nur ccache-2.4-old/ccache.c ccache-2.4/ccache.c
--- ccache-2.4-old/ccache.c	2009-02-06 10:36:25.000000000 +0200
+++ ccache-2.4/ccache.c	2009-02-06 10:39:22.000000000 +0200
@@ -675,9 +675,14 @@
 
 		/* these are too hard */
 		if (strcmp(argv[i], "-fbranch-probabilities")==0 ||
-		    strcmp(argv[i], "-M") == 0 ||
-		    strcmp(argv[i], "-MM") == 0 ||
-		    strcmp(argv[i], "-x") == 0) {
+			strcmp(argv[i], "-fprofile-arcs") == 0 ||
+			strcmp(argv[i], "-fprofile-generate") == 0 ||
+			strcmp(argv[i], "-fprofile-use") == 0 ||
+			strcmp(argv[i], "-ftest-coverage") == 0 ||
+			strcmp(argv[i], "--coverage") == 0 ||
+			strcmp(argv[i], "-M") == 0 ||
+			strcmp(argv[i], "-MM") == 0 ||
+			strcmp(argv[i], "-x") == 0) {
 			cc_log("argument %s is unsupported\n", argv[i]);
 			stats_update(STATS_UNSUPPORTED);
 			failed();
