diff -Nur ccache-2.4-old/Makefile.in ccache-2.4/Makefile.in
--- ccache-2.4-old/Makefile.in	2009-02-06 10:28:06.000000000 +0200
+++ ccache-2.4/Makefile.in	2009-02-06 10:30:57.000000000 +0200
@@ -21,7 +21,7 @@
 docs: ccache.1 web/ccache-man.html
 
 ccache$(EXEEXT): $(OBJS) $(HEADERS)
-	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(OBJS) $(LIBS)
 
 ccache.1: ccache.yo
 	-yodl2man -o ccache.1 ccache.yo
