diff -p -up nasm-2.07rc6/ndisasm.c.format-strings nasm-2.07rc6/ndisasm.c
--- nasm-2.07rc6/ndisasm.c.format-strings	2009-07-15 22:44:06.000000000 +0200
+++ nasm-2.07rc6/ndisasm.c	2009-07-16 10:57:04.000000000 +0200
@@ -114,7 +114,7 @@ int main(int argc, char **argv)
                     p++;
                     break;
                 case 'h':
-                    fprintf(stderr, help);
+                    fprintf(stderr, "%s", help);
                     return 0;
                 case 'r':
                 case 'v':
diff -p -up nasm-2.07rc6/output/outieee.c.format-strings nasm-2.07rc6/output/outieee.c
diff -p -up nasm-2.07rc6/rdoff/rdflib.c.format-strings nasm-2.07rc6/rdoff/rdflib.c
--- nasm-2.07rc6/rdoff/rdflib.c.format-strings	2009-07-15 22:44:06.000000000 +0200
+++ nasm-2.07rc6/rdoff/rdflib.c	2009-07-16 10:57:04.000000000 +0200
@@ -166,7 +166,7 @@ int main(int argc, char **argv)
 
     if (argc < 3 || !strncmp(argv[1], "-h", 2)
         || !strncmp(argv[1], "--h", 3)) {
-        printf(usage);
+        printf("%s", usage);
         exit(1);
     }
 
diff -p -up nasm-2.07rc6/output/outieee.c~ nasm-2.07rc6/output/outieee.c
--- nasm-2.07rc6/output/outieee.c~	2009-07-15 22:44:06.000000000 +0200
+++ nasm-2.07rc6/output/outieee.c	2009-07-16 11:09:43.000000000 +0200
@@ -1157,7 +1157,7 @@ static void ieee_putascii(char *format, 
         if ((uint8_t)buffer[i] > 31)
             checksum += buffer[i];
     va_end(ap);
-    fprintf(ofp, buffer);
+    fprintf(ofp, "%s", buffer);
 }
 
 /*
