Index: qt-everywhere-opensource-src-4.7.0-beta2/tools/assistant/tools/assistant/assistant.pro
===================================================================
--- a/tools/assistant/tools/assistant/assistant.pro
+++ a/tools/assistant/tools/assistant/assistant.pro
@@ -13,6 +13,8 @@ target.path = $$[QT_INSTALL_BINS]
 INSTALLS += target
 DEPENDPATH += ../shared
 
+LIBS += -lphonon -lQtDBus -lQtCLucene
+
 # ## Work around a qmake issue when statically linking to
 # ## not-yet-installed plugins
 QMAKE_LIBDIR += $$QT_BUILD_TREE/plugins/sqldrivers
--- a/tools/assistant/tools/qhelpgenerator/qhelpgenerator.pro.backup	2010-08-20 16:40:31.460000182 +0300
+++ a/tools/assistant/tools/qhelpgenerator/qhelpgenerator.pro	2010-08-20 16:40:41.903000402 +0300
@@ -5,6 +5,8 @@
 CONFIG -= app_bundle
 QT += network
 
+LIBS += -lQtCLucene
+
 target.path=$$[QT_INSTALL_BINS]
 INSTALLS += target
 
--- a/tools/assistant/tools/qcollectiongenerator/qcollectiongenerator.pro.backup	2010-08-20 16:42:41.020002930 +0300
+++ a/tools/assistant/tools/qcollectiongenerator/qcollectiongenerator.pro	2010-08-20 16:43:00.997997639 +0300
@@ -7,6 +7,9 @@
     warn_on \
     help \
     console
+
+LIBS += -lQtCLucene
+
 CONFIG -= app_bundle
 target.path = $$[QT_INSTALL_BINS]
 INSTALLS += target
