Index: qt-everywhere-opensource-src-4.7.1/mkspecs/common/g++.conf
===================================================================
--- qt-everywhere-opensource-src-4.7.1.orig/mkspecs/common/g++.conf
+++ qt-everywhere-opensource-src-4.7.1/mkspecs/common/g++.conf
@@ -2,12 +2,12 @@
 # qmake configuration for common gcc
 #
 
-QMAKE_CC		= gcc
+QMAKE_CC		= PARDUS_CC
 QMAKE_CFLAGS		+= -pipe
 QMAKE_CFLAGS_DEPS	+= -M
 QMAKE_CFLAGS_WARN_ON	+= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	+= -w
-QMAKE_CFLAGS_RELEASE	+= -O2
+QMAKE_CFLAGS_RELEASE	+= PARDUS_CFLAGS
 QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -O2 -g
 QMAKE_CFLAGS_DEBUG	+= -g
 QMAKE_CFLAGS_SHLIB	+= -fPIC
@@ -17,7 +17,7 @@ QMAKE_CFLAGS_HIDESYMS   += -fvisibility=
 QMAKE_CFLAGS_PRECOMPILE += -x c-header -c ${QMAKE_PCH_INPUT} -o ${QMAKE_PCH_OUTPUT}
 QMAKE_CFLAGS_USE_PRECOMPILE += -include ${QMAKE_PCH_OUTPUT_BASE}
 
-QMAKE_CXX		= g++
+QMAKE_CXX		= PARDUS_CXX
 QMAKE_CXXFLAGS		+= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	+= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	+= $$QMAKE_CFLAGS_WARN_ON
@@ -32,12 +32,12 @@ QMAKE_CXXFLAGS_HIDESYMS += $$QMAKE_CFLAG
 QMAKE_CXXFLAGS_PRECOMPILE += -x c++-header -c ${QMAKE_PCH_INPUT} -o ${QMAKE_PCH_OUTPUT}
 QMAKE_CXXFLAGS_USE_PRECOMPILE = $$QMAKE_CFLAGS_USE_PRECOMPILE
 
-QMAKE_LINK		= g++
-QMAKE_LINK_SHLIB	= g++
-QMAKE_LINK_C		= gcc
-QMAKE_LINK_C_SHLIB	= gcc
+QMAKE_LINK		= PARDUS_CXX
+QMAKE_LINK_SHLIB	= PARDUS_CXX
+QMAKE_LINK_C		= PARDUS_CC
+QMAKE_LINK_C_SHLIB	= PARDUS_CC
 QMAKE_LFLAGS		+=
-QMAKE_LFLAGS_RELEASE	+= -Wl,-O1
+QMAKE_LFLAGS_RELEASE	+= PARDUS_LDFLAGS
 QMAKE_LFLAGS_DEBUG	+=
 QMAKE_LFLAGS_APP	+=
 QMAKE_LFLAGS_SHLIB	+= -shared
Index: qt-everywhere-opensource-src-4.7.1/mkspecs/common/linux.conf
===================================================================
--- qt-everywhere-opensource-src-4.7.1.orig/mkspecs/common/linux.conf
+++ qt-everywhere-opensource-src-4.7.1/mkspecs/common/linux.conf
@@ -51,7 +51,7 @@ QMAKE_COPY_DIR        = $(COPY) -r
 QMAKE_MOVE            = mv -f
 QMAKE_DEL_FILE        = rm -f
 QMAKE_DEL_DIR         = rmdir
-QMAKE_STRIP           = strip
+QMAKE_STRIP           =
 QMAKE_STRIPFLAGS_LIB += --strip-unneeded
 QMAKE_CHK_DIR_EXISTS  = test -d
 QMAKE_MKDIR           = mkdir -p
