diff -Naur iscan-2.11.0/configure.ac iscan-2.11.0-patched/configure.ac
--- iscan-2.11.0/configure.ac	2008-02-18 02:53:41.000000000 +0200
+++ iscan-2.11.0-patched/configure.ac	2008-09-07 23:42:45.000000000 +0300
@@ -74,8 +74,8 @@
     AC_DEFINE([HAVE_GTK_2], 1,
 	      [Define to 1 if you have the gtk+-2.0 library.])
    ],[
-    PKG_CHECK_MODULES(GTK, gtk+) 
-    PKG_CHECK_MODULES(GDK_IMLIB, imlibgdk)
+    AC_DEFINE([HAVE_GTK_2], 0) 
+    #PKG_CHECK_MODULES(GDK_IMLIB, imlibgdk)
    ])
 
 PKG_CHECK_MODULES(GIMP, gimp-2.0,
diff -Naur iscan-2.11.0/doc/Makefile.am iscan-2.11.0-patched/doc/Makefile.am
--- iscan-2.11.0/doc/Makefile.am	2008-02-18 02:53:40.000000000 +0200
+++ iscan-2.11.0-patched/doc/Makefile.am	2008-09-07 23:42:45.000000000 +0300
@@ -23,7 +23,7 @@
 man_MANS = sane-epkowa.5
 
 if ENABLE_FRONTEND
-man_MANS += iscan.1
+man_MANS += 
 endif
 
 
diff -Naur iscan-2.11.0/doc/Makefile.in iscan-2.11.0-patched/doc/Makefile.in
--- iscan-2.11.0/doc/Makefile.in	2008-02-18 02:54:18.000000000 +0200
+++ iscan-2.11.0-patched/doc/Makefile.in	2008-09-07 23:42:45.000000000 +0300
@@ -35,7 +35,7 @@
 POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
-@ENABLE_FRONTEND_TRUE@am__append_1 = iscan.1
+@ENABLE_FRONTEND_TRUE@am__append_1 = 
 subdir = doc
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
diff -Naur iscan-2.11.0/Makefile.am iscan-2.11.0-patched/Makefile.am
--- iscan-2.11.0/Makefile.am	2008-02-18 02:53:41.000000000 +0200
+++ iscan-2.11.0-patched/Makefile.am	2008-09-07 23:42:45.000000000 +0300
@@ -29,8 +29,8 @@
 	lib \
 	sanei \
 	backend \
-	non-free \
-	frontend \
+	\
+	\
 	utils \
 	intl \
 	doc
