Index: kdebase-4.3.0/apps/dolphin/src/settings/dolphin_generalsettings.kcfg
===================================================================
--- kdebase-4.3.0.orig/apps/dolphin/src/settings/dolphin_generalsettings.kcfg
+++ kdebase-4.3.0/apps/dolphin/src/settings/dolphin_generalsettings.kcfg
@@ -47,7 +47,7 @@
         </entry>
         <entry name="RenameInline" type="Bool">
             <label>Rename inline</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="ShowSelectionToggle" type="Bool">
             <label>Show selection toggle</label>
@@ -59,7 +59,7 @@
         </entry>
         <entry name="ShowCopyMoveMenu" type="Bool">
             <label>Show 'Copy To' and 'Move To' commands in context menu</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
         <entry name="ViewPropsTimestamp" type="DateTime" >
             <label>Timestamp since when the view properties are valid</label>
@@ -74,7 +74,7 @@
         </entry>
         <entry name="ShowSpaceInfo" type="Bool">
             <label>Show the space information in the statusbar</label>
-            <default>false</default>
+            <default>true</default>
         </entry>
     </group>
 </kcfg>
