Index: qbittorrent-2.7.0/src/preferences/preferences.h
===================================================================
--- qbittorrent-2.7.0.orig/src/preferences/preferences.h
+++ qbittorrent-2.7.0/src/preferences/preferences.h
@@ -142,7 +142,7 @@ public:
   }
 
   bool closeToTray() const {
-    return value(QString::fromUtf8("Preferences/General/CloseToTray"), false).toBool();
+    return value(QString::fromUtf8("Preferences/General/CloseToTray"), true).toBool();
   }
 
   void setCloseToTray(bool b) {
@@ -158,7 +158,7 @@ public:
   }
 
   bool isSlashScreenDisabled() const {
-    return value(QString::fromUtf8("Preferences/General/NoSplashScreen"), false).toBool();
+    return value(QString::fromUtf8("Preferences/General/NoSplashScreen"), true).toBool();
   }
 
   void setSplashScreenDisabled(bool b) {
@@ -1034,7 +1034,7 @@ public:
   }
 
   bool useMonochromeTrayIcon() const {
-    return value(QString::fromUtf8("Preferences/Advanced/useMonochromeTrayIcon"), false).toBool();
+    return value(QString::fromUtf8("Preferences/Advanced/useMonochromeTrayIcon"), true).toBool();
   }
   void setUseMonochromeTrayIcon(bool use) {
     setValue(QString::fromUtf8("Preferences/Advanced/useMonochromeTrayIcon"), use);
