diff -Nur make-3.81/tests/scripts/misc/close_stdout make-3.81-new/tests/scripts/misc/close_stdout
--- make-3.81/tests/scripts/misc/close_stdout	2005-06-13 01:22:07.000000000 +0300
+++ make-3.81-new/tests/scripts/misc/close_stdout	2009-02-06 12:31:48.000000000 +0200
@@ -3,7 +3,7 @@
 $description = "Make sure make exits with an error if stdout is full.";
 
 if (-e '/dev/full') {
-  run_make_test('', '-v > /dev/full', '#MAKE#: write error', 256);
+  run_make_test('', '-v > /dev/full', '#MAKE#: write error: No space left on device', 256);
 }
 
 1;
