diff -Nur gfxtheme-pardus-0.11.1-old/src/common.inc gfxtheme-pardus-0.11.1/src/common.inc
--- gfxtheme-pardus-0.11.1-old/src/common.inc	2010-09-16 15:11:48.322478165 +0300
+++ gfxtheme-pardus-0.11.1/src/common.inc	2010-09-16 15:11:56.568440327 +0300
@@ -340,21 +340,6 @@
     menu.texts menu.entry get "%s " bc.cmd sprintf
   } ifelse
 
-  syslinux {
-    /splash_file 32 string def
-
-    video.modes.list xmenu.video .xm_current get get
-    dup .vm_mode get 0 ge {
-      dup .vm_height get exch .vm_width get
-      ",%04d%04d.spl" splash_file sprintf
-    } { pop } ifelse
-
-    % check whether splash file exists
-    splash_file 1 add filesize .undef eq {
-      splash_file 0 0 put	% empty string
-    } if
-  } if
-
   xmenu.kernelopts {
     kernelopts.options xmenu.kernelopts .xm_current get get dup "" ne {
       "%s " bc.cmd dup length add sprintf
@@ -402,25 +387,6 @@
 
   /splash_file xfree
 
-  syslinux {
-    video.modes.list xmenu.video .xm_current get get .vm_mode get
-
-    dup 0 ge {
-      0x200 add "vga=0x%x " bc.cmd dup length add sprintf
-    } {
-      -1 eq {
-        "textmode=1 " bc.cmd dup length add sprintf
-      } if
-    } ifelse
-
-    v_impaired 1 ge {
-      "braille=1 " bc.cmd dup length add sprintf
-    } if
-
-    v_impaired 2 ge {
-      "linemode=1 " bc.cmd dup length add sprintf
-    } if
-  } if
 
     config.lang {
       %config.lang "en_US" ne {
@@ -1246,11 +1212,6 @@
       sound.getsamplerate { sound.getsamplerate sound.setsamplerate } if
     } ifelse
 
-    xmenu.video .xm_current 0 put
-    % redraw unless another menu is still open
-    window.current {
-      window.current .xmenu get .undef eq { video.update } if
-    } if
 
     .undef 0 idle
     kroete.file free /kroete.file .undef def
diff -Nur gfxtheme-pardus-0.11.1-old/src/panel.inc gfxtheme-pardus-0.11.1/src/panel.inc
--- gfxtheme-pardus-0.11.1-old/src/panel.inc	2010-09-16 15:11:48.322478165 +0300
+++ gfxtheme-pardus-0.11.1/src/panel.inc	2010-09-16 15:12:28.995542406 +0300
@@ -44,12 +44,12 @@
       livecd {
 
         [ keyF2 0 "F2" /panel.lang /panel.lang.width /panel.lang.update /lang.init ]
-        [ keyF3 0 "F3" /panel.video /panel.video.width /panel.video.update /video.init ]
+        %[ keyF3 0 "F3" /panel.video /panel.video.width /panel.video.update /video.init ]
 
       } {
 
         [ keyF2 0 "F2" /panel.lang /panel.lang.width /panel.lang.update /lang.init ]
-        [ keyF3 0 "F3" /panel.video /panel.video.width /panel.video.update /video.init ]
+        %[ keyF3 0 "F3" /panel.video /panel.video.width /panel.video.update /video.init ]
         % [ keyF4 0 "F4" /panel.install /panel.install.width /panel.install.update /install.init ]
         [ keyF4 0 "F4" /panel.zorg /panel.zorg.width /panel.zorg.update /zorg.init ]
         [ keyF5 0 "F5" /panel.kernelopts /panel.kernelopts.width /panel.kernelopts.update /kernelopts.init ]
