diff -Nur gfxtheme-pardus-0.11.1-old/src/bsplash.inc gfxtheme-pardus-0.11.1/src/bsplash.inc
--- gfxtheme-pardus-0.11.1-old/src/bsplash.inc	2010-10-21 14:41:56.971196119 +0300
+++ gfxtheme-pardus-0.11.1/src/bsplash.inc	2010-10-21 14:42:39.497165764 +0300
@@ -220,7 +220,7 @@
   /b2_text_tmp b2_text imgsize 0 0 moveto savescreen def
   % /b2_spot_tmp b2_spot imgsize 0 0 moveto savescreen def
 
-  250 170 moveto
+  318 226 moveto
   /b2_orig b2_text imgsize savescreen def
   /b2_buf b2_text imgsize savescreen def
 
@@ -283,7 +283,7 @@
   0 0 moveto
   0xffffff b2_text_tmp b2_buf blend
 
-  250 170 moveto b2_buf restorescreen
+  318 226 moveto b2_buf restorescreen
 
   false
 
