diff -Nur gfxtheme-pardus-0.11.3-old/src/dia_kernelopts.inc gfxtheme-pardus-0.11.3/src/dia_kernelopts.inc
--- gfxtheme-pardus-0.11.3-old/src/dia_kernelopts.inc	2011-01-18 10:27:32.502433528 +0200
+++ gfxtheme-pardus-0.11.3/src/dia_kernelopts.inc	2011-01-18 10:27:13.646372022 +0200
@@ -11,6 +11,7 @@
 /kernelopts.default 0 def
 
 /kernelopts.options [
+  "splash quiet"
   ""
   "apm=off acpi=off mce=off barrier=off ide=nodma idewait=50 i8042.nomux psmouse.proto=bare irqpoll pci=nommconf"
   "acpi=off"
@@ -32,6 +33,7 @@
   xmenu .xm_current kernelopts.default put
   xmenu .xm_list [
     /txt_kernel_default
+    /txt_zorg_off
     /txt_kernel_safe
     /txt_kernel_noacpi
     /txt_kernel_nolapic
