diff -Naur exo-0.3.105/exo-mount/mount.rc exo-0.3.105.tpg/exo-mount/mount.rc
--- exo-0.3.105/exo-mount/mount.rc	2009-10-18 15:53:04.000000000 +0000
+++ exo-0.3.105.tpg/exo-mount/mount.rc	2009-10-23 22:52:19.000000000 +0000
@@ -34,12 +34,14 @@
 
 [vfat]
 uid=<auto>
+utf8=true
 shortname=winnt
 # FreeBSD specific option
 longnames=true
 
 [iso9660]
 uid=<auto>
+utf8=true
 
 [udf]
 uid=<auto>
@@ -47,6 +49,7 @@
 
 [ntfs]
 uid=<auto>
+utf8=true
 
 [ntfs-3g]
 uid=<auto>
