diff -Naur exo-0.3.105/exo-mount-notify/main.c exo-0.3.105.tpg/exo-mount-notify/main.c
--- exo-0.3.105/exo-mount-notify/main.c	2009-10-18 15:53:04.000000000 +0000
+++ exo-0.3.105.tpg/exo-mount-notify/main.c	2009-10-23 22:56:17.000000000 +0000
@@ -247,7 +247,7 @@
   g_signal_connect (G_OBJECT (notification), "closed", G_CALLBACK (gtk_main_quit), NULL);
   g_object_add_weak_pointer (G_OBJECT (notification), (gpointer) &notification);
   notify_notification_set_urgency (notification, NOTIFY_URGENCY_CRITICAL);
-  notify_notification_set_timeout (notification, NOTIFY_EXPIRES_NEVER);
+  notify_notification_set_timeout (notification, NOTIFY_EXPIRES_DEFAULT);
   notify_notification_show (notification, NULL);
   g_free (message);
 
