diff -Naur exo-0.3.4/exo/Makefile.am exo-0.3.4.oden/exo/Makefile.am
--- exo-0.3.4/exo/Makefile.am	2007-12-02 13:33:44.000000000 +0100
+++ exo-0.3.4.oden/exo/Makefile.am	2008-08-20 10:21:53.000000000 +0200
@@ -107,7 +107,7 @@
 
 libexo_0_3_la_LIBADD =							\
 	$(GTK_LIBS)							\
-	$(LIBXFCE4UTIL_LIBS)
+	$(LIBXFCE4UTIL_LIBS) -lm -lX11
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = exo-0.3.pc
diff -Naur exo-0.3.4/exo/Makefile.in exo-0.3.4.oden/exo/Makefile.in
--- exo-0.3.4/exo/Makefile.in	2007-12-02 13:37:19.000000000 +0100
+++ exo-0.3.4.oden/exo/Makefile.in	2008-08-20 10:22:24.000000000 +0200
@@ -437,7 +437,7 @@
 
 libexo_0_3_la_LIBADD = \
 	$(GTK_LIBS)							\
-	$(LIBXFCE4UTIL_LIBS)
+	$(LIBXFCE4UTIL_LIBS) -lm -lX11
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = exo-0.3.pc
