Index: curl-7.20.1/lib/urldata.h
===================================================================
--- curl-7.20.1.orig/lib/urldata.h
+++ curl-7.20.1/lib/urldata.h
@@ -52,7 +52,7 @@
 #define DICT_DEFINE3 "/LOOKUP:"
 
 #define CURL_DEFAULT_USER "anonymous"
-#define CURL_DEFAULT_PASSWORD "ftp@example.com"
+#define CURL_DEFAULT_PASSWORD "-curl@"
 
 /* length of longest IPv6 address string including the trailing null */
 #define MAX_IPADR_LEN sizeof("ffff:ffff:ffff:ffff:ffff:ffff:255.255.255.255")
