--- browser/components/shell/src/nsGNOMEShellService.cpp
+++ browser/components/shell/src/nsGNOMEShellService.cpp
@@ -133,7 +133,11 @@
   rv = appPath->AppendNative(NS_LITERAL_CSTRING(MOZ_APP_NAME));
   NS_ENSURE_SUCCESS(rv, rv);
 
-  return appPath->GetNativePath(mAppPath);
+  /* This path to firefox is hardcoded for the default path setting */
+  mAppPath.Assign ("/usr/bin/firefox");
+  
+  //return appPath->GetNativePath(mAppPath);
+  return NS_OK;
 }
 
 NS_IMPL_ISUPPORTS2(nsGNOMEShellService, nsIShellService, nsIShellService_MOZILLA_1_8_BRANCH)
