Index: qtermwidget/qtermwidget.sip
===================================================================
--- qtermwidget.orig/qtermwidget.sip
+++ qtermwidget/qtermwidget.sip
@@ -19,6 +19,7 @@ public:
         ScrollBarLeft=1,
         ScrollBarRight=2
     };
+    void startShellProgram();
 	void setTerminalFont(QFont &font);
 	void setShellProgram(QString &progname);
 	void setArgs(QStringList &args);
