Index: gyachi-1.2.10/gyachi.desktop
===================================================================
--- gyachi-1.2.10.orig/gyachi.desktop
+++ gyachi-1.2.10/gyachi.desktop
@@ -1,12 +1,14 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=GYachE Improved
 Name[sr]=Усавршени GYachE
+GenericName=GYachE Improved
+GenericName[tr]=Gelişmiş GYachE
+Comment[tr]=Yahoo! İstemcisi
 Comment[de]=Yahoo! Klient
 Comment[sr]=Програм за Yahoo!
 Comment=Yahoo! Client
 Exec=gyachi
-Icon=gyachi.png
+Icon=gyachi
 Terminal=false
 Type=Application
-Categories=Application;Network;
+Categories=Network;
