diff -up gyachi-1.2.10/webcam/gyacheupload-v4l.c.libv4l1 gyachi-1.2.10/webcam/gyacheupload-v4l.c
--- gyachi-1.2.10/webcam/gyacheupload-v4l.c.libv4l1	2011-05-17 10:50:46.813232393 -0400
+++ gyachi-1.2.10/webcam/gyacheupload-v4l.c	2011-05-17 10:51:11.372330954 -0400
@@ -112,7 +112,7 @@
 #endif
 
 /* V4L1 */
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 /*  V4L2 */
 #include <linux/videodev2.h>
diff -up gyachi-1.2.10/webcam/Makefile.am.libv4l1 gyachi-1.2.10/webcam/Makefile.am
