Index: k9copy-2.3.6-Source/k9copy_assistant.desktop
===================================================================
--- k9copy-2.3.6-Source.orig/k9copy_assistant.desktop
+++ k9copy-2.3.6-Source/k9copy_assistant.desktop
@@ -6,11 +6,11 @@ Name=k9copy assistant
 Exec=k9copy --assistant -caption "%c" %i %m  %u
 Comment=DVD9 to DVD5 backup tool
 Comment[fr]=Backup de DVD9 en DVD5
-Comment[tr]=DVD9 DVD5 yedekleme aracı
+Comment[tr]=DVD9'dan DVD5'e yedekleme aracı
 Comment[ru]=Инструмент для копирования с DVD9 на DVD5
 GenericName=Video DVD Backup
 GenericName[fr]=Backup de DVD Video
-GenericName[tr]=DVD Vidyo Yedekleme Aracı
+GenericName[tr]=Video Yedekleme Aracı
 GenericName[ru]=Копирование с видео-DVD
 MimeTypes=
 Terminal=false
