Index: network-manager-applet-0.8.1/nm-applet.desktop.in
===================================================================
--- network-manager-applet-0.8.1.orig/nm-applet.desktop.in
+++ network-manager-applet-0.8.1/nm-applet.desktop.in
@@ -10,3 +10,4 @@ X-GNOME-Bugzilla-Product=NetworkManager
 X-GNOME-Bugzilla-Component=general
 X-GNOME-Autostart-enabled=true
 Hidden[$e]=$(grep -q 'DEFAULT="False"' /etc/conf.d/NetworkManager && echo true || echo false)
+NotShowIn=KDE
