Index: network-manager-applet-0.8.1/src/utils/nma-bling-spinner.c
===================================================================
--- network-manager-applet-0.8.1.orig/src/utils/nma-bling-spinner.c
+++ network-manager-applet-0.8.1/src/utils/nma-bling-spinner.c
@@ -154,7 +154,7 @@ nma_bling_spinner_init (NmaBlingSpinner
 	priv->current = 0;
 	priv->timeout = 0;
 
-	GTK_WIDGET_SET_FLAGS (GTK_WIDGET (spinner), GTK_NO_WINDOW);
+	gtk_widget_set_has_window(GTK_WIDGET (spinner), FALSE);
 }
 
 static gboolean
Index: network-manager-applet-0.8.1/src/wireless-dialog.c
===================================================================
--- network-manager-applet-0.8.1.orig/src/wireless-dialog.c
+++ network-manager-applet-0.8.1/src/wireless-dialog.c
@@ -1069,7 +1069,6 @@ internal_init (NMAWirelessDialog *self,
 	gtk_container_set_border_width (GTK_CONTAINER (self), 6);
 	gtk_window_set_default_size (GTK_WINDOW (self), 488, -1);
 	gtk_window_set_resizable (GTK_WINDOW (self), FALSE);
-	gtk_dialog_set_has_separator (GTK_DIALOG (self), FALSE);
 
 	priv->auth_only = auth_only;
 	if (auth_only)
