Index: network-manager-applet-0.8.1/src/utils/nma-bling-spinner.c
===================================================================
--- network-manager-applet-0.8.1.orig/src/utils/nma-bling-spinner.c
+++ network-manager-applet-0.8.1/src/utils/nma-bling-spinner.c
@@ -154,7 +154,7 @@ nma_bling_spinner_init (NmaBlingSpinner
 	priv->current = 0;
 	priv->timeout = 0;
 
-	GTK_WIDGET_SET_FLAGS (GTK_WIDGET (spinner), GTK_NO_WINDOW);
+	gtk_widget_set_can_default(GTK_WIDGET (spinner), TRUE);
 }
 
 static gboolean
