Index: network-manager-applet-0.8.1/src/utils/tests/test-utils.c
===================================================================
--- network-manager-applet-0.8.1.orig/src/utils/tests/test-utils.c
+++ network-manager-applet-0.8.1/src/utils/tests/test-utils.c
@@ -413,9 +413,7 @@ test_ap_hash_foobar_asdf11_adhoc_wpa_rsn
 	g_assert (strcmp (d->foobar_adhoc_wpa_rsn, d->asdf11_adhoc_wpa_rsn));
 }
 
-typedef void (*TCFunc)(void);
-
-#define TESTCASE(t, d) g_test_create_case (#t, 0, d, NULL, (TCFunc) t, NULL)
+#define TESTCASE(t, d) g_test_create_case (#t, 0, d, NULL, (GTestFixtureFunc) t, NULL)
 
 int main (int argc, char **argv)
 {
