--- ocaml-3.11.0+beta1.orig/tools/Makefile.shared	2007-11-22 22:14:43.000000000 +0000
+++ ocaml-3.11.0+beta1/tools/Makefile.shared	2008-11-18 15:42:44.000000000 +0000
@@ -107,9 +107,6 @@
 	sed -e "s|%%BINDIR%%|$(BINDIR)|" \
             -e "s|%%SUPPORTS_SHARED_LIBRARIES%%|$(SUPPORTS_SHARED_LIBRARIES)|" \
             -e "s|%%MKSHAREDLIB%%|$(MKSHAREDLIB)|" \
-            -e "s|%%BYTECCRPATH%%|$(BYTECCRPATH)|" \
-            -e "s|%%NATIVECCRPATH%%|$(NATIVECCRPATH)|" \
-            -e "s|%%MKSHAREDLIBRPATH%%|$(MKSHAREDLIBRPATH)|" \
             -e "s|%%RANLIB%%|$(RANLIB)|" \
           ocamlmklib.mlp >> ocamlmklib.ml
 
