--- src/rekonq.kcfg.orig	2010-07-05 23:13:01.000000000 +0200
+++ src/rekonq.kcfg	2010-07-09 19:48:44.000000000 +0200
@@ -36,10 +36,10 @@
 <!-- New Tab Page Settings -->
 <group name="NewTabPage">
     <entry name="previewNames" type="StringList">
-        <default>KDE Homepage,UserBase,rekonq site</default>
+        <default>Pardus Homepage,Pardus Users Portal</default>
     </entry>
     <entry name="previewUrls" type="StringList">
-        <default>http://www.kde.org,http://userbase.kde.org,http://rekonq.sourceforge.net</default>
+	    <default>http://www.pardus.org.tr,http://www.ozgurlukicin.com/</default>
     </entry>
 </group>
 
@@ -56,7 +56,7 @@
         <default>0</default>
     </entry>
     <entry name="homePage" type="String">
-        <default>http://www.kde.org/</default>
+        <default>http://www.pardus.org.tr/</default>
     </entry>
     <entry name="useNewTabPage" type="Bool">
         <default>false</default>
