Index: pardus-default-settings-0.1.3/usr/share/kde4/config/kdeglobals
===================================================================
--- pardus-default-settings-0.1.3.orig/usr/share/kde4/config/kdeglobals
+++ pardus-default-settings-0.1.3/usr/share/kde4/config/kdeglobals
@@ -1,6 +1,6 @@
 [General]
 widgetStyle=qtcurve
-BrowserApplication[$e]=mozillafirefox.desktop
+BrowserApplication[$e]=!firefox
 
 [Icons]
 Theme=milky
