From 57bef052ce4c9e44592bf7afd5106ae8a3b71d77 Mon Sep 17 00:00:00 2001
From: Ionut Biru <ibiru@archlinux.org>
Date: Mon, 08 Nov 2010 16:53:46 +0000
Subject: test: install nm-online binary

---
diff --git a/test/Makefile.am b/test/Makefile.am
index 8e235e4..7bdaf0d 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -10,9 +10,9 @@ AM_CPPFLAGS = \
 	-DNM_RUN_DIR=\"$(rundir)\" \
 	-DDATADIR=\"$(datadir)\"
 
-bin_PROGRAMS = nm-tool
+bin_PROGRAMS = nm-tool nm-online
 
-noinst_PROGRAMS = nm-online libnm-glib-test
+noinst_PROGRAMS = libnm-glib-test
 
 nm_tool_SOURCES = nm-tool.c
 nm_tool_LDADD = \
--
cgit v0.8.3-6-g21f6
