From cbb0d6d116e2d5e844c2c118f2f707d10e6b6fc4 Mon Sep 17 00:00:00 2001
From: Dan Williams <dcbw@redhat.com>
Date: Mon, 08 Nov 2010 15:09:54 +0000
Subject: build: create NM state directory at install time

---
diff --git a/src/Makefile.am b/src/Makefile.am
index 8c01df0..d7dccaa 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -275,8 +275,10 @@ EXTRA_DIST = \
 	$(NetworkManager_DATA)
 
 rundir=$(localstatedir)/run/NetworkManager
+statedir=$(localstatedir)/lib/NetworkManager
 install-data-hook:
 	$(mkinstalldirs) -m 0700 $(DESTDIR)$(rundir)
+	$(mkinstalldirs) -m 0700 $(DESTDIR)$(statedir)
 
 CLEANFILES = $(BUILT_SOURCES)
 
--
cgit v0.8.3-6-g21f6
