Index: NetworkManager-0.8/configure.ac
===================================================================
--- NetworkManager-0.8.orig/configure.ac
+++ NetworkManager-0.8/configure.ac
@@ -417,8 +417,6 @@ AC_SUBST(SYSTEM_CA_PATH)
 
 NM_COMPILER_WARNINGS
 
-GTK_DOC_CHECK(1.0)
-
 dnl
 dnl Tests
 dnl
@@ -501,9 +499,6 @@ man/nm-system-settings.conf.5
 man/nm-tool.1
 po/Makefile.in
 policy/Makefile
-docs/Makefile
-docs/libnm-glib/Makefile
-docs/libnm-util/Makefile
 NetworkManager.pc
 examples/Makefile
 examples/python/Makefile
Index: NetworkManager-0.8/Makefile.am
===================================================================
--- NetworkManager-0.8.orig/Makefile.am
+++ NetworkManager-0.8/Makefile.am
@@ -14,7 +14,6 @@ SUBDIRS = \
 	test \
 	po \
 	man \
-	docs \
 	examples
 
 EXTRA_DIST = \
