Index: NetworkManager-0.8.2/system-settings/plugins/keyfile/tests/test-keyfile.c
===================================================================
--- NetworkManager-0.8.2.orig/system-settings/plugins/keyfile/tests/test-keyfile.c
+++ NetworkManager-0.8.2/system-settings/plugins/keyfile/tests/test-keyfile.c
@@ -2022,7 +2022,8 @@ int main (int argc, char **argv)
 	base = g_path_get_basename (argv[0]);
 	fprintf (stdout, "%s: SUCCESS\n", base);
 	g_free (base);
-	dbus_g_connection_unref (bus);
+	if (bus)
+		dbus_g_connection_unref (bus);
 	return 0;
 }
 
