Index: avogadro-1.0.1/libavogadro/src/python/sip.cpp
===================================================================
--- avogadro-1.0.1.orig/libavogadro/src/python/sip.cpp
+++ avogadro-1.0.1/libavogadro/src/python/sip.cpp
@@ -42,32 +42,13 @@ const sipAPIDef *sip_API = 0;
 
 bool init_sip_api()
 {
-  // import the sip module
-  object sip_module = import("sip");
-  if (!sip_module.ptr()) {
-    std::cout << "Could not import sip python module." << std::endl;
-    return false;
-  }
+  sip_API = reinterpret_cast<const sipAPIDef*>(PyCapsule_Import("sip._C_API", 0));
 
-  // get the dictionary
-  dict sip_dict = extract<dict>(sip_module.attr("__dict__"));
-  if (!sip_dict.ptr()) {
-    std::cout << "Could not find the __dict__ attribute in the sip python module." << std::endl;
-    return false;
-  }
-  // get the _C_API object from the dictionary
-  object sip_capi_obj = sip_dict.get("_C_API");
-  if (!sip_capi_obj.ptr()) {
-    std::cout << "Could not find the _C_API entry in the sip python module dictionary." << std::endl;
-    return false;
+  if (!sip_API)
+  {
+      std::cout << "Could not find SIP C_API." << std::endl;
+      return false;
   }
-  
-  if (!PyCObject_Check(sip_capi_obj.ptr())) {
-    std::cout << "The _C_API object in the sip python module is invalid." << std::endl;
-    return false;
-  }
-
-  sip_API = reinterpret_cast<const sipAPIDef*>(PyCObject_AsVoidPtr(sip_capi_obj.ptr()));
 
   return true;
 }
