Index: vlc-0.9.6/modules/misc/freetype.c
===================================================================
--- vlc-0.9.6.orig/modules/misc/freetype.c
+++ vlc-0.9.6/modules/misc/freetype.c
@@ -61,7 +61,7 @@
 #define DEFAULT_FONT "" /* Default font found at run-time */
 #define FC_DEFAULT_FONT "Arial"
 #else
-#define DEFAULT_FONT "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
+#define DEFAULT_FONT "/usr/share/fonts/dejavu/DejaVuSans.ttf"
 #define FC_DEFAULT_FONT "Serif Bold"
 #endif
 
