--- vlc-1.1.10/projects/mozilla/support/npunix.cpp.new-xulrunner	2011-05-30 17:03:50.000000000 +0200
+++ vlc-1.1.10/projects/mozilla/support/npunix.cpp	2011-06-06 13:25:30.000000000 +0200
@@ -277,7 +277,7 @@ void NPN_ReloadPlugins(NPBool reloadPage
 #endif
 }
 
-#ifdef OJI
+#ifdef OJI_OLD
 JRIEnv* NPN_GetJavaEnv()
 {
 #if (((NP_VERSION_MAJOR << 8) + NP_VERSION_MINOR) < 20)
@@ -636,7 +636,7 @@ void Private_URLNotify(NPP instance, con
 void Private_Print(NPP instance, NPPrint* platformPrint);
 NPError Private_GetValue(NPP instance, NPPVariable variable, void *r_value);
 NPError Private_SetValue(NPP instance, NPNVariable variable, void *r_value);
-#ifdef OJI
+#ifdef OJI_OLD
 JRIGlobalRef Private_GetJavaClass(void);
 #endif
 
@@ -742,7 +742,7 @@ Private_SetValue(NPP instance, NPNVariab
     return NPP_SetValue(instance, variable, r_value);
 }
 
-#ifdef OJI
+#ifdef OJI_OLD
 JRIGlobalRef
 Private_GetJavaClass(void)
 {
@@ -767,11 +767,13 @@ Private_GetJavaClass(void)
  *  - Netscape uses the return value to identify when an object instance
  *    of this plugin should be created.
  */
+/*
 char *
 NP_GetMIMEDescription(void)
 {
     return NPP_GetMIMEDescription();
 }
+*/
 
 /*
  * NP_GetValue [optional]
@@ -859,7 +861,7 @@ NP_Initialize(NPNetscapeFuncs* nsTable,
         gNetscapeFuncs.pluginthreadasynccall =
             nsTable->pluginthreadasynccall;
 #endif
-#ifdef OJI
+#ifdef OJI_OLD
         if( minor >= NPVERS_HAS_LIVECONNECT )
         {
             gNetscapeFuncs.getJavaEnv    = nsTable->getJavaEnv;
@@ -982,7 +984,7 @@ NP_Initialize(NPNetscapeFuncs* nsTable,
             pluginFuncs->urlnotify = (NPP_URLNotifyProcPtr)(Private_URLNotify);
 #endif
         }
-#ifdef OJI
+#ifdef OJI_OLD
         if( minor >= NPVERS_HAS_LIVECONNECT )
             pluginFuncs->javaClass  = Private_GetJavaClass();
         else
@@ -1004,6 +1006,7 @@ NP_Initialize(NPNetscapeFuncs* nsTable,
  *    the last object of this kind has been destroyed.
  *
  */
+/*
 NPError
 NP_Shutdown(void)
 {
@@ -1011,3 +1014,4 @@ NP_Shutdown(void)
     NPP_Shutdown();
     return NPERR_NO_ERROR;
 }
+*/
--- vlc-1.1.10/projects/mozilla/vlcshell.cpp.new-xulrunner	2010-05-03 11:26:24.000000000 +0200
+++ vlc-1.1.10/projects/mozilla/vlcshell.cpp	2011-06-06 13:18:44.000000000 +0200
@@ -270,7 +270,7 @@ NPError NPP_Initialize( void )
 }
 
 #ifdef OJI
-jref NPP_GetJavaClass( void )
+void* NPP_GetJavaClass( void )
 {
     return NULL;
 }
--- vlc-1.1.10/projects/mozilla/vlcshell.h.new-xulrunner	2010-04-27 23:44:27.000000000 +0200
+++ vlc-1.1.10/projects/mozilla/vlcshell.h	2011-06-06 13:18:44.000000000 +0200
@@ -29,7 +29,7 @@ char * NPP_GetMIMEDescription( void );
 NPError NPP_Initialize( void );
 
 #ifdef OJI 
-jref NPP_GetJavaClass( void );
+void* NPP_GetJavaClass( void );
 #endif
 void NPP_Shutdown( void );
 
