diff -Nur binutils-2.20.51.0.10-old/ld/ldmain.c binutils-2.20.51.0.10/ld/ldmain.c
--- binutils-2.20.51.0.10-old/ld/ldmain.c	2010-07-20 16:05:40.488560601 +0300
+++ binutils-2.20.51.0.10/ld/ldmain.c	2010-07-20 16:05:16.171622223 +0300
@@ -293,6 +293,9 @@
   link_info.path_separator = ':';
   link_info.sharable_sections = FALSE;
 
+  if (getenv ("PARDUS_ASNEEDED") && atoi(getenv ("PARDUS_ASNEEDED")) > 0)
+    as_needed = TRUE;
+
   ldfile_add_arch ("");
   emulation = get_emulation (argc, argv);
   ldemul_choose_mode (emulation);
