diff -Nur binutils-2.19.51.0.9-old/bfd/Makefile.am binutils-2.19.51.0.9/bfd/Makefile.am
--- binutils-2.19.51.0.9-old/bfd/Makefile.am	2009-06-13 16:14:29.711183399 +0300
+++ binutils-2.19.51.0.9/bfd/Makefile.am	2009-06-13 16:14:51.972433050 +0300
@@ -800,7 +800,7 @@
 libbfd_la_SOURCES = $(BFD32_LIBS_CFILES)
 libbfd_la_DEPENDENCIES = $(OFILES) ofiles
 libbfd_la_LIBADD = `cat ofiles` @SHARED_LIBADD@ $(LIBDL)
-libbfd_la_LDFLAGS = -release `cat libtool-soversion` @SHARED_LDFLAGS@
+libbfd_la_LDFLAGS = -Wl,-Bsymbolic-functions -release `cat libtool-soversion` @SHARED_LDFLAGS@
 
 # libtool will build .libs/libbfd.a.  We create libbfd.a in the build
 # directory so that we don't have to convert all the programs that use
diff -Nur binutils-2.19.51.0.9-old/bfd/Makefile.in binutils-2.19.51.0.9/bfd/Makefile.in
--- binutils-2.19.51.0.9-old/bfd/Makefile.in	2009-06-13 16:14:29.755183418 +0300
+++ binutils-2.19.51.0.9/bfd/Makefile.in	2009-06-13 16:14:52.006433121 +0300
@@ -984,7 +984,7 @@
 libbfd_la_SOURCES = $(BFD32_LIBS_CFILES)
 libbfd_la_DEPENDENCIES = $(OFILES) ofiles
 libbfd_la_LIBADD = `cat ofiles` @SHARED_LIBADD@ $(LIBDL)
-libbfd_la_LDFLAGS = -release `cat libtool-soversion` @SHARED_LDFLAGS@
+libbfd_la_LDFLAGS = -Wl,-Bsymbolic-functions -release `cat libtool-soversion` @SHARED_LDFLAGS@
 
 # libtool will build .libs/libbfd.a.  We create libbfd.a in the build
 # directory so that we don't have to convert all the programs that use
diff -Nur binutils-2.19.51.0.9-old/opcodes/Makefile.am binutils-2.19.51.0.9/opcodes/Makefile.am
--- binutils-2.19.51.0.9-old/opcodes/Makefile.am	2009-06-13 16:14:29.593194272 +0300
+++ binutils-2.19.51.0.9/opcodes/Makefile.am	2009-06-13 16:15:04.603434028 +0300
@@ -394,7 +394,7 @@
 # hardcoded path to libbfd.so instead of relying on the entries in libbfd.la.
 libopcodes_la_DEPENDENCIES = $(OFILES) @SHARED_DEPENDENCIES@
 libopcodes_la_LIBADD = $(OFILES) @SHARED_LIBADD@
-libopcodes_la_LDFLAGS = -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
+libopcodes_la_LDFLAGS = -Wl,-Bsymbolic-functions -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
 
 # libtool will build .libs/libopcodes.a.  We create libopcodes.a in
 # the build directory so that we don't have to convert all the
diff -Nur binutils-2.19.51.0.9-old/opcodes/Makefile.in binutils-2.19.51.0.9/opcodes/Makefile.in
--- binutils-2.19.51.0.9-old/opcodes/Makefile.in	2009-06-13 16:14:29.638183455 +0300
+++ binutils-2.19.51.0.9/opcodes/Makefile.in	2009-06-13 16:15:04.625433059 +0300
@@ -628,7 +628,7 @@
 # hardcoded path to libbfd.so instead of relying on the entries in libbfd.la.
 libopcodes_la_DEPENDENCIES = $(OFILES) @SHARED_DEPENDENCIES@
 libopcodes_la_LIBADD = $(OFILES) @SHARED_LIBADD@
-libopcodes_la_LDFLAGS = -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
+libopcodes_la_LDFLAGS = -Wl,-Bsymbolic-functions -release `cat ../bfd/libtool-soversion` @SHARED_LDFLAGS@
 
 # libtool will build .libs/libopcodes.a.  We create libopcodes.a in
 # the build directory so that we don't have to convert all the
