--- binutils-2.17.50/ld/ld.texinfo
+++ binutils-2.17.50/ld/ld.texinfo
@@ -1941,7 +1941,7 @@
 systems may not understand them. If you specify
 @option{--enable-new-dtags}, the dynamic tags will be created as needed.
 If you specify @option{--disable-new-dtags}, no new dynamic tags will be
-created. By default, the new dynamic tags are not created. Note that
+created. By default, the new dynamic tags are created. Note that
 those options are only available for ELF systems.
 
 @kindex --hash-size=@var{number}
--- binutils-2.17.50/ld/ldmain.c
+++ binutils-2.17.50/ld/ldmain.c
@@ -263,6 +263,7 @@
 
   link_info.allow_undefined_version = TRUE;
   link_info.keep_memory = TRUE;
+  link_info.new_dtags = TRUE;
   link_info.combreloc = TRUE;
   link_info.strip_discarded = TRUE;
   link_info.callbacks = &link_callbacks;
