diff -Nur binutils-2.19.51.0.9-old/ld/configure.host binutils-2.19.51.0.9/ld/configure.host
--- binutils-2.19.51.0.9-old/ld/configure.host	2010-03-03 19:14:22.320328438 +0200
+++ binutils-2.19.51.0.9/ld/configure.host	2010-03-03 19:14:37.092331352 +0200
@@ -230,7 +230,7 @@
   ;;
 
 x86_64-*-linux-*)
-  HOSTING_CRT0=`echo "$HOSTING_CRT0" | sed -e "s,\\\`specs.*\"\\\`,/lib64/ld-linux-x86-64.so.2,"`
+  HOSTING_CRT0=`echo "$HOSTING_CRT0" | sed -e "s,\\\`specs.*\"\\\`,/lib/ld-linux-x86-64.so.2,"`
   ;;
 
 *-*-freebsd* | *-*-kfreebsd*-gnu | *-*-dragonfly*)
