Index: selective.exp
===================================================================
RCS file: /cvs/src/src/ld/testsuite/ld-selective/selective.exp,v
retrieving revision 1.26
diff -c -3 -p -r1.26 selective.exp
*** ld/testsuite/ld-selective/selective.exp	10 Aug 2002 12:48:52 -0000	1.26
--- ld/testsuite/ld-selective/selective.exp	19 Aug 2003 08:00:20 -0000
*************** set seltests {
*** 59,66 ****
      {A::foo() B::foo() dropme1() dropme2()} {*-*-*}}
  }
  
! set cflags "-w -O -ffunction-sections -fdata-sections"
! set cxxflags "-fvtable-gc -fno-exceptions -fno-rtti"
  set ldflags "--gc-sections -Bstatic"
  
  if [istarget sh64*-*-elf] {
--- 59,66 ----
      {A::foo() B::foo() dropme1() dropme2()} {*-*-*}}
  }
  
! set cflags "-w -O -ffunction-sections -fdata-sections -fno-unit-at-a-time"
! set cxxflags "-fvtable-gc -fno-exceptions -fno-rtti -fno-unit-at-a-time"
  set ldflags "--gc-sections -Bstatic"
  
  if [istarget sh64*-*-elf] {

