diff -Naur PyQt-x11-gpl-snapshot-4.8-eac5dd92c907-orig//sip/QtCore/qglobal.sip PyQt-x11-gpl-snapshot-4.8-eac5dd92c907/sip/QtCore/qglobal.sip
--- PyQt-x11-gpl-snapshot-4.8-eac5dd92c907-orig//sip/QtCore/qglobal.sip	2010-09-28 14:06:40.682000047 +0300
+++ PyQt-x11-gpl-snapshot-4.8-eac5dd92c907/sip/QtCore/qglobal.sip	2010-09-28 14:07:22.952000047 +0300
@@ -304,22 +304,6 @@
     //QFlags operator&(ENUM f) const;
     QFlags operator~() const;
 
-    // These are here to ensure consistency between, for example:
-    //   Qt.AlignLeft | Qt.AlignTop | Qt.TextWordWrap and
-    //   Qt.AlignLeft | Qt.TextWordWrap | Qt.AlignTop
-    // In the first of the above Qt.TextWordWrap is ored with a
-    // Qt.AlignmentFlag enum.  In the second it is being ored with a
-    // Qt.Alignment class.
-    QFlags operator|(int f);
-%MethodCode
-        sipRes = new QFlags(*a0 | (ENUM(a1)));
-%End
-
-    QFlags operator^(int f);
-%MethodCode
-        sipRes = new QFlags(*a0 ^ (ENUM(a1)));
-%End
-
     // These are necessary to prevent Python comparing object IDs.
     bool operator==(const QFlags &f) const;
 %MethodCode
