From 1f1bba573a0bc1e27264c870b7822a8d183f80f3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Mon, 25 Jul 2011 11:03:31 +0100
Subject: [PATCH] Resolves: rhbz#725144 wrong csh syntax

---
 odk/setsdkenv_unix.csh.in |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/odk/setsdkenv_unix.csh.in b/odk/setsdkenv_unix.csh.in
index cb2b0f5..f48be7b 100644
--- a/odk/setsdkenv_unix.csh.in
+++ b/odk/setsdkenv_unix.csh.in
@@ -13,8 +13,7 @@ setenv OO_SDK_HOME '@OO_SDK_HOME@'
 
 # Office installation directory.
 # Example: set OFFICE_HOME=/opt/libreoffice
-set OFFICE_HOME='@OFFICE_HOME@'
-OFFICE_BASE_HOME='@OFFICE_BASE_HOME@'
+setenv OFFICE_HOME '@OFFICE_HOME@'
 
 # URE installation directory.
 # Example: setenv OO_SDK_URE_HOME /opt/libreoffice/ure
@@ -227,7 +226,7 @@ if ( $?OO_SDK_JAVA_HOME == 0) then
 endif
 
 
-if [ "${platform}" == "MacOSX" ] then
+if ( "${platform}" == "MacOSX" ) then
 #    For URE, prepare symbolic links for libraries:
 #    Only necessary on MacOSX, on other Unix systems the links are already prepared
 #    in the SDK installation.
-- 
1.7.6

