From 194da0076d3b4a6968496261fd2eaca9e8e955f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Fri, 24 Jun 2011 10:29:09 +0100
Subject: [PATCH] bad merge, fix to enable extensions to build again

---
 odk/settings/settings.mk |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/odk/settings/settings.mk b/odk/settings/settings.mk
index bbd18ee..e4fc05a 100644
--- a/odk/settings/settings.mk
+++ b/odk/settings/settings.mk
@@ -502,11 +502,9 @@ PLATFORM=kfreebsd
 ifeq "$(PROCTYPE)" "x86_64"
 PACKAGE_LIB_DIR=kfreebsd_x86_64.plt
 UNOPKG_PLATFORM=kFreeBSD_x86_64
-STLPORT=no
 else
 PACKAGE_LIB_DIR=kfreebsd_x86.plt
 UNOPKG_PLATFORM=kFreeBSD_x86
-STLPORT=no
 endif
 else
 PLATFORM=freebsd
@@ -570,10 +568,6 @@ CPPUHELPERLIB=-luno_cppuhelper$(COMID)
 SALHELPERLIB=-luno_salhelper$(COMID)
 REGLIB=-lreg
 STORELIB=-lstore
-ifeq "$(STLPORT)" "yes"
-endif
-else
-STLPORTLIB=
 
 EMPTYSTRING=
 PATH_SEPARATOR=:
-- 
1.7.5.4

