Index: libreoffice-bootstrap-3.4.1.3/solenv/bin/modules/installer/scriptitems.pm
===================================================================
--- libreoffice-bootstrap-3.4.1.3.orig/solenv/bin/modules/installer/scriptitems.pm
+++ libreoffice-bootstrap-3.4.1.3/solenv/bin/modules/installer/scriptitems.pm
@@ -1066,7 +1066,7 @@ sub get_sourcepath_from_filename_and_inc
             }
             else
             {
-                $infoline = "ERROR: Source for $$searchfilenameref not found!\n";	 # Important message in log file				
+				$infoline = "WARNING: Source for $$searchfilenameref not found!\n";	 # Important message in log file				
             }
 
             push( @installer::globals::logfileinfo, $infoline);
@@ -1144,7 +1144,7 @@ sub get_sourcepath_from_filename_and_inc
             }
             else
             {
-                $infoline = "ERROR: Source for $$searchfilenameref not found!\n";	 # Important message in log file				
+				$infoline = "WARNING: Source for $$searchfilenameref not found!\n";	 # Important message in log file				
             }
 
             push( @installer::globals::logfileinfo, $infoline);
@@ -1356,11 +1356,10 @@ sub remove_Files_Without_Sourcedirectory
 
             if ( ! $installer::globals::languagepack && !$installer::globals::helppack)
             {
-                $infoline = "ERROR: Removing file $filename from file list.\n";
+                $infoline = "WARNING: Removing file $filename from file list.\n";
                 push( @installer::globals::logfileinfo, $infoline);
 
-                push(@missingfiles, "ERROR: File not found: $filename\n");	
-                $error_occurred = 1;
+                push(@missingfiles, "WARNING: File not found: $filename\n");	
 
                 next;	# removing this file from list, if sourcepath is empty
             }
@@ -1368,11 +1367,10 @@ sub remove_Files_Without_Sourcedirectory
             {
                 if (( $onefile->{'ismultilingual'} ) || ( $styles =~ /\bFORCELANGUAGEPACK\b/ ))
                 {
-                    $infoline = "ERROR: Removing file $filename from file list.\n";
+                    $infoline = "WARNING: Removing file $filename from file list.\n";
                     push( @installer::globals::logfileinfo, $infoline);
 
-                    push(@missingfiles, "ERROR: File not found: $filename\n");	
-                    $error_occurred = 1;
+                    push(@missingfiles, "WARNING: File not found: $filename\n");	
 
                     next;	# removing this file from list, if sourcepath is empty		
                 }
@@ -1390,11 +1388,10 @@ sub remove_Files_Without_Sourcedirectory
             {
                 if (( $onefile->{'ismultilingual'} ) || ( $styles =~ /\bFORCEHELPPACK\b/ ))
                 {
-                    $infoline = "ERROR: Removing file $filename from file list.\n";
+                    $infoline = "WARNING: Removing file $filename from file list.\n";
                     push( @installer::globals::logfileinfo, $infoline);
 
-                    push(@missingfiles, "ERROR: File not found: $filename\n");	
-                    $error_occured = 1;
+                    push(@missingfiles, "WARNING: File not found: $filename\n");	
 
                     next;	# removing this file from list, if sourcepath is empty		
                 }
Index: libreoffice-bootstrap-3.4.1.3/solenv/bin/modules/installer/simplepackage.pm
===================================================================
--- libreoffice-bootstrap-3.4.1.3.orig/solenv/bin/modules/installer/simplepackage.pm
+++ libreoffice-bootstrap-3.4.1.3/solenv/bin/modules/installer/simplepackage.pm
@@ -52,7 +52,7 @@ sub check_simple_packager_project
         ( $installer::globals::packageformat eq "archive" ))
     {
         $installer::globals::is_simple_packager_project = 1;
-        $installer::globals::patch_user_dir = 1;
+        $installer::globals::patch_user_dir = 0;
     }
     elsif( $installer::globals::packageformat eq "dmg" )
     {
