Index: libreoffice-bootstrap-3.4.1.3/sfx2/source/doc/guisaveas.cxx
===================================================================
--- libreoffice-bootstrap-3.4.1.3.orig/sfx2/source/doc/guisaveas.cxx
+++ libreoffice-bootstrap-3.4.1.3/sfx2/source/doc/guisaveas.cxx
@@ -303,7 +303,7 @@ public:
                                 sal_Bool bSetStandardName,
                                 ::rtl::OUString& aSuggestedName,
                                 sal_Bool bPreselectPassword,
-                                const ::rtl::OUString& aSuggestedDir,
+                                ::rtl::OUString& aSuggestedDir,
                                 sal_Int16 nDialog,
                                 const ::rtl::OUString& rStandardDir,
                                 const ::com::sun::star::uno::Sequence< ::rtl::OUString >& rBlackList
@@ -808,7 +808,7 @@ sal_Bool ModelData_Impl::OutputFileDialo
                                             sal_Bool bSetStandardName,
                                             ::rtl::OUString& aSuggestedName,
                                             sal_Bool bPreselectPassword,
-                                            const ::rtl::OUString& aSuggestedDir,
+                                            ::rtl::OUString& aSuggestedDir,
                                             sal_Int16 nDialog,
                                             const ::rtl::OUString& rStandardDir,
                                             const ::com::sun::star::uno::Sequence< ::rtl::OUString >& rBlackList)
@@ -1016,6 +1016,7 @@ sal_Bool ModelData_Impl::OutputFileDialo
     INetURLObject aURL( pFileDlg->GetPath() );
     // the path should be provided outside since it might be used for further calls to the dialog
     aSuggestedName = aURL.GetName( INetURLObject::DECODE_WITH_CHARSET );
+       aSuggestedDir = pFileDlg->GetDisplayDirectory();
 
     // old filter options should be cleared in case different filter is used
 
@@ -1501,10 +1502,8 @@ sal_Bool SfxStoringHelper::GUIStoreModel
         sal_Bool bExit = sal_False;
         while ( !bExit )
         {
-            bUseFilterOptions = aModelData.OutputFileDialog( nStoreMode, aFilterProps, bSetStandardName, aSuggestedName, bPreselectPassword, aSuggestedDir, nDialog, sStandardDir, aBlackList );
-
             // in case the dialog is opend a second time the folder should be the same as before, not what was handed over by parameters
-            aSuggestedDir = ::rtl::OUString();
+            bUseFilterOptions = aModelData.OutputFileDialog( nStoreMode, aFilterProps, bSetStandardName, aSuggestedName, bPreselectPassword, aSuggestedDir, nDialog, sStandardDir, aBlackList );
             if ( nStoreMode == SAVEAS_REQUESTED )
             {
                 // in case of saving check filter for possible alien warning
