--- unzip-5.52/list.c.pom	2006-02-06 09:39:47.622466528 +0100
+++ unzip-5.52/list.c	2006-02-06 09:57:00.813397632 +0100
@@ -76,7 +76,7 @@
    static ZCONST char Far ShortHdrStats[] =
      "%9lu  %02u%c%02u%c%02u %02u:%02u  %c";
    static ZCONST char Far ShortFileTrailer[] = " --------       \
-            -------\n%9lu                   %lu file%s\n";
+            -------\n%9ju                   %lu file%s\n";
 #endif /* ?OS2_EAS */
 #endif /* !WINDLL */
 
