--- unzip-5.52/fileio.c.pom	2007-08-30 11:36:05.000000000 +0200
+++ unzip-5.52/fileio.c	2007-09-04 10:37:41.000000000 +0200
@@ -53,7 +53,6 @@
 
   ---------------------------------------------------------------------------*/
 
-
 #define __FILEIO_C      /* identifies this source module */
 #define UNZIP_INTERNAL
 #include "unzip.h"
@@ -425,7 +424,7 @@ int open_outfile(__G)         /* return 
     Trace((stderr, "open_outfile:  doing fopen(%s) for writing\n",
       FnFilter1(G.filename)));
     
-    fd = open(G.filename, O_WRONLY | O_LARGEFILE | O_CREAT);
+    fd = open(G.filename, O_WRONLY | O_LARGEFILE | O_CREAT, 0600);
     if ((G.outfile = fdopen(fd, FOPW)) == (FILE *)NULL) {
         Info(slide, 0x401, ((char *)slide, LoadFarString(CannotCreateFile),
           FnFilter1(G.filename)));
