--- cmake/Ts.cmake.orig	2009-07-17 11:52:39.000000000 +0300
+++ cmake/Ts.cmake	2009-07-17 11:53:03.000000000 +0300
@@ -15,7 +15,7 @@
 
 MACRO(FIND_LRELEASE)
     IF(NOT LRELEASE_EXECUTABLE AND NOT LRELEASE_NOT_FOUND)
-		FIND_PROGRAM(LRELEASE_EXECUTABLE lrelease PATHS
+		FIND_PROGRAM(LRELEASE_EXECUTABLE lrelease-qt4 PATHS
 			"[HKEY_CURRENT_USER\\Software\\Trolltech\\Qt3Versions\\4.0.0;InstallDir]/bin"
 			"[HKEY_CURRENT_USER\\Software\\Trolltech\\Versions\\4.0.0;InstallDir]/bin"
 			$ENV{QTDIR}/bin)
