diff -ur libsoup-2.32.0.orig//configure.ac libsoup-2.32.0/configure.ac
--- libsoup-2.32.0.orig//configure.ac	2010-09-28 03:34:03.000000000 +0300
+++ libsoup-2.32.0/configure.ac	2010-10-29 15:19:56.123000077 +0300
@@ -220,7 +220,7 @@
 dnl *****************************
 dnl *** gobject-introspection ***
 dnl *****************************
-GOBJECT_INTROSPECTION_CHECK([0.9.5])
+GOBJECT_INTROSPECTION_CHECK([0.9.3])
 
 dnl *************************************
 dnl *** Warnings to show if using GCC ***
diff -ur libsoup-2.32.0.orig//libsoup/Makefile.am libsoup-2.32.0/libsoup/Makefile.am
--- libsoup-2.32.0.orig//libsoup/Makefile.am	2010-09-17 21:35:31.000000000 +0300
+++ libsoup-2.32.0/libsoup/Makefile.am	2010-10-29 15:25:16.796000077 +0300
@@ -238,7 +238,7 @@
 		          $(libsoupgnomeinclude_HEADERS) \
 	                  $(filter-out %.h, $(libsoup_gnome_2_4_la_SOURCES)))
 SoupGNOME-2.4.gir: libsoup-gnome-2.4.la Soup-2.4.gir
-SoupGNOME_2_4_gir_SCANNERFLAGS = --identifier-prefix=Soup --symbol-prefix=soup
+SoupGNOME_2_4_gir_SCANNERFLAGS = --strip-prefix=soup
 SoupGNOME_2_4_gir_INCLUDES = Soup-2.4
 SoupGNOME_2_4_gir_CFLAGS = $(INCLUDES)
 SoupGNOME_2_4_gir_LIBS = libsoup-gnome-2.4.la
