Index: miro-3.5.1/linux/miro.desktop
===================================================================
--- miro-3.5.1.orig/linux/miro.desktop
+++ miro-3.5.1/linux/miro.desktop
@@ -2,11 +2,13 @@
 Type=Application
 Name=Miro Internet TV
 GenericName=Internet TV
+GenericName[tr]=Internet TV
 Comment=Watch online video
 Comment[de]=Multimedia Player
 Comment[es]=Reproductor multimedia
 Comment[fr]=Lecteur multimedia
 Comment[it]=Lettore multimediale
+Comment[tr]=Çevrimiçi video izleyin, indirin
 Icon=miro
 TryExec=miro
 Exec=miro %F
