--- setup.py.orig	2009-12-04 17:34:31.000000000 +0200
+++ setup.py	2009-12-04 17:53:12.000000000 +0200
@@ -62,9 +62,6 @@
                os.path.join('bin', 'bbimage')],
       data_files=[(os.path.join(bbdir, 'conf'), [os.path.join('conf', 'bitbake.conf')]),
                   (os.path.join(bbdir, 'classes'), [os.path.join('classes', 'base.bbclass')]),
-                  (os.path.join(docdir, 'bitbake-%s' % bb.__version__, 'html'), glob.glob(os.path.join('doc', 'manual', 'html', '*.html'))),
-                  (os.path.join(docdir, 'bitbake-%s' % bb.__version__, 'pdf'), glob.glob(os.path.join('doc', 'manual', 'pdf', '*.pdf'))),],
+                  (os.path.join(docdir, 'bitbake', 'html'), glob.glob(os.path.join('doc', 'manual', 'html', '*.html'))),
+                  (os.path.join(docdir, 'bitbake', 'pdf'), glob.glob(os.path.join('doc', 'manual', 'pdf', '*.pdf'))),],
      )
-
-if 'bdist' in sys.argv[1:]:
-    clean_doc('html')
