diff -up scim-1.4.7/src/scim_utility.cpp.bz462820 scim-1.4.7/src/scim_utility.cpp
--- scim-1.4.7/src/scim_utility.cpp.bz462820	2008-09-19 16:27:18.000000000 +0800
+++ scim-1.4.7/src/scim_utility.cpp	2008-09-19 16:33:50.000000000 +0800
@@ -765,6 +765,8 @@ static __Language __languages [] = {
     { "pt_PT",    NULL, N_("Portuguese"), "Português", "@euro" },
     { "ro_RO",    NULL, N_("Romanian"), "Română", NULL },
     { "ru_RU",    NULL, N_("Russian"), "русский", ".koi8r" },
+    { "sd",    "sd_IN", N_("Sindhi"), "ﺲﻧڌﻱ", NULL },
+    { "sd_IN",    NULL, N_("Sindhi"), "सिन्धी", "@devanagari" },
     { "si_LK",    NULL, N_("Sinhala"), "සිංහල", NULL },
     { "sk_SK",    NULL, N_("Slovak"), "Slovenský", NULL },
     { "sl_SI",    NULL, N_("Slovenian"), "Slovenščina", NULL },
