--- scim-1.4.5/extras/panel/scim_panel_gtk.cpp.9-icon-size	2006-09-30 17:40:15.000000000 +1000
+++ scim-1.4.5/extras/panel/scim_panel_gtk.cpp	2006-11-03 15:12:49.000000000 +1000
@@ -108,7 +108,7 @@
 #define SCIM_PIN_DOWN_ICON_FILE     (SCIM_ICONDIR "/pin-down.png")
 
 #define TOOLBAR_ICON_SIZE                     16 
-#define TRAY_ICON_SIZE                        18
+#define TRAY_ICON_SIZE                        11
 #define LOOKUP_ICON_SIZE                      12
 
 /////////////////////////////////////////////////////////////////////////////
