diff -u scim-1.4.5/extras/gtk2_immodule/gtkimcontextscim.cpp~ scim-1.4.5/extras/gtk2_immodule/gtkimcontextscim.cpp
--- scim-1.4.5/extras/gtk2_immodule/gtkimcontextscim.cpp~	2006-11-24 16:44:04.000000000 +1000
+++ scim-1.4.5/extras/gtk2_immodule/gtkimcontextscim.cpp	2006-11-24 16:44:04.000000000 +1000
@@ -306,7 +306,7 @@
 
 static bool                                             _on_the_spot                = true;
 static bool                                             _shared_input_method        = false;
-static bool                                             _use_key_snooper            = true;
+static bool                                             _use_key_snooper            = false;
 
 // A hack to shutdown the immodule cleanly even if im_module_exit () is not called when exiting.
 class FinalizeHandler
