--- scim-1.4.4.orig/modules/FrontEnd/scim_x11_frontend.cpp	2006-10-06 01:05:49.000000000 +0900
+++ scim-1.4.4/modules/FrontEnd/scim_x11_frontend.cpp	2006-10-06 10:29:55.000000000 +0900
@@ -1094,7 +1094,8 @@
     // If the ic is not focused, then return.
     if (!is_focused_ic (ic)) {
         SCIM_DEBUG_FRONTEND(1) << "IC " << call_data->icid << " is not focused, focus it first.\n";
-        ims_set_ic_focus_handler (ims, (IMChangeFocusStruct *) call_data);
+//        ims_set_ic_focus_handler (ims, (IMChangeFocusStruct *) call_data);
+	return 1;
     }
 
     XKeyEvent *event = (XKeyEvent*) &(call_data->event);
