diff -up scim-1.4.7/src/Makefile.am.17-rpath~ scim-1.4.7/src/Makefile.am
--- scim-1.4.7/src/Makefile.am.17-rpath~	2007-06-26 22:31:38.000000000 +0800
+++ scim-1.4.7/src/Makefile.am	2008-07-17 14:56:54.000000000 +0800
@@ -130,7 +130,6 @@ libscim@SCIM_EPOCH@_la_SOURCES = scim_ba
 
 libscim@SCIM_EPOCH@_la_LDFLAGS= -version-info $(SCIM_CURRENT):$(SCIM_REVISION):$(SCIM_AGE) \
 			  -export-dynamic \
-			  -rpath $(libdir) \
 			  $(LD_VERSION_SCRIPT_OPTION) \
 			  @LIBTOOL_EXPORT_OPTIONS@ \
 			  @LIBICONV@ \
diff -up scim-1.4.7/utils/Makefile.am.17-rpath~ scim-1.4.7/utils/Makefile.am
--- scim-1.4.7/utils/Makefile.am.17-rpath~	2007-06-26 22:31:56.000000000 +0800
+++ scim-1.4.7/utils/Makefile.am	2008-07-17 14:56:54.000000000 +0800
@@ -53,7 +53,6 @@ libscim_gtkutils@SCIM_EPOCH@_la_CXXFLAGS
 
 libscim_gtkutils@SCIM_EPOCH@_la_LDFLAGS= -version-info $(SCIM_CURRENT):$(SCIM_REVISION):$(SCIM_AGE) \
 				  -export-dynamic \
-				  -rpath $(libdir) \
 				  @LIBTOOL_EXPORT_OPTIONS@ \
 				  @LTLIBINTL@ \
 				  @GTK2_LIBS@ 
@@ -68,7 +67,6 @@ libscim_x11utils@SCIM_EPOCH@_la_CXXFLAGS
 
 libscim_x11utils@SCIM_EPOCH@_la_LDFLAGS= -version-info $(SCIM_CURRENT):$(SCIM_REVISION):$(SCIM_AGE) \
 				  -export-dynamic \
-				  -rpath $(libdir) \
 				  @LIBTOOL_EXPORT_OPTIONS@ \
 				  @LTLIBINTL@ \
 				  @X_LIBS@ 
