--- /home/kane/projects/inputs/scim-1.4.5/configs/global	2005-06-26 19:35:08.000000000 +0300
+++ scim-1.4.5/configs/global	2006-10-16 19:13:52.000000000 +0300
@@ -1,6 +1,6 @@
 /SupportedUnicodeLocales = en_US.UTF-8
-/DefaultPanelProgram = scim-panel-gtk
-/DefaultConfigModule = simple
+/DefaultPanelProgram = scim-panel-dbus
+/DefaultConfigModule = simple
 /DefaultSocketFrontEndAddress = local:/tmp/scim-socket-frontend
 /DefaultSocketIMEngineAddress = local:/tmp/scim-socket-frontend
 /DefaultSocketConfigAddress = local:/tmp/scim-socket-frontend
