diff -Naur bpython-0.9.6.2.orig/data/bpython.desktop bpython-0.9.6.2/data/bpython.desktop
--- bpython-0.9.6.2.orig/data/bpython.desktop	2009-12-30 14:07:06.000000000 +0200
+++ bpython-0.9.6.2/data/bpython.desktop	2010-01-12 16:36:16.197719069 +0200
@@ -1,7 +1,13 @@
 [Desktop Entry]
-Icon=/usr/share/pixmaps/python.xpm
+Type=Application
+Encoding=UTF-8
+Icon=bpython
 Name=bpython
+Name[tr]=bpython
+GenericName=Python interpreter
+GenericName[tr]=Python yorumlayıcısı
 Comment=A fancy interface to the python interpreter!
+Comment[tr]=Python yorumlayıcısı olarak farklı bir arayüz alternatifi
 Exec=/usr/bin/bpython
 Terminal=true
 Type=Application
