Index: bpython-0.10/setup.py
===================================================================
--- bpython-0.10.orig/setup.py
+++ bpython-0.10/setup.py
@@ -21,12 +21,12 @@ try:
 except ImportError:
     from distutils.command.build_py import build_py
 
-try:
-    from babel.messages.frontend import compile_catalog as _compile_catalog
-    from babel.messages.frontend import extract_messages
-    using_translations = True
-except ImportError:
-    using_translations = False
+#try:
+#    from babel.messages.frontend import compile_catalog as _compile_catalog
+#    from babel.messages.frontend import extract_messages
+#    using_translations = True
+#except ImportError:
+using_translations = False
 
 
 cmdclass = dict(build_py=build_py, build=build)
