--- ppp-2.4.2/pppd/Makefile.linux.pie	2004-05-14 13:46:14.188327258 +0200
+++ ppp-2.4.2/pppd/Makefile.linux	2004-05-14 13:46:16.147058249 +0200
@@ -206,7 +206,7 @@
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
-	$(CC) $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
+	$(CC) $(CFLAGS) -pie $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
 
 srp-entry:	srp-entry.c
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ srp-entry.c $(LIBS)
--- ppp-2.4.2/chat/Makefile.linux.pie	2004-05-14 13:46:32.699784925 +0200
+++ ppp-2.4.2/chat/Makefile.linux	2004-05-14 13:46:45.071085866 +0200
@@ -14,7 +14,7 @@
 all:	chat
 
 chat:	chat.o
-	$(CC) -o chat chat.o
+	$(CC) -pie -o chat chat.o
 
 chat.o:	chat.c
 	$(CC) -c $(CFLAGS) -o chat.o chat.c
