Index: translate-toolkit-1.7.0/translate/convert/po2html.py
===================================================================
--- translate-toolkit-1.7.0.orig/translate/convert/po2html.py
+++ translate-toolkit-1.7.0/translate/convert/po2html.py
@@ -85,7 +85,7 @@ class po2html:
                 htmlresult = htmlresult.replace(msgid, msgstr, 1)
         htmlresult = htmlresult.encode('utf-8')
         if self.tidy:
-            htmlresult = str(tidy.parseString(htmlresult))
+            htmlresult = str(tidy.parseString(htmlresult, **{'char_encoding': "utf8"}))
         return htmlresult
 
 def converthtml(inputfile, outputfile, templatefile, wrap=None, includefuzzy=False, usetidy=True):
