Index: yakuake-2.9.7/app/config/yakuake.kcfg
===================================================================
--- yakuake-2.9.7.orig/app/config/yakuake.kcfg
+++ yakuake-2.9.7/app/config/yakuake.kcfg
@@ -128,7 +128,7 @@
     <entry name="Frames" type="Int">
       <label context="@label">Animation frames</label>
       <whatsthis context="@info:whatsthis">The number of frames of the open/retract animation.</whatsthis>
-      <default>20</default>
+      <default>0</default>
       <min>0</min>
       <max>50</max>
     </entry>
