Index: lokalize/src/main.cpp
===================================================================
--- lokalize/src/main.cpp	(revision 1088844)
+++ lokalize/src/main.cpp	(working copy)
@@ -103,13 +103,14 @@
             Project::instance()->load( QFileInfo(path).absoluteFilePath() );
         }
         LokalizeMainWindow* lmw=new LokalizeMainWindow;
-        kWarning()<<"showing LokalizeMainWindow";
-        lmw->show();
-        kWarning()<<"LokalizeMainWindow shown";
         int j=args->count();
         while (--j>=0)
             lmw->fileOpen(args->url(j));
 
+        kWarning()<<"showing LokalizeMainWindow";
+        lmw->show();
+        kWarning()<<"LokalizeMainWindow shown";
+
         Project::instance()->model()->setCompleteScan(args->isSet("projectscan"));
         args->clear();
     }
