Index: src/incremen.c
===================================================================
--- src/incremen.c
+++ src/incremen.c	2010-07-19 18:19:20.000000000 +0200
@@ -258,7 +258,7 @@
   directory->dump = directory->idump = NULL;
   directory->orig = NULL;
   directory->flags = false;
-  if (namelen && ISSLASH (name[namelen - 1]))
+  if (namelen > 1 && ISSLASH (name[namelen - 1]))
     namelen--;
   directory->name = xmalloc (namelen + 1);
   memcpy (directory->name, name, namelen);
