Index: tar-1.23/tests/testsuite.at
===================================================================
--- tar-1.23.orig/tests/testsuite.at
+++ tar-1.23/tests/testsuite.at
@@ -157,7 +157,6 @@ m4_include([incremental.at])
 m4_include([incr01.at])
 m4_include([incr02.at])
 m4_include([listed01.at])
-m4_include([listed02.at])
 m4_include([incr03.at])
 m4_include([incr04.at])
 m4_include([incr05.at])
