Index: busybox-1.14.1/networking/udhcp/options.c
===================================================================
--- busybox-1.14.1.orig/networking/udhcp/options.c
+++ busybox-1.14.1/networking/udhcp/options.c
@@ -27,7 +27,7 @@ const struct dhcp_option dhcp_options[] 
 	{ OPTION_U16                              , 0x0d }, /* DHCP_BOOT_SIZE     */
 	{ OPTION_STRING | OPTION_LIST | OPTION_REQ, 0x0f }, /* DHCP_DOMAIN_NAME   */
 	{ OPTION_IP                               , 0x10 }, /* DHCP_SWAP_SERVER   */
-	{ OPTION_STRING                           , 0x11 }, /* DHCP_ROOT_PATH     */
+	{ OPTION_STRING               | OPTION_REQ, 0x11 }, /* DHCP_ROOT_PATH     */
 	{ OPTION_U8                               , 0x17 }, /* DHCP_IP_TTL        */
 	{ OPTION_U16                              , 0x1a }, /* DHCP_MTU           */
 	{ OPTION_IP                   | OPTION_REQ, 0x1c }, /* DHCP_BROADCAST     */
