Index: work/texmf-dist/tex/latex/latexconfig/lualatex.ini
===================================================================
--- work.orig/texmf-dist/tex/latex/latexconfig/lualatex.ini
+++ work/texmf-dist/tex/latex/latexconfig/lualatex.ini
@@ -1,6 +1,7 @@
 % $Id: lualatex.ini 9276 2008-07-05 18:04:46Z karl $
 % Karl Berry, 2008. public domain.
 %
+\input luatexiniconfig.tex
 \input pdftexconfig
 \pdfoutput=0  % dvi output by default
 %
Index: work/texmf-dist/tex/latex/latexconfig/pdflualatex.ini
===================================================================
--- work.orig/texmf-dist/tex/latex/latexconfig/pdflualatex.ini
+++ work/texmf-dist/tex/latex/latexconfig/pdflualatex.ini
@@ -1,4 +1,5 @@
 % Karl Berry, 2008. public domain.
+\input luatexiniconfig.tex
 \input pdftexconfig.tex
 \scrollmode
 \input latex.ltx
