diff -Nur mplayer-old/libvo/vo_vaapi.c mplayer/libvo/vo_vaapi.c
--- mplayer-old/libvo/vo_vaapi.c	2010-08-19 21:17:33.259523038 +0300
+++ mplayer/libvo/vo_vaapi.c	2010-08-19 21:19:55.601526187 +0300
@@ -2351,10 +2351,10 @@
     enable_osd();
 }
 
-static void draw_eosd(mp_eosd_images_t *imgs)
+static void draw_eosd(EOSD_ImageList *imgs)
 {
-    ass_image_t *img = imgs->imgs;
-    ass_image_t *i;
+    ASS_Image *img = imgs->imgs;
+    ASS_Image *i;
     VAStatus status;
 
     if (!va_eosd_draw_alpha)
