diff --git a/mkfs.c b/mkfs.c
index d664254..138bcc9 100644
--- a/mkfs.c
+++ b/mkfs.c
@@ -294,7 +294,6 @@ static u64 parse_profile(char *s)
 
 static char *parse_label(char *input)
 {
-	int i;
 	int len = strlen(input);
 
 	if (len > BTRFS_LABEL_SIZE) {
@@ -302,12 +301,6 @@ static char *parse_label(char *input)
 			BTRFS_LABEL_SIZE);
 		exit(1);
 	}
-	for (i = 0; i < len; i++) {
-		if (input[i] == '/' || input[i] == '\\') {
-			fprintf(stderr, "invalid label %s\n", input);
-			exit(1);
-		}
-	}
 	return strdup(input);
 }
 
