diff -Nuar libgsf-1.14.15.orig/python/Makefile.am libgsf-1.14.15/python/Makefile.am
--- libgsf-1.14.15.orig/python/Makefile.am	2009-07-02 16:10:10.000000000 +0300
+++ libgsf-1.14.15/python/Makefile.am	2009-07-02 17:09:19.000000000 +0300
@@ -18,7 +18,7 @@
 gnomemodule_la_CFLAGS = \
 	-I$(top_srcdir)		\
 	$(LIBGSF_CFLAGS) $(LIBGSF_GNOME_CFLAGS) $(ORBIT_PYTHON_CFLAGS)
-gnomemodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex initgnome
+gnomemodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex initgnome `python-config --libs`
 gnomemodule_la_LIBADD = \
 	$(top_builddir)/gsf-gnome/libgsf-gnome-1.la	\
 	$(top_builddir)/gsf/libgsf-1.la			\
@@ -32,7 +32,7 @@
 _gsfmodule_la_CFLAGS = 		\
 	-I$(top_srcdir)		\
 	$(LIBGSF_CFLAGS)
-_gsfmodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_gsf
+_gsfmodule_la_LDFLAGS = -module -avoid-version -export-symbols-regex init_gsf `python-config --libs`
 _gsfmodule_la_LIBADD = $(top_builddir)/gsf/libgsf-1.la $(LIBGSF_LIBS)
 _gsfmodule_la_SOURCES = gsfmodule.c
 nodist__gsfmodule_la_SOURCES = gsf.c
