Index: alsa-driver/configure
===================================================================
--- alsa-driver.orig/configure
+++ alsa-driver/configure
@@ -8486,7 +8486,7 @@ $as_echo "yes" >&6; }
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
 $as_echo "no" >&6; }
     if test ! -f include/pcmcia/cs_types.h; then
-      mkdir -p include/linux include/asm include/media
+      mkdir -p include/linux include/asm include/media include/pcmcia
       mkdir -p include/linux/regulator include/linux/usb
       if test -z "#define SND_HAVE_DUMMY_CS_TYPES_H" ; then
         echo "Creating a dummy <pcmcia/cs_types.h>..."
