Index: texinfo-4.13/makeinfo/sectioning.c
===================================================================
--- texinfo-4.13.orig/makeinfo/sectioning.c
+++ texinfo-4.13/makeinfo/sectioning.c
@@ -257,13 +257,13 @@ current_chapter_number (void)
   else if (enum_marker == APPENDIX_MAGIC)
     {
       char s[2];
-      sprintf (s, "%c", numbers[0] + 64);
+      snprintf (s, sizeof(s), "%c", numbers[0] + 64);
       return xstrdup (s);
     }
   else
     {
       char s[5];
-      sprintf (s, "%4d", numbers[0]);
+      snprintf (s, sizeof(s), "%d", numbers[0]);
       return xstrdup (s);
     }
 }
