Index: clementine-0.6_2629/src/core/backgroundstreams.cpp
===================================================================
--- clementine-0.6_2629.orig/src/core/backgroundstreams.cpp
+++ clementine-0.6_2629/src/core/backgroundstreams.cpp
@@ -7,7 +7,8 @@
 
 const char* BackgroundStreams::kSettingsGroup = "BackgroundStreams";
 const char* BackgroundStreams::kHypnotoadUrl = "hypnotoad:///";
-const char* BackgroundStreams::kRainUrl = "http://data.clementine-player.org/rainymood";
+//const char* BackgroundStreams::kRainUrl = "http://data.clementine-player.org/rainymood";
+const char* BackgroundStreams::kRainUrl = "http://cekirdek.pardus.org.tr/~ibrahim/rainymood.mp3";
 
 BackgroundStreams::BackgroundStreams(EngineBase* engine, QObject* parent)
     : QObject(parent),
