Index: clementine-0.7.1_3440/dist/clementine.desktop
===================================================================
--- clementine-0.7.1_3440.orig/dist/clementine.desktop
+++ clementine-0.7.1_3440/dist/clementine.desktop
@@ -4,15 +4,18 @@ Type=Application
 Name=Clementine
 GenericName=Clementine Music Player
 GenericName[pt]=Reprodutor de músicas Clementine
+GenericName[tr]=Müzik Oynatıcı
 Comment=Plays music and last.fm streams
 Comment[pt]=Reprodução de músicas e emissões last.fm
+Comment[tr]=Müzik ve Last.fm yayınlarını dinleyin
 Exec=clementine
 TryExec=clementine
-Icon=application-x-clementine
+Icon=clementine
 Terminal=false
 Categories=AudioVideo;Player;Qt;
-StartupNotify=false
-MimeType=application/ogg;application/x-ogg;application/x-ogm-audio;audio/aac;audio/mp4;audio/mpeg;audio/mpegurl;audio/ogg;audio/vnd.rn-realaudio;audio/vorbis;audio/x-flac;audio/x-mp3;audio/x-mpeg;audio/x-mpegurl;audio/x-ms-wma;audio/x-musepack;audio/x-oggflac;audio/x-pn-realaudio;audio/x-scpls;audio/x-speex;audio/x-vorbis;audio/x-vorbis+ogg;audio/x-wav;video/x-ms-asf;x-content/audio-player;
+InitialPreference=15
+StartupNotify=true
+MimeType=application/ogg;application/x-ogg;application/x-ogm-audio;audio/aac;audio/x-aac;audio/x-mp4;audio/mpeg;audio/mpegurl;audio/ogg;audio/x-ogg;audio/vnd.rn-realaudio;audio/vorbis;audio/flac;audio/x-flac;audio/x-mp3;audio/mp3;audio/x-mpeg;audio/x-mpegurl;audio/ms-wma;audio/x-ms-wma;audio/x-musepack;audio/x-oggflac;audio/x-pn-realaudio;audio/x-scpls;audio/x-speex;audio/x-vorbis;audio/x-vorbis+ogg;audio/x-wav;video/x-ms-asf;x-content/audio-player;
 X-Ayatana-Desktop-Shortcuts=Play;Pause;Stop;Previous;Next;
 
 [Play Shortcut Group] 
