diff -Naur clementine-0.1.orig/dist/clementine.desktop clementine-0.1/dist/clementine.desktop
--- clementine-0.1.orig/dist/clementine.desktop	2010-02-22 22:59:32.709855237 +0200
+++ clementine-0.1/dist/clementine.desktop	2010-02-22 23:56:02.531099824 +0200
@@ -2,11 +2,13 @@
 Version=1.0
 Type=Application
 Name=Clementine
-GenericName=Clementine Music Player
+GenericName=Music Player
+GenericName[tr]=Müzik Oynatıcı
 Comment=Plays music and last.fm streams
+Comment[tr]=Müzik ve Last.fm yayınlarını dinleyin
 Exec=clementine
 TryExec=clementine
-Icon=application-x-clementine
+Icon=clementine
 Terminal=false
 Categories=AudioVideo;Player;Qt;
 StartupNotify=false
