Index: clementine-0.5.90/dist/clementine.desktop
===================================================================
--- clementine-0.5.90.orig/dist/clementine.desktop
+++ clementine-0.5.90/dist/clementine.desktop
@@ -4,11 +4,13 @@ Type=Application
 Name=Clementine
 GenericName=Clementine Music Player
 GenericName[pt]=Reprodutor de músicas Clementine
++GenericName[tr]=Müzik Oynatıcı
 Comment=Plays music and last.fm streams
 Comment[pt]=Reprodução de músicas e emissões last.fm
++Comment[tr]=Müzik ve Last.fm yayınlarını dinleyin
 Exec=clementine
 TryExec=clementine
-Icon=application-x-clementine
++Icon=clementine
 Terminal=false
 Categories=AudioVideo;Player;Qt;
 StartupNotify=false
