Index: clementine-0.6_2629/src/playlist/playlistview.cpp
===================================================================
--- clementine-0.6_2629.orig/src/playlist/playlistview.cpp
+++ clementine-0.6_2629/src/playlist/playlistview.cpp
@@ -200,6 +200,7 @@ void PlaylistView::LoadGeometry() {
     header_->HideSection(Playlist::Column_BPM);
     header_->HideSection(Playlist::Column_Bitrate);
     header_->HideSection(Playlist::Column_Samplerate);
+    header_->HideSection(Playlist::Column_BaseFilename);
     header_->HideSection(Playlist::Column_Filename);
     header_->HideSection(Playlist::Column_Filesize);
     header_->HideSection(Playlist::Column_Filetype);
