Index: clementine-0.6_2629/src/widgets/fancytabwidget.cpp
===================================================================
--- clementine-0.6_2629.orig/src/widgets/fancytabwidget.cpp
+++ clementine-0.6_2629/src/widgets/fancytabwidget.cpp
@@ -700,7 +700,15 @@ void FancyTabWidget::MakeTabBar(QTabBar:
     if (icons && text)
       tab_id = bar->addTab(item.tab_icon_, label);
     else if (icons)
-      tab_id = bar->addTab(item.tab_icon_, QString());
+      {
+      tab_id = bar->addTab(QString());
+      QLabel *label;
+      label = new QLabel(bar);
+      label->setPixmap(item.tab_icon_.pixmap(16));
+      label->setAlignment(Qt::AlignHCenter);
+      label->setStyleSheet("margin-left:20px");
+      bar->setTabButton(tab_id, QTabBar::LeftSide, label);
+      }
     else if (text)
       tab_id = bar->addTab(label);
 
