Index: clementine-0.6_2629/src/core/player.cpp
===================================================================
--- clementine-0.6_2629.orig/src/core/player.cpp
+++ clementine-0.6_2629/src/core/player.cpp
@@ -292,6 +292,14 @@ void Player::Seek(int seconds) {
   emit Seeked(msec * 1000);
 }
 
+void Player::SeekForward() {
+  Seek(engine()->position() / 1000 + 5);
+}
+
+void Player::SeekBackward() {
+  Seek(engine()->position() / 1000 - 5);
+}
+
 void Player::EngineMetadataReceived(const Engine::SimpleMetaBundle& bundle) {
   PlaylistItemPtr item = playlists_->active()->current_item();
   if (!item)
Index: clementine-0.6_2629/src/core/player.h
===================================================================
--- clementine-0.6_2629.orig/src/core/player.h
+++ clementine-0.6_2629/src/core/player.h
@@ -76,8 +76,10 @@ class Player : public QObject {
   void VolumeUp() { SetVolume(GetVolume() + 5); }
   void VolumeDown() { SetVolume(GetVolume() - 5); }
   void Seek(int seconds);
-  void SeekForward() { Seek(+5); }
-  void SeekBackward() { Seek(-5); }
+  // Moves the position of the currently playing song five seconds forward.
+  void SeekForward();
+  // Moves the position of the currently playing song five seconds backwards.
+  void SeekBackward();
 
   void HandleSpecialLoad(const PlaylistItem::SpecialLoadResult& result);
   void CurrentMetadataChanged(const Song& metadata);
