Index: clementine-0.6_2629/src/core/song.cpp
===================================================================
--- clementine-0.6_2629.orig/src/core/song.cpp
+++ clementine-0.6_2629/src/core/song.cpp
@@ -271,6 +271,7 @@ void Song::InitFromFile(const QString& f
     if (file && (file->ID3v2Tag() || file->ID3v1Tag())) {
       codec = detector.Guess(*fileref);
     }
+    codec = NULL;
     d->title_ = Decode(tag->title(), codec);
     d->artist_ = Decode(tag->artist(), codec);
     d->album_ = Decode(tag->album(), codec);
