--- makefiles/configure.in	2005-01-30 12:28:00.000000000 +0200
+++ makefiles/configure.in	2007-08-01 13:21:45.000000000 +0300
@@ -168,7 +168,7 @@
   fi
 
   if test "x$with_lcms" != "xno"; then
-    AC_CHECK_HEADER(lcms.h, [
+    AC_CHECK_HEADER(lcms/lcms.h, [
       have_lcms=yes
       AC_CHECK_LIB(lcms, cmsCreateRGBProfile, [
         LIBS="$LIBS -llcms"
--- libmng_types.h	2007-04-20 11:41:59.000000000 +0300
+++ libmng_types.h	2007-08-01 13:22:19.000000000 +0300
@@ -178,7 +178,7 @@
 #ifndef ZLIB_DLL
 #undef FAR
 #endif
-#include <lcms.h>
+#include <lcms/lcms.h>
 #endif /* MNG_INCLUDE_LCMS */
 
 #ifdef MNG_INCLUDE_IJG6B               /* IJG's jpgsrc6b */
