--- kcm_gtk/kcmgtk.cpp.orig	2009-02-20 17:25:42.000000000 +0200
+++ kcm_gtk/kcmgtk.cpp	2009-02-20 17:25:51.000000000 +0200
@@ -37,7 +37,7 @@
 #include <sys/stat.h>
 
 
-const QString KcmGtk::k_gtkRcFileName(QDir::homePath() + "/.gtkrc-2.0-kde4");
+const QString KcmGtk::k_gtkRcFileName(QDir::homePath() + "/.gtkrc-2.0");
 const QString KcmGtk::k_envFileName(KGlobal::dirs()->localkdedir() + "/env/gtk-qt-engine.rc.sh");
 const QString KcmGtk::k_qtThemeName("Qt4");
 
