Index: cfitsio/Makefile.in
===================================================================
--- cfitsio.orig/Makefile.in
+++ cfitsio/Makefile.in
@@ -88,7 +88,7 @@ libcfitsio.a:	${OBJECTS}
 shared: libcfitsio${SHLIB_SUFFIX}
 
 libcfitsio${SHLIB_SUFFIX}: ${OBJECTS}
-		${SHLIB_LD} ${LDFLAGS} -o $@ ${OBJECTS}
+		${SHLIB_LD} ${LDFLAGS} -o $@ ${OBJECTS} -lm
 
 install:	libcfitsio.a $(INSTALL_DIRS)
 		@if [ -f libcfitsio.a ]; then \
diff -Nuar cfitsio.orig//Makefile.in cfitsio/Makefile.in
--- cfitsio.orig//Makefile.in	2010-11-29 11:44:22.601167343 +0200
+++ cfitsio/Makefile.in	2010-11-29 11:45:20.833170678 +0200
@@ -113,16 +113,16 @@
 		${CC} $(CFLAGS) $(DEFS) -o funpack funpack.o fpackutil.o -L. -lcfitsio -lm ${LIBS}
 
 fitscopy:	fitscopy.o libcfitsio.a ${OBJECTS}
-		${CC} $(CFLAGS) $(DEFS) -o fitscopy fitscopy.o -L. -lcfitsio -lm ${LIBS}
+		${CC} $(CFLAGS) $(DEFS) -o fitscopy fitscopy.o -L. -lcfitsio  ${LIBS}
 
 speed:		speed.o libcfitsio.a ${OBJECTS}
 		${CC} $(CFLAGS) $(DEFS) -o speed speed.o -L. -lcfitsio -lm ${LIBS}
 
 imcopy:		imcopy.o libcfitsio.a ${OBJECTS}
-		${CC} $(CFLAGS) $(DEFS) -o imcopy imcopy.o -L. -lcfitsio -lm ${LIBS}
+		${CC} $(CFLAGS) $(DEFS) -o imcopy imcopy.o -L. -lcfitsio ${LIBS}
 
 listhead:	listhead.o libcfitsio.a ${OBJECTS}
-		${CC} $(CFLAGS) $(DEFS) -o listhead listhead.o -L. -lcfitsio -lm ${LIBS}
+		${CC} $(CFLAGS) $(DEFS) -o listhead listhead.o -L. -lcfitsio ${LIBS}
 
 cookbook:	cookbook.o libcfitsio.a ${OBJECTS}
 		${CC} $(CFLAGS) $(DEFS) -o cookbook cookbook.o -L. -lcfitsio -lm ${LIBS}
