Index: kdebase/workspace/plasma/desktop/shell/controllerwindow.cpp
===================================================================
--- kdebase/workspace/plasma/desktop/shell/controllerwindow.cpp	(revision 1179174)
+++ kdebase/workspace/plasma/desktop/shell/controllerwindow.cpp	(revision 1179175)
@@ -160,6 +160,10 @@
 
 void ControllerWindow::setGraphicsWidget(QGraphicsWidget *widget)
 {
+    if (m_graphicsWidget == widget) {
+        return;
+    }
+
     if (m_graphicsWidget) {
         m_graphicsWidget->removeEventFilter(this);
         if (m_graphicsWidget == m_widgetExplorer) {
