Index: kdebase/workspace/kwin/events.cpp
===================================================================
--- kdebase/workspace/kwin/events.cpp	(revision 1175747)
+++ kdebase/workspace/kwin/events.cpp	(revision 1175748)
@@ -1698,6 +1698,7 @@
             sync_timeout = NULL;
             if( sync_resize_pending )
                 performMoveResize();
+            sync_resize_pending = false;
             }
         }
     }
Index: kdebase/workspace/kwin/geometry.cpp
===================================================================
--- kdebase/workspace/kwin/geometry.cpp	(revision 1175747)
+++ kdebase/workspace/kwin/geometry.cpp	(revision 1175748)
@@ -3299,7 +3299,6 @@
         sendSyncRequest();
         }
 #endif
-    sync_resize_pending = false;
     if( transparent )
         {
         if( !haveResizeEffect )
