Index: kdebase-workspace-4.5.5/solid/hal/halpower.cpp
===================================================================
--- kdebase-workspace-4.5.5.orig/solid/hal/halpower.cpp
+++ kdebase-workspace-4.5.5/solid/hal/halpower.cpp
@@ -34,7 +34,7 @@
 
 HalPower::HalPower(QObject *parent, const QVariantList  & /*args */)
     : PowerManager(parent),
-      m_brightnessInHardware(true),
+      m_brightnessInHardware(false),
       m_halComputer("org.freedesktop.Hal",
                      "/org/freedesktop/Hal/devices/computer",
                      "org.freedesktop.Hal.Device",
