Index: kdebase-workspace-4.5.2/systemsettings/categories/settings-display.desktop
===================================================================
--- kdebase-workspace-4.5.2.orig/systemsettings/categories/settings-display.desktop
+++ kdebase-workspace-4.5.2/systemsettings/categories/settings-display.desktop
@@ -51,7 +51,7 @@ Name[sr@ijekavianlatin]=Prikaz i monitor
 Name[sr@latin]=Prikaz i monitor
 Name[sv]=Bildskärm
 Name[th]=ระบบแสดงผลและจอภาพ
-Name[tr]=Ekran ve Moniör
+Name[tr]=Ekran ve Monitör
 Name[uk]=Показ зображення і монітор
 Name[x-test]=xxDisplay and Monitorxx
 Name[zh_CN]=显示
