diff -Nur kdebase-runtime-4.5.5-orig//kcontrol/locale/CMakeLists.txt kdebase-runtime-4.5.5/kcontrol/locale/CMakeLists.txt
--- kdebase-runtime-4.5.5-orig//kcontrol/locale/CMakeLists.txt	2011-01-13 09:19:53.388000016 +0200
+++ kdebase-runtime-4.5.5/kcontrol/locale/CMakeLists.txt	2011-01-13 09:20:16.789000019 +0200
@@ -26,6 +26,16 @@
 install(TARGETS kcm_locale  DESTINATION ${PLUGIN_INSTALL_DIR} )
 
 
+#### KAuth helper and actions ####
+kde4_add_executable(kcmlocalehelper helper.cpp)
+target_link_libraries(kcmlocalehelper ${KDE4_KDECORE_LIBS})
+install(TARGETS kcmlocalehelper DESTINATION ${LIBEXEC_INSTALL_DIR})
+
+kde4_install_auth_helper_files(kcmlocalehelper org.kde.kcontrol.kcmlocale root)
+
+kde4_install_auth_actions(org.kde.kcontrol.kcmlocale kcmlocale.actions)
+
+
 ########### install files ###############
 
 install( FILES language.desktop  DESTINATION  ${SERVICES_INSTALL_DIR} )
diff -Nur kdebase-runtime-4.5.5-orig//kcontrol/locale/helper.cpp kdebase-runtime-4.5.5/kcontrol/locale/helper.cpp
--- kdebase-runtime-4.5.5-orig//kcontrol/locale/helper.cpp	1970-01-01 02:00:00.000000000 +0200
+++ kdebase-runtime-4.5.5/kcontrol/locale/helper.cpp	2011-01-13 09:20:16.795000016 +0200
@@ -0,0 +1,77 @@
+#include "helper.h"
+#include <iostream>
+#include <QStringList>
+#include <QFile>
+#include <QTextStream>
+#include <QDebug>
+
+ActionReply Helper::createReply(int code, const QVariantMap *returnData)
+{
+    ActionReply reply;
+
+    if (code) {
+        reply = ActionReply::HelperError;
+        reply.setErrorCode(code);
+    } else {
+        reply = ActionReply::SuccessReply;
+    }
+
+    if (returnData)
+        reply.setData(*returnData);
+
+    return reply;
+}
+
+
+bool Helper::writeLocale(const QString &locale)
+{
+    QString mudurFile = "/etc/conf.d/mudur";
+    QString mudurNewFile = "/etc/conf.d/mudurnew";
+
+    // open file to read
+    QFile file(mudurFile);
+    if( !file.open( QIODevice::ReadOnly | QIODevice::Text)) {
+        qDebug() << "Failed to read.";
+        return false;
+    }
+
+    // read whole file, replace language string and create a new string
+    QString output;
+    QTextStream in(&file);
+    while (!in.atEnd()) {
+        QString line = in.readLine();
+        if (line.contains("language=")) {
+            QString newline = "language=\"" + locale + "\"\n";
+            qDebug() << newline << endl;
+            output.append(newline);
+        } else {
+            QString newline = line + "\n";
+            output.append(newline);
+        }
+    }
+    file.close();
+
+    // create a new file to write
+    QFile fileNew(mudurNewFile);
+    if( !fileNew.open( QIODevice::WriteOnly | QIODevice::Text)) {
+        qDebug() << "Failed to write.";
+        return false;
+    }
+    QTextStream out(&fileNew);
+    out << output; //write the output to the new file
+    fileNew.close();
+
+    return true;
+}
+
+ActionReply Helper::managelocale(QVariantMap args)
+{
+    int code = 0;
+
+    QString locale = args.value("locale").toString();
+
+    code = (writeLocale(locale) ? 0 : WriteLocaleError);
+    return createReply(code);
+}
+
+KDE4_AUTH_HELPER_MAIN("org.kde.kcontrol.kcmlocale", Helper)
diff -Nur kdebase-runtime-4.5.5-orig//kcontrol/locale/helper.h kdebase-runtime-4.5.5/kcontrol/locale/helper.h
--- kdebase-runtime-4.5.5-orig//kcontrol/locale/helper.h	1970-01-01 02:00:00.000000000 +0200
+++ kdebase-runtime-4.5.5/kcontrol/locale/helper.h	2011-01-13 09:20:16.805000015 +0200
@@ -0,0 +1,17 @@
+#include <kauth.h>
+#define WriteLocaleError 1
+
+using namespace KAuth;
+
+class Helper : public QObject
+{
+    Q_OBJECT
+
+    public slots:
+        ActionReply managelocale(QVariantMap args);
+
+    private:
+        bool writeLocale(const QString &locale);
+        ActionReply createReply(int code, const QVariantMap *returnData = 0);
+
+};
diff -Nur kdebase-runtime-4.5.5-orig//kcontrol/locale/kcmlocale.actions kdebase-runtime-4.5.5/kcontrol/locale/kcmlocale.actions
--- kdebase-runtime-4.5.5-orig//kcontrol/locale/kcmlocale.actions	1970-01-01 02:00:00.000000000 +0200
+++ kdebase-runtime-4.5.5/kcontrol/locale/kcmlocale.actions	2011-01-13 09:20:16.810000016 +0200
@@ -0,0 +1,9 @@
+[Domain]
+Name=Langage Manager Control Module
+Icon=preferences-system-login
+
+[org.kde.kcontrol.kcmlocale.managelocale]
+Name=Save the Locale settings
+Description=Administrator authorization is required to change the Locale settings
+Policy=auth_admin
+Persistence=session
diff -Nur kdebase-runtime-4.5.5-orig//kcontrol/locale/kcmlocale.cpp kdebase-runtime-4.5.5/kcontrol/locale/kcmlocale.cpp
--- kdebase-runtime-4.5.5-orig//kcontrol/locale/kcmlocale.cpp	2011-01-13 09:19:53.389000016 +0200
+++ kdebase-runtime-4.5.5/kcontrol/locale/kcmlocale.cpp	2011-01-13 09:20:16.810000016 +0200
@@ -24,13 +24,14 @@
  */
 
 #include "kcmlocale.h"
-
+#include "helper.h"
 #include <QPushButton>
 
 #include <KDebug>
 #include <KConfig>
 #include <KConfigGroup>
 #include <KStandardDirs>
+#include <KMessageBox>
 
 #include "kcmlocale.moc"
 #include "toplevel.h"
@@ -73,6 +74,7 @@
 
     m_crLabel->setObjectName(I18N_NOOP("Country or region:"));
     m_languagesLabel->setObjectName(I18N_NOOP("Languages:"));
+    m_systemWide->setObjectName(I18N_NOOP("Set the first language as the system default"));
     languageRemove->setObjectName(I18N_NOOP("Remove"));
     m_upButton->setObjectName(QString());
     m_downButton->setObjectName(QString());
@@ -88,6 +90,7 @@
     connect(m_selectedCountryLabel, SIGNAL(linkActivated(const QString &)), SLOT(changeCountry()));
     connect(languageAdd, SIGNAL(activated(const QString &)), SLOT(slotAddLanguage(const QString &)));
     connect(m_languages, SIGNAL(itemSelectionChanged()), SLOT(slotCheckButtons()));
+    connect(m_systemWide, SIGNAL(clicked()), this, SIGNAL(localeChanged()));
 
     m_upButton->setIcon(KIcon("arrow-up"));
     m_downButton->setIcon(KIcon("arrow-down"));
@@ -197,8 +200,33 @@
                          m_languageList.join(":"), KConfig::Persistent|KConfig::Global);
 
   config->sync();
+
+  if (m_systemWide->isChecked()) {
+      int replyErrorCode = saveSystemWide();
+      qDebug() << replyErrorCode << endl;
+      if (replyErrorCode != 0)
+          KMessageBox::error(this, i18n("KAuth returned an error code: %1", replyErrorCode));
+  };
+}
+
+int  KLocaleConfig::saveSystemWide()
+{
+  KAuth::Action action("org.kde.kcontrol.kcmlocale.managelocale");
+  action.setHelperID("org.kde.kcontrol.kcmlocale");
+
+  QVariantMap helperargs;
+  helperargs["locale"] = m_languageList.at(0);
+
+  qDebug() << m_languageList.at(0);
+
+  action.setArguments(helperargs);
+
+  KAuth::ActionReply reply = action.execute();
+  return reply.errorCode();
+
 }
 
+
 void KLocaleConfig::slotCheckButtons()
 {
   languageRemove->setEnabled( selectedRow() != -1 && m_languages->count() > 1 );
@@ -280,3 +308,4 @@
   emit localeChanged();
   emit languageChanged();
 }
+
diff -Nur kdebase-runtime-4.5.5-orig//kcontrol/locale/kcmlocale.h kdebase-runtime-4.5.5/kcontrol/locale/kcmlocale.h
--- kdebase-runtime-4.5.5-orig//kcontrol/locale/kcmlocale.h	2011-01-13 09:19:53.388000016 +0200
+++ kdebase-runtime-4.5.5/kcontrol/locale/kcmlocale.h	2011-01-13 09:20:16.821000011 +0200
@@ -68,7 +68,7 @@
   void languageMove(Direction direcition);
   int selectedRow() const;
 
-
+  int saveSystemWide();
   // NOTE: we need to mantain our own language list instead of using KLocale's
   // because KLocale does not add a language if there is no translation
   // for the current application so it would not be possible to set
diff -Nur kdebase-runtime-4.5.5-orig//kcontrol/locale/language.ui kdebase-runtime-4.5.5/kcontrol/locale/language.ui
--- kdebase-runtime-4.5.5-orig//kcontrol/locale/language.ui	2011-01-13 09:19:53.389000016 +0200
+++ kdebase-runtime-4.5.5/kcontrol/locale/language.ui	2011-01-13 09:20:16.832000034 +0200
@@ -1,53 +1,54 @@
-<ui version="4.0" >
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
  <class>Language</class>
- <widget class="QWidget" name="Language" >
-  <property name="geometry" >
+ <widget class="QWidget" name="Language">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
-    <width>482</width>
-    <height>303</height>
+    <width>1031</width>
+    <height>595</height>
    </rect>
   </property>
-  <layout class="QGridLayout" >
-   <item row="0" column="0" >
-    <layout class="QVBoxLayout" >
+  <layout class="QGridLayout">
+   <item row="0" column="0">
+    <layout class="QVBoxLayout">
      <item>
-      <layout class="QHBoxLayout" >
+      <layout class="QHBoxLayout">
        <item>
-        <layout class="QGridLayout" >
-         <item row="0" column="0" colspan="3" >
-          <widget class="QLabel" name="m_crLabel" >
-           <property name="sizePolicy" >
-            <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
+        <layout class="QGridLayout">
+         <item row="0" column="0" colspan="3">
+          <widget class="QLabel" name="m_crLabel">
+           <property name="sizePolicy">
+            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
-           <property name="text" >
+           <property name="text">
             <string>Country or region:</string>
            </property>
-           <property name="alignment" >
+           <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
-         <item row="1" column="0" colspan="3" >
-          <widget class="QLabel" name="m_languagesLabel" >
-           <property name="text" >
+         <item row="1" column="0" colspan="3">
+          <widget class="QLabel" name="m_languagesLabel">
+           <property name="text">
             <string>Languages:</string>
            </property>
-           <property name="alignment" >
+           <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTop|Qt::AlignTrailing</set>
            </property>
           </widget>
          </item>
-         <item row="3" column="0" colspan="2" >
+         <item row="3" column="0" colspan="2">
           <spacer>
-           <property name="orientation" >
+           <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
-           <property name="sizeHint" stdset="0" >
+           <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
@@ -55,96 +56,119 @@
            </property>
           </spacer>
          </item>
-         <item row="3" column="2" >
-          <layout class="QVBoxLayout" >
+         <item row="2" column="0">
+          <spacer name="verticalSpacer">
+           <property name="orientation">
+            <enum>Qt::Vertical</enum>
+           </property>
+           <property name="sizeHint" stdset="0">
+            <size>
+             <width>20</width>
+             <height>40</height>
+            </size>
+           </property>
+          </spacer>
+         </item>
+         <item row="3" column="2">
+          <layout class="QVBoxLayout">
            <item>
-            <widget class="QPushButton" name="m_upButton" >
-             <property name="text" >
+            <widget class="QPushButton" name="m_upButton">
+             <property name="text">
               <string/>
              </property>
-             <property name="icon" >
+             <property name="icon">
               <iconset>
                <normaloff>../../../../../../../../opt/kde3/share/icons/crystalsvg/32x32/actions/up.png</normaloff>../../../../../../../../opt/kde3/share/icons/crystalsvg/32x32/actions/up.png</iconset>
              </property>
             </widget>
            </item>
            <item>
-            <widget class="QPushButton" name="m_downButton" >
-             <property name="text" >
+            <widget class="QPushButton" name="m_downButton">
+             <property name="text">
               <string/>
              </property>
-             <property name="icon" >
+             <property name="icon">
               <iconset>
                <normaloff>../../../../../../../../opt/kde3/share/icons/crystalsvg/32x32/actions/down.png</normaloff>../../../../../../../../opt/kde3/share/icons/crystalsvg/32x32/actions/down.png</iconset>
              </property>
             </widget>
            </item>
+           <item>
+            <spacer name="verticalSpacer_2">
+             <property name="orientation">
+              <enum>Qt::Vertical</enum>
+             </property>
+             <property name="sizeType">
+              <enum>QSizePolicy::Fixed</enum>
+             </property>
+             <property name="sizeHint" stdset="0">
+              <size>
+               <width>20</width>
+               <height>30</height>
+              </size>
+             </property>
+            </spacer>
+           </item>
           </layout>
          </item>
-         <item row="2" column="0" >
-          <spacer name="verticalSpacer" >
-           <property name="orientation" >
-            <enum>Qt::Vertical</enum>
-           </property>
-           <property name="sizeHint" stdset="0" >
-            <size>
-             <width>20</width>
-             <height>40</height>
-            </size>
-           </property>
-          </spacer>
-         </item>
         </layout>
        </item>
        <item>
-        <layout class="QVBoxLayout" >
+        <layout class="QVBoxLayout">
          <item>
-          <widget class="QLabel" name="m_selectedCountryLabel" >
-           <property name="sizePolicy" >
-            <sizepolicy vsizetype="Preferred" hsizetype="Preferred" >
+          <widget class="QLabel" name="m_selectedCountryLabel">
+           <property name="sizePolicy">
+            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
-           <property name="text" >
-            <string>None selected (&lt;a href="/change">change...&lt;/a>)</string>
+           <property name="text">
+            <string>None selected (&lt;a href=&quot;/change&quot;&gt;change...&lt;/a&gt;)</string>
            </property>
           </widget>
          </item>
          <item>
-          <widget class="QListWidget" name="m_languages" />
+          <widget class="QListWidget" name="m_languages"/>
+         </item>
+         <item>
+          <layout class="QHBoxLayout">
+           <item>
+            <widget class="QCheckBox" name="m_systemWide">
+             <property name="text">
+              <string>Set the first language as the system default</string>
+             </property>
+            </widget>
+           </item>
+           <item>
+            <spacer>
+             <property name="orientation">
+              <enum>Qt::Horizontal</enum>
+             </property>
+             <property name="sizeHint" stdset="0">
+              <size>
+               <width>40</width>
+               <height>20</height>
+              </size>
+             </property>
+            </spacer>
+           </item>
+           <item>
+            <widget class="KLanguageButton" name="languageAdd"/>
+           </item>
+           <item>
+            <widget class="QPushButton" name="languageRemove">
+             <property name="text">
+              <string>Remove</string>
+             </property>
+            </widget>
+           </item>
+          </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
-     <item>
-      <layout class="QHBoxLayout" >
-       <item>
-        <spacer>
-         <property name="orientation" >
-          <enum>Qt::Horizontal</enum>
-         </property>
-         <property name="sizeHint" stdset="0" >
-          <size>
-           <width>40</width>
-           <height>20</height>
-          </size>
-         </property>
-        </spacer>
-       </item>
-       <item>
-        <widget class="KLanguageButton" name="languageAdd" />
-       </item>
-       <item>
-        <widget class="QPushButton" name="languageRemove" >
-         <property name="text" >
-          <string>Remove</string>
-         </property>
-        </widget>
-       </item>
-      </layout>
-     </item>
     </layout>
    </item>
   </layout>
