--- glew-1.5.8/Makefile	2011-03-23 15:07:32.456140865 -0700
+++ glew-1.5.8/Makefile.new	2011-03-23 15:08:29.169450772 -0700
@@ -131,7 +131,7 @@
 	$(AR) cr $@ $^
 
 lib/$(LIB.SHARED.MX): $(LIB.SOBJS.MX)
-	$(LD) $(LDFLAGS.SO) -o $@ $^ $(LIB.LDFLAGS) $(LIB.LIBS)
+	$(LD) $(LDFLAGS.SO.MX) -o $@ $^ $(LIB.LDFLAGS) $(LIB.LIBS)
 ifeq ($(filter-out mingw% cygwin,$(SYSTEM)),)
 else
 	$(LN) $(LIB.SHARED.MX) lib/$(LIB.SONAME.MX)
--- glew-1.5.8/config/Makefile.linux	2011-03-23 15:07:32.570143498 -0700
+++ glew-1.5.8/config/Makefile.linux.new	2011-03-23 15:08:09.650999994 -0700
@@ -3,6 +3,7 @@
 LD = cc
 PICFLAG = -fPIC
 LDFLAGS.SO = -shared -Wl,-soname=$(LIB.SONAME)
+LDFLAGS.SO.MX = -shared -Wl,-soname=$(LIB.SONAME.MX)
 M_ARCH ?= $(shell uname -m)
 LIBDIR = $(GLEW_DEST)$(libdir)
 LDFLAGS.GL = -lGL -lX11 -lGLU
