Index: konq-plugins/imagerotation/imageconverter.desktop
===================================================================
--- konq-plugins.orig/imagerotation/imageconverter.desktop
+++ konq-plugins/imagerotation/imageconverter.desktop
@@ -228,7 +228,7 @@ Name[x-test]=xxTIFFxx
 Name[zh_CN]=TIFF
 Name[zh_TW]=TIFF
 Icon=image-tiff
-Exec=convert %f "`echo %f | perl -pe 's/\\.[^.]+$//'`.tif"
+Exec=convert %f "`echo %f | perl -pe 's/\\.[^.]+$//'`.tiff"
 
 [Desktop Action convToGIF]
 Name=GIF
