diff -Naur konq-plugins-4.3.1-orig/babelfish/plugin_babelfish.cpp konq-plugins-4.3.1/babelfish/plugin_babelfish.cpp
--- konq-plugins-4.3.1-orig/babelfish/plugin_babelfish.cpp	2009-11-25 11:04:46.767229747 +0200
+++ konq-plugins-4.3.1/babelfish/plugin_babelfish.cpp	2009-11-25 11:05:44.644854717 +0200
@@ -41,14 +41,8 @@
 #include <KConfigGroup>
 #include <KHTMLPart>
 
-#ifdef HAVE_WEBKITKDE
+#include <QWebView>
 #include <webkitpart.h>
-#ifdef HAVE_WEBVIEW
-#include <webview.h>
-#else
-#include <webkitview.h>
-#endif
-#endif
 
 static const KAboutData aboutdata("babelfish", 0, ki18n("Translate Web Page") , "1.0" );
 K_PLUGIN_FACTORY(BabelFishFactory, registerPlugin<PluginBabelFish>();)
