Index: Root_Actions_2.5.92/rootactions-servicemenu.pl
===================================================================
--- Root_Actions_2.5.92/rootactions-servicemenu.pl.orig
+++ Root_Actions_2.5.92/rootactions-servicemenu.pl
@@ -837,11 +837,7 @@ if ( $#ARGV lt 0 ) {
 &get_kde_path ;
 
 # Check whether kdesudo is installed, and use it instead of kdesu to get root privileges.
-if ( `which kdesudo` ) {
-	$SUCOMMAND = "$KDEBINPATH"."kdesudo -d --noignorebutton --"}
-elsif ( `which kdesu` ) {
-	$SUCOMMAND = "$KDEBINPATH"."kdesu -d --"}
-else { $SUCOMMAND = "xdg-su -c"}
+$SUCOMMAND = "xdg-su -c";
 
 $DIALOGCOMMAND = "$KDEBINPATH"."kdialog" ;
 $CONFIGCOMMAND = "$KDEBINPATH"."kreadconfig" ;
