diff -up grub-0.97/acinclude.m4.build-id grub-0.97/acinclude.m4
--- grub-0.97/acinclude.m4.build-id	2008-01-01 19:45:11.000000000 +0900
+++ grub-0.97/acinclude.m4	2008-01-01 20:03:56.000000000 +0900
@@ -59,7 +59,7 @@ else
 fi
 grub_cv_prog_objcopy_absolute=yes
 for link_addr in 2000 8000 7C00; do
-  if AC_TRY_COMMAND([${CC-cc} ${CFLAGS} -nostdlib -Wl,-N -Wl,-Ttext -Wl,$link_addr conftest.o -o conftest.exec]); then :
+  if AC_TRY_COMMAND([${CC-cc} ${CFLAGS} ${LDFLAGS} -nostdlib -Wl,-N -Wl,-Ttext -Wl,$link_addr conftest.o -o conftest.exec]); then :
   else
     AC_MSG_ERROR([${CC-cc} cannot link at address $link_addr])
   fi
diff -up grub-0.97/stage1/Makefile.am.build-id grub-0.97/stage1/Makefile.am
--- grub-0.97/stage1/Makefile.am.build-id	2008-01-01 19:45:11.000000000 +0900
+++ grub-0.97/stage1/Makefile.am	2008-01-01 20:05:24.000000000 +0900
@@ -5,7 +5,7 @@ CLEANFILES = $(nodist_pkgdata_DATA)
 
 # We can't use builtins or standard includes.
 AM_CCASFLAGS = $(STAGE1_CFLAGS) -fno-builtin -nostdinc
-LDFLAGS = -nostdlib -Wl,-N,-Ttext,7C00
+stage1_exec_LDFLAGS = -nostdlib -Wl,-N,-Ttext,7C00
 
 noinst_PROGRAMS = stage1.exec
 stage1_exec_SOURCES = stage1.S stage1.h
diff -up grub-0.97/configure.ac.build-id grub-0.97/configure.ac
--- grub-0.97/configure.ac.build-id	2008-01-01 19:45:45.000000000 +0900
+++ grub-0.97/configure.ac	2008-01-01 20:02:44.000000000 +0900
@@ -149,6 +149,17 @@ fi
 if test "x$undef_flag" = xyes; then
   CPPFLAGS="$CPPFLAGS -Wundef"
 fi
+ 
+# Check if build ID can be disabled in the linker
+AC_MSG_CHECKING([whether linker accepts `--build-id=none'])
+save_LDFLAGS="$LDFLAGS"
+LDFLAGS="$LDFLAGS -Wl,--build-id=none"
+AC_TRY_LINK(, , build_id_flag=yes, build_id_flag=no)
+AC_MSG_RESULT([$build_id_flag])
+LDFLAGS="$save_LDFLAGS"
+if test "x$build_id_flag" = xyes; then
+  LDFLAGS="$LDFLAGS -Wl,--build-id=none"
+fi
 
 if test "x$with_binutils" != x; then
 dnl  AC_PATH_TOOL(OBJCOPY, objcopy, , "$with_binutils:$PATH")
