diff -Nur grub-0.97-old/lib/device.c grub-0.97/lib/device.c
--- grub-0.97-old/lib/device.c	2006-10-11 17:12:42.833956530 +0300
+++ grub-0.97/lib/device.c	2006-10-11 17:12:55.967959858 +0300
@@ -1037,6 +1037,8 @@
   else
     {
       if ((strncmp (dev, "/dev/ataraid/", 13) == 0) ||
+         (strncmp (dev, "/dev/ida/", 9) == 0) ||
+         (strncmp (dev, "/dev/cciss/", 11) == 0) ||
          (strncmp (dev, "/dev/rd/", 8) == 0))
         strcpy (dev + strlen(dev), "p");
     }
