Index: kdelibs-4.7.0/plasma/theme.cpp
===================================================================
--- kdelibs-4.7.0.orig/plasma/theme.cpp
+++ kdelibs-4.7.0/plasma/theme.cpp
@@ -54,8 +54,8 @@ namespace Plasma
 {
 
 //NOTE: Default wallpaper can be set from the theme configuration
-#define DEFAULT_WALLPAPER_THEME "default"
-#define DEFAULT_WALLPAPER_SUFFIX ".png"
+#define DEFAULT_WALLPAPER_THEME "Cervus.elaphus"
+#define DEFAULT_WALLPAPER_SUFFIX ".jpg"
 static const int DEFAULT_WALLPAPER_WIDTH = 1920;
 static const int DEFAULT_WALLPAPER_HEIGHT = 1200;
 
