Index: kdelibs-4.6.1/plasma/theme.cpp
===================================================================
--- kdelibs-4.6.1.orig/plasma/theme.cpp
+++ kdelibs-4.6.1/plasma/theme.cpp
@@ -52,7 +52,7 @@
 namespace Plasma
 {
 
-#define DEFAULT_WALLPAPER_THEME "Horos"
+#define DEFAULT_WALLPAPER_THEME "Pardus_Mood"
 #define DEFAULT_WALLPAPER_SUFFIX ".png"
 static const int DEFAULT_WALLPAPER_WIDTH = 1920;
 static const int DEFAULT_WALLPAPER_HEIGHT = 1200;
