Index: kdelibs-4.7.0/kio/kio/kprotocolmanager.cpp
===================================================================
--- kdelibs-4.7.0.orig/kio/kio/kprotocolmanager.cpp
+++ kdelibs-4.7.0/kio/kio/kprotocolmanager.cpp
@@ -644,6 +644,7 @@ QString KProtocolManager::defaultUserAge
     d->useragent += QString::number(KDE::versionMajor());
     d->useragent += QL1C('.');
     d->useragent += QString::number(KDE::versionMinor());
+    d->useragent += QL1S(" Pardus");
   }
   else
   {
