Index: kde4libs-4.5.80/kio/kio/kprotocolmanager.cpp
===================================================================
--- kde4libs-4.5.80.orig/kio/kio/kprotocolmanager.cpp	2010-11-05 06:48:26.000000000 -0400
+++ kde4libs-4.5.80/kio/kio/kprotocolmanager.cpp	2010-11-21 15:54:23.000000000 -0500
@@ -87,7 +87,7 @@
 QString::number(KDE::versionMajor()) % QL1C('.') % QString::number(KDE::versionMinor()) % \
 X % QL1S(") KHTML/") % \
 QString::number(KDE::versionMajor()) % QL1C('.') % QString::number(KDE::versionMinor()) % \
-QL1C('.') % QString::number(KDE::versionRelease()) % QL1S(" (like Gecko)")
+QL1C('.') % QString::number(KDE::versionRelease()) % QL1S(" (like Gecko) Pardus")
 
 #define PRIVATE_DATA \
 KProtocolManagerPrivate *d = kProtocolManagerPrivate
