Index: kdelibs-4.6.5/kfile/kfilepreviewgenerator.cpp
===================================================================
--- kdelibs-4.6.5.orig/kfile/kfilepreviewgenerator.cpp
+++ kdelibs-4.6.5/kfile/kfilepreviewgenerator.cpp
@@ -1032,7 +1032,9 @@ void KFilePreviewGenerator::Private::sta
             m_enabledPlugins = globalConfig.readEntry("Plugins", QStringList()
                                                                  << "directorythumbnail"
                                                                  << "imagethumbnail"
-                                                                 << "jpegthumbnail");
+                                                                 << "svgthumbnail"
+                                                                 << "ffmpegthumbs"
+                                                                 << "jpegrotatedthumbnail");
         }
 
         KIO::PreviewJob* job = KIO::filePreview(items, width, height, 0, 70, true, true, &m_enabledPlugins);
