--- kdelibs-4.2.3-orig/kded/kbuildservicefactory.cpp	2009-05-21 12:23:16.161424770 +0300
+++ kdelibs-4.2.3/kded/kbuildservicefactory.cpp	2009-05-21 12:23:33.554399343 +0300
@@ -266,7 +266,7 @@
 #endif
 
             if (!serviceType) {
-                kDebug(7021) << service->entryPath() << "specifies undefined mimetype/servicetype" << stName;
+                //kDebug(7021) << service->entryPath() << "specifies undefined mimetype/servicetype" << stName;
                 continue;
             }
 
