=== modified file 'util/mtp-hotplug.c'
--- util/mtp-hotplug.c	2011-01-09 23:18:55 +0000
+++ util/mtp-hotplug.c	2011-01-09 23:20:10 +0000
@@ -63,7 +63,7 @@
    * with ENV{ID_MEDIA_PLAYER}=1 writable for the console
    * user.
    */
-  char default_udev_action[] = "SYMLINK+=\"libmtp-%k\", ENV{ID_MTP_DEVICE}=\"1\", ENV{ID_MEDIA_PLAYER}=\"1\"";
+  char default_udev_action[] = "SYMLINK+=\"libmtp-%k\", ENV{ID_MTP_DEVICE}=\"1\", ENV{ID_MEDIA_PLAYER}=\"1\", MODE=\"0660\", GROUP=\"plugdev\"";
   char *action; // To hold the action actually used.
   uint16_t last_vendor = 0x0000U;
 

 
