diff -Nur lirc-0.8.3pre1-old/configure.ac lirc-0.8.3pre1/configure.ac
--- lirc-0.8.3pre1-old/configure.ac	2007-12-05 19:09:41.334145542 +0200
+++ lirc-0.8.3pre1/configure.ac	2007-12-05 19:11:39.227223071 +0200
@@ -1585,11 +1585,7 @@
 ])
 
 AH_BOTTOM([
-#ifdef LIRC_HAVE_DEVFS
-#define LIRC_DRIVER_DEVICE	DEVDIR "/" DEV_LIRC "/0"
-#else
-#define LIRC_DRIVER_DEVICE      DEVDIR "/" DEV_LIRC
-#endif /* LIRC_HAVE_DEVFS */
+#define LIRC_DRIVER_DEVICE    "/dev/lirc0"
 
 #define LIRC_IRTTY		DEVDIR "/" "ttyS0"
 
