diff -Nur lirc-0.8.4a-old/drivers/lirc_parallel/Makefile.am lirc-0.8.4a/drivers/lirc_parallel/Makefile.am
--- lirc-0.8.4a-old/drivers/lirc_parallel/Makefile.am	2009-04-21 19:38:06.000000000 +0300
+++ lirc-0.8.4a/drivers/lirc_parallel/Makefile.am	2009-04-21 19:38:41.000000000 +0300
@@ -8,6 +8,6 @@
 automake_dummy_SOURCES = lirc_parallel.c lirc_parallel.h
 
 ## there is no *just* object file support in automake.  This is close enough
-module_DATA = lirc_parallel.o
+module_DATA = 
 
 include ../Makefile.common
