Index: kdegraphics-4.3.3/okular/core/fileprinter.cpp
===================================================================
--- kdegraphics-4.3.3.orig/okular/core/fileprinter.cpp
+++ kdegraphics-4.3.3/okular/core/fileprinter.cpp
@@ -528,7 +528,7 @@ QStringList FilePrinter::optionOrientati
 {
     switch ( printer.orientation() ) {
     case QPrinter::Portrait:   return QStringList("-o") << "portrait";
-    case QPrinter::Landscape:  return QStringList("-o") << "landscape";
+    case QPrinter::Landscape:  return QStringList("-o") << "portrait";
     default:                   return QStringList();
     }
 }
