Index: gwenview/app/gwenview.desktop
===================================================================
--- gwenview/app/gwenview.desktop.orig
+++ gwenview/app/gwenview.desktop
@@ -126,7 +126,7 @@ Terminal=false
 Icon=gwenview
 Type=Application
 Categories=Qt;KDE;Graphics;Viewer;Photography;
-MimeType=inode/directory;image/gif;image/jpeg;image/png;image/bmp;image/x-eps;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-xbitmap;image/x-xpixmap;
+MimeType=inode/directory;image/gif;image/jpeg;image/png;image/bmp;image/x-eps;image/x-ico;image/x-portable-bitmap;image/x-portable-pixmap;image/x-xbitmap;image/x-xpixmap;image/svg+xml;image/svg+xml-compressed;
 X-DocPath=gwenview/index.html
 # InitialPreference should be greater than Okular so that Gwenview is the
 # primary application associated with images, but less than Konqueror or Dolphin
