Index: gwenview/app/gwenviewui.rc
===================================================================
--- gwenview/app/gwenviewui.rc.orig
+++ gwenview/app/gwenviewui.rc
@@ -123,11 +123,17 @@
 	<Action name="view" />
 	<Action name="fullscreen" />
 	<Separator/>
+	<Action name="go_up" />
 	<Action name="go_previous" />
 	<Action name="go_next" />
 	<Separator/>
 	<Action name="rotate_left" />
 	<Action name="rotate_right" />
+    <Separator/>
+    <Action name="view_actual_size"/>
+    <Action name="view_zoom_in"/>
+    <Action name="view_zoom_out"/>
+    <Action name="view_zoom_to_fit"/>
 </ToolBar>
 
 </kpartgui>
