Index: kfile/kfileplacesmodel.cpp
===================================================================
--- kfile/kfileplacesmodel.cpp	(révision 1085497)
+++ kfile/kfileplacesmodel.cpp	(copie de travail)
@@ -20,6 +20,7 @@
 #include "kfileplacesmodel.h"
 #include "kfileplacesitem_p.h"
 #include "kfileplacessharedbookmarks_p.h"
+#include "kglobalsettings.h"
 
 #include <QtCore/QMimeData>
 #include <QtCore/QTimer>
@@ -110,7 +111,22 @@
         KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
                                               "Home", I18N_NOOP2("KFile System Bookmarks", "Home"),
                                               KUrl(KUser().homeDir()), "user-home");
+        // We add Music, download, Images, Video
         KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Music", I18N_NOOP2("KFile System Bookmarks", "Music"),
+                                              KUrl(KGlobalSettings::musicPath()), "folder-sound");
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Downloads", I18N_NOOP2("KFile System Bookmarks", "Downloads"),
+                                              KUrl(KGlobalSettings::downloadPath()), "folder-downloads");
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Videos", I18N_NOOP2("KFile System Bookmarks", "Videos"),
+                                              KUrl(KGlobalSettings::videosPath()), "folder-video");
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
+                                              "Pictures", I18N_NOOP2("KFile System Bookmarks", "Pictures"),
+                                              KUrl(KGlobalSettings::picturesPath()), "folder-image");
+
+
+        KFilePlacesItem::createSystemBookmark(d->bookmarkManager,
                                               "Network", I18N_NOOP2("KFile System Bookmarks", "Network"),
                                               KUrl("remote:/"), "network-workgroup");
 #ifdef Q_OS_WIN
