Index: ecore-1.0.0/m4/efl_threads.m4
===================================================================
--- ecore-1.0.0.orig/m4/efl_threads.m4
+++ ecore-1.0.0/m4/efl_threads.m4
@@ -89,8 +89,8 @@ case "$host_os" in
       _efl_threads_libs="-mt"
       ;;
    *)
-      _efl_threads_cflags="-pthread"
-      _efl_threads_libs="-pthread"
+      _efl_threads_cflags="-lpthread"
+      _efl_threads_libs="-lpthread"
       ;;
 esac
 
