diff -uNrp FlightGear-2.0.0.orig/src/ATCDCL/ATCmgr.cxx FlightGear-2.0.0/src/ATCDCL/ATCmgr.cxx
--- FlightGear-2.0.0.orig/src/ATCDCL/ATCmgr.cxx	2010-02-17 17:11:56.000000000 +0100
+++ FlightGear-2.0.0/src/ATCDCL/ATCmgr.cxx	2010-07-18 22:38:35.907740944 +0200
@@ -303,7 +303,9 @@ void FGATCMgr::ZapOtherService(const str
         //cout << "Eradicating service: '" << svc->first << "'" << endl;
     svc->second->SetNoDisplay();
     svc->second->Update(0);     // one last update
-    delete svc->second;
+    //TODO FIXME! AIPlanes keep private copies obtained by "FGATCMgr::GetATCPointer".
+    // Deleting the object causes stale pointers. => For now, rather accept a tiny memory leak...
+    //delete svc->second;
     atc_list->erase(svc);
 // ALL pointers into the ATC list are now invalid,
 // so let's reset them:
