diff -uNrp FlightGear-2.0.0.orig/src/Main/CameraGroup.cxx FlightGear-2.0.0/src/Main/CameraGroup.cxx
--- FlightGear-2.0.0.orig/src/Main/CameraGroup.cxx	2010-02-17 17:12:13.000000000 +0100
+++ FlightGear-2.0.0/src/Main/CameraGroup.cxx	2010-06-02 16:27:18.217898705 +0200
@@ -411,7 +411,9 @@ CameraInfo* CameraGroup::buildGUICamera(
         if (windowNode) {
             // New style window declaration / definition
             window = wBuild->buildWindow(windowNode);
-            
+	    if (!window) {
+	        return 0;
+	    }
         } else {
             return 0;
         }
