From 7241cc479a0f4a148ae60336add6d7be80da9ff0 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 8 Jul 2010 14:11:18 +0200
Subject: [PATCH 20/39] spice-display: draw.h is internal now

---
 spice-display.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/spice-display.h b/spice-display.h
index 70a7be4..b55e7ea 100644
--- a/spice-display.h
+++ b/spice-display.h
@@ -1,5 +1,5 @@
 #include <spice/ipc_ring.h>
-#include <spice/draw.h>
+#include <spice/enums.h>
 #include <spice/qxl_dev.h>

 #include "pflib.h"
-- 
1.7.2.3

