From c269c8b87769b25c9d69d40944de0e883458af86 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 8 Jul 2010 14:31:10 +0200
Subject: [PATCH 21/39] spice-display: disable debug

---
 spice-display.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/spice-display.c b/spice-display.c
index a749e64..2291cc7 100644
--- a/spice-display.c
+++ b/spice-display.c
@@ -15,7 +15,7 @@

 #include "spice-display.h"

-static int debug = 1;
+static int debug = 0;

 int qemu_spice_rect_is_empty(const QXLRect* r)
 {
-- 
1.7.2.3

