From b0f3b268c79c532725ae23af56621ab2948ac480 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 16 Sep 2010 12:07:51 +0000
Subject: [PATCH 40/40] qxl: flip default to stable pci revision.

---
 hw/qxl.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hw/qxl.c b/hw/qxl.c
index 8448893..d2a0c16 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -1455,7 +1455,7 @@ static PCIDeviceInfo qxl_info = {
     .qdev.props = (Property[]) {
         DEFINE_PROP_UINT32("ram_size", PCIQXLDevice, vga.vram_size, 64 * 1024 * 1024),
         DEFINE_PROP_UINT32("vram_size", PCIQXLDevice, vram_size, 64 * 1024 * 1024),
-        DEFINE_PROP_UINT32("revision", PCIQXLDevice, revision, 3),
+        DEFINE_PROP_UINT32("revision", PCIQXLDevice, revision, 2),
         DEFINE_PROP_UINT32("debug", PCIQXLDevice, debug, 0),
         DEFINE_PROP_UINT32("cmdlog", PCIQXLDevice, cmdlog, 0),
         DEFINE_PROP_END_OF_LIST(),
-- 
1.7.2.3

