Index: pidgin-2.6.5/pidgin/gtkprefs.c
===================================================================
--- pidgin-2.6.5.orig/pidgin/gtkprefs.c
+++ pidgin-2.6.5/pidgin/gtkprefs.c
@@ -2820,8 +2820,8 @@ pidgin_prefs_init(void)
 	/* Browsers */
 	purple_prefs_add_none(PIDGIN_PREFS_ROOT "/browsers");
 	purple_prefs_add_int(PIDGIN_PREFS_ROOT "/browsers/place", PIDGIN_BROWSER_DEFAULT);
-	purple_prefs_add_path(PIDGIN_PREFS_ROOT "/browsers/command", "");
-	purple_prefs_add_string(PIDGIN_PREFS_ROOT "/browsers/browser", "mozilla");
+	purple_prefs_add_path(PIDGIN_PREFS_ROOT "/browsers/command", "xdg-open %s");
+	purple_prefs_add_string(PIDGIN_PREFS_ROOT "/browsers/browser", "custom");
 #endif
 
 	/* Plugins */
