Index: alienarena-7.45/source/Makefile
===================================================================
--- alienarena-7.45.orig/source/Makefile
+++ alienarena-7.45/source/Makefile
@@ -26,6 +26,8 @@ BUILD?=ALL
 WITH_DATADIR?=no
 WITH_LIBDIR?=no
 
+WITH_CURRENTDIR?=yes
+
 # End of configurable options.
 
 VERSION=		1.40
@@ -68,6 +70,10 @@ LIBDIR?=$(PREFIX)/lib/alienarena
 BASE_CFLAGS+=-DLIBDIR='\"$(LIBDIR)\"'
 endif
 
+ifeq ($(strip $(WITH_CURRENTDIR)),yes)
+BASE_CFLAGS+=-DCURRENTDIR='\".\"'
+endif
+
 RELEASE_CFLAGS=$(BASE_CFLAGS)
 
 ifeq ($(strip $(OPTIMIZED_CFLAGS)),yes)
Index: alienarena-7.45/source/qcommon/files.c
===================================================================
--- alienarena-7.45.orig/source/qcommon/files.c
+++ alienarena-7.45/source/qcommon/files.c
@@ -727,7 +727,9 @@ void FS_SetGamedir (char *dir)
 #ifdef LIBDIR
 		FS_AddGameDirectory (va("%s/%s", LIBDIR, dir) );
 #endif
+#ifdef CURRENTDIR
 		FS_AddGameDirectory (va("%s/%s", fs_basedir->string, dir) );
+#endif
 #ifdef __unix__
 		FS_AddHomeAsGameDirectory(dir);
 #endif
