Index: alienarena-7.45/source/qcommon/files.c
===================================================================
--- alienarena-7.45.orig/source/qcommon/files.c
+++ alienarena-7.45/source/qcommon/files.c
@@ -568,7 +568,7 @@ void FS_AddGameDirectory (char *dir)
 ================
 FS_AddHomeAsGameDirectory
 
-Use ~/.codered/dir as fs_gamedir
+Use ~/.alienarena/dir as fs_gamedir
 ================
 */
 void FS_AddHomeAsGameDirectory (char *dir)
@@ -586,7 +586,7 @@ void FS_AddHomeAsGameDirectory (char *di
 		homedir = getenv("HOME");
 		if (!homedir)
 			return;
-		len = snprintf(gdir,sizeof(gdir),"%s/.codered/%s/", homedir, dir);
+		len = snprintf(gdir,sizeof(gdir),"%s/.alienarena/%s/", homedir, dir);
 	}
 
 	Com_Printf("using %s for writing\n",gdir);
