--- xorg-server-1.9.0/hw/xfree86/vbe/vbe.c.orig	2010-08-23 15:55:52.000000000 +0000
+++ xorg-server-1.9.0/hw/xfree86/vbe/vbe.c	2010-08-23 15:56:40.000000000 +0000
@@ -586,10 +586,10 @@ VBEGetModeInfo(vbeInfoPtr pVbe, int mode
 	    memcpy(&block->Reserved2, ((char*)pVbe->memory) + 66, 188);
 	}
 	else
-	memcpy(&block->LinBytesPerScanLine, ((char*)pVbe->memory) + 50, 206);
+	memcpy(&block->LinBytesPerScanLine, ((char*)pVbe->memory) + 50, 205);
     }
     else
-	memcpy(&block->PhysBasePtr, ((char*)pVbe->memory) + 40, 216);
+	memcpy(&block->PhysBasePtr, ((char*)pVbe->memory) + 40, 215);
 
     return block;
 }
