--- scummvm-1.2.0/common/translation.cpp.orig	2010-12-21 09:25:08.085000100 +0200
+++ scummvm-1.2.0/common/translation.cpp	2010-12-21 09:25:38.471000156 +0200
@@ -87,7 +87,7 @@
 	}
 #else // WIN32
 	// Activating current locale settings
-	const char *locale = setlocale(LC_ALL, "");
+	const char *locale = setlocale(LC_ALL, "en_US.UTF-8");
 
 	// Detect the language from the locale
 	if (!locale) {
