Index: cellwriter-1.3.4/src/window.c
===================================================================
--- cellwriter-1.3.4.orig/src/window.c
+++ cellwriter-1.3.4/src/window.c
@@ -817,7 +817,7 @@ UnicodeBlock unicode_blocks[] =
 {
         { TRUE,  0x0000, 0x007F, "Basic Latin" },
         { TRUE,  0x0080, 0x00FF, "Latin-1 Supplement" },
-        { FALSE, 0x0100, 0x017F, "Latin Extended-A" },
+        { TRUE,  0x0100, 0x017F, "Latin Extended-A" },
         { FALSE, 0x0180, 0x024F, "Latin Extended-B" },
         { FALSE, 0x0250, 0x02AF, "IPA Extensions" },
         { FALSE, 0x02B0, 0x02FF, "Spacing Modifier Letters" },
