Index: polkit-0.98/src/polkitagent/Makefile.am
===================================================================
--- polkit-0.98.orig/src/polkitagent/Makefile.am
+++ polkit-0.98/src/polkitagent/Makefile.am
@@ -46,6 +46,7 @@ libpolkit_agent_1includedir=$(includedir
 
 libpolkit_agent_1include_HEADERS =                        				\
 	polkitagent.h									\
+	polkitagentenumtypes.h								\
 	polkitagenttypes.h								\
 	polkitagentsession.h								\
 	polkitagentlistener.h								\
Index: polkit-0.98/src/polkitagent/Makefile.in
===================================================================
--- polkit-0.98.orig/src/polkitagent/Makefile.in
+++ polkit-0.98/src/polkitagent/Makefile.in
@@ -338,6 +338,7 @@ lib_LTLIBRARIES = libpolkit-agent-1.la
 libpolkit_agent_1includedir = $(includedir)/polkit-1/polkitagent
 libpolkit_agent_1include_HEADERS = \
 	polkitagent.h									\
+	polkitagentenumtypes.h								\
 	polkitagenttypes.h								\
 	polkitagentsession.h								\
 	polkitagentlistener.h								\
