diff -Nuar pygtk-2.16.0.orig/gtk/Makefile.in pygtk-2.16.0/gtk/Makefile.in
--- pygtk-2.16.0.orig/gtk/Makefile.in	2009-08-24 14:32:24.670468410 +0300
+++ pygtk-2.16.0/gtk/Makefile.in	2009-08-24 14:36:44.780471236 +0300
@@ -395,7 +395,7 @@
 	gtkwindow.override
 
 _gtk_la_CFLAGS = $(PYCAIRO_CFLAGS) $(GTK_CFLAGS) $(am__append_21)
-_gtk_la_LDFLAGS = $(common_ldflags) -export-symbols-regex init_gtk
+_gtk_la_LDFLAGS = $(common_ldflags) -export-symbols-regex init_gtk `python-config --libs`
 _gtk_la_LIBADD = $(GTK_LIBS)
 _gtk_la_SOURCES = \
 	gtk-types.c 		\
@@ -414,12 +414,12 @@
 LIBGLADE_DEFS = libglade.defs
 glade_la_CFLAGS = $(LIBGLADE_CFLAGS)
 glade_la_LIBADD = $(LIBGLADE_LIBS)
-glade_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initglade
+glade_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initglade `python-config --libs`
 glade_la_SOURCES = libglademodule.c
 nodist_glade_la_SOURCES = libglade.c
 gtkunixprint_la_CFLAGS = $(PYCAIRO_CFLAGS) $(GTKUNIXPRINT_CFLAGS)
 gtkunixprint_la_LIBADD = $(GTKUNIXPRINT_LIBS)
-gtkunixprint_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initgtkunixprint
+gtkunixprint_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initgtkunixprint `python-config --libs`
 gtkunixprint_la_SOURCES = gtkunixprintmodule.c
 nodist_gtkunixprint_la_SOURCES = gtkunixprint.c
 @BUILD_GTKUNIXPRINT_TRUE@pkgpyexec_LTLIBRARIES = gtkunixprint.la
diff -Nuar pygtk-2.16.0.orig/Makefile.in pygtk-2.16.0/Makefile.in
--- pygtk-2.16.0.orig/Makefile.in	2009-08-24 14:32:24.427475753 +0300
+++ pygtk-2.16.0/Makefile.in	2009-08-24 14:38:15.523470531 +0300
@@ -356,7 +356,7 @@
 ATK_OVERRIDES = atk.override atkrectangle.override
 atk_la_CFLAGS = $(ATK_CFLAGS)
 atk_la_LIBADD = $(ATK_LIBS)
-atk_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initatk
+atk_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initatk `python-config --libs`
 atk_la_SOURCES = atkmodule.c
 nodist_atk_la_SOURCES = atk.c
 
@@ -364,14 +364,14 @@
 PANGO_DEFS = pango.defs pango-types.defs
 pango_la_CFLAGS = $(PANGO_CFLAGS)
 pango_la_LIBADD = $(PANGO_LIBS)
-pango_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initpango
+pango_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initpango `python-config --libs`
 pango_la_SOURCES = pangomodule.c
 nodist_pango_la_SOURCES = pango.c
 
 # pangocairo module
 PANGOCAIRO_DEFS = pangocairo.defs pangocairo.override
 pangocairo_la_CFLAGS = $(PYCAIRO_CFLAGS) $(PANGOCAIRO_CFLAGS)
-pangocairo_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initpangocairo
+pangocairo_la_LDFLAGS = $(common_ldflags) -export-symbols-regex initpangocairo `python-config --libs`
 pangocairo_la_LIBADD = $(PANGOCAIRO_LIBS)
 pangocairo_la_SOURCES = pangocairomodule.c
 nodist_pangocairo_la_SOURCES = pangocairo.c
