--- vmpk.pro.orig	2010-10-04 19:32:16.000000000 +0300
+++ vmpk.pro	2010-11-07 19:14:21.716000044 +0200
@@ -53,7 +53,7 @@
         data/help_de.html \
         data/help_es.html \
         data/help_ru.html \
-#       data/help_tr.html \
+        data/help_tr.html \
         data/gmgsxg.ins \
         data/spanish.xml \
         data/german.xml \
@@ -67,13 +67,14 @@
         $$[QT_INSTALL_TRANSLATIONS]/qt_es.qm \
         $$[QT_INSTALL_TRANSLATIONS]/qt_fr.qm \
         $$[QT_INSTALL_TRANSLATIONS]/qt_ru.qm \
+        $$[QT_INSTALL_TRANSLATIONS]/qt_tr.qm \
         $$[QT_INSTALL_TRANSLATIONS]/qt_zh_CN.qm \
         $$DESTDIR/translations/vmpk_cs.qm \
         $$DESTDIR/translations/vmpk_de.qm \
         $$DESTDIR/translations/vmpk_es.qm \
         $$DESTDIR/translations/vmpk_fr.qm \
         $$DESTDIR/translations/vmpk_ru.qm \
-#       $$DESTDIR/translations/vmpk_tr.qm \
+        $$DESTDIR/translations/vmpk_tr.qm \
         $$DESTDIR/translations/vmpk_zh_CN.qm
     BUNDLE_RES.path = Contents/Resources
     QMAKE_BUNDLE_DATA += BUNDLE_RES
@@ -163,6 +164,7 @@
     translations/vmpk_es.ts \
     translations/vmpk_fr.ts \
     translations/vmpk_ru.ts \
+    translations/vmpk_tr.ts \
     translations/vmpk_zh_CN.ts
 
 include(updateqm.pri)
