Index: apmd-3.2.2.orig/Makefile
===================================================================
--- apmd-3.2.2.orig.orig/Makefile
+++ apmd-3.2.2.orig/Makefile
@@ -60,7 +60,7 @@ RANLIB=ranlib
 
 LIBTOOL=libtool --quiet
 LT_COMPILE = $(LIBTOOL) --mode=compile $(CC)
-LT_LINK = $(LIBTOOL) --mode=link $(CC)
+LT_LINK = $(LIBTOOL) --mode=link $(CC) -static
 LT_INSTALL = $(LIBTOOL) --mode=install install
 LT_CLEAN = $(LIBTOOL) --mode=clean rm
 
