diff -ur apmd-3.2.1.orig.orig/Makefile apmd-3.2.1.orig/Makefile
--- apmd-3.2.1.orig.orig/Makefile	2004-01-04 10:13:18.000000000 +0200
+++ apmd-3.2.1.orig/Makefile	2005-12-13 02:07:15.000000000 +0200
@@ -29,7 +29,7 @@
 .SUFFIXES:
 
 OBJS=apmlib.o
-EXES=apm apmd xapm apmsleep
+EXES=apm apmd  apmsleep
 HEADERS=apm.h
 
 PREFIX=/usr
@@ -105,7 +105,7 @@
 	$(LT_INSTALL) -m 644 $(LIBAPM)   $(DESTDIR)$(LIBDIR)
 	$(LT_INSTALL)        apm         $(DESTDIR)$(BINDIR)
 	$(LT_INSTALL)        apmd        $(DESTDIR)$(SBINDIR)
-	$(LT_INSTALL)        xapm        $(DESTDIR)$(BINDIR)
+	echo
 	$(LT_INSTALL)        apmsleep    $(DESTDIR)$(BINDIR)
 	install              on_ac_power $(DESTDIR)$(BINDIR)
 	install       -m 644 apm.h       $(DESTDIR)$(INCDIR)
