diff -Naur lash-0.5.4-orig/configure.ac lash-0.5.4/configure.ac
--- lash-0.5.4-orig/configure.ac	2009-07-29 17:17:55.265442760 +0300
+++ lash-0.5.4/configure.ac	2009-07-29 17:19:20.279443361 +0300
@@ -2,6 +2,7 @@
 AC_CONFIG_SRCDIR([lash/types.h])
 AC_CONFIG_HEADER([config.h])
 AM_INIT_AUTOMAKE
+AC_CONFIG_MACRO_DIR([m4])
 AM_ACLOCAL_INCLUDE([m4])
 
 ### Check for programs ###
diff -Naur lash-0.5.4-orig/Makefile.am lash-0.5.4/Makefile.am
--- lash-0.5.4-orig/Makefile.am	2009-07-29 17:17:55.266442800 +0300
+++ lash-0.5.4/Makefile.am	2009-07-29 17:18:39.279442681 +0300
@@ -3,6 +3,8 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = lash-1.0.pc
 
+ACLOCAL_AMFLAGS = -I m4
+
 EXTRA_DIST = common.am README.SECURITY \
 	lash-configure-template ChangeLog.old
 
