Index: mixxx-1.7.0~beta2/src/SConscript
===================================================================
--- mixxx-1.7.0~beta2.orig/src/SConscript
+++ mixxx-1.7.0~beta2/src/SConscript
@@ -40,7 +40,8 @@ def getSVNRevision(): # GPL code taken f
 	return ""
 
 def getBZRRevision():
-	return os.popen("bzr revno").readline().strip()
+    #return os.popen("bzr revno").readline().strip()
+    return "1.7.0-beta2"
 
 def getMixxxVersion():
 	defs = File('#src/defs.h') #have to handle out-of-tree building, that's why the '#' :(
