Index: mixxx-1.7.0~beta2/src/SConscript
===================================================================
--- mixxx-1.7.0~beta2.orig/src/SConscript
+++ mixxx-1.7.0~beta2/src/SConscript
@@ -233,7 +233,7 @@ else:
 vars.Update(env)
 Help(vars.GenerateHelpText(env))
 
-for getenv in ['CXXFLAGS', 'CCFLAGS', 'LINKFLAGS', 'LIBPATH', 'CPPPATH']:
+for getenv in ['CXXFLAGS', 'CFLAGS', 'LINKFLAGS', 'LIBPATH', 'CPPPATH']:
 	kwargs = {}
 	if os.environ.has_key(getenv):
 		kwargs[getenv] = SCons.Util.CLVar( os.environ[getenv] )
