Index: mixxx-1.7.0/src/m4a/mp4-mixxx.cpp
===================================================================
--- mixxx-1.7.0.orig/src/m4a/mp4-mixxx.cpp
+++ mixxx-1.7.0/src/m4a/mp4-mixxx.cpp
@@ -150,7 +150,7 @@ static int mp4_open(struct input_plugin_
 	}
 
 	/* init decoder according to mpeg-4 audio config */
-	if (faacDecInit2(priv->decoder, buf, buf_size, (uint32_t*) &priv->sample_rate, &priv->channels) < 0) {
+	if (faacDecInit2(priv->decoder, buf, buf_size, &priv->sample_rate, &priv->channels) < 0) {
 		free(buf);
 		goto out;
 	}
