diff -Nur mixxx-1.7.0-old/src/m4a/mp4.c mixxx-1.7.0/src/m4a/mp4.c
--- mixxx-1.7.0-old/src/m4a/mp4.c	2010-02-10 23:09:39.482692463 +0200
+++ mixxx-1.7.0/src/m4a/mp4.c	2010-02-10 23:10:59.804696410 +0200
@@ -22,7 +22,7 @@
 #include "debug.h"
 #include "file.h"
 
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 #include <faad.h>
 
 #include <errno.h>
diff -Nur mixxx-1.7.0-old/src/m4a/mp4-mixxx.cpp mixxx-1.7.0/src/m4a/mp4-mixxx.cpp
--- mixxx-1.7.0-old/src/m4a/mp4-mixxx.cpp	2010-02-10 23:09:39.482692463 +0200
+++ mixxx-1.7.0/src/m4a/mp4-mixxx.cpp	2010-02-10 23:10:59.804696410 +0200
@@ -33,7 +33,7 @@
 // #include "debug.h"
 // #include "file.h"
 
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 #include <faad.h>
 
 #include <errno.h>
diff -Nur mixxx-1.7.0-old/src/soundsourcem4a.h mixxx-1.7.0/src/soundsourcem4a.h
--- mixxx-1.7.0-old/src/soundsourcem4a.h	2010-02-10 23:09:39.487692542 +0200
+++ mixxx-1.7.0/src/soundsourcem4a.h	2010-02-10 23:10:37.480692585 +0200
@@ -19,7 +19,7 @@
 
 #include <qstring.h>
 #include "soundsource.h"
-#include "mp4.h"
+#include <mp4v2/mp4v2.h>
 
 #include "neaacdec.h"
 #include "m4a/ip.h"
