Index: mixxx-1.7.0~beta2/src/SConscript
===================================================================
--- mixxx-1.7.0~beta2.orig/src/SConscript
+++ mixxx-1.7.0~beta2/src/SConscript
@@ -609,9 +609,6 @@ if platform == 'linux':
 	flags_djconsole = getFlags(env, 'djconsole', 0)
 	flags_djconsole_legacy = getFlags(env, 'djconsole_legacy', 0)
 	if int(flags_djconsole):
-		if not conf.CheckLibWithHeader('djconsole', 'libdjconsole/djconsole.h', 'C++'):
-			print "Did not find libdjconsole or it\'s development headers, exiting!"
-			Exit(1)
 		env.ParseConfig('pkg-config libdjconsole --silence-errors --cflags --libs')
 		env.Append(CPPDEFINES = '__LIBDJCONSOLE__')
 	elif int(flags_djconsole_legacy):
