--- RendererModules/directfbRenderer/directfb-renderer.cpp~	2008-11-12 13:48:04.000000000 +0200
+++ RendererModules/directfbRenderer/directfb-renderer.cpp	2009-01-03 16:41:04.000000000 +0200
@@ -22,6 +22,7 @@
  *   ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
  *   OTHER DEALINGS IN THE SOFTWARE.
  ***************************************************************************/
+#include <algorithm>
 #ifdef HAVE_CONFIG_H
 #include "config.h"
 #endif
