diff -Naur CEGUI-0.6.2/src/Makefile.am CEGUI-0.6.2.tpg/src/Makefile.am
--- CEGUI-0.6.2/src/Makefile.am	2008-06-24 19:01:53.000000000 +0000
+++ CEGUI-0.6.2.tpg/src/Makefile.am	2008-12-03 20:29:14.000000000 +0000
@@ -10,7 +10,7 @@
 endif
 
 libCEGUIBase_la_LDFLAGS += -version-info @CEGUI_VERSION_INFO@
-libCEGUIBase_la_LIBADD += $(freetype2_LIBS) $(pcre_LIBS)
+libCEGUIBase_la_LIBADD += $(freetype2_LIBS) $(pcre_LIBS) -ldl
 
 lib_LTLIBRARIES=libCEGUIBase.la
 
