--- include/elements/Makefile.am.orig	2010-11-05 16:51:37.052000029 +0200
+++ include/elements/Makefile.am	2010-11-05 16:52:28.730000029 +0200
@@ -21,8 +21,6 @@
                      CEGUIItemListBaseProperties.h \
                      CEGUIItemListbox.h \
                      CEGUIItemListboxProperties.h \
-                     CEGUIListHeader.h \
-                     CEGUIListHeaderProperties.h \
                      CEGUIListbox.h \
                      CEGUIListboxItem.h \
                      CEGUIListboxProperties.h \
