diff -up qt-everywhere-opensource-src-4.7.0-beta1/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h.s390x qt-everywhere-opensource-src-4.7.0-beta1/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h
--- qt-everywhere-opensource-src-4.7.0-beta1/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h.s390x	2010-06-16 13:54:50.000000000 +0200
+++ qt-everywhere-opensource-src-4.7.0-beta1/src/3rdparty/javascriptcore/JavaScriptCore/wtf/Platform.h	2010-06-16 13:53:43.000000000 +0200
@@ -179,6 +179,18 @@
 #define WTF_CPU_SPARC 1
 #endif
 
+/* CPU(S390X) - S390 64-bit */                                                                                  
+#if defined(__s390x__)                                                                                          
+#define WTF_CPU_S390X 1                                                                                         
+#define WTF_CPU_BIG_ENDIAN 1                                                                                    
+#endif                                                                                                          
+                                                                                                                
+/* CPU(S390) - S390 32-bit */                                                                                   
+#if defined(__s390__)                                                                                           
+#define WTF_CPU_S390 1                                                                                          
+#define WTF_CPU_BIG_ENDIAN 1                                                                                    
+#endif                                                                                                          
+   
 /* CPU(X86) - i386 / x86 32-bit */
 #if   defined(__i386__) \
     || defined(i386)     \
@@ -860,7 +872,7 @@
 #endif
 
 #if !defined(WTF_USE_JSVALUE64) && !defined(WTF_USE_JSVALUE32) && !defined(WTF_USE_JSVALUE32_64)
-#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS) || OS(SOLARIS) || OS(HPUX))) || (CPU(IA64) && !CPU(IA64_32)) || CPU(ALPHA) || CPU(AIX64) || CPU(SPARC64)
+#if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS) || OS(SOLARIS) || OS(HPUX))) || (CPU(IA64) && !CPU(IA64_32)) || CPU(ALPHA) || CPU(AIX64) || CPU(SPARC64) || CPU(S390X)
 #define WTF_USE_JSVALUE64 1
 #elif CPU(ARM) || CPU(PPC64)
 #define WTF_USE_JSVALUE32 1
diff -up qt-everywhere-opensource-src-4.7.0-beta1/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h.s390x qt-everywhere-opensource-src-4.7.0-beta1/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h
--- qt-everywhere-opensource-src-4.7.0-beta1/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h.s390x	2010-06-16 13:49:00.000000000 +0200
+++ qt-everywhere-opensource-src-4.7.0-beta1/src/3rdparty/webkit/JavaScriptCore/wtf/Platform.h	2010-06-16 13:50:01.000000000 +0200
@@ -169,6 +169,18 @@
 #define WTF_CPU_SPARC 1
 #endif
 
+/* CPU(S390X) - S390 64-bit */                                                                                  
+#if defined(__s390x__)                                                                                          
+#define WTF_CPU_S390X 1                                                                                         
+#define WTF_CPU_BIG_ENDIAN 1                                                                                    
+#endif                                                                                                          
+                                                                                                                
+/* CPU(S390) - S390 32-bit */                                                                                   
+#if defined(__s390__)                                                                                           
+#define WTF_CPU_S390 1                                                                                          
+#define WTF_CPU_BIG_ENDIAN 1                                                                                    
+#endif                                                                                                          
+
 /* CPU(X86) - i386 / x86 32-bit */
 #if   defined(__i386__) \
     || defined(i386)     \
@@ -874,6 +886,7 @@
 #if (CPU(X86_64) && (OS(UNIX) || OS(WINDOWS))) \
     || (CPU(IA64) && !CPU(IA64_32)) \
     || CPU(ALPHA) \
+    || CPU(S390X) \
     || CPU(SPARC64)
 #define WTF_USE_JSVALUE64 1
 #elif CPU(ARM) || CPU(PPC64) || CPU(MIPS)
