--- wpa_supplicant-0.6.3/wpa_supplicant/Makefile.qmake	2007-07-30 22:17:29.000000000 -0400
+++ wpa_supplicant-0.6.3/wpa_supplicant/Makefile	2007-07-30 22:17:46.000000000 -0400
@@ -818,7 +818,7 @@ windows-bin: $(WINALL)
 	$(STRIP) $(WINALL)
 
 wpa_gui/Makefile:
-	qmake -o wpa_gui/Makefile wpa_gui/wpa_gui.pro 
+	$(QTDIR)/bin/qmake -o wpa_gui/Makefile wpa_gui/wpa_gui.pro 
 
 wpa_gui: wpa_gui/Makefile
 	$(MAKE) -C wpa_gui
