diff -up wpa_supplicant-0.6.8/src/crypto/tls_openssl.c.ossl-init wpa_supplicant-0.6.8/src/crypto/tls_openssl.c
--- wpa_supplicant-0.6.8/src/crypto/tls_openssl.c.ossl-init	2009-02-15 19:00:00.000000000 +0100
+++ wpa_supplicant-0.6.8/src/crypto/tls_openssl.c	2009-12-15 11:10:31.000000000 +0100
@@ -734,6 +734,7 @@ void * tls_init(const struct tls_config 
 
 	if (tls_openssl_ref_count == 0) {
 		SSL_load_error_strings();
+		OpenSSL_add_all_algorithms();
 		SSL_library_init();
 		/* TODO: if /dev/urandom is available, PRNG is seeded
 		 * automatically. If this is not the case, random data should
