Index: bash-4.2/execute_cmd.c
===================================================================
--- bash-4.2.orig/execute_cmd.c
+++ bash-4.2/execute_cmd.c
@@ -4644,6 +4644,7 @@ execute_disk_command (words, redirects,
 {
   char *pathname, *command, **args;
   int nofork, result;
+  int fval;
   pid_t pid;
   SHELL_VAR *hookf;
   WORD_LIST *wl;
@@ -4756,7 +4757,10 @@ execute_disk_command (words, redirects,
 	    }
 
 	  wl = make_word_list (make_word (NOTFOUND_HOOK), words);
-	  exit (execute_shell_function (hookf, wl));
+	  fval = execute_shell_function (hookf, wl);
+	  if (fval == EX_NOTFOUND)
+	      internal_error (_("%s: command not found"), pathname);
+	  exit(fval);
 	}
 
       /* Execve expects the command name to be in args[0].  So we
