diff -Nur ode-0.11.1-old/drawstuff/src/Makefile.in ode-0.11.1/drawstuff/src/Makefile.in
--- ode-0.11.1-old/drawstuff/src/Makefile.in	2009-07-22 12:06:11.984531307 +0300
+++ ode-0.11.1/drawstuff/src/Makefile.in	2009-07-22 12:07:40.751780302 +0300
@@ -223,7 +223,7 @@
 libdrawstuff_la_SOURCES = drawstuff.cpp internal.h $(am__append_1) \
 	$(am__append_2) $(am__append_3)
 AM_CPPFLAGS = -I$(top_srcdir)/include \
-        -DDEFAULT_PATH_TO_TEXTURES='"$(top_srcdir)/drawstuff/textures/"' \
+        -DDEFAULT_PATH_TO_TEXTURES='"/usr/share/doc/ode/examples/"' \
         $(X_CFLAGS)
 
 @WIN32_TRUE@libdrawstuff_la_LIBADD = -lwinmm -lgdi32
diff -Nur ode-0.11.1-old/ode/demo/Makefile.in ode-0.11.1/ode/demo/Makefile.in
--- ode-0.11.1-old/ode/demo/Makefile.in	2009-07-22 12:06:12.027655475 +0300
+++ ode-0.11.1/ode/demo/Makefile.in	2009-07-22 12:07:40.761780217 +0300
@@ -459,7 +459,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AM_CPPFLAGS = -I$(abs_top_srcdir)/include \
-	-DDRAWSTUFF_TEXTURE_PATH="\"$(abs_top_srcdir)/drawstuff/textures\"" \
+	-DDRAWSTUFF_TEXTURE_PATH="\"/usr/share/doc/ode/examples\"" \
 	$(am__append_2)
 @X11_TRUE@AM_LDFLAGS = $(X_PRE_LIBS) $(X_LIBS) $(X_EXTRA_LIBS)
 
