diff -up cairo-1.8.6/src/cairo-xlib-surface.c.jx cairo-1.8.6/src/cairo-xlib-surface.c
--- cairo-1.8.6/src/cairo-xlib-surface.c.jx	2008-12-12 07:48:05.000000000 -0500
+++ cairo-1.8.6/src/cairo-xlib-surface.c	2009-04-08 15:06:25.000000000 -0400
@@ -1425,7 +1425,11 @@ _cairo_xlib_surface_set_attributes (cair
 	_cairo_xlib_surface_set_repeat (surface, RepeatNormal);
 	break;
     case CAIRO_EXTEND_REFLECT:
+	_cairo_xlib_surface_set_repeat (surface, RepeatReflect);
+	break;
     case CAIRO_EXTEND_PAD:
+	_cairo_xlib_surface_set_repeat (surface, RepeatPad);
+	break;
     default:
 	return CAIRO_INT_STATUS_UNSUPPORTED;
     }
