--- Docs/sample/gnokiirc	2007-05-07 21:06:36.000000000 +0300
+++ Docs/sample/gnokiirc.patched	2007-05-11 00:21:02.000000000 +0300
@@ -7,7 +7,7 @@
 
 # Set port to the physical port used to connect to your phone.
 # Linux version is:
-port = /dev/ttyS0
+#port = /dev/ttyS0
 #
 # For MacOSX you will need something like:
 # port = /dev/cu.USA28X1P1.1
@@ -24,7 +24,10 @@
 # or similiar.
 #
 # Use this setting also for the Bluetooth connection:
-# port = aa:bb:cc:dd:ee:ff
+
+port = aa:bb:cc:dd:ee:ff
+rfcomm_channel = 14
+
 # when using it with AT driver set it to:
 # port = /dev/rfcomm0
 # or similiar.
@@ -45,7 +48,9 @@
 # model = AT
 # If you can't figure out what to put here read the FAQ.
 # If it still doesn't help, consult gnokii-ml or #gnokii at freenode.
-model = 6510
+
+model = symbian
+
 # There are few main models that should make use of the certain drivers.
 # These are: 6110, 7110, 6510, 3110, 2110, 6160.
 
@@ -81,7 +86,9 @@
 # you don't use model = AT) you may want to use 'dlr3p' value here.
 # Note that it is recommended and currently the best way to use this
 # cable with nk6510 driver.
-connection = serial
+
+connection = bluetooth
+
 # With DKU-2 cable use the following setting if you want to libusb driver
 # (recommended):
 # connection = dku2libusb
@@ -102,7 +109,8 @@
 # port software (eg. minicom). If you have wrong permissions for the
 # directory, a warning will be generated. If you don't want a lockfile, set
 # it to 'no'.
-use_locking = yes
+
+use_locking = no
 
 # Baudrate to use on serial port connections.
 # Currently used only by models AT and BIP/CIMD. Defaults to 19200.
@@ -163,7 +171,8 @@
 # permissions 4750, owned by root, group gnokii.  Ensure you
 # are in the gnokii group and that the group exists...
 [gnokiid]
-bindir = /usr/local/sbin/
+
+bindir = /usr/sbin/
 
 # Any entries in the following two sections will be set as environment
 # variables when running the scripts.
