Index: mozilla/security/coreconf/headers.mk
===================================================================
--- mozilla.orig/security/coreconf/headers.mk
+++ mozilla/security/coreconf/headers.mk
@@ -44,7 +44,8 @@
 #  (xp) include paths
 #
 
-INCLUDES += -I$(SOURCE_MDHEADERS_DIR) -I$(SOURCE_XPHEADERS_DIR)
+INCLUDES += -I$(SOURCE_MDHEADERS_DIR) -I$(SOURCE_XPHEADERS_DIR) -I/usr/include/nspr -I$(DIST)/include/dbm
+
 
 #
 #  Only append source-side private cross-platform include paths for
