diff -Nur tremulous-1.1.0-src-old/src/tools/asm/Makefile tremulous-1.1.0-src/src/tools/asm/Makefile
--- tremulous-1.1.0-src-old/src/tools/asm/Makefile	2009-07-24 14:49:37.337328204 +0300
+++ tremulous-1.1.0-src/src/tools/asm/Makefile	2009-07-24 14:49:39.008077897 +0300
@@ -13,7 +13,7 @@
 endif
 
 CC=gcc
-Q3ASM_CFLAGS=-O2 -Wall -Werror -fno-strict-aliasing
+Q3ASM_CFLAGS=-O2 -Wall -fno-strict-aliasing
 
 ifeq ($(PLATFORM),darwin)
   LCC_CFLAGS += -DMACOS_X=1
