diff -Nur file-4.26-old/magic/Magdir/softquad file-4.26/magic/Magdir/softquad
--- file-4.26-old/magic/Magdir/softquad	2009-09-15 23:03:07.835291105 +0300
+++ file-4.26/magic/Magdir/softquad	2009-09-15 23:10:21.578293518 +0300
@@ -20,12 +20,6 @@
 # Bitmaps...
 0	search/1	SQ\ BITMAP1	SoftQuad Raster Format text
 #0	string		SQ\ BITMAP2	SoftQuad Raster Format data
-# sqtroff intermediate language (replacement for ditroff int. lang.)
-0	string		X\ 		SoftQuad troff Context intermediate
->2	string		495		for AT&T 495 laser printer
->2	string		hp		for Hewlett-Packard LaserJet
->2	string		impr		for IMAGEN imPRESS
->2	string		ps		for PostScript
 
 # From: Michael Piefel <piefel@debian.org>
 # sqtroff intermediate language (replacement for ditroff int. lang.)
