diff -Naur file-4.26-orig/magic/Magdir/python file-4.26/magic/Magdir/python
--- file-4.26-orig/magic/Magdir/python	2009-05-07 10:43:05.613119995 +0300
+++ file-4.26/magic/Magdir/python	2009-05-07 10:43:46.428244895 +0300
@@ -14,6 +14,7 @@
 0	belong		0x3bf20d0a	python 2.3 byte-compiled
 0	belong		0x6df20d0a	python 2.4 byte-compiled
 0	belong		0xb3f20d0a	python 2.5 byte-compiled
+0	belong		0xd1f20d0a	python 2.6 byte-compiled
 
 0	string/b  #!\ /usr/bin/python	python script text executable
 
