Index: wireshark-1.4.6/wireshark.desktop
===================================================================
--- wireshark-1.4.6.orig/wireshark.desktop
+++ wireshark-1.4.6/wireshark.desktop
@@ -68,10 +68,11 @@ Comment=Network traffic analyzer
 Comment[fi]=Verkkoliikenne analysaattori
 Comment[fr]=Analyseur de trafic réseau
 Comment[sv]=Nätverkstrafikanalysator
-Icon=wireshark.png
+Icon=wireshark
 TryExec=wireshark
-Exec=wireshark
+Exec=wireshark %f
 Terminal=false
 # Categorie entry according to version 1.0 of
 # http://www.freedesktop.org/wiki/Specifications/menu-spec
 Categories=System;Monitor;GTK;
+MimeType=application/x-pcap;
