Index: readcd/readcd.c
===================================================================
--- readcd/readcd.c.orig
+++ readcd/readcd.c
@@ -370,7 +370,7 @@ main(ac, av)
 	if (help)
 		usage(0);
 	if (pversion) {
-		printf("readcd %s (%s-%s-%s) Copyright (C) 1987, 1995-2010 Jörg Schilling\n",
+		printf("readcd %s (%s-%s-%s) Copyright (C) 1987, 1995-2010 Jorg Schilling\n",
 								cdr_version,
 								HOST_CPU, HOST_VENDOR, HOST_OS);
 		exit(0);
Index: cdrecord/cdrecord.c
===================================================================
--- cdrecord/cdrecord.c.orig
+++ cdrecord/cdrecord.c
@@ -376,7 +376,7 @@ main(ac, av)
 #	define	CLONE_TITLE	""
 #endif
 	if ((flags & F_MSINFO) == 0 || lverbose || flags & F_VERSION) {
-		printf("Cdrecord%s%s%s %s (%s-%s-%s) Copyright (C) 1995-2010 Jörg Schilling\n",
+		printf("Cdrecord%s%s%s %s (%s-%s-%s) Copyright (C) 1995-2010 Jorg Schilling\n",
 								PRODVD_TITLE,
 								PROBD_TITLE,
 								CLONE_TITLE,
